{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create combined schema of tables\n",
    "\n",
    "import sqlite3\n",
    "import json\n",
    "import config\n",
    "import importlib\n",
    "importlib.reload(config)\n",
    "from config import config, reset_config\n",
    "from sqlalchemy import create_engine, MetaData, Table, Column, String\n",
    "import re\n",
    "import time\n",
    "import logging\n",
    "\n",
    "logging.basicConfig(level=logging.INFO)\n",
    "log = logging.getLogger(__name__)\n",
    "\n",
    "def combine_schemas(db_files):\n",
    "    combined_schema = {}\n",
    "\n",
    "    for db_file in db_files:\n",
    "        engine = create_engine('sqlite:///' + db_file)\n",
    "        conn = sqlite3.connect(db_file)\n",
    "        cursor = conn.cursor()\n",
    "\n",
    "        # Get all tables\n",
    "        cursor.execute(\"SELECT name FROM sqlite_master WHERE type='table';\")\n",
    "        tables = cursor.fetchall()\n",
    "\n",
    "        for table in tables:\n",
    "            table_name = table[0]\n",
    "            metadata_obj = MetaData()\n",
    "            \n",
    "            # Get table schema\n",
    "            cursor.execute(f\"PRAGMA table_info({table_name});\")\n",
    "            schema = cursor.fetchall()\n",
    "\n",
    "            # Create a Table object to store schema info\n",
    "            table_obj = Table(table_name, metadata_obj)\n",
    "\n",
    "            for column in schema:\n",
    "                col_name, col_type = column[1], column[2]\n",
    "                # Add column to the table object\n",
    "                table_obj.append_column(Column(col_name, String))\n",
    "\n",
    "            # Serialize table schema\n",
    "            schema_info = [{\"column_name\": col.name, \"data_type\": str(col.type)} for col in table_obj.columns]\n",
    "            combined_schema[f\"{table_name} in {db_file}\"] = schema_info\n",
    "\n",
    "        conn.close()\n",
    "\n",
    "    return combined_schema\n",
    "\n",
    "def save_schema_to_json(combined_schema, filename=config.db_schema):\n",
    "    with open(filename, \"w\") as file:\n",
    "        json.dump(combined_schema, file, indent=4)\n",
    "\n",
    "# SQL check\n",
    "def extract_sql(llm_response: str) -> str:\n",
    "    # If the llm_response contains a markdown code block, with or without the sql tag, extract the sql from it\n",
    "    sql = re.search(r\"```sql\\n(.*)```\", llm_response, re.DOTALL)\n",
    "    if sql:\n",
    "        log(f\"Output from LLM: {llm_response} \\nExtracted SQL: {sql.group(1)}\")\n",
    "        return sql.group(1)\n",
    "\n",
    "    sql = re.search(r\"```(.*)```\", llm_response, re.DOTALL)\n",
    "    if sql:\n",
    "        log(f\"Output from LLM: {llm_response} \\nExtracted SQL: {sql.group(1)}\")\n",
    "        return sql.group(1)\n",
    "\n",
    "    return llm_response\n",
    "\n",
    "def is_sql_valid(sql: str) -> bool:\n",
    "    # This is a check to see the SQL is valid and should be run\n",
    "    # This simple function just checks if the SQL contains a SELECT statement\n",
    "\n",
    "    if \"SELECT\" in sql.upper():\n",
    "        return True\n",
    "    else:\n",
    "        return False\n",
    "\n",
    "# Paths to your database files\n",
    "db_files = [\"files/db/CCLEGisticCNDB.db\" , 'files/db/CCLEMutDB.db', 'files/db/CCLEVarDB.db']\n",
    "all_schemas = combine_schemas(db_files)\n",
    "save_schema_to_json(all_schemas)\n",
    "# Now the schema is saved in 'combined_schema.json'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No duplicates found.\n",
      "You are an exceptional computational biologist and genomics expert and know everything about drug discovery.. Please write the appropriate SQL query using these three tables. The schemas are as {'ccle_cn_gistic in files/db/CCLEGisticCNDB.db': [{'column_name': 'Hugo_Symbol', 'data_type': 'VARCHAR'}, {'column_name': 'CCLE_Name', 'data_type': 'VARCHAR'}, {'column_name': 'gistic_cn', 'data_type': 'VARCHAR'}, {'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}], 'ccle_mutation in files/db/CCLEMutDB.db': [{'column_name': 'Hugo_Symbol', 'data_type': 'VARCHAR'}, {'column_name': 'Entrez_Gene_Id', 'data_type': 'VARCHAR'}, {'column_name': 'NCBI_Build', 'data_type': 'VARCHAR'}, {'column_name': 'Chromosome', 'data_type': 'VARCHAR'}, {'column_name': 'Start_position', 'data_type': 'VARCHAR'}, {'column_name': 'End_position', 'data_type': 'VARCHAR'}, {'column_name': 'Strand', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_Classification', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_Type', 'data_type': 'VARCHAR'}, {'column_name': 'Reference_Allele', 'data_type': 'VARCHAR'}, {'column_name': 'Alternate_Allele', 'data_type': 'VARCHAR'}, {'column_name': 'dbSNP_RS', 'data_type': 'VARCHAR'}, {'column_name': 'dbSNP_Val_Status', 'data_type': 'VARCHAR'}, {'column_name': 'Genome_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Annotation_Transcript', 'data_type': 'VARCHAR'}, {'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}, {'column_name': 'cDNA_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Codon_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Protein_Change', 'data_type': 'VARCHAR'}, {'column_name': 'isDeleterious', 'data_type': 'VARCHAR'}, {'column_name': 'isTCGAhotspot', 'data_type': 'VARCHAR'}, {'column_name': 'TCGAhsCnt', 'data_type': 'VARCHAR'}, {'column_name': 'isCOSMIChotspot', 'data_type': 'VARCHAR'}, {'column_name': 'COSMIChsCnt', 'data_type': 'VARCHAR'}, {'column_name': 'ExAC_AF', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_annotation', 'data_type': 'VARCHAR'}, {'column_name': 'CGA_WES_AC', 'data_type': 'VARCHAR'}, {'column_name': 'HC_AC', 'data_type': 'VARCHAR'}, {'column_name': 'RD_AC', 'data_type': 'VARCHAR'}, {'column_name': 'RNAseq_AC', 'data_type': 'VARCHAR'}, {'column_name': 'SangerWES_AC', 'data_type': 'VARCHAR'}, {'column_name': 'WGS_AC', 'data_type': 'VARCHAR'}], 'ccle_variants in files/db/CCLEVarDB.db': [{'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}, {'column_name': 'Chrom', 'data_type': 'VARCHAR'}, {'column_name': 'Pos', 'data_type': 'VARCHAR'}, {'column_name': 'Ref', 'data_type': 'VARCHAR'}, {'column_name': 'Alt', 'data_type': 'VARCHAR'}, {'column_name': 'AF', 'data_type': 'VARCHAR'}, {'column_name': 'VariantType', 'data_type': 'VARCHAR'}, {'column_name': 'VariantInfo', 'data_type': 'VARCHAR'}, {'column_name': 'DNAChange', 'data_type': 'VARCHAR'}, {'column_name': 'ProteinChange', 'data_type': 'VARCHAR'}, {'column_name': 'gene_name', 'data_type': 'VARCHAR'}, {'column_name': 'Transcript', 'data_type': 'VARCHAR'}, {'column_name': 'CCLEDeleterious', 'data_type': 'VARCHAR'}, {'column_name': 'CosmicHotspot', 'data_type': 'VARCHAR'}, {'column_name': 'AssociatedWith', 'data_type': 'VARCHAR'}, {'column_name': 'LoF', 'data_type': 'VARCHAR'}, {'column_name': 'Driver', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyDriver', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyGoF', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyLoF', 'data_type': 'VARCHAR'}, {'column_name': 'VariantAnnotation', 'data_type': 'VARCHAR'}, {'column_name': 'copy_number', 'data_type': 'VARCHAR'}]}. The tables can be understood as utils/database_description.json. Try to answer the following question. Only the SQL should be returned within ''' SQL query '''. \"Which cell lines have high dependency for the target of interest for the gene POLR3E?\"\n",
      "You are an exceptional computational biologist and genomics expert and know everything about drug discovery.. Please write the appropriate SQL query using these three tables. The schemas are as {'ccle_cn_gistic in files/db/CCLEGisticCNDB.db': [{'column_name': 'Hugo_Symbol', 'data_type': 'VARCHAR'}, {'column_name': 'CCLE_Name', 'data_type': 'VARCHAR'}, {'column_name': 'gistic_cn', 'data_type': 'VARCHAR'}, {'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}], 'ccle_mutation in files/db/CCLEMutDB.db': [{'column_name': 'Hugo_Symbol', 'data_type': 'VARCHAR'}, {'column_name': 'Entrez_Gene_Id', 'data_type': 'VARCHAR'}, {'column_name': 'NCBI_Build', 'data_type': 'VARCHAR'}, {'column_name': 'Chromosome', 'data_type': 'VARCHAR'}, {'column_name': 'Start_position', 'data_type': 'VARCHAR'}, {'column_name': 'End_position', 'data_type': 'VARCHAR'}, {'column_name': 'Strand', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_Classification', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_Type', 'data_type': 'VARCHAR'}, {'column_name': 'Reference_Allele', 'data_type': 'VARCHAR'}, {'column_name': 'Alternate_Allele', 'data_type': 'VARCHAR'}, {'column_name': 'dbSNP_RS', 'data_type': 'VARCHAR'}, {'column_name': 'dbSNP_Val_Status', 'data_type': 'VARCHAR'}, {'column_name': 'Genome_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Annotation_Transcript', 'data_type': 'VARCHAR'}, {'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}, {'column_name': 'cDNA_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Codon_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Protein_Change', 'data_type': 'VARCHAR'}, {'column_name': 'isDeleterious', 'data_type': 'VARCHAR'}, {'column_name': 'isTCGAhotspot', 'data_type': 'VARCHAR'}, {'column_name': 'TCGAhsCnt', 'data_type': 'VARCHAR'}, {'column_name': 'isCOSMIChotspot', 'data_type': 'VARCHAR'}, {'column_name': 'COSMIChsCnt', 'data_type': 'VARCHAR'}, {'column_name': 'ExAC_AF', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_annotation', 'data_type': 'VARCHAR'}, {'column_name': 'CGA_WES_AC', 'data_type': 'VARCHAR'}, {'column_name': 'HC_AC', 'data_type': 'VARCHAR'}, {'column_name': 'RD_AC', 'data_type': 'VARCHAR'}, {'column_name': 'RNAseq_AC', 'data_type': 'VARCHAR'}, {'column_name': 'SangerWES_AC', 'data_type': 'VARCHAR'}, {'column_name': 'WGS_AC', 'data_type': 'VARCHAR'}], 'ccle_variants in files/db/CCLEVarDB.db': [{'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}, {'column_name': 'Chrom', 'data_type': 'VARCHAR'}, {'column_name': 'Pos', 'data_type': 'VARCHAR'}, {'column_name': 'Ref', 'data_type': 'VARCHAR'}, {'column_name': 'Alt', 'data_type': 'VARCHAR'}, {'column_name': 'AF', 'data_type': 'VARCHAR'}, {'column_name': 'VariantType', 'data_type': 'VARCHAR'}, {'column_name': 'VariantInfo', 'data_type': 'VARCHAR'}, {'column_name': 'DNAChange', 'data_type': 'VARCHAR'}, {'column_name': 'ProteinChange', 'data_type': 'VARCHAR'}, {'column_name': 'gene_name', 'data_type': 'VARCHAR'}, {'column_name': 'Transcript', 'data_type': 'VARCHAR'}, {'column_name': 'CCLEDeleterious', 'data_type': 'VARCHAR'}, {'column_name': 'CosmicHotspot', 'data_type': 'VARCHAR'}, {'column_name': 'AssociatedWith', 'data_type': 'VARCHAR'}, {'column_name': 'LoF', 'data_type': 'VARCHAR'}, {'column_name': 'Driver', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyDriver', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyGoF', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyLoF', 'data_type': 'VARCHAR'}, {'column_name': 'VariantAnnotation', 'data_type': 'VARCHAR'}, {'column_name': 'copy_number', 'data_type': 'VARCHAR'}]}. The tables can be understood as utils/database_description.json. Try to answer the following question. Only the SQL should be returned within ''' SQL query '''. \"For the gene POLR3E, identify indications with significant differential expression in cancer vs normal tissue. Is it overexpressed or underexpressed in tumor tissue?\"\n",
      "You are an exceptional computational biologist and genomics expert and know everything about drug discovery.. Please write the appropriate SQL query using these three tables. The schemas are as {'ccle_cn_gistic in files/db/CCLEGisticCNDB.db': [{'column_name': 'Hugo_Symbol', 'data_type': 'VARCHAR'}, {'column_name': 'CCLE_Name', 'data_type': 'VARCHAR'}, {'column_name': 'gistic_cn', 'data_type': 'VARCHAR'}, {'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}], 'ccle_mutation in files/db/CCLEMutDB.db': [{'column_name': 'Hugo_Symbol', 'data_type': 'VARCHAR'}, {'column_name': 'Entrez_Gene_Id', 'data_type': 'VARCHAR'}, {'column_name': 'NCBI_Build', 'data_type': 'VARCHAR'}, {'column_name': 'Chromosome', 'data_type': 'VARCHAR'}, {'column_name': 'Start_position', 'data_type': 'VARCHAR'}, {'column_name': 'End_position', 'data_type': 'VARCHAR'}, {'column_name': 'Strand', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_Classification', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_Type', 'data_type': 'VARCHAR'}, {'column_name': 'Reference_Allele', 'data_type': 'VARCHAR'}, {'column_name': 'Alternate_Allele', 'data_type': 'VARCHAR'}, {'column_name': 'dbSNP_RS', 'data_type': 'VARCHAR'}, {'column_name': 'dbSNP_Val_Status', 'data_type': 'VARCHAR'}, {'column_name': 'Genome_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Annotation_Transcript', 'data_type': 'VARCHAR'}, {'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}, {'column_name': 'cDNA_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Codon_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Protein_Change', 'data_type': 'VARCHAR'}, {'column_name': 'isDeleterious', 'data_type': 'VARCHAR'}, {'column_name': 'isTCGAhotspot', 'data_type': 'VARCHAR'}, {'column_name': 'TCGAhsCnt', 'data_type': 'VARCHAR'}, {'column_name': 'isCOSMIChotspot', 'data_type': 'VARCHAR'}, {'column_name': 'COSMIChsCnt', 'data_type': 'VARCHAR'}, {'column_name': 'ExAC_AF', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_annotation', 'data_type': 'VARCHAR'}, {'column_name': 'CGA_WES_AC', 'data_type': 'VARCHAR'}, {'column_name': 'HC_AC', 'data_type': 'VARCHAR'}, {'column_name': 'RD_AC', 'data_type': 'VARCHAR'}, {'column_name': 'RNAseq_AC', 'data_type': 'VARCHAR'}, {'column_name': 'SangerWES_AC', 'data_type': 'VARCHAR'}, {'column_name': 'WGS_AC', 'data_type': 'VARCHAR'}], 'ccle_variants in files/db/CCLEVarDB.db': [{'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}, {'column_name': 'Chrom', 'data_type': 'VARCHAR'}, {'column_name': 'Pos', 'data_type': 'VARCHAR'}, {'column_name': 'Ref', 'data_type': 'VARCHAR'}, {'column_name': 'Alt', 'data_type': 'VARCHAR'}, {'column_name': 'AF', 'data_type': 'VARCHAR'}, {'column_name': 'VariantType', 'data_type': 'VARCHAR'}, {'column_name': 'VariantInfo', 'data_type': 'VARCHAR'}, {'column_name': 'DNAChange', 'data_type': 'VARCHAR'}, {'column_name': 'ProteinChange', 'data_type': 'VARCHAR'}, {'column_name': 'gene_name', 'data_type': 'VARCHAR'}, {'column_name': 'Transcript', 'data_type': 'VARCHAR'}, {'column_name': 'CCLEDeleterious', 'data_type': 'VARCHAR'}, {'column_name': 'CosmicHotspot', 'data_type': 'VARCHAR'}, {'column_name': 'AssociatedWith', 'data_type': 'VARCHAR'}, {'column_name': 'LoF', 'data_type': 'VARCHAR'}, {'column_name': 'Driver', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyDriver', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyGoF', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyLoF', 'data_type': 'VARCHAR'}, {'column_name': 'VariantAnnotation', 'data_type': 'VARCHAR'}, {'column_name': 'copy_number', 'data_type': 'VARCHAR'}]}. The tables can be understood as utils/database_description.json. Try to answer the following question. Only the SQL should be returned within ''' SQL query '''. \"Are there specific regions of the protein of the gene POLR3E where mutations occur at a higher frequency?\"\n",
      "You are an exceptional computational biologist and genomics expert and know everything about drug discovery.. Please write the appropriate SQL query using these three tables. The schemas are as {'ccle_cn_gistic in files/db/CCLEGisticCNDB.db': [{'column_name': 'Hugo_Symbol', 'data_type': 'VARCHAR'}, {'column_name': 'CCLE_Name', 'data_type': 'VARCHAR'}, {'column_name': 'gistic_cn', 'data_type': 'VARCHAR'}, {'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}], 'ccle_mutation in files/db/CCLEMutDB.db': [{'column_name': 'Hugo_Symbol', 'data_type': 'VARCHAR'}, {'column_name': 'Entrez_Gene_Id', 'data_type': 'VARCHAR'}, {'column_name': 'NCBI_Build', 'data_type': 'VARCHAR'}, {'column_name': 'Chromosome', 'data_type': 'VARCHAR'}, {'column_name': 'Start_position', 'data_type': 'VARCHAR'}, {'column_name': 'End_position', 'data_type': 'VARCHAR'}, {'column_name': 'Strand', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_Classification', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_Type', 'data_type': 'VARCHAR'}, {'column_name': 'Reference_Allele', 'data_type': 'VARCHAR'}, {'column_name': 'Alternate_Allele', 'data_type': 'VARCHAR'}, {'column_name': 'dbSNP_RS', 'data_type': 'VARCHAR'}, {'column_name': 'dbSNP_Val_Status', 'data_type': 'VARCHAR'}, {'column_name': 'Genome_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Annotation_Transcript', 'data_type': 'VARCHAR'}, {'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}, {'column_name': 'cDNA_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Codon_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Protein_Change', 'data_type': 'VARCHAR'}, {'column_name': 'isDeleterious', 'data_type': 'VARCHAR'}, {'column_name': 'isTCGAhotspot', 'data_type': 'VARCHAR'}, {'column_name': 'TCGAhsCnt', 'data_type': 'VARCHAR'}, {'column_name': 'isCOSMIChotspot', 'data_type': 'VARCHAR'}, {'column_name': 'COSMIChsCnt', 'data_type': 'VARCHAR'}, {'column_name': 'ExAC_AF', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_annotation', 'data_type': 'VARCHAR'}, {'column_name': 'CGA_WES_AC', 'data_type': 'VARCHAR'}, {'column_name': 'HC_AC', 'data_type': 'VARCHAR'}, {'column_name': 'RD_AC', 'data_type': 'VARCHAR'}, {'column_name': 'RNAseq_AC', 'data_type': 'VARCHAR'}, {'column_name': 'SangerWES_AC', 'data_type': 'VARCHAR'}, {'column_name': 'WGS_AC', 'data_type': 'VARCHAR'}], 'ccle_variants in files/db/CCLEVarDB.db': [{'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}, {'column_name': 'Chrom', 'data_type': 'VARCHAR'}, {'column_name': 'Pos', 'data_type': 'VARCHAR'}, {'column_name': 'Ref', 'data_type': 'VARCHAR'}, {'column_name': 'Alt', 'data_type': 'VARCHAR'}, {'column_name': 'AF', 'data_type': 'VARCHAR'}, {'column_name': 'VariantType', 'data_type': 'VARCHAR'}, {'column_name': 'VariantInfo', 'data_type': 'VARCHAR'}, {'column_name': 'DNAChange', 'data_type': 'VARCHAR'}, {'column_name': 'ProteinChange', 'data_type': 'VARCHAR'}, {'column_name': 'gene_name', 'data_type': 'VARCHAR'}, {'column_name': 'Transcript', 'data_type': 'VARCHAR'}, {'column_name': 'CCLEDeleterious', 'data_type': 'VARCHAR'}, {'column_name': 'CosmicHotspot', 'data_type': 'VARCHAR'}, {'column_name': 'AssociatedWith', 'data_type': 'VARCHAR'}, {'column_name': 'LoF', 'data_type': 'VARCHAR'}, {'column_name': 'Driver', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyDriver', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyGoF', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyLoF', 'data_type': 'VARCHAR'}, {'column_name': 'VariantAnnotation', 'data_type': 'VARCHAR'}, {'column_name': 'copy_number', 'data_type': 'VARCHAR'}]}. The tables can be understood as utils/database_description.json. Try to answer the following question. Only the SQL should be returned within ''' SQL query '''. \"For the gene POLR3E, determine how many of the mutations are pathogenic.\"\n",
      "You are an exceptional computational biologist and genomics expert and know everything about drug discovery.. Please write the appropriate SQL query using these three tables. The schemas are as {'ccle_cn_gistic in files/db/CCLEGisticCNDB.db': [{'column_name': 'Hugo_Symbol', 'data_type': 'VARCHAR'}, {'column_name': 'CCLE_Name', 'data_type': 'VARCHAR'}, {'column_name': 'gistic_cn', 'data_type': 'VARCHAR'}, {'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}], 'ccle_mutation in files/db/CCLEMutDB.db': [{'column_name': 'Hugo_Symbol', 'data_type': 'VARCHAR'}, {'column_name': 'Entrez_Gene_Id', 'data_type': 'VARCHAR'}, {'column_name': 'NCBI_Build', 'data_type': 'VARCHAR'}, {'column_name': 'Chromosome', 'data_type': 'VARCHAR'}, {'column_name': 'Start_position', 'data_type': 'VARCHAR'}, {'column_name': 'End_position', 'data_type': 'VARCHAR'}, {'column_name': 'Strand', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_Classification', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_Type', 'data_type': 'VARCHAR'}, {'column_name': 'Reference_Allele', 'data_type': 'VARCHAR'}, {'column_name': 'Alternate_Allele', 'data_type': 'VARCHAR'}, {'column_name': 'dbSNP_RS', 'data_type': 'VARCHAR'}, {'column_name': 'dbSNP_Val_Status', 'data_type': 'VARCHAR'}, {'column_name': 'Genome_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Annotation_Transcript', 'data_type': 'VARCHAR'}, {'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}, {'column_name': 'cDNA_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Codon_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Protein_Change', 'data_type': 'VARCHAR'}, {'column_name': 'isDeleterious', 'data_type': 'VARCHAR'}, {'column_name': 'isTCGAhotspot', 'data_type': 'VARCHAR'}, {'column_name': 'TCGAhsCnt', 'data_type': 'VARCHAR'}, {'column_name': 'isCOSMIChotspot', 'data_type': 'VARCHAR'}, {'column_name': 'COSMIChsCnt', 'data_type': 'VARCHAR'}, {'column_name': 'ExAC_AF', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_annotation', 'data_type': 'VARCHAR'}, {'column_name': 'CGA_WES_AC', 'data_type': 'VARCHAR'}, {'column_name': 'HC_AC', 'data_type': 'VARCHAR'}, {'column_name': 'RD_AC', 'data_type': 'VARCHAR'}, {'column_name': 'RNAseq_AC', 'data_type': 'VARCHAR'}, {'column_name': 'SangerWES_AC', 'data_type': 'VARCHAR'}, {'column_name': 'WGS_AC', 'data_type': 'VARCHAR'}], 'ccle_variants in files/db/CCLEVarDB.db': [{'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}, {'column_name': 'Chrom', 'data_type': 'VARCHAR'}, {'column_name': 'Pos', 'data_type': 'VARCHAR'}, {'column_name': 'Ref', 'data_type': 'VARCHAR'}, {'column_name': 'Alt', 'data_type': 'VARCHAR'}, {'column_name': 'AF', 'data_type': 'VARCHAR'}, {'column_name': 'VariantType', 'data_type': 'VARCHAR'}, {'column_name': 'VariantInfo', 'data_type': 'VARCHAR'}, {'column_name': 'DNAChange', 'data_type': 'VARCHAR'}, {'column_name': 'ProteinChange', 'data_type': 'VARCHAR'}, {'column_name': 'gene_name', 'data_type': 'VARCHAR'}, {'column_name': 'Transcript', 'data_type': 'VARCHAR'}, {'column_name': 'CCLEDeleterious', 'data_type': 'VARCHAR'}, {'column_name': 'CosmicHotspot', 'data_type': 'VARCHAR'}, {'column_name': 'AssociatedWith', 'data_type': 'VARCHAR'}, {'column_name': 'LoF', 'data_type': 'VARCHAR'}, {'column_name': 'Driver', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyDriver', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyGoF', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyLoF', 'data_type': 'VARCHAR'}, {'column_name': 'VariantAnnotation', 'data_type': 'VARCHAR'}, {'column_name': 'copy_number', 'data_type': 'VARCHAR'}]}. The tables can be understood as utils/database_description.json. Try to answer the following question. Only the SQL should be returned within ''' SQL query '''. \"Which mutations in PAS domain for the target protein exhibit high selectivity for the prostate cancer cell lines?\"\n",
      "You are an exceptional computational biologist and genomics expert and know everything about drug discovery.. Please write the appropriate SQL query using these three tables. The schemas are as {'ccle_cn_gistic in files/db/CCLEGisticCNDB.db': [{'column_name': 'Hugo_Symbol', 'data_type': 'VARCHAR'}, {'column_name': 'CCLE_Name', 'data_type': 'VARCHAR'}, {'column_name': 'gistic_cn', 'data_type': 'VARCHAR'}, {'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}], 'ccle_mutation in files/db/CCLEMutDB.db': [{'column_name': 'Hugo_Symbol', 'data_type': 'VARCHAR'}, {'column_name': 'Entrez_Gene_Id', 'data_type': 'VARCHAR'}, {'column_name': 'NCBI_Build', 'data_type': 'VARCHAR'}, {'column_name': 'Chromosome', 'data_type': 'VARCHAR'}, {'column_name': 'Start_position', 'data_type': 'VARCHAR'}, {'column_name': 'End_position', 'data_type': 'VARCHAR'}, {'column_name': 'Strand', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_Classification', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_Type', 'data_type': 'VARCHAR'}, {'column_name': 'Reference_Allele', 'data_type': 'VARCHAR'}, {'column_name': 'Alternate_Allele', 'data_type': 'VARCHAR'}, {'column_name': 'dbSNP_RS', 'data_type': 'VARCHAR'}, {'column_name': 'dbSNP_Val_Status', 'data_type': 'VARCHAR'}, {'column_name': 'Genome_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Annotation_Transcript', 'data_type': 'VARCHAR'}, {'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}, {'column_name': 'cDNA_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Codon_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Protein_Change', 'data_type': 'VARCHAR'}, {'column_name': 'isDeleterious', 'data_type': 'VARCHAR'}, {'column_name': 'isTCGAhotspot', 'data_type': 'VARCHAR'}, {'column_name': 'TCGAhsCnt', 'data_type': 'VARCHAR'}, {'column_name': 'isCOSMIChotspot', 'data_type': 'VARCHAR'}, {'column_name': 'COSMIChsCnt', 'data_type': 'VARCHAR'}, {'column_name': 'ExAC_AF', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_annotation', 'data_type': 'VARCHAR'}, {'column_name': 'CGA_WES_AC', 'data_type': 'VARCHAR'}, {'column_name': 'HC_AC', 'data_type': 'VARCHAR'}, {'column_name': 'RD_AC', 'data_type': 'VARCHAR'}, {'column_name': 'RNAseq_AC', 'data_type': 'VARCHAR'}, {'column_name': 'SangerWES_AC', 'data_type': 'VARCHAR'}, {'column_name': 'WGS_AC', 'data_type': 'VARCHAR'}], 'ccle_variants in files/db/CCLEVarDB.db': [{'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}, {'column_name': 'Chrom', 'data_type': 'VARCHAR'}, {'column_name': 'Pos', 'data_type': 'VARCHAR'}, {'column_name': 'Ref', 'data_type': 'VARCHAR'}, {'column_name': 'Alt', 'data_type': 'VARCHAR'}, {'column_name': 'AF', 'data_type': 'VARCHAR'}, {'column_name': 'VariantType', 'data_type': 'VARCHAR'}, {'column_name': 'VariantInfo', 'data_type': 'VARCHAR'}, {'column_name': 'DNAChange', 'data_type': 'VARCHAR'}, {'column_name': 'ProteinChange', 'data_type': 'VARCHAR'}, {'column_name': 'gene_name', 'data_type': 'VARCHAR'}, {'column_name': 'Transcript', 'data_type': 'VARCHAR'}, {'column_name': 'CCLEDeleterious', 'data_type': 'VARCHAR'}, {'column_name': 'CosmicHotspot', 'data_type': 'VARCHAR'}, {'column_name': 'AssociatedWith', 'data_type': 'VARCHAR'}, {'column_name': 'LoF', 'data_type': 'VARCHAR'}, {'column_name': 'Driver', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyDriver', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyGoF', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyLoF', 'data_type': 'VARCHAR'}, {'column_name': 'VariantAnnotation', 'data_type': 'VARCHAR'}, {'column_name': 'copy_number', 'data_type': 'VARCHAR'}]}. The tables can be understood as utils/database_description.json. Try to answer the following question. Only the SQL should be returned within ''' SQL query '''. \"Relationship between dependency scores of JAK1 in DepMapDB.\"\n"
     ]
    }
   ],
   "source": [
    "# LLM checks to write a SQL query\n",
    "\n",
    "import replicate\n",
    "import pandas as pd\n",
    "import json\n",
    "import os\n",
    "import config\n",
    "import importlib\n",
    "importlib.reload(config)\n",
    "from config import config, reset_config\n",
    "from dotenv import load_dotenv\n",
    "load_dotenv()\n",
    "import time\n",
    "\n",
    "config.set_mode(\"dbs\")\n",
    "\n",
    "folder_path = 'files'\n",
    "if not os.path.exists(folder_path):\n",
    "    os.makedirs(folder_path)\n",
    "\n",
    "INSTRUCTION = config.INSTRUCTION\n",
    "F_NAME = config.F_NAME\n",
    "\n",
    "def load_file(file_path):\n",
    "    with open(file_path, 'r') as file:\n",
    "        return json.load(file)\n",
    "\n",
    "db_desc = load_file(config.db_layout)\n",
    "all_schemas = load_file(config.db_schema)\n",
    "\n",
    "df = pd.read_excel(config.questions)\n",
    "df.to_excel(config.q_original, index=False)\n",
    "\n",
    "df['Question'] = df['Question'].str.strip()  # Removes leading/trailing whitespace\n",
    "\n",
    "# Check for duplicate questions\n",
    "duplicates = df.duplicated(subset=['Question'], keep=False)\n",
    "if duplicates.any():\n",
    "    print(\"Duplicates found. Removing duplicates.\")\n",
    "    df = df.drop_duplicates(subset=['Question'], keep='first')\n",
    "    df.to_excel(config.q_db, index=False)\n",
    "else:\n",
    "    print(\"No duplicates found.\")\n",
    "\n",
    "# DataFrame to store the results\n",
    "results_df = pd.DataFrame(columns=['Model', 'Question', 'Response', 'Latency'])\n",
    "\n",
    "models = {\n",
    "    \"qwen-14b\": \"nomagick/qwen-14b-chat:f9e1ed25e2073f72ff9a3f46545d909b1078e674da543e791dec79218072ae70\",\n",
    "    \"falcon-40b\": \"joehoover/falcon-40b-instruct:7d58d6bddc53c23fa451c403b2b5373b1e0fa094e4e0d1b98c3d02931aa07173\",\n",
    "    # \"yi-34b\": \"01-ai/yi-34b-chat:914692bbe8a8e2b91a4e44203e70d170c9c5ccc1359b283c84b0ec8d47819a46\",\n",
    "    \"mistral-7b\": \"mistralai/mistral-7b-instruct-v0.2:f5701ad84de5715051cb99d550539719f8a7fbcf65e0e62a3d1eb3f94720764e\",\n",
    "    \"llama2-70b\": \"meta/llama-2-70b-chat\",\n",
    "    # \"openhermes2\": \"antoinelyset/openhermes-2.5-mistral-7b:d7ccd25700fb11c1787c25b580ac8d715d2b677202fe54b77f9b4a1eb7d73e2b\",\n",
    "    \"mixtral-instruct\": \"mistralai/mixtral-8x7b-instruct-v0.1:2b56576fcfbe32fa0526897d8385dd3fb3d36ba6fd0dbe033c72886b81ade93e\",\n",
    "    \"deepseek_33bq\": \"kcaverly/deepseek-coder-33b-instruct-gguf:ea964345066a8868e43aca432f314822660b72e29cab6b4b904b779014fe58fd\",\n",
    "    }\n",
    "\n",
    "prompt_for_qwen=\"\"\"<|im_start|>system\\n {INSTRUCTION}. Please write the appropriate SQL query using these three tables. The schemas are as {all_schemas}. The tables can be understood as {config.db_layout}. Try to answer the following question. The SQL should be returned within ''' SQL query '''. <|im_end|>\\n<|im_start|>user\\n{question}<|im_end|>\\n<|im_start|>assistant\\n\"\"\"\n",
    "prompt_for_hermes = \"\"\"[\n",
    "{{\n",
    "  \"role\": \"system\",\n",
    "  \"content\": \"{INSTRUCTION}. Please write the appropriate SQL query using these three tables. The schemas are as {all_schemas}. The tables can be understood as {config.db_layout}. Try to answer the following question. The SQL should be returned within ''' SQL query ''' \" \n",
    "}},\n",
    "{{\n",
    "  \"role\": \"user\",\n",
    "  \"content\": {question}\n",
    "}}\n",
    "]\"\"\"\n",
    "\n",
    "# Iterate through each model\n",
    "for model_key, model_value in models.items():\n",
    "    responses = []\n",
    "\n",
    "    for index, row in df.iterrows():\n",
    "        qn = row['Question']\n",
    "        question = json.dumps(qn)\n",
    "\n",
    "        if model_key == \"yi-34b\":  # Yi model\n",
    "            prompt = prompt_for_qwen.format(INSTRUCTION=INSTRUCTION, question=question)\n",
    "        if model_key == \"qwen-14b\":  # Qwen model\n",
    "            prompt = prompt_for_qwen.format(INSTRUCTION=INSTRUCTION, question=question)\n",
    "        elif model_key == \"openhermes2\":  # Hermes model\n",
    "            prompt = prompt_for_hermes.format(INSTRUCTION=INSTRUCTION, question=question)\n",
    "        else:\n",
    "            prompt = f\"{INSTRUCTION}. Please write the appropriate SQL query using these three different dbs, and their schemas are as {all_schemas}. The tables can be understood as {config.db_layout}. Try to answer the following question. Only the SQL should be returned within ''' SQL query '''. {question}\"\n",
    "\n",
    "        start_time = time.time()  # Record the start time\n",
    "\n",
    "        try:\n",
    "            print(prompt)\n",
    "            output = replicate.run(\n",
    "                model_value,\n",
    "                input={\n",
    "                  \"debug\": False,\n",
    "                #   \"top_k\": 50,\n",
    "                  \"top_p\": 0.9,\n",
    "                  \"prompt\": prompt,\n",
    "                  \"temperature\": 0.7,\n",
    "                  \"max_new_tokens\": 500,\n",
    "                  \"min_new_tokens\": -1\n",
    "                }\n",
    "            )\n",
    "            response = \"\"\n",
    "            response_parts = []  # Initialize an empty list to collect string representations\n",
    "            for item in output:\n",
    "                item_str = str(item).strip()  # Convert item to string\n",
    "                response += item_str # if len(item_str) == 1 else f\" {item_str}\"\n",
    "\n",
    "            response = response.strip()\n",
    "            extracted_sql = extract_sql(response) # Get only the SQL query\n",
    "            valid = is_sql_valid(response) # Check if the SQL query is valid\n",
    "            response = {\n",
    "                \"response\": extracted_sql,\n",
    "                \"is_valid\": valid\n",
    "            }\n",
    "\n",
    "        except Exception as e:\n",
    "            response = f\"Error: {e}\"\n",
    "\n",
    "        end_time = time.time()  # Record the end time\n",
    "        latency = end_time - start_time  # Calculate latency\n",
    "\n",
    "        new_row = pd.DataFrame({'Model': [model_key], 'Question': [qn], 'Response': [extracted_sql], 'Latency': [latency]})\n",
    "        results_df = pd.concat([results_df, new_row], ignore_index=True)\n",
    "\n",
    "        if index % 20 == 0:  # Save every 10 questions, adjust as needed\n",
    "            results_df.to_excel(config.llmresults_file_path, index=False, sheet_name='Sheet1')\n",
    "\n",
    "results_df.to_excel(config.llmresults_file_path, index=False, sheet_name='Sheet1')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# GPT-4 writes a SQL query\n",
    "\n",
    "import pandas as pd\n",
    "import json\n",
    "import openai\n",
    "import requests\n",
    "from openai import OpenAI\n",
    "import time\n",
    "from dotenv import load_dotenv\n",
    "load_dotenv()\n",
    "import os\n",
    "from config import config\n",
    "config.set_mode(\"dbs\")\n",
    "\n",
    "with open('config.json', 'r') as config_file:\n",
    "    config = json.load(config_file)\n",
    "\n",
    "folder_path = 'files'\n",
    "if not os.path.exists(folder_path):\n",
    "    os.makedirs(folder_path)\n",
    "\n",
    "INSTRUCTION = config.INSTRUCTION\n",
    "F_NAME = config.F_NAME\n",
    "GPT_MODEL = config.GPT_MODEL\n",
    "INPUT_CSV_PATH = config.questions\n",
    "OUTPUT_CSV_PATH = config.gpt4results_csv_path\n",
    "\n",
    "client = OpenAI()\n",
    "def show_json(obj):\n",
    "    print(json.loads(obj.model_dump_json()))\n",
    "\n",
    "assistant = client.beta.assistants.create(\n",
    "    name=f\"{F_NAME} AI Evaluator via reading DB\",\n",
    "    instructions=INSTRUCTION,\n",
    "    model=GPT_MODEL,\n",
    ")\n",
    "show_json(assistant)\n",
    "\n",
    "# Utility functions\n",
    "def read_csv(file_path):\n",
    "    return pd.read_excel(file_path)\n",
    "\n",
    "def process_data_for_gpt(data):\n",
    "    prompts = []\n",
    "    for _, row in data.iterrows():\n",
    "        question = row['Question']\n",
    "        prompt = f\"Please write the appropriate SQL query using these three table schemas {all_schemas} to answer the following question. The SQL should be returned within ''' SQL query '''.:\\n\\n{question}\"\n",
    "        prompts.append(prompt)\n",
    "    return prompts\n",
    "\n",
    "def submit_message_and_create_run(assistant_id, prompt):\n",
    "    thread = client.beta.threads.create() # If you replace this globally it appends all answers to the one before.\n",
    "    client.beta.threads.messages.create(thread_id=thread.id, role=\"user\", content=prompt)\n",
    "    return client.beta.threads.runs.create(thread_id=thread.id, assistant_id=assistant_id), thread\n",
    "\n",
    "def wait_on_run_and_get_response(run, thread):\n",
    "    while run.status == \"queued\" or run.status == \"in_progress\":\n",
    "        run = client.beta.threads.runs.retrieve(thread_id=thread.id, run_id=run.id)\n",
    "        time.sleep(0.5)\n",
    "    messages = client.beta.threads.messages.list(thread_id=thread.id, order=\"asc\")\n",
    "    return [m.content[0].text.value for m in messages if m.role == 'assistant']\n",
    "\n",
    "def create_output_csv(data, responses, model_name, interim_csv_path):\n",
    "    new_rows = []\n",
    "    for question, response in zip(data['Question'], responses):\n",
    "        new_rows.append({'Model': model_name, 'Question': question, 'Response': response})\n",
    "    new_data = pd.DataFrame(new_rows)\n",
    "    new_data.to_excel(interim_csv_path, index=False)\n",
    "\n",
    "data = read_csv(INPUT_CSV_PATH)\n",
    "prompts = process_data_for_gpt(data)\n",
    "ASSISTANT_ID = assistant.id\n",
    "\n",
    "responses = []\n",
    "latencies = []  # Initialize a list to store latencies\n",
    "\n",
    "for prompt in prompts:\n",
    "    start_time = time.time()  # Capture start time\n",
    "    run, thread = submit_message_and_create_run(ASSISTANT_ID, prompt)\n",
    "    response = wait_on_run_and_get_response(run, thread)\n",
    "    if isinstance(response, list):\n",
    "        response = ' '.join(map(str, response))\n",
    "    response = response.replace(\"\\\\\\\\n\", \"\\\\n\")\n",
    "    response = response.strip()\n",
    "    extracted_sql = extract_sql(response) # Get only the SQL query\n",
    "    valid = is_sql_valid(response) # Check if the SQL query is valid\n",
    "    response = {\n",
    "        \"response\": extracted_sql,\n",
    "        \"is_valid\": valid\n",
    "    }\n",
    "    print(response)\n",
    "    responses.append(response)\n",
    "    end_time = time.time()  # Capture end time\n",
    "    latency = end_time - start_time  # Calculate latency\n",
    "    latencies.append(latency)  # Store latency\n",
    "\n",
    "create_output_csv(data, responses, GPT_MODEL, OUTPUT_CSV_PATH)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
