{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create combined schema of tables\n",
    "\n",
    "import sqlite3\n",
    "import json\n",
    "import config\n",
    "import importlib\n",
    "importlib.reload(config)\n",
    "from config import config, reset_config\n",
    "from sqlalchemy import create_engine, MetaData, Table, Column, String\n",
    "import re\n",
    "import logging\n",
    "\n",
    "logging.basicConfig(level=logging.WARNING)\n",
    "log = logging.getLogger(__name__)\n",
    "\n",
    "def combine_schemas(db_files):\n",
    "    combined_schema = {}\n",
    "\n",
    "    for db_file in db_files:\n",
    "        engine = create_engine('sqlite:///' + db_file)\n",
    "        conn = sqlite3.connect(db_file)\n",
    "        cursor = conn.cursor()\n",
    "\n",
    "        # Get all tables\n",
    "        cursor.execute(\"SELECT name FROM sqlite_master WHERE type='table';\")\n",
    "        tables = cursor.fetchall()\n",
    "\n",
    "        for table in tables:\n",
    "            table_name = table[0]\n",
    "            metadata_obj = MetaData()\n",
    "            \n",
    "            # Get table schema\n",
    "            cursor.execute(f\"PRAGMA table_info({table_name});\")\n",
    "            schema = cursor.fetchall()\n",
    "\n",
    "            # Create a Table object to store schema info\n",
    "            table_obj = Table(table_name, metadata_obj)\n",
    "\n",
    "            for column in schema:\n",
    "                col_name, col_type = column[1], column[2]\n",
    "                # Add column to the table object\n",
    "                table_obj.append_column(Column(col_name, String))\n",
    "\n",
    "            # Serialize table schema\n",
    "            schema_info = [{\"column_name\": col.name, \"data_type\": str(col.type)} for col in table_obj.columns]\n",
    "            combined_schema[f\"{table_name} in {db_file}\"] = schema_info\n",
    "\n",
    "        conn.close()\n",
    "\n",
    "    return combined_schema\n",
    "\n",
    "def save_schema_to_json(combined_schema, filename=config.db_schema):\n",
    "    with open(filename, \"w\") as file:\n",
    "        json.dump(combined_schema, file, indent=4)\n",
    "\n",
    "# SQL check\n",
    "def extract_sql(llm_response: str) -> str:\n",
    "    # If the llm_response contains a markdown code block, with or without the sql tag, extract the sql from it\n",
    "    sql = re.search(r\"```sql\\n(.*)```\", llm_response, re.DOTALL)\n",
    "    if sql:\n",
    "        log.info(f\"Output from LLM: {llm_response} \\nExtracted SQL: {sql.group(1)}\")\n",
    "        return sql.group(1)\n",
    "\n",
    "    sql = re.search(r\"```(.*)```\", llm_response, re.DOTALL)\n",
    "    if sql:\n",
    "        log.info(f\"Output from LLM: {llm_response} \\nExtracted SQL: {sql.group(1)}\")\n",
    "        return sql.group(1)\n",
    "\n",
    "    return llm_response\n",
    "\n",
    "def is_sql_valid(sql: str) -> bool:\n",
    "    # This is a check to see the SQL is valid and should be run\n",
    "    # This simple function just checks if the SQL contains a SELECT statement\n",
    "\n",
    "    if \"SELECT\" in sql.upper():\n",
    "        return True\n",
    "    else:\n",
    "        return False\n",
    "\n",
    "# Paths to your database files\n",
    "db_files = [\"files/db/CCLEGisticCNDB.db\" , 'files/db/CCLEMutDB.db', 'files/db/CCLEVarDB.db']\n",
    "all_schemas = combine_schemas(db_files)\n",
    "save_schema_to_json(all_schemas)\n",
    "# Now the schema is saved in 'combined_schema.json'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No duplicates found.\n",
      "You are an exceptional computational biologist and genomics expert and know everything about drug discovery.. Please write the appropriate SQL query using these three different dbs, and their schemas are as {'ccle_cn_gistic in files/db/CCLEGisticCNDB.db': [{'column_name': 'Hugo_Symbol', 'data_type': 'VARCHAR'}, {'column_name': 'CCLE_Name', 'data_type': 'VARCHAR'}, {'column_name': 'gistic_cn', 'data_type': 'VARCHAR'}, {'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}], 'ccle_mutation in files/db/CCLEMutDB.db': [{'column_name': 'Hugo_Symbol', 'data_type': 'VARCHAR'}, {'column_name': 'Entrez_Gene_Id', 'data_type': 'VARCHAR'}, {'column_name': 'NCBI_Build', 'data_type': 'VARCHAR'}, {'column_name': 'Chromosome', 'data_type': 'VARCHAR'}, {'column_name': 'Start_position', 'data_type': 'VARCHAR'}, {'column_name': 'End_position', 'data_type': 'VARCHAR'}, {'column_name': 'Strand', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_Classification', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_Type', 'data_type': 'VARCHAR'}, {'column_name': 'Reference_Allele', 'data_type': 'VARCHAR'}, {'column_name': 'Alternate_Allele', 'data_type': 'VARCHAR'}, {'column_name': 'dbSNP_RS', 'data_type': 'VARCHAR'}, {'column_name': 'dbSNP_Val_Status', 'data_type': 'VARCHAR'}, {'column_name': 'Genome_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Annotation_Transcript', 'data_type': 'VARCHAR'}, {'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}, {'column_name': 'cDNA_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Codon_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Protein_Change', 'data_type': 'VARCHAR'}, {'column_name': 'isDeleterious', 'data_type': 'VARCHAR'}, {'column_name': 'isTCGAhotspot', 'data_type': 'VARCHAR'}, {'column_name': 'TCGAhsCnt', 'data_type': 'VARCHAR'}, {'column_name': 'isCOSMIChotspot', 'data_type': 'VARCHAR'}, {'column_name': 'COSMIChsCnt', 'data_type': 'VARCHAR'}, {'column_name': 'ExAC_AF', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_annotation', 'data_type': 'VARCHAR'}, {'column_name': 'CGA_WES_AC', 'data_type': 'VARCHAR'}, {'column_name': 'HC_AC', 'data_type': 'VARCHAR'}, {'column_name': 'RD_AC', 'data_type': 'VARCHAR'}, {'column_name': 'RNAseq_AC', 'data_type': 'VARCHAR'}, {'column_name': 'SangerWES_AC', 'data_type': 'VARCHAR'}, {'column_name': 'WGS_AC', 'data_type': 'VARCHAR'}], 'ccle_variants in files/db/CCLEVarDB.db': [{'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}, {'column_name': 'Chrom', 'data_type': 'VARCHAR'}, {'column_name': 'Pos', 'data_type': 'VARCHAR'}, {'column_name': 'Ref', 'data_type': 'VARCHAR'}, {'column_name': 'Alt', 'data_type': 'VARCHAR'}, {'column_name': 'AF', 'data_type': 'VARCHAR'}, {'column_name': 'VariantType', 'data_type': 'VARCHAR'}, {'column_name': 'VariantInfo', 'data_type': 'VARCHAR'}, {'column_name': 'DNAChange', 'data_type': 'VARCHAR'}, {'column_name': 'ProteinChange', 'data_type': 'VARCHAR'}, {'column_name': 'gene_name', 'data_type': 'VARCHAR'}, {'column_name': 'Transcript', 'data_type': 'VARCHAR'}, {'column_name': 'CCLEDeleterious', 'data_type': 'VARCHAR'}, {'column_name': 'CosmicHotspot', 'data_type': 'VARCHAR'}, {'column_name': 'AssociatedWith', 'data_type': 'VARCHAR'}, {'column_name': 'LoF', 'data_type': 'VARCHAR'}, {'column_name': 'Driver', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyDriver', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyGoF', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyLoF', 'data_type': 'VARCHAR'}, {'column_name': 'VariantAnnotation', 'data_type': 'VARCHAR'}, {'column_name': 'copy_number', 'data_type': 'VARCHAR'}]}. Try to answer the following question. Only the SQL should be returned within ''' SQL query '''. \"Which cell lines have high dependency for the target of interest for the gene POLR3E?\"\n",
      "Response is: {'response': \"```sql--AssumingthetargetofinterestisaspecificgenewithHugo_Symbol'POLR3E'--JointhethreetablesonDepMap_IDSELECTccle_gistic.CCLE_NameFROMccle_cn_gisticASccle_gisticJOINccle_mutationASccle_mutationONccle_gistic.DepMap_ID=ccle_mutation.DepMap_IDWHEREccle_mutation.Hugo_Symbol='POLR3E'--FilterbygenesymbolAND(ccle_gistic.gistic_cnISNOTNULL)--OnlyconsiderCNalterationsAND(ccle_gistic.gistic_cnLIKE'%high%')--Filterforhighcopynumberalterations--Alternatively,youcanusetheccle_variantstableifyou'relookingformutationsinsteadofcopynumberalterations--AND(ccle_mutation.Variant_Classification='Missense_Mutation')--Filterformissensemutationsifthat'sthetypeofmutationofinterest--AND(ccle_mutation.isDeleterious='True')--Filterfordeleteriousmutations--AND(ccle_mutation.isTCGAhotspot='True')--Filterforhotspotmutations--OR(ccle_mutation.isCOSMIChotspot='True')--FilterforCOSMIChotspotmutations--Thisqueryassumesthatthere'saconsistentDepMap_IDacrossallthreedatabases--Ifthereisn't,youmayneedtouseajoinkeythat'scommontoallthreedatabases--oruseatemporarytabletomergethedatabeforerunningthisquery'''\", 'is_valid': True}\n",
      "You are an exceptional computational biologist and genomics expert and know everything about drug discovery.. Please write the appropriate SQL query using these three different dbs, and their schemas are as {'ccle_cn_gistic in files/db/CCLEGisticCNDB.db': [{'column_name': 'Hugo_Symbol', 'data_type': 'VARCHAR'}, {'column_name': 'CCLE_Name', 'data_type': 'VARCHAR'}, {'column_name': 'gistic_cn', 'data_type': 'VARCHAR'}, {'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}], 'ccle_mutation in files/db/CCLEMutDB.db': [{'column_name': 'Hugo_Symbol', 'data_type': 'VARCHAR'}, {'column_name': 'Entrez_Gene_Id', 'data_type': 'VARCHAR'}, {'column_name': 'NCBI_Build', 'data_type': 'VARCHAR'}, {'column_name': 'Chromosome', 'data_type': 'VARCHAR'}, {'column_name': 'Start_position', 'data_type': 'VARCHAR'}, {'column_name': 'End_position', 'data_type': 'VARCHAR'}, {'column_name': 'Strand', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_Classification', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_Type', 'data_type': 'VARCHAR'}, {'column_name': 'Reference_Allele', 'data_type': 'VARCHAR'}, {'column_name': 'Alternate_Allele', 'data_type': 'VARCHAR'}, {'column_name': 'dbSNP_RS', 'data_type': 'VARCHAR'}, {'column_name': 'dbSNP_Val_Status', 'data_type': 'VARCHAR'}, {'column_name': 'Genome_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Annotation_Transcript', 'data_type': 'VARCHAR'}, {'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}, {'column_name': 'cDNA_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Codon_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Protein_Change', 'data_type': 'VARCHAR'}, {'column_name': 'isDeleterious', 'data_type': 'VARCHAR'}, {'column_name': 'isTCGAhotspot', 'data_type': 'VARCHAR'}, {'column_name': 'TCGAhsCnt', 'data_type': 'VARCHAR'}, {'column_name': 'isCOSMIChotspot', 'data_type': 'VARCHAR'}, {'column_name': 'COSMIChsCnt', 'data_type': 'VARCHAR'}, {'column_name': 'ExAC_AF', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_annotation', 'data_type': 'VARCHAR'}, {'column_name': 'CGA_WES_AC', 'data_type': 'VARCHAR'}, {'column_name': 'HC_AC', 'data_type': 'VARCHAR'}, {'column_name': 'RD_AC', 'data_type': 'VARCHAR'}, {'column_name': 'RNAseq_AC', 'data_type': 'VARCHAR'}, {'column_name': 'SangerWES_AC', 'data_type': 'VARCHAR'}, {'column_name': 'WGS_AC', 'data_type': 'VARCHAR'}], 'ccle_variants in files/db/CCLEVarDB.db': [{'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}, {'column_name': 'Chrom', 'data_type': 'VARCHAR'}, {'column_name': 'Pos', 'data_type': 'VARCHAR'}, {'column_name': 'Ref', 'data_type': 'VARCHAR'}, {'column_name': 'Alt', 'data_type': 'VARCHAR'}, {'column_name': 'AF', 'data_type': 'VARCHAR'}, {'column_name': 'VariantType', 'data_type': 'VARCHAR'}, {'column_name': 'VariantInfo', 'data_type': 'VARCHAR'}, {'column_name': 'DNAChange', 'data_type': 'VARCHAR'}, {'column_name': 'ProteinChange', 'data_type': 'VARCHAR'}, {'column_name': 'gene_name', 'data_type': 'VARCHAR'}, {'column_name': 'Transcript', 'data_type': 'VARCHAR'}, {'column_name': 'CCLEDeleterious', 'data_type': 'VARCHAR'}, {'column_name': 'CosmicHotspot', 'data_type': 'VARCHAR'}, {'column_name': 'AssociatedWith', 'data_type': 'VARCHAR'}, {'column_name': 'LoF', 'data_type': 'VARCHAR'}, {'column_name': 'Driver', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyDriver', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyGoF', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyLoF', 'data_type': 'VARCHAR'}, {'column_name': 'VariantAnnotation', 'data_type': 'VARCHAR'}, {'column_name': 'copy_number', 'data_type': 'VARCHAR'}]}. Try to answer the following question. Only the SQL should be returned within ''' SQL query '''. \"For the gene POLR3E, identify indications with significant differential expression in cancer vs normal tissue. Is it overexpressed or underexpressed in tumor tissue?\"\n",
      "Response is: Error: name 'log' is not defined\n",
      "You are an exceptional computational biologist and genomics expert and know everything about drug discovery.. Please write the appropriate SQL query using these three different dbs, and their schemas are as {'ccle_cn_gistic in files/db/CCLEGisticCNDB.db': [{'column_name': 'Hugo_Symbol', 'data_type': 'VARCHAR'}, {'column_name': 'CCLE_Name', 'data_type': 'VARCHAR'}, {'column_name': 'gistic_cn', 'data_type': 'VARCHAR'}, {'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}], 'ccle_mutation in files/db/CCLEMutDB.db': [{'column_name': 'Hugo_Symbol', 'data_type': 'VARCHAR'}, {'column_name': 'Entrez_Gene_Id', 'data_type': 'VARCHAR'}, {'column_name': 'NCBI_Build', 'data_type': 'VARCHAR'}, {'column_name': 'Chromosome', 'data_type': 'VARCHAR'}, {'column_name': 'Start_position', 'data_type': 'VARCHAR'}, {'column_name': 'End_position', 'data_type': 'VARCHAR'}, {'column_name': 'Strand', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_Classification', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_Type', 'data_type': 'VARCHAR'}, {'column_name': 'Reference_Allele', 'data_type': 'VARCHAR'}, {'column_name': 'Alternate_Allele', 'data_type': 'VARCHAR'}, {'column_name': 'dbSNP_RS', 'data_type': 'VARCHAR'}, {'column_name': 'dbSNP_Val_Status', 'data_type': 'VARCHAR'}, {'column_name': 'Genome_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Annotation_Transcript', 'data_type': 'VARCHAR'}, {'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}, {'column_name': 'cDNA_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Codon_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Protein_Change', 'data_type': 'VARCHAR'}, {'column_name': 'isDeleterious', 'data_type': 'VARCHAR'}, {'column_name': 'isTCGAhotspot', 'data_type': 'VARCHAR'}, {'column_name': 'TCGAhsCnt', 'data_type': 'VARCHAR'}, {'column_name': 'isCOSMIChotspot', 'data_type': 'VARCHAR'}, {'column_name': 'COSMIChsCnt', 'data_type': 'VARCHAR'}, {'column_name': 'ExAC_AF', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_annotation', 'data_type': 'VARCHAR'}, {'column_name': 'CGA_WES_AC', 'data_type': 'VARCHAR'}, {'column_name': 'HC_AC', 'data_type': 'VARCHAR'}, {'column_name': 'RD_AC', 'data_type': 'VARCHAR'}, {'column_name': 'RNAseq_AC', 'data_type': 'VARCHAR'}, {'column_name': 'SangerWES_AC', 'data_type': 'VARCHAR'}, {'column_name': 'WGS_AC', 'data_type': 'VARCHAR'}], 'ccle_variants in files/db/CCLEVarDB.db': [{'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}, {'column_name': 'Chrom', 'data_type': 'VARCHAR'}, {'column_name': 'Pos', 'data_type': 'VARCHAR'}, {'column_name': 'Ref', 'data_type': 'VARCHAR'}, {'column_name': 'Alt', 'data_type': 'VARCHAR'}, {'column_name': 'AF', 'data_type': 'VARCHAR'}, {'column_name': 'VariantType', 'data_type': 'VARCHAR'}, {'column_name': 'VariantInfo', 'data_type': 'VARCHAR'}, {'column_name': 'DNAChange', 'data_type': 'VARCHAR'}, {'column_name': 'ProteinChange', 'data_type': 'VARCHAR'}, {'column_name': 'gene_name', 'data_type': 'VARCHAR'}, {'column_name': 'Transcript', 'data_type': 'VARCHAR'}, {'column_name': 'CCLEDeleterious', 'data_type': 'VARCHAR'}, {'column_name': 'CosmicHotspot', 'data_type': 'VARCHAR'}, {'column_name': 'AssociatedWith', 'data_type': 'VARCHAR'}, {'column_name': 'LoF', 'data_type': 'VARCHAR'}, {'column_name': 'Driver', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyDriver', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyGoF', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyLoF', 'data_type': 'VARCHAR'}, {'column_name': 'VariantAnnotation', 'data_type': 'VARCHAR'}, {'column_name': 'copy_number', 'data_type': 'VARCHAR'}]}. Try to answer the following question. Only the SQL should be returned within ''' SQL query '''. \"Are there specific regions of the protein of the gene POLR3E where mutations occur at a higher frequency?\"\n",
      "Response is: Error: name 'log' is not defined\n",
      "You are an exceptional computational biologist and genomics expert and know everything about drug discovery.. Please write the appropriate SQL query using these three different dbs, and their schemas are as {'ccle_cn_gistic in files/db/CCLEGisticCNDB.db': [{'column_name': 'Hugo_Symbol', 'data_type': 'VARCHAR'}, {'column_name': 'CCLE_Name', 'data_type': 'VARCHAR'}, {'column_name': 'gistic_cn', 'data_type': 'VARCHAR'}, {'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}], 'ccle_mutation in files/db/CCLEMutDB.db': [{'column_name': 'Hugo_Symbol', 'data_type': 'VARCHAR'}, {'column_name': 'Entrez_Gene_Id', 'data_type': 'VARCHAR'}, {'column_name': 'NCBI_Build', 'data_type': 'VARCHAR'}, {'column_name': 'Chromosome', 'data_type': 'VARCHAR'}, {'column_name': 'Start_position', 'data_type': 'VARCHAR'}, {'column_name': 'End_position', 'data_type': 'VARCHAR'}, {'column_name': 'Strand', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_Classification', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_Type', 'data_type': 'VARCHAR'}, {'column_name': 'Reference_Allele', 'data_type': 'VARCHAR'}, {'column_name': 'Alternate_Allele', 'data_type': 'VARCHAR'}, {'column_name': 'dbSNP_RS', 'data_type': 'VARCHAR'}, {'column_name': 'dbSNP_Val_Status', 'data_type': 'VARCHAR'}, {'column_name': 'Genome_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Annotation_Transcript', 'data_type': 'VARCHAR'}, {'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}, {'column_name': 'cDNA_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Codon_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Protein_Change', 'data_type': 'VARCHAR'}, {'column_name': 'isDeleterious', 'data_type': 'VARCHAR'}, {'column_name': 'isTCGAhotspot', 'data_type': 'VARCHAR'}, {'column_name': 'TCGAhsCnt', 'data_type': 'VARCHAR'}, {'column_name': 'isCOSMIChotspot', 'data_type': 'VARCHAR'}, {'column_name': 'COSMIChsCnt', 'data_type': 'VARCHAR'}, {'column_name': 'ExAC_AF', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_annotation', 'data_type': 'VARCHAR'}, {'column_name': 'CGA_WES_AC', 'data_type': 'VARCHAR'}, {'column_name': 'HC_AC', 'data_type': 'VARCHAR'}, {'column_name': 'RD_AC', 'data_type': 'VARCHAR'}, {'column_name': 'RNAseq_AC', 'data_type': 'VARCHAR'}, {'column_name': 'SangerWES_AC', 'data_type': 'VARCHAR'}, {'column_name': 'WGS_AC', 'data_type': 'VARCHAR'}], 'ccle_variants in files/db/CCLEVarDB.db': [{'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}, {'column_name': 'Chrom', 'data_type': 'VARCHAR'}, {'column_name': 'Pos', 'data_type': 'VARCHAR'}, {'column_name': 'Ref', 'data_type': 'VARCHAR'}, {'column_name': 'Alt', 'data_type': 'VARCHAR'}, {'column_name': 'AF', 'data_type': 'VARCHAR'}, {'column_name': 'VariantType', 'data_type': 'VARCHAR'}, {'column_name': 'VariantInfo', 'data_type': 'VARCHAR'}, {'column_name': 'DNAChange', 'data_type': 'VARCHAR'}, {'column_name': 'ProteinChange', 'data_type': 'VARCHAR'}, {'column_name': 'gene_name', 'data_type': 'VARCHAR'}, {'column_name': 'Transcript', 'data_type': 'VARCHAR'}, {'column_name': 'CCLEDeleterious', 'data_type': 'VARCHAR'}, {'column_name': 'CosmicHotspot', 'data_type': 'VARCHAR'}, {'column_name': 'AssociatedWith', 'data_type': 'VARCHAR'}, {'column_name': 'LoF', 'data_type': 'VARCHAR'}, {'column_name': 'Driver', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyDriver', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyGoF', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyLoF', 'data_type': 'VARCHAR'}, {'column_name': 'VariantAnnotation', 'data_type': 'VARCHAR'}, {'column_name': 'copy_number', 'data_type': 'VARCHAR'}]}. Try to answer the following question. Only the SQL should be returned within ''' SQL query '''. \"For the gene POLR3E, determine how many of the mutations are pathogenic.\"\n",
      "Response is: Error: name 'log' is not defined\n",
      "You are an exceptional computational biologist and genomics expert and know everything about drug discovery.. Please write the appropriate SQL query using these three different dbs, and their schemas are as {'ccle_cn_gistic in files/db/CCLEGisticCNDB.db': [{'column_name': 'Hugo_Symbol', 'data_type': 'VARCHAR'}, {'column_name': 'CCLE_Name', 'data_type': 'VARCHAR'}, {'column_name': 'gistic_cn', 'data_type': 'VARCHAR'}, {'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}], 'ccle_mutation in files/db/CCLEMutDB.db': [{'column_name': 'Hugo_Symbol', 'data_type': 'VARCHAR'}, {'column_name': 'Entrez_Gene_Id', 'data_type': 'VARCHAR'}, {'column_name': 'NCBI_Build', 'data_type': 'VARCHAR'}, {'column_name': 'Chromosome', 'data_type': 'VARCHAR'}, {'column_name': 'Start_position', 'data_type': 'VARCHAR'}, {'column_name': 'End_position', 'data_type': 'VARCHAR'}, {'column_name': 'Strand', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_Classification', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_Type', 'data_type': 'VARCHAR'}, {'column_name': 'Reference_Allele', 'data_type': 'VARCHAR'}, {'column_name': 'Alternate_Allele', 'data_type': 'VARCHAR'}, {'column_name': 'dbSNP_RS', 'data_type': 'VARCHAR'}, {'column_name': 'dbSNP_Val_Status', 'data_type': 'VARCHAR'}, {'column_name': 'Genome_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Annotation_Transcript', 'data_type': 'VARCHAR'}, {'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}, {'column_name': 'cDNA_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Codon_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Protein_Change', 'data_type': 'VARCHAR'}, {'column_name': 'isDeleterious', 'data_type': 'VARCHAR'}, {'column_name': 'isTCGAhotspot', 'data_type': 'VARCHAR'}, {'column_name': 'TCGAhsCnt', 'data_type': 'VARCHAR'}, {'column_name': 'isCOSMIChotspot', 'data_type': 'VARCHAR'}, {'column_name': 'COSMIChsCnt', 'data_type': 'VARCHAR'}, {'column_name': 'ExAC_AF', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_annotation', 'data_type': 'VARCHAR'}, {'column_name': 'CGA_WES_AC', 'data_type': 'VARCHAR'}, {'column_name': 'HC_AC', 'data_type': 'VARCHAR'}, {'column_name': 'RD_AC', 'data_type': 'VARCHAR'}, {'column_name': 'RNAseq_AC', 'data_type': 'VARCHAR'}, {'column_name': 'SangerWES_AC', 'data_type': 'VARCHAR'}, {'column_name': 'WGS_AC', 'data_type': 'VARCHAR'}], 'ccle_variants in files/db/CCLEVarDB.db': [{'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}, {'column_name': 'Chrom', 'data_type': 'VARCHAR'}, {'column_name': 'Pos', 'data_type': 'VARCHAR'}, {'column_name': 'Ref', 'data_type': 'VARCHAR'}, {'column_name': 'Alt', 'data_type': 'VARCHAR'}, {'column_name': 'AF', 'data_type': 'VARCHAR'}, {'column_name': 'VariantType', 'data_type': 'VARCHAR'}, {'column_name': 'VariantInfo', 'data_type': 'VARCHAR'}, {'column_name': 'DNAChange', 'data_type': 'VARCHAR'}, {'column_name': 'ProteinChange', 'data_type': 'VARCHAR'}, {'column_name': 'gene_name', 'data_type': 'VARCHAR'}, {'column_name': 'Transcript', 'data_type': 'VARCHAR'}, {'column_name': 'CCLEDeleterious', 'data_type': 'VARCHAR'}, {'column_name': 'CosmicHotspot', 'data_type': 'VARCHAR'}, {'column_name': 'AssociatedWith', 'data_type': 'VARCHAR'}, {'column_name': 'LoF', 'data_type': 'VARCHAR'}, {'column_name': 'Driver', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyDriver', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyGoF', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyLoF', 'data_type': 'VARCHAR'}, {'column_name': 'VariantAnnotation', 'data_type': 'VARCHAR'}, {'column_name': 'copy_number', 'data_type': 'VARCHAR'}]}. Try to answer the following question. Only the SQL should be returned within ''' SQL query '''. \"Which mutations in PAS domain for the target protein exhibit high selectivity for the prostate cancer cell lines?\"\n",
      "Response is: Error: name 'log' is not defined\n",
      "You are an exceptional computational biologist and genomics expert and know everything about drug discovery.. Please write the appropriate SQL query using these three different dbs, and their schemas are as {'ccle_cn_gistic in files/db/CCLEGisticCNDB.db': [{'column_name': 'Hugo_Symbol', 'data_type': 'VARCHAR'}, {'column_name': 'CCLE_Name', 'data_type': 'VARCHAR'}, {'column_name': 'gistic_cn', 'data_type': 'VARCHAR'}, {'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}], 'ccle_mutation in files/db/CCLEMutDB.db': [{'column_name': 'Hugo_Symbol', 'data_type': 'VARCHAR'}, {'column_name': 'Entrez_Gene_Id', 'data_type': 'VARCHAR'}, {'column_name': 'NCBI_Build', 'data_type': 'VARCHAR'}, {'column_name': 'Chromosome', 'data_type': 'VARCHAR'}, {'column_name': 'Start_position', 'data_type': 'VARCHAR'}, {'column_name': 'End_position', 'data_type': 'VARCHAR'}, {'column_name': 'Strand', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_Classification', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_Type', 'data_type': 'VARCHAR'}, {'column_name': 'Reference_Allele', 'data_type': 'VARCHAR'}, {'column_name': 'Alternate_Allele', 'data_type': 'VARCHAR'}, {'column_name': 'dbSNP_RS', 'data_type': 'VARCHAR'}, {'column_name': 'dbSNP_Val_Status', 'data_type': 'VARCHAR'}, {'column_name': 'Genome_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Annotation_Transcript', 'data_type': 'VARCHAR'}, {'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}, {'column_name': 'cDNA_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Codon_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Protein_Change', 'data_type': 'VARCHAR'}, {'column_name': 'isDeleterious', 'data_type': 'VARCHAR'}, {'column_name': 'isTCGAhotspot', 'data_type': 'VARCHAR'}, {'column_name': 'TCGAhsCnt', 'data_type': 'VARCHAR'}, {'column_name': 'isCOSMIChotspot', 'data_type': 'VARCHAR'}, {'column_name': 'COSMIChsCnt', 'data_type': 'VARCHAR'}, {'column_name': 'ExAC_AF', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_annotation', 'data_type': 'VARCHAR'}, {'column_name': 'CGA_WES_AC', 'data_type': 'VARCHAR'}, {'column_name': 'HC_AC', 'data_type': 'VARCHAR'}, {'column_name': 'RD_AC', 'data_type': 'VARCHAR'}, {'column_name': 'RNAseq_AC', 'data_type': 'VARCHAR'}, {'column_name': 'SangerWES_AC', 'data_type': 'VARCHAR'}, {'column_name': 'WGS_AC', 'data_type': 'VARCHAR'}], 'ccle_variants in files/db/CCLEVarDB.db': [{'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}, {'column_name': 'Chrom', 'data_type': 'VARCHAR'}, {'column_name': 'Pos', 'data_type': 'VARCHAR'}, {'column_name': 'Ref', 'data_type': 'VARCHAR'}, {'column_name': 'Alt', 'data_type': 'VARCHAR'}, {'column_name': 'AF', 'data_type': 'VARCHAR'}, {'column_name': 'VariantType', 'data_type': 'VARCHAR'}, {'column_name': 'VariantInfo', 'data_type': 'VARCHAR'}, {'column_name': 'DNAChange', 'data_type': 'VARCHAR'}, {'column_name': 'ProteinChange', 'data_type': 'VARCHAR'}, {'column_name': 'gene_name', 'data_type': 'VARCHAR'}, {'column_name': 'Transcript', 'data_type': 'VARCHAR'}, {'column_name': 'CCLEDeleterious', 'data_type': 'VARCHAR'}, {'column_name': 'CosmicHotspot', 'data_type': 'VARCHAR'}, {'column_name': 'AssociatedWith', 'data_type': 'VARCHAR'}, {'column_name': 'LoF', 'data_type': 'VARCHAR'}, {'column_name': 'Driver', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyDriver', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyGoF', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyLoF', 'data_type': 'VARCHAR'}, {'column_name': 'VariantAnnotation', 'data_type': 'VARCHAR'}, {'column_name': 'copy_number', 'data_type': 'VARCHAR'}]}. Try to answer the following question. Only the SQL should be returned within ''' SQL query '''. \"Relationship between dependency scores of JAK1 in DepMapDB.\"\n",
      "Response is: Error: name 'log' is not defined\n",
      "You are an exceptional computational biologist and genomics expert and know everything about drug discovery.. Please write the appropriate SQL query using these three different dbs, and their schemas are as {'ccle_cn_gistic in files/db/CCLEGisticCNDB.db': [{'column_name': 'Hugo_Symbol', 'data_type': 'VARCHAR'}, {'column_name': 'CCLE_Name', 'data_type': 'VARCHAR'}, {'column_name': 'gistic_cn', 'data_type': 'VARCHAR'}, {'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}], 'ccle_mutation in files/db/CCLEMutDB.db': [{'column_name': 'Hugo_Symbol', 'data_type': 'VARCHAR'}, {'column_name': 'Entrez_Gene_Id', 'data_type': 'VARCHAR'}, {'column_name': 'NCBI_Build', 'data_type': 'VARCHAR'}, {'column_name': 'Chromosome', 'data_type': 'VARCHAR'}, {'column_name': 'Start_position', 'data_type': 'VARCHAR'}, {'column_name': 'End_position', 'data_type': 'VARCHAR'}, {'column_name': 'Strand', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_Classification', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_Type', 'data_type': 'VARCHAR'}, {'column_name': 'Reference_Allele', 'data_type': 'VARCHAR'}, {'column_name': 'Alternate_Allele', 'data_type': 'VARCHAR'}, {'column_name': 'dbSNP_RS', 'data_type': 'VARCHAR'}, {'column_name': 'dbSNP_Val_Status', 'data_type': 'VARCHAR'}, {'column_name': 'Genome_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Annotation_Transcript', 'data_type': 'VARCHAR'}, {'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}, {'column_name': 'cDNA_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Codon_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Protein_Change', 'data_type': 'VARCHAR'}, {'column_name': 'isDeleterious', 'data_type': 'VARCHAR'}, {'column_name': 'isTCGAhotspot', 'data_type': 'VARCHAR'}, {'column_name': 'TCGAhsCnt', 'data_type': 'VARCHAR'}, {'column_name': 'isCOSMIChotspot', 'data_type': 'VARCHAR'}, {'column_name': 'COSMIChsCnt', 'data_type': 'VARCHAR'}, {'column_name': 'ExAC_AF', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_annotation', 'data_type': 'VARCHAR'}, {'column_name': 'CGA_WES_AC', 'data_type': 'VARCHAR'}, {'column_name': 'HC_AC', 'data_type': 'VARCHAR'}, {'column_name': 'RD_AC', 'data_type': 'VARCHAR'}, {'column_name': 'RNAseq_AC', 'data_type': 'VARCHAR'}, {'column_name': 'SangerWES_AC', 'data_type': 'VARCHAR'}, {'column_name': 'WGS_AC', 'data_type': 'VARCHAR'}], 'ccle_variants in files/db/CCLEVarDB.db': [{'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}, {'column_name': 'Chrom', 'data_type': 'VARCHAR'}, {'column_name': 'Pos', 'data_type': 'VARCHAR'}, {'column_name': 'Ref', 'data_type': 'VARCHAR'}, {'column_name': 'Alt', 'data_type': 'VARCHAR'}, {'column_name': 'AF', 'data_type': 'VARCHAR'}, {'column_name': 'VariantType', 'data_type': 'VARCHAR'}, {'column_name': 'VariantInfo', 'data_type': 'VARCHAR'}, {'column_name': 'DNAChange', 'data_type': 'VARCHAR'}, {'column_name': 'ProteinChange', 'data_type': 'VARCHAR'}, {'column_name': 'gene_name', 'data_type': 'VARCHAR'}, {'column_name': 'Transcript', 'data_type': 'VARCHAR'}, {'column_name': 'CCLEDeleterious', 'data_type': 'VARCHAR'}, {'column_name': 'CosmicHotspot', 'data_type': 'VARCHAR'}, {'column_name': 'AssociatedWith', 'data_type': 'VARCHAR'}, {'column_name': 'LoF', 'data_type': 'VARCHAR'}, {'column_name': 'Driver', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyDriver', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyGoF', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyLoF', 'data_type': 'VARCHAR'}, {'column_name': 'VariantAnnotation', 'data_type': 'VARCHAR'}, {'column_name': 'copy_number', 'data_type': 'VARCHAR'}]}. Try to answer the following question. Only the SQL should be returned within ''' SQL query '''. \"Which cell lines have high dependency for the target of interest for the gene POLR3E?\"\n",
      "Response is: Error: name 'log' is not defined\n",
      "You are an exceptional computational biologist and genomics expert and know everything about drug discovery.. Please write the appropriate SQL query using these three different dbs, and their schemas are as {'ccle_cn_gistic in files/db/CCLEGisticCNDB.db': [{'column_name': 'Hugo_Symbol', 'data_type': 'VARCHAR'}, {'column_name': 'CCLE_Name', 'data_type': 'VARCHAR'}, {'column_name': 'gistic_cn', 'data_type': 'VARCHAR'}, {'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}], 'ccle_mutation in files/db/CCLEMutDB.db': [{'column_name': 'Hugo_Symbol', 'data_type': 'VARCHAR'}, {'column_name': 'Entrez_Gene_Id', 'data_type': 'VARCHAR'}, {'column_name': 'NCBI_Build', 'data_type': 'VARCHAR'}, {'column_name': 'Chromosome', 'data_type': 'VARCHAR'}, {'column_name': 'Start_position', 'data_type': 'VARCHAR'}, {'column_name': 'End_position', 'data_type': 'VARCHAR'}, {'column_name': 'Strand', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_Classification', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_Type', 'data_type': 'VARCHAR'}, {'column_name': 'Reference_Allele', 'data_type': 'VARCHAR'}, {'column_name': 'Alternate_Allele', 'data_type': 'VARCHAR'}, {'column_name': 'dbSNP_RS', 'data_type': 'VARCHAR'}, {'column_name': 'dbSNP_Val_Status', 'data_type': 'VARCHAR'}, {'column_name': 'Genome_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Annotation_Transcript', 'data_type': 'VARCHAR'}, {'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}, {'column_name': 'cDNA_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Codon_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Protein_Change', 'data_type': 'VARCHAR'}, {'column_name': 'isDeleterious', 'data_type': 'VARCHAR'}, {'column_name': 'isTCGAhotspot', 'data_type': 'VARCHAR'}, {'column_name': 'TCGAhsCnt', 'data_type': 'VARCHAR'}, {'column_name': 'isCOSMIChotspot', 'data_type': 'VARCHAR'}, {'column_name': 'COSMIChsCnt', 'data_type': 'VARCHAR'}, {'column_name': 'ExAC_AF', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_annotation', 'data_type': 'VARCHAR'}, {'column_name': 'CGA_WES_AC', 'data_type': 'VARCHAR'}, {'column_name': 'HC_AC', 'data_type': 'VARCHAR'}, {'column_name': 'RD_AC', 'data_type': 'VARCHAR'}, {'column_name': 'RNAseq_AC', 'data_type': 'VARCHAR'}, {'column_name': 'SangerWES_AC', 'data_type': 'VARCHAR'}, {'column_name': 'WGS_AC', 'data_type': 'VARCHAR'}], 'ccle_variants in files/db/CCLEVarDB.db': [{'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}, {'column_name': 'Chrom', 'data_type': 'VARCHAR'}, {'column_name': 'Pos', 'data_type': 'VARCHAR'}, {'column_name': 'Ref', 'data_type': 'VARCHAR'}, {'column_name': 'Alt', 'data_type': 'VARCHAR'}, {'column_name': 'AF', 'data_type': 'VARCHAR'}, {'column_name': 'VariantType', 'data_type': 'VARCHAR'}, {'column_name': 'VariantInfo', 'data_type': 'VARCHAR'}, {'column_name': 'DNAChange', 'data_type': 'VARCHAR'}, {'column_name': 'ProteinChange', 'data_type': 'VARCHAR'}, {'column_name': 'gene_name', 'data_type': 'VARCHAR'}, {'column_name': 'Transcript', 'data_type': 'VARCHAR'}, {'column_name': 'CCLEDeleterious', 'data_type': 'VARCHAR'}, {'column_name': 'CosmicHotspot', 'data_type': 'VARCHAR'}, {'column_name': 'AssociatedWith', 'data_type': 'VARCHAR'}, {'column_name': 'LoF', 'data_type': 'VARCHAR'}, {'column_name': 'Driver', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyDriver', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyGoF', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyLoF', 'data_type': 'VARCHAR'}, {'column_name': 'VariantAnnotation', 'data_type': 'VARCHAR'}, {'column_name': 'copy_number', 'data_type': 'VARCHAR'}]}. Try to answer the following question. Only the SQL should be returned within ''' SQL query '''. \"For the gene POLR3E, identify indications with significant differential expression in cancer vs normal tissue. Is it overexpressed or underexpressed in tumor tissue?\"\n",
      "Response is: Error: name 'log' is not defined\n",
      "You are an exceptional computational biologist and genomics expert and know everything about drug discovery.. Please write the appropriate SQL query using these three different dbs, and their schemas are as {'ccle_cn_gistic in files/db/CCLEGisticCNDB.db': [{'column_name': 'Hugo_Symbol', 'data_type': 'VARCHAR'}, {'column_name': 'CCLE_Name', 'data_type': 'VARCHAR'}, {'column_name': 'gistic_cn', 'data_type': 'VARCHAR'}, {'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}], 'ccle_mutation in files/db/CCLEMutDB.db': [{'column_name': 'Hugo_Symbol', 'data_type': 'VARCHAR'}, {'column_name': 'Entrez_Gene_Id', 'data_type': 'VARCHAR'}, {'column_name': 'NCBI_Build', 'data_type': 'VARCHAR'}, {'column_name': 'Chromosome', 'data_type': 'VARCHAR'}, {'column_name': 'Start_position', 'data_type': 'VARCHAR'}, {'column_name': 'End_position', 'data_type': 'VARCHAR'}, {'column_name': 'Strand', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_Classification', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_Type', 'data_type': 'VARCHAR'}, {'column_name': 'Reference_Allele', 'data_type': 'VARCHAR'}, {'column_name': 'Alternate_Allele', 'data_type': 'VARCHAR'}, {'column_name': 'dbSNP_RS', 'data_type': 'VARCHAR'}, {'column_name': 'dbSNP_Val_Status', 'data_type': 'VARCHAR'}, {'column_name': 'Genome_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Annotation_Transcript', 'data_type': 'VARCHAR'}, {'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}, {'column_name': 'cDNA_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Codon_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Protein_Change', 'data_type': 'VARCHAR'}, {'column_name': 'isDeleterious', 'data_type': 'VARCHAR'}, {'column_name': 'isTCGAhotspot', 'data_type': 'VARCHAR'}, {'column_name': 'TCGAhsCnt', 'data_type': 'VARCHAR'}, {'column_name': 'isCOSMIChotspot', 'data_type': 'VARCHAR'}, {'column_name': 'COSMIChsCnt', 'data_type': 'VARCHAR'}, {'column_name': 'ExAC_AF', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_annotation', 'data_type': 'VARCHAR'}, {'column_name': 'CGA_WES_AC', 'data_type': 'VARCHAR'}, {'column_name': 'HC_AC', 'data_type': 'VARCHAR'}, {'column_name': 'RD_AC', 'data_type': 'VARCHAR'}, {'column_name': 'RNAseq_AC', 'data_type': 'VARCHAR'}, {'column_name': 'SangerWES_AC', 'data_type': 'VARCHAR'}, {'column_name': 'WGS_AC', 'data_type': 'VARCHAR'}], 'ccle_variants in files/db/CCLEVarDB.db': [{'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}, {'column_name': 'Chrom', 'data_type': 'VARCHAR'}, {'column_name': 'Pos', 'data_type': 'VARCHAR'}, {'column_name': 'Ref', 'data_type': 'VARCHAR'}, {'column_name': 'Alt', 'data_type': 'VARCHAR'}, {'column_name': 'AF', 'data_type': 'VARCHAR'}, {'column_name': 'VariantType', 'data_type': 'VARCHAR'}, {'column_name': 'VariantInfo', 'data_type': 'VARCHAR'}, {'column_name': 'DNAChange', 'data_type': 'VARCHAR'}, {'column_name': 'ProteinChange', 'data_type': 'VARCHAR'}, {'column_name': 'gene_name', 'data_type': 'VARCHAR'}, {'column_name': 'Transcript', 'data_type': 'VARCHAR'}, {'column_name': 'CCLEDeleterious', 'data_type': 'VARCHAR'}, {'column_name': 'CosmicHotspot', 'data_type': 'VARCHAR'}, {'column_name': 'AssociatedWith', 'data_type': 'VARCHAR'}, {'column_name': 'LoF', 'data_type': 'VARCHAR'}, {'column_name': 'Driver', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyDriver', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyGoF', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyLoF', 'data_type': 'VARCHAR'}, {'column_name': 'VariantAnnotation', 'data_type': 'VARCHAR'}, {'column_name': 'copy_number', 'data_type': 'VARCHAR'}]}. Try to answer the following question. Only the SQL should be returned within ''' SQL query '''. \"Are there specific regions of the protein of the gene POLR3E where mutations occur at a higher frequency?\"\n",
      "Response is: Error: name 'log' is not defined\n",
      "You are an exceptional computational biologist and genomics expert and know everything about drug discovery.. Please write the appropriate SQL query using these three different dbs, and their schemas are as {'ccle_cn_gistic in files/db/CCLEGisticCNDB.db': [{'column_name': 'Hugo_Symbol', 'data_type': 'VARCHAR'}, {'column_name': 'CCLE_Name', 'data_type': 'VARCHAR'}, {'column_name': 'gistic_cn', 'data_type': 'VARCHAR'}, {'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}], 'ccle_mutation in files/db/CCLEMutDB.db': [{'column_name': 'Hugo_Symbol', 'data_type': 'VARCHAR'}, {'column_name': 'Entrez_Gene_Id', 'data_type': 'VARCHAR'}, {'column_name': 'NCBI_Build', 'data_type': 'VARCHAR'}, {'column_name': 'Chromosome', 'data_type': 'VARCHAR'}, {'column_name': 'Start_position', 'data_type': 'VARCHAR'}, {'column_name': 'End_position', 'data_type': 'VARCHAR'}, {'column_name': 'Strand', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_Classification', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_Type', 'data_type': 'VARCHAR'}, {'column_name': 'Reference_Allele', 'data_type': 'VARCHAR'}, {'column_name': 'Alternate_Allele', 'data_type': 'VARCHAR'}, {'column_name': 'dbSNP_RS', 'data_type': 'VARCHAR'}, {'column_name': 'dbSNP_Val_Status', 'data_type': 'VARCHAR'}, {'column_name': 'Genome_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Annotation_Transcript', 'data_type': 'VARCHAR'}, {'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}, {'column_name': 'cDNA_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Codon_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Protein_Change', 'data_type': 'VARCHAR'}, {'column_name': 'isDeleterious', 'data_type': 'VARCHAR'}, {'column_name': 'isTCGAhotspot', 'data_type': 'VARCHAR'}, {'column_name': 'TCGAhsCnt', 'data_type': 'VARCHAR'}, {'column_name': 'isCOSMIChotspot', 'data_type': 'VARCHAR'}, {'column_name': 'COSMIChsCnt', 'data_type': 'VARCHAR'}, {'column_name': 'ExAC_AF', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_annotation', 'data_type': 'VARCHAR'}, {'column_name': 'CGA_WES_AC', 'data_type': 'VARCHAR'}, {'column_name': 'HC_AC', 'data_type': 'VARCHAR'}, {'column_name': 'RD_AC', 'data_type': 'VARCHAR'}, {'column_name': 'RNAseq_AC', 'data_type': 'VARCHAR'}, {'column_name': 'SangerWES_AC', 'data_type': 'VARCHAR'}, {'column_name': 'WGS_AC', 'data_type': 'VARCHAR'}], 'ccle_variants in files/db/CCLEVarDB.db': [{'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}, {'column_name': 'Chrom', 'data_type': 'VARCHAR'}, {'column_name': 'Pos', 'data_type': 'VARCHAR'}, {'column_name': 'Ref', 'data_type': 'VARCHAR'}, {'column_name': 'Alt', 'data_type': 'VARCHAR'}, {'column_name': 'AF', 'data_type': 'VARCHAR'}, {'column_name': 'VariantType', 'data_type': 'VARCHAR'}, {'column_name': 'VariantInfo', 'data_type': 'VARCHAR'}, {'column_name': 'DNAChange', 'data_type': 'VARCHAR'}, {'column_name': 'ProteinChange', 'data_type': 'VARCHAR'}, {'column_name': 'gene_name', 'data_type': 'VARCHAR'}, {'column_name': 'Transcript', 'data_type': 'VARCHAR'}, {'column_name': 'CCLEDeleterious', 'data_type': 'VARCHAR'}, {'column_name': 'CosmicHotspot', 'data_type': 'VARCHAR'}, {'column_name': 'AssociatedWith', 'data_type': 'VARCHAR'}, {'column_name': 'LoF', 'data_type': 'VARCHAR'}, {'column_name': 'Driver', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyDriver', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyGoF', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyLoF', 'data_type': 'VARCHAR'}, {'column_name': 'VariantAnnotation', 'data_type': 'VARCHAR'}, {'column_name': 'copy_number', 'data_type': 'VARCHAR'}]}. Try to answer the following question. Only the SQL should be returned within ''' SQL query '''. \"For the gene POLR3E, determine how many of the mutations are pathogenic.\"\n",
      "Response is: Error: name 'log' is not defined\n",
      "You are an exceptional computational biologist and genomics expert and know everything about drug discovery.. Please write the appropriate SQL query using these three different dbs, and their schemas are as {'ccle_cn_gistic in files/db/CCLEGisticCNDB.db': [{'column_name': 'Hugo_Symbol', 'data_type': 'VARCHAR'}, {'column_name': 'CCLE_Name', 'data_type': 'VARCHAR'}, {'column_name': 'gistic_cn', 'data_type': 'VARCHAR'}, {'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}], 'ccle_mutation in files/db/CCLEMutDB.db': [{'column_name': 'Hugo_Symbol', 'data_type': 'VARCHAR'}, {'column_name': 'Entrez_Gene_Id', 'data_type': 'VARCHAR'}, {'column_name': 'NCBI_Build', 'data_type': 'VARCHAR'}, {'column_name': 'Chromosome', 'data_type': 'VARCHAR'}, {'column_name': 'Start_position', 'data_type': 'VARCHAR'}, {'column_name': 'End_position', 'data_type': 'VARCHAR'}, {'column_name': 'Strand', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_Classification', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_Type', 'data_type': 'VARCHAR'}, {'column_name': 'Reference_Allele', 'data_type': 'VARCHAR'}, {'column_name': 'Alternate_Allele', 'data_type': 'VARCHAR'}, {'column_name': 'dbSNP_RS', 'data_type': 'VARCHAR'}, {'column_name': 'dbSNP_Val_Status', 'data_type': 'VARCHAR'}, {'column_name': 'Genome_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Annotation_Transcript', 'data_type': 'VARCHAR'}, {'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}, {'column_name': 'cDNA_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Codon_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Protein_Change', 'data_type': 'VARCHAR'}, {'column_name': 'isDeleterious', 'data_type': 'VARCHAR'}, {'column_name': 'isTCGAhotspot', 'data_type': 'VARCHAR'}, {'column_name': 'TCGAhsCnt', 'data_type': 'VARCHAR'}, {'column_name': 'isCOSMIChotspot', 'data_type': 'VARCHAR'}, {'column_name': 'COSMIChsCnt', 'data_type': 'VARCHAR'}, {'column_name': 'ExAC_AF', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_annotation', 'data_type': 'VARCHAR'}, {'column_name': 'CGA_WES_AC', 'data_type': 'VARCHAR'}, {'column_name': 'HC_AC', 'data_type': 'VARCHAR'}, {'column_name': 'RD_AC', 'data_type': 'VARCHAR'}, {'column_name': 'RNAseq_AC', 'data_type': 'VARCHAR'}, {'column_name': 'SangerWES_AC', 'data_type': 'VARCHAR'}, {'column_name': 'WGS_AC', 'data_type': 'VARCHAR'}], 'ccle_variants in files/db/CCLEVarDB.db': [{'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}, {'column_name': 'Chrom', 'data_type': 'VARCHAR'}, {'column_name': 'Pos', 'data_type': 'VARCHAR'}, {'column_name': 'Ref', 'data_type': 'VARCHAR'}, {'column_name': 'Alt', 'data_type': 'VARCHAR'}, {'column_name': 'AF', 'data_type': 'VARCHAR'}, {'column_name': 'VariantType', 'data_type': 'VARCHAR'}, {'column_name': 'VariantInfo', 'data_type': 'VARCHAR'}, {'column_name': 'DNAChange', 'data_type': 'VARCHAR'}, {'column_name': 'ProteinChange', 'data_type': 'VARCHAR'}, {'column_name': 'gene_name', 'data_type': 'VARCHAR'}, {'column_name': 'Transcript', 'data_type': 'VARCHAR'}, {'column_name': 'CCLEDeleterious', 'data_type': 'VARCHAR'}, {'column_name': 'CosmicHotspot', 'data_type': 'VARCHAR'}, {'column_name': 'AssociatedWith', 'data_type': 'VARCHAR'}, {'column_name': 'LoF', 'data_type': 'VARCHAR'}, {'column_name': 'Driver', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyDriver', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyGoF', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyLoF', 'data_type': 'VARCHAR'}, {'column_name': 'VariantAnnotation', 'data_type': 'VARCHAR'}, {'column_name': 'copy_number', 'data_type': 'VARCHAR'}]}. Try to answer the following question. Only the SQL should be returned within ''' SQL query '''. \"Which mutations in PAS domain for the target protein exhibit high selectivity for the prostate cancer cell lines?\"\n",
      "Response is: Error: name 'log' is not defined\n",
      "You are an exceptional computational biologist and genomics expert and know everything about drug discovery.. Please write the appropriate SQL query using these three different dbs, and their schemas are as {'ccle_cn_gistic in files/db/CCLEGisticCNDB.db': [{'column_name': 'Hugo_Symbol', 'data_type': 'VARCHAR'}, {'column_name': 'CCLE_Name', 'data_type': 'VARCHAR'}, {'column_name': 'gistic_cn', 'data_type': 'VARCHAR'}, {'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}], 'ccle_mutation in files/db/CCLEMutDB.db': [{'column_name': 'Hugo_Symbol', 'data_type': 'VARCHAR'}, {'column_name': 'Entrez_Gene_Id', 'data_type': 'VARCHAR'}, {'column_name': 'NCBI_Build', 'data_type': 'VARCHAR'}, {'column_name': 'Chromosome', 'data_type': 'VARCHAR'}, {'column_name': 'Start_position', 'data_type': 'VARCHAR'}, {'column_name': 'End_position', 'data_type': 'VARCHAR'}, {'column_name': 'Strand', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_Classification', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_Type', 'data_type': 'VARCHAR'}, {'column_name': 'Reference_Allele', 'data_type': 'VARCHAR'}, {'column_name': 'Alternate_Allele', 'data_type': 'VARCHAR'}, {'column_name': 'dbSNP_RS', 'data_type': 'VARCHAR'}, {'column_name': 'dbSNP_Val_Status', 'data_type': 'VARCHAR'}, {'column_name': 'Genome_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Annotation_Transcript', 'data_type': 'VARCHAR'}, {'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}, {'column_name': 'cDNA_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Codon_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Protein_Change', 'data_type': 'VARCHAR'}, {'column_name': 'isDeleterious', 'data_type': 'VARCHAR'}, {'column_name': 'isTCGAhotspot', 'data_type': 'VARCHAR'}, {'column_name': 'TCGAhsCnt', 'data_type': 'VARCHAR'}, {'column_name': 'isCOSMIChotspot', 'data_type': 'VARCHAR'}, {'column_name': 'COSMIChsCnt', 'data_type': 'VARCHAR'}, {'column_name': 'ExAC_AF', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_annotation', 'data_type': 'VARCHAR'}, {'column_name': 'CGA_WES_AC', 'data_type': 'VARCHAR'}, {'column_name': 'HC_AC', 'data_type': 'VARCHAR'}, {'column_name': 'RD_AC', 'data_type': 'VARCHAR'}, {'column_name': 'RNAseq_AC', 'data_type': 'VARCHAR'}, {'column_name': 'SangerWES_AC', 'data_type': 'VARCHAR'}, {'column_name': 'WGS_AC', 'data_type': 'VARCHAR'}], 'ccle_variants in files/db/CCLEVarDB.db': [{'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}, {'column_name': 'Chrom', 'data_type': 'VARCHAR'}, {'column_name': 'Pos', 'data_type': 'VARCHAR'}, {'column_name': 'Ref', 'data_type': 'VARCHAR'}, {'column_name': 'Alt', 'data_type': 'VARCHAR'}, {'column_name': 'AF', 'data_type': 'VARCHAR'}, {'column_name': 'VariantType', 'data_type': 'VARCHAR'}, {'column_name': 'VariantInfo', 'data_type': 'VARCHAR'}, {'column_name': 'DNAChange', 'data_type': 'VARCHAR'}, {'column_name': 'ProteinChange', 'data_type': 'VARCHAR'}, {'column_name': 'gene_name', 'data_type': 'VARCHAR'}, {'column_name': 'Transcript', 'data_type': 'VARCHAR'}, {'column_name': 'CCLEDeleterious', 'data_type': 'VARCHAR'}, {'column_name': 'CosmicHotspot', 'data_type': 'VARCHAR'}, {'column_name': 'AssociatedWith', 'data_type': 'VARCHAR'}, {'column_name': 'LoF', 'data_type': 'VARCHAR'}, {'column_name': 'Driver', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyDriver', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyGoF', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyLoF', 'data_type': 'VARCHAR'}, {'column_name': 'VariantAnnotation', 'data_type': 'VARCHAR'}, {'column_name': 'copy_number', 'data_type': 'VARCHAR'}]}. Try to answer the following question. Only the SQL should be returned within ''' SQL query '''. \"Relationship between dependency scores of JAK1 in DepMapDB.\"\n",
      "Response is: Error: name 'log' is not defined\n",
      "You are an exceptional computational biologist and genomics expert and know everything about drug discovery.. Please write the appropriate SQL query using these three different dbs, and their schemas are as {'ccle_cn_gistic in files/db/CCLEGisticCNDB.db': [{'column_name': 'Hugo_Symbol', 'data_type': 'VARCHAR'}, {'column_name': 'CCLE_Name', 'data_type': 'VARCHAR'}, {'column_name': 'gistic_cn', 'data_type': 'VARCHAR'}, {'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}], 'ccle_mutation in files/db/CCLEMutDB.db': [{'column_name': 'Hugo_Symbol', 'data_type': 'VARCHAR'}, {'column_name': 'Entrez_Gene_Id', 'data_type': 'VARCHAR'}, {'column_name': 'NCBI_Build', 'data_type': 'VARCHAR'}, {'column_name': 'Chromosome', 'data_type': 'VARCHAR'}, {'column_name': 'Start_position', 'data_type': 'VARCHAR'}, {'column_name': 'End_position', 'data_type': 'VARCHAR'}, {'column_name': 'Strand', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_Classification', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_Type', 'data_type': 'VARCHAR'}, {'column_name': 'Reference_Allele', 'data_type': 'VARCHAR'}, {'column_name': 'Alternate_Allele', 'data_type': 'VARCHAR'}, {'column_name': 'dbSNP_RS', 'data_type': 'VARCHAR'}, {'column_name': 'dbSNP_Val_Status', 'data_type': 'VARCHAR'}, {'column_name': 'Genome_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Annotation_Transcript', 'data_type': 'VARCHAR'}, {'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}, {'column_name': 'cDNA_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Codon_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Protein_Change', 'data_type': 'VARCHAR'}, {'column_name': 'isDeleterious', 'data_type': 'VARCHAR'}, {'column_name': 'isTCGAhotspot', 'data_type': 'VARCHAR'}, {'column_name': 'TCGAhsCnt', 'data_type': 'VARCHAR'}, {'column_name': 'isCOSMIChotspot', 'data_type': 'VARCHAR'}, {'column_name': 'COSMIChsCnt', 'data_type': 'VARCHAR'}, {'column_name': 'ExAC_AF', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_annotation', 'data_type': 'VARCHAR'}, {'column_name': 'CGA_WES_AC', 'data_type': 'VARCHAR'}, {'column_name': 'HC_AC', 'data_type': 'VARCHAR'}, {'column_name': 'RD_AC', 'data_type': 'VARCHAR'}, {'column_name': 'RNAseq_AC', 'data_type': 'VARCHAR'}, {'column_name': 'SangerWES_AC', 'data_type': 'VARCHAR'}, {'column_name': 'WGS_AC', 'data_type': 'VARCHAR'}], 'ccle_variants in files/db/CCLEVarDB.db': [{'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}, {'column_name': 'Chrom', 'data_type': 'VARCHAR'}, {'column_name': 'Pos', 'data_type': 'VARCHAR'}, {'column_name': 'Ref', 'data_type': 'VARCHAR'}, {'column_name': 'Alt', 'data_type': 'VARCHAR'}, {'column_name': 'AF', 'data_type': 'VARCHAR'}, {'column_name': 'VariantType', 'data_type': 'VARCHAR'}, {'column_name': 'VariantInfo', 'data_type': 'VARCHAR'}, {'column_name': 'DNAChange', 'data_type': 'VARCHAR'}, {'column_name': 'ProteinChange', 'data_type': 'VARCHAR'}, {'column_name': 'gene_name', 'data_type': 'VARCHAR'}, {'column_name': 'Transcript', 'data_type': 'VARCHAR'}, {'column_name': 'CCLEDeleterious', 'data_type': 'VARCHAR'}, {'column_name': 'CosmicHotspot', 'data_type': 'VARCHAR'}, {'column_name': 'AssociatedWith', 'data_type': 'VARCHAR'}, {'column_name': 'LoF', 'data_type': 'VARCHAR'}, {'column_name': 'Driver', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyDriver', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyGoF', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyLoF', 'data_type': 'VARCHAR'}, {'column_name': 'VariantAnnotation', 'data_type': 'VARCHAR'}, {'column_name': 'copy_number', 'data_type': 'VARCHAR'}]}. Try to answer the following question. Only the SQL should be returned within ''' SQL query '''. \"Which cell lines have high dependency for the target of interest for the gene POLR3E?\"\n",
      "Response is: {'response': \"SQLquery:SELECTDISTINCTcvd.DepMap_IDFROMccle_variantscvdJOINccle_mutationcmONcvd.gene_name=cm.Hugo_SymbolJOINccle_cn_gisticccgONcvd.gene_name=ccg.Hugo_SymbolWHEREcvd.gene_name='POLR3E'ANDccg.gistic_cn='High'ANDcm.Variant_Classification='Missense_Mutation';\", 'is_valid': True}\n",
      "You are an exceptional computational biologist and genomics expert and know everything about drug discovery.. Please write the appropriate SQL query using these three different dbs, and their schemas are as {'ccle_cn_gistic in files/db/CCLEGisticCNDB.db': [{'column_name': 'Hugo_Symbol', 'data_type': 'VARCHAR'}, {'column_name': 'CCLE_Name', 'data_type': 'VARCHAR'}, {'column_name': 'gistic_cn', 'data_type': 'VARCHAR'}, {'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}], 'ccle_mutation in files/db/CCLEMutDB.db': [{'column_name': 'Hugo_Symbol', 'data_type': 'VARCHAR'}, {'column_name': 'Entrez_Gene_Id', 'data_type': 'VARCHAR'}, {'column_name': 'NCBI_Build', 'data_type': 'VARCHAR'}, {'column_name': 'Chromosome', 'data_type': 'VARCHAR'}, {'column_name': 'Start_position', 'data_type': 'VARCHAR'}, {'column_name': 'End_position', 'data_type': 'VARCHAR'}, {'column_name': 'Strand', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_Classification', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_Type', 'data_type': 'VARCHAR'}, {'column_name': 'Reference_Allele', 'data_type': 'VARCHAR'}, {'column_name': 'Alternate_Allele', 'data_type': 'VARCHAR'}, {'column_name': 'dbSNP_RS', 'data_type': 'VARCHAR'}, {'column_name': 'dbSNP_Val_Status', 'data_type': 'VARCHAR'}, {'column_name': 'Genome_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Annotation_Transcript', 'data_type': 'VARCHAR'}, {'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}, {'column_name': 'cDNA_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Codon_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Protein_Change', 'data_type': 'VARCHAR'}, {'column_name': 'isDeleterious', 'data_type': 'VARCHAR'}, {'column_name': 'isTCGAhotspot', 'data_type': 'VARCHAR'}, {'column_name': 'TCGAhsCnt', 'data_type': 'VARCHAR'}, {'column_name': 'isCOSMIChotspot', 'data_type': 'VARCHAR'}, {'column_name': 'COSMIChsCnt', 'data_type': 'VARCHAR'}, {'column_name': 'ExAC_AF', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_annotation', 'data_type': 'VARCHAR'}, {'column_name': 'CGA_WES_AC', 'data_type': 'VARCHAR'}, {'column_name': 'HC_AC', 'data_type': 'VARCHAR'}, {'column_name': 'RD_AC', 'data_type': 'VARCHAR'}, {'column_name': 'RNAseq_AC', 'data_type': 'VARCHAR'}, {'column_name': 'SangerWES_AC', 'data_type': 'VARCHAR'}, {'column_name': 'WGS_AC', 'data_type': 'VARCHAR'}], 'ccle_variants in files/db/CCLEVarDB.db': [{'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}, {'column_name': 'Chrom', 'data_type': 'VARCHAR'}, {'column_name': 'Pos', 'data_type': 'VARCHAR'}, {'column_name': 'Ref', 'data_type': 'VARCHAR'}, {'column_name': 'Alt', 'data_type': 'VARCHAR'}, {'column_name': 'AF', 'data_type': 'VARCHAR'}, {'column_name': 'VariantType', 'data_type': 'VARCHAR'}, {'column_name': 'VariantInfo', 'data_type': 'VARCHAR'}, {'column_name': 'DNAChange', 'data_type': 'VARCHAR'}, {'column_name': 'ProteinChange', 'data_type': 'VARCHAR'}, {'column_name': 'gene_name', 'data_type': 'VARCHAR'}, {'column_name': 'Transcript', 'data_type': 'VARCHAR'}, {'column_name': 'CCLEDeleterious', 'data_type': 'VARCHAR'}, {'column_name': 'CosmicHotspot', 'data_type': 'VARCHAR'}, {'column_name': 'AssociatedWith', 'data_type': 'VARCHAR'}, {'column_name': 'LoF', 'data_type': 'VARCHAR'}, {'column_name': 'Driver', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyDriver', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyGoF', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyLoF', 'data_type': 'VARCHAR'}, {'column_name': 'VariantAnnotation', 'data_type': 'VARCHAR'}, {'column_name': 'copy_number', 'data_type': 'VARCHAR'}]}. Try to answer the following question. Only the SQL should be returned within ''' SQL query '''. \"For the gene POLR3E, identify indications with significant differential expression in cancer vs normal tissue. Is it overexpressed or underexpressed in tumor tissue?\"\n",
      "Response is: Error: name 'log' is not defined\n",
      "You are an exceptional computational biologist and genomics expert and know everything about drug discovery.. Please write the appropriate SQL query using these three different dbs, and their schemas are as {'ccle_cn_gistic in files/db/CCLEGisticCNDB.db': [{'column_name': 'Hugo_Symbol', 'data_type': 'VARCHAR'}, {'column_name': 'CCLE_Name', 'data_type': 'VARCHAR'}, {'column_name': 'gistic_cn', 'data_type': 'VARCHAR'}, {'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}], 'ccle_mutation in files/db/CCLEMutDB.db': [{'column_name': 'Hugo_Symbol', 'data_type': 'VARCHAR'}, {'column_name': 'Entrez_Gene_Id', 'data_type': 'VARCHAR'}, {'column_name': 'NCBI_Build', 'data_type': 'VARCHAR'}, {'column_name': 'Chromosome', 'data_type': 'VARCHAR'}, {'column_name': 'Start_position', 'data_type': 'VARCHAR'}, {'column_name': 'End_position', 'data_type': 'VARCHAR'}, {'column_name': 'Strand', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_Classification', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_Type', 'data_type': 'VARCHAR'}, {'column_name': 'Reference_Allele', 'data_type': 'VARCHAR'}, {'column_name': 'Alternate_Allele', 'data_type': 'VARCHAR'}, {'column_name': 'dbSNP_RS', 'data_type': 'VARCHAR'}, {'column_name': 'dbSNP_Val_Status', 'data_type': 'VARCHAR'}, {'column_name': 'Genome_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Annotation_Transcript', 'data_type': 'VARCHAR'}, {'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}, {'column_name': 'cDNA_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Codon_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Protein_Change', 'data_type': 'VARCHAR'}, {'column_name': 'isDeleterious', 'data_type': 'VARCHAR'}, {'column_name': 'isTCGAhotspot', 'data_type': 'VARCHAR'}, {'column_name': 'TCGAhsCnt', 'data_type': 'VARCHAR'}, {'column_name': 'isCOSMIChotspot', 'data_type': 'VARCHAR'}, {'column_name': 'COSMIChsCnt', 'data_type': 'VARCHAR'}, {'column_name': 'ExAC_AF', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_annotation', 'data_type': 'VARCHAR'}, {'column_name': 'CGA_WES_AC', 'data_type': 'VARCHAR'}, {'column_name': 'HC_AC', 'data_type': 'VARCHAR'}, {'column_name': 'RD_AC', 'data_type': 'VARCHAR'}, {'column_name': 'RNAseq_AC', 'data_type': 'VARCHAR'}, {'column_name': 'SangerWES_AC', 'data_type': 'VARCHAR'}, {'column_name': 'WGS_AC', 'data_type': 'VARCHAR'}], 'ccle_variants in files/db/CCLEVarDB.db': [{'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}, {'column_name': 'Chrom', 'data_type': 'VARCHAR'}, {'column_name': 'Pos', 'data_type': 'VARCHAR'}, {'column_name': 'Ref', 'data_type': 'VARCHAR'}, {'column_name': 'Alt', 'data_type': 'VARCHAR'}, {'column_name': 'AF', 'data_type': 'VARCHAR'}, {'column_name': 'VariantType', 'data_type': 'VARCHAR'}, {'column_name': 'VariantInfo', 'data_type': 'VARCHAR'}, {'column_name': 'DNAChange', 'data_type': 'VARCHAR'}, {'column_name': 'ProteinChange', 'data_type': 'VARCHAR'}, {'column_name': 'gene_name', 'data_type': 'VARCHAR'}, {'column_name': 'Transcript', 'data_type': 'VARCHAR'}, {'column_name': 'CCLEDeleterious', 'data_type': 'VARCHAR'}, {'column_name': 'CosmicHotspot', 'data_type': 'VARCHAR'}, {'column_name': 'AssociatedWith', 'data_type': 'VARCHAR'}, {'column_name': 'LoF', 'data_type': 'VARCHAR'}, {'column_name': 'Driver', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyDriver', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyGoF', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyLoF', 'data_type': 'VARCHAR'}, {'column_name': 'VariantAnnotation', 'data_type': 'VARCHAR'}, {'column_name': 'copy_number', 'data_type': 'VARCHAR'}]}. Try to answer the following question. Only the SQL should be returned within ''' SQL query '''. \"Are there specific regions of the protein of the gene POLR3E where mutations occur at a higher frequency?\"\n",
      "Response is: {'response': \"'''SELECTv.DepMap\\\\_ID,v.Chrom,v.Pos,v.Ref,v.Alt,v.AF,v.VariantType,v.DNAChange,v.ProteinChange,v.gene\\\\_name,v.Transcript,v.CCLEDeleterious,v.CosmicHotspot,v.AssociatedWith,v.LoF,v.Driver,v.LikelyDriver,v.LikelyGoF,v.LikelyLoF,v.VariantAnnotation,c.gistic\\\\_cnFROMccle\\\\_variantsvJOINccle\\\\_cn\\\\_gisticcONv.DepMap\\\\_ID=c.DepMap\\\\_IDWHEREv.gene\\\\_name='POLR3E'ORDERBYv.Pos;Thisqueryselectsthenecessarycolumnsfromthe'ccle\\\\_variants'and'ccle\\\\_cn\\\\_gistic'tables,joinsthemonthe'DepMap\\\\_ID'column,filtersforthegeneofinterest(POLR3E),andorderstheresultsbyposition.The'gistic\\\\_cn'columnfromthe'ccle\\\\_cn\\\\_gistic'tablewillindicatewhethertherearespecificregionsoftheproteinwheremutationsoccuratahigherfrequency.\", 'is_valid': True}\n",
      "You are an exceptional computational biologist and genomics expert and know everything about drug discovery.. Please write the appropriate SQL query using these three different dbs, and their schemas are as {'ccle_cn_gistic in files/db/CCLEGisticCNDB.db': [{'column_name': 'Hugo_Symbol', 'data_type': 'VARCHAR'}, {'column_name': 'CCLE_Name', 'data_type': 'VARCHAR'}, {'column_name': 'gistic_cn', 'data_type': 'VARCHAR'}, {'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}], 'ccle_mutation in files/db/CCLEMutDB.db': [{'column_name': 'Hugo_Symbol', 'data_type': 'VARCHAR'}, {'column_name': 'Entrez_Gene_Id', 'data_type': 'VARCHAR'}, {'column_name': 'NCBI_Build', 'data_type': 'VARCHAR'}, {'column_name': 'Chromosome', 'data_type': 'VARCHAR'}, {'column_name': 'Start_position', 'data_type': 'VARCHAR'}, {'column_name': 'End_position', 'data_type': 'VARCHAR'}, {'column_name': 'Strand', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_Classification', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_Type', 'data_type': 'VARCHAR'}, {'column_name': 'Reference_Allele', 'data_type': 'VARCHAR'}, {'column_name': 'Alternate_Allele', 'data_type': 'VARCHAR'}, {'column_name': 'dbSNP_RS', 'data_type': 'VARCHAR'}, {'column_name': 'dbSNP_Val_Status', 'data_type': 'VARCHAR'}, {'column_name': 'Genome_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Annotation_Transcript', 'data_type': 'VARCHAR'}, {'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}, {'column_name': 'cDNA_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Codon_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Protein_Change', 'data_type': 'VARCHAR'}, {'column_name': 'isDeleterious', 'data_type': 'VARCHAR'}, {'column_name': 'isTCGAhotspot', 'data_type': 'VARCHAR'}, {'column_name': 'TCGAhsCnt', 'data_type': 'VARCHAR'}, {'column_name': 'isCOSMIChotspot', 'data_type': 'VARCHAR'}, {'column_name': 'COSMIChsCnt', 'data_type': 'VARCHAR'}, {'column_name': 'ExAC_AF', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_annotation', 'data_type': 'VARCHAR'}, {'column_name': 'CGA_WES_AC', 'data_type': 'VARCHAR'}, {'column_name': 'HC_AC', 'data_type': 'VARCHAR'}, {'column_name': 'RD_AC', 'data_type': 'VARCHAR'}, {'column_name': 'RNAseq_AC', 'data_type': 'VARCHAR'}, {'column_name': 'SangerWES_AC', 'data_type': 'VARCHAR'}, {'column_name': 'WGS_AC', 'data_type': 'VARCHAR'}], 'ccle_variants in files/db/CCLEVarDB.db': [{'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}, {'column_name': 'Chrom', 'data_type': 'VARCHAR'}, {'column_name': 'Pos', 'data_type': 'VARCHAR'}, {'column_name': 'Ref', 'data_type': 'VARCHAR'}, {'column_name': 'Alt', 'data_type': 'VARCHAR'}, {'column_name': 'AF', 'data_type': 'VARCHAR'}, {'column_name': 'VariantType', 'data_type': 'VARCHAR'}, {'column_name': 'VariantInfo', 'data_type': 'VARCHAR'}, {'column_name': 'DNAChange', 'data_type': 'VARCHAR'}, {'column_name': 'ProteinChange', 'data_type': 'VARCHAR'}, {'column_name': 'gene_name', 'data_type': 'VARCHAR'}, {'column_name': 'Transcript', 'data_type': 'VARCHAR'}, {'column_name': 'CCLEDeleterious', 'data_type': 'VARCHAR'}, {'column_name': 'CosmicHotspot', 'data_type': 'VARCHAR'}, {'column_name': 'AssociatedWith', 'data_type': 'VARCHAR'}, {'column_name': 'LoF', 'data_type': 'VARCHAR'}, {'column_name': 'Driver', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyDriver', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyGoF', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyLoF', 'data_type': 'VARCHAR'}, {'column_name': 'VariantAnnotation', 'data_type': 'VARCHAR'}, {'column_name': 'copy_number', 'data_type': 'VARCHAR'}]}. Try to answer the following question. Only the SQL should be returned within ''' SQL query '''. \"For the gene POLR3E, determine how many of the mutations are pathogenic.\"\n",
      "Response is: Error: name 'log' is not defined\n",
      "You are an exceptional computational biologist and genomics expert and know everything about drug discovery.. Please write the appropriate SQL query using these three different dbs, and their schemas are as {'ccle_cn_gistic in files/db/CCLEGisticCNDB.db': [{'column_name': 'Hugo_Symbol', 'data_type': 'VARCHAR'}, {'column_name': 'CCLE_Name', 'data_type': 'VARCHAR'}, {'column_name': 'gistic_cn', 'data_type': 'VARCHAR'}, {'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}], 'ccle_mutation in files/db/CCLEMutDB.db': [{'column_name': 'Hugo_Symbol', 'data_type': 'VARCHAR'}, {'column_name': 'Entrez_Gene_Id', 'data_type': 'VARCHAR'}, {'column_name': 'NCBI_Build', 'data_type': 'VARCHAR'}, {'column_name': 'Chromosome', 'data_type': 'VARCHAR'}, {'column_name': 'Start_position', 'data_type': 'VARCHAR'}, {'column_name': 'End_position', 'data_type': 'VARCHAR'}, {'column_name': 'Strand', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_Classification', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_Type', 'data_type': 'VARCHAR'}, {'column_name': 'Reference_Allele', 'data_type': 'VARCHAR'}, {'column_name': 'Alternate_Allele', 'data_type': 'VARCHAR'}, {'column_name': 'dbSNP_RS', 'data_type': 'VARCHAR'}, {'column_name': 'dbSNP_Val_Status', 'data_type': 'VARCHAR'}, {'column_name': 'Genome_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Annotation_Transcript', 'data_type': 'VARCHAR'}, {'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}, {'column_name': 'cDNA_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Codon_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Protein_Change', 'data_type': 'VARCHAR'}, {'column_name': 'isDeleterious', 'data_type': 'VARCHAR'}, {'column_name': 'isTCGAhotspot', 'data_type': 'VARCHAR'}, {'column_name': 'TCGAhsCnt', 'data_type': 'VARCHAR'}, {'column_name': 'isCOSMIChotspot', 'data_type': 'VARCHAR'}, {'column_name': 'COSMIChsCnt', 'data_type': 'VARCHAR'}, {'column_name': 'ExAC_AF', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_annotation', 'data_type': 'VARCHAR'}, {'column_name': 'CGA_WES_AC', 'data_type': 'VARCHAR'}, {'column_name': 'HC_AC', 'data_type': 'VARCHAR'}, {'column_name': 'RD_AC', 'data_type': 'VARCHAR'}, {'column_name': 'RNAseq_AC', 'data_type': 'VARCHAR'}, {'column_name': 'SangerWES_AC', 'data_type': 'VARCHAR'}, {'column_name': 'WGS_AC', 'data_type': 'VARCHAR'}], 'ccle_variants in files/db/CCLEVarDB.db': [{'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}, {'column_name': 'Chrom', 'data_type': 'VARCHAR'}, {'column_name': 'Pos', 'data_type': 'VARCHAR'}, {'column_name': 'Ref', 'data_type': 'VARCHAR'}, {'column_name': 'Alt', 'data_type': 'VARCHAR'}, {'column_name': 'AF', 'data_type': 'VARCHAR'}, {'column_name': 'VariantType', 'data_type': 'VARCHAR'}, {'column_name': 'VariantInfo', 'data_type': 'VARCHAR'}, {'column_name': 'DNAChange', 'data_type': 'VARCHAR'}, {'column_name': 'ProteinChange', 'data_type': 'VARCHAR'}, {'column_name': 'gene_name', 'data_type': 'VARCHAR'}, {'column_name': 'Transcript', 'data_type': 'VARCHAR'}, {'column_name': 'CCLEDeleterious', 'data_type': 'VARCHAR'}, {'column_name': 'CosmicHotspot', 'data_type': 'VARCHAR'}, {'column_name': 'AssociatedWith', 'data_type': 'VARCHAR'}, {'column_name': 'LoF', 'data_type': 'VARCHAR'}, {'column_name': 'Driver', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyDriver', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyGoF', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyLoF', 'data_type': 'VARCHAR'}, {'column_name': 'VariantAnnotation', 'data_type': 'VARCHAR'}, {'column_name': 'copy_number', 'data_type': 'VARCHAR'}]}. Try to answer the following question. Only the SQL should be returned within ''' SQL query '''. \"Which mutations in PAS domain for the target protein exhibit high selectivity for the prostate cancer cell lines?\"\n",
      "Response is: {'response': \"SELECTccg.Hugo_Symbol,ccm.Entrez_Gene_Id,ccm.NCBI_Build,ccm.Chromosome,ccm.Start_position,ccm.End_position,ccm.Strand,ccm.Variant_Classification,ccm.Variant_Type,ccm.Reference_Allele,ccm.Alternate_Allele,ccm.dbSNP_RS,ccm.dbSNP_Val_Status,ccm.Genome_Change,ccm.Annotation_Transcript,ccm.DepMap_ID,ccm.cDNA_Change,ccm.Codon_Change,ccm.Protein_Change,ccm.isDeleterious,ccm.isTCGAhotspot,ccm.TCGAhsCnt,ccm.isCOSMIChotspot,ccm.COSMIChsCnt,ccm.ExAC_AF,ccm.Variant_annotation,ccm.CGA_WES_AC,ccm.HC_AC,ccm.RD_AC,ccm.RNAseq_AC,ccm.SangerWES_AC,ccm.WGS_AC,ccv.Chrom,ccv.Pos,ccv.Ref,ccv.Alt,ccv.AF,ccv.VariantType,ccv.VariantInfo,ccv.DNAChange,ccv.ProteinChange,ccv.gene_name,ccv.Transcript,ccv.CCLEDeleterious,ccv.CosmicHotspot,ccv.AssociatedWith,ccv.LoF,ccv.Driver,ccv.LikelyDriver,ccv.LikelyGoF,ccv.LikelyLoF,ccv.VariantAnnotation,ccv.copy_numberFROMccle_mutationccmJOINccle_variantsccvONccm.DepMap_ID=ccv.DepMap_IDJOINccle_cn_gisticccgONccm.DepMap_ID=ccg.DepMap_IDWHEREccg.Hugo_Symbol='PAS'ANDc\", 'is_valid': True}\n",
      "You are an exceptional computational biologist and genomics expert and know everything about drug discovery.. Please write the appropriate SQL query using these three different dbs, and their schemas are as {'ccle_cn_gistic in files/db/CCLEGisticCNDB.db': [{'column_name': 'Hugo_Symbol', 'data_type': 'VARCHAR'}, {'column_name': 'CCLE_Name', 'data_type': 'VARCHAR'}, {'column_name': 'gistic_cn', 'data_type': 'VARCHAR'}, {'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}], 'ccle_mutation in files/db/CCLEMutDB.db': [{'column_name': 'Hugo_Symbol', 'data_type': 'VARCHAR'}, {'column_name': 'Entrez_Gene_Id', 'data_type': 'VARCHAR'}, {'column_name': 'NCBI_Build', 'data_type': 'VARCHAR'}, {'column_name': 'Chromosome', 'data_type': 'VARCHAR'}, {'column_name': 'Start_position', 'data_type': 'VARCHAR'}, {'column_name': 'End_position', 'data_type': 'VARCHAR'}, {'column_name': 'Strand', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_Classification', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_Type', 'data_type': 'VARCHAR'}, {'column_name': 'Reference_Allele', 'data_type': 'VARCHAR'}, {'column_name': 'Alternate_Allele', 'data_type': 'VARCHAR'}, {'column_name': 'dbSNP_RS', 'data_type': 'VARCHAR'}, {'column_name': 'dbSNP_Val_Status', 'data_type': 'VARCHAR'}, {'column_name': 'Genome_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Annotation_Transcript', 'data_type': 'VARCHAR'}, {'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}, {'column_name': 'cDNA_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Codon_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Protein_Change', 'data_type': 'VARCHAR'}, {'column_name': 'isDeleterious', 'data_type': 'VARCHAR'}, {'column_name': 'isTCGAhotspot', 'data_type': 'VARCHAR'}, {'column_name': 'TCGAhsCnt', 'data_type': 'VARCHAR'}, {'column_name': 'isCOSMIChotspot', 'data_type': 'VARCHAR'}, {'column_name': 'COSMIChsCnt', 'data_type': 'VARCHAR'}, {'column_name': 'ExAC_AF', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_annotation', 'data_type': 'VARCHAR'}, {'column_name': 'CGA_WES_AC', 'data_type': 'VARCHAR'}, {'column_name': 'HC_AC', 'data_type': 'VARCHAR'}, {'column_name': 'RD_AC', 'data_type': 'VARCHAR'}, {'column_name': 'RNAseq_AC', 'data_type': 'VARCHAR'}, {'column_name': 'SangerWES_AC', 'data_type': 'VARCHAR'}, {'column_name': 'WGS_AC', 'data_type': 'VARCHAR'}], 'ccle_variants in files/db/CCLEVarDB.db': [{'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}, {'column_name': 'Chrom', 'data_type': 'VARCHAR'}, {'column_name': 'Pos', 'data_type': 'VARCHAR'}, {'column_name': 'Ref', 'data_type': 'VARCHAR'}, {'column_name': 'Alt', 'data_type': 'VARCHAR'}, {'column_name': 'AF', 'data_type': 'VARCHAR'}, {'column_name': 'VariantType', 'data_type': 'VARCHAR'}, {'column_name': 'VariantInfo', 'data_type': 'VARCHAR'}, {'column_name': 'DNAChange', 'data_type': 'VARCHAR'}, {'column_name': 'ProteinChange', 'data_type': 'VARCHAR'}, {'column_name': 'gene_name', 'data_type': 'VARCHAR'}, {'column_name': 'Transcript', 'data_type': 'VARCHAR'}, {'column_name': 'CCLEDeleterious', 'data_type': 'VARCHAR'}, {'column_name': 'CosmicHotspot', 'data_type': 'VARCHAR'}, {'column_name': 'AssociatedWith', 'data_type': 'VARCHAR'}, {'column_name': 'LoF', 'data_type': 'VARCHAR'}, {'column_name': 'Driver', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyDriver', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyGoF', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyLoF', 'data_type': 'VARCHAR'}, {'column_name': 'VariantAnnotation', 'data_type': 'VARCHAR'}, {'column_name': 'copy_number', 'data_type': 'VARCHAR'}]}. Try to answer the following question. Only the SQL should be returned within ''' SQL query '''. \"Relationship between dependency scores of JAK1 in DepMapDB.\"\n",
      "Response is: {'response': \"Toanswerthequestion,youneedtoperformaself-joinoperationonthe'ccle_mutation'tableinthe'CCLEMutDB.db'database.Theself-joinoperationisnecessarytocomparethedependencyscoresofJAK1betweendifferentsamples.Here'stheSQLquery:'''SQLquerySELECTa.DepMap\\\\_ID,a.Hugo\\\\_Symbol,a.isDeleterious,a.isTCGAhotspot,a.TCGAhsCnt,a.isCOSMIChotspot,a.COSMIChsCnt,a.ExAC\\\\_AF,a.Variant\\\\_annotation,a.CGA\\\\_WES\\\\_AC,a.HC\\\\_AC,a.RD\\\\_AC,a.RNAseq\\\\_AC,a.SangerWES\\\\_AC,a.WGS\\\\_AC,b.DepMap\\\\_ID,b.CCLE\\\\_Name,b.gistic\\\\_cnFROMccle\\\\_mutationaJOINccle\\\\_cn\\\\_gisticbONa.Hugo\\\\_Symbol='JAK1'ANDa.DepMap\\\\_ID=b.DepMap\\\\_IDORDERBYa.DepMap\\\\_ID,b.gistic\\\\_cn;'''Thisquerywillreturnallthecolumnsfromthe'ccle\\\\_mutation'tableandthe'gistic\\\\_cn'columnfromthe'ccle\\\\_cn\\\\_gistic'table,filteredbytheHugo\\\\_Symbol'JAK1'.TheresultwillbesortedbyDepMap\\\\_IDandgistic\\\\_cn,showingtherelationshipbetweendependencyscoresofJAK1inDepMapDB.\", 'is_valid': True}\n",
      "You are an exceptional computational biologist and genomics expert and know everything about drug discovery.. Please write the appropriate SQL query using these three different dbs, and their schemas are as {'ccle_cn_gistic in files/db/CCLEGisticCNDB.db': [{'column_name': 'Hugo_Symbol', 'data_type': 'VARCHAR'}, {'column_name': 'CCLE_Name', 'data_type': 'VARCHAR'}, {'column_name': 'gistic_cn', 'data_type': 'VARCHAR'}, {'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}], 'ccle_mutation in files/db/CCLEMutDB.db': [{'column_name': 'Hugo_Symbol', 'data_type': 'VARCHAR'}, {'column_name': 'Entrez_Gene_Id', 'data_type': 'VARCHAR'}, {'column_name': 'NCBI_Build', 'data_type': 'VARCHAR'}, {'column_name': 'Chromosome', 'data_type': 'VARCHAR'}, {'column_name': 'Start_position', 'data_type': 'VARCHAR'}, {'column_name': 'End_position', 'data_type': 'VARCHAR'}, {'column_name': 'Strand', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_Classification', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_Type', 'data_type': 'VARCHAR'}, {'column_name': 'Reference_Allele', 'data_type': 'VARCHAR'}, {'column_name': 'Alternate_Allele', 'data_type': 'VARCHAR'}, {'column_name': 'dbSNP_RS', 'data_type': 'VARCHAR'}, {'column_name': 'dbSNP_Val_Status', 'data_type': 'VARCHAR'}, {'column_name': 'Genome_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Annotation_Transcript', 'data_type': 'VARCHAR'}, {'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}, {'column_name': 'cDNA_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Codon_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Protein_Change', 'data_type': 'VARCHAR'}, {'column_name': 'isDeleterious', 'data_type': 'VARCHAR'}, {'column_name': 'isTCGAhotspot', 'data_type': 'VARCHAR'}, {'column_name': 'TCGAhsCnt', 'data_type': 'VARCHAR'}, {'column_name': 'isCOSMIChotspot', 'data_type': 'VARCHAR'}, {'column_name': 'COSMIChsCnt', 'data_type': 'VARCHAR'}, {'column_name': 'ExAC_AF', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_annotation', 'data_type': 'VARCHAR'}, {'column_name': 'CGA_WES_AC', 'data_type': 'VARCHAR'}, {'column_name': 'HC_AC', 'data_type': 'VARCHAR'}, {'column_name': 'RD_AC', 'data_type': 'VARCHAR'}, {'column_name': 'RNAseq_AC', 'data_type': 'VARCHAR'}, {'column_name': 'SangerWES_AC', 'data_type': 'VARCHAR'}, {'column_name': 'WGS_AC', 'data_type': 'VARCHAR'}], 'ccle_variants in files/db/CCLEVarDB.db': [{'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}, {'column_name': 'Chrom', 'data_type': 'VARCHAR'}, {'column_name': 'Pos', 'data_type': 'VARCHAR'}, {'column_name': 'Ref', 'data_type': 'VARCHAR'}, {'column_name': 'Alt', 'data_type': 'VARCHAR'}, {'column_name': 'AF', 'data_type': 'VARCHAR'}, {'column_name': 'VariantType', 'data_type': 'VARCHAR'}, {'column_name': 'VariantInfo', 'data_type': 'VARCHAR'}, {'column_name': 'DNAChange', 'data_type': 'VARCHAR'}, {'column_name': 'ProteinChange', 'data_type': 'VARCHAR'}, {'column_name': 'gene_name', 'data_type': 'VARCHAR'}, {'column_name': 'Transcript', 'data_type': 'VARCHAR'}, {'column_name': 'CCLEDeleterious', 'data_type': 'VARCHAR'}, {'column_name': 'CosmicHotspot', 'data_type': 'VARCHAR'}, {'column_name': 'AssociatedWith', 'data_type': 'VARCHAR'}, {'column_name': 'LoF', 'data_type': 'VARCHAR'}, {'column_name': 'Driver', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyDriver', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyGoF', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyLoF', 'data_type': 'VARCHAR'}, {'column_name': 'VariantAnnotation', 'data_type': 'VARCHAR'}, {'column_name': 'copy_number', 'data_type': 'VARCHAR'}]}. Try to answer the following question. Only the SQL should be returned within ''' SQL query '''. \"Which cell lines have high dependency for the target of interest for the gene POLR3E?\"\n",
      "Response is: Error: name 'log' is not defined\n",
      "You are an exceptional computational biologist and genomics expert and know everything about drug discovery.. Please write the appropriate SQL query using these three different dbs, and their schemas are as {'ccle_cn_gistic in files/db/CCLEGisticCNDB.db': [{'column_name': 'Hugo_Symbol', 'data_type': 'VARCHAR'}, {'column_name': 'CCLE_Name', 'data_type': 'VARCHAR'}, {'column_name': 'gistic_cn', 'data_type': 'VARCHAR'}, {'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}], 'ccle_mutation in files/db/CCLEMutDB.db': [{'column_name': 'Hugo_Symbol', 'data_type': 'VARCHAR'}, {'column_name': 'Entrez_Gene_Id', 'data_type': 'VARCHAR'}, {'column_name': 'NCBI_Build', 'data_type': 'VARCHAR'}, {'column_name': 'Chromosome', 'data_type': 'VARCHAR'}, {'column_name': 'Start_position', 'data_type': 'VARCHAR'}, {'column_name': 'End_position', 'data_type': 'VARCHAR'}, {'column_name': 'Strand', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_Classification', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_Type', 'data_type': 'VARCHAR'}, {'column_name': 'Reference_Allele', 'data_type': 'VARCHAR'}, {'column_name': 'Alternate_Allele', 'data_type': 'VARCHAR'}, {'column_name': 'dbSNP_RS', 'data_type': 'VARCHAR'}, {'column_name': 'dbSNP_Val_Status', 'data_type': 'VARCHAR'}, {'column_name': 'Genome_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Annotation_Transcript', 'data_type': 'VARCHAR'}, {'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}, {'column_name': 'cDNA_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Codon_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Protein_Change', 'data_type': 'VARCHAR'}, {'column_name': 'isDeleterious', 'data_type': 'VARCHAR'}, {'column_name': 'isTCGAhotspot', 'data_type': 'VARCHAR'}, {'column_name': 'TCGAhsCnt', 'data_type': 'VARCHAR'}, {'column_name': 'isCOSMIChotspot', 'data_type': 'VARCHAR'}, {'column_name': 'COSMIChsCnt', 'data_type': 'VARCHAR'}, {'column_name': 'ExAC_AF', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_annotation', 'data_type': 'VARCHAR'}, {'column_name': 'CGA_WES_AC', 'data_type': 'VARCHAR'}, {'column_name': 'HC_AC', 'data_type': 'VARCHAR'}, {'column_name': 'RD_AC', 'data_type': 'VARCHAR'}, {'column_name': 'RNAseq_AC', 'data_type': 'VARCHAR'}, {'column_name': 'SangerWES_AC', 'data_type': 'VARCHAR'}, {'column_name': 'WGS_AC', 'data_type': 'VARCHAR'}], 'ccle_variants in files/db/CCLEVarDB.db': [{'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}, {'column_name': 'Chrom', 'data_type': 'VARCHAR'}, {'column_name': 'Pos', 'data_type': 'VARCHAR'}, {'column_name': 'Ref', 'data_type': 'VARCHAR'}, {'column_name': 'Alt', 'data_type': 'VARCHAR'}, {'column_name': 'AF', 'data_type': 'VARCHAR'}, {'column_name': 'VariantType', 'data_type': 'VARCHAR'}, {'column_name': 'VariantInfo', 'data_type': 'VARCHAR'}, {'column_name': 'DNAChange', 'data_type': 'VARCHAR'}, {'column_name': 'ProteinChange', 'data_type': 'VARCHAR'}, {'column_name': 'gene_name', 'data_type': 'VARCHAR'}, {'column_name': 'Transcript', 'data_type': 'VARCHAR'}, {'column_name': 'CCLEDeleterious', 'data_type': 'VARCHAR'}, {'column_name': 'CosmicHotspot', 'data_type': 'VARCHAR'}, {'column_name': 'AssociatedWith', 'data_type': 'VARCHAR'}, {'column_name': 'LoF', 'data_type': 'VARCHAR'}, {'column_name': 'Driver', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyDriver', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyGoF', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyLoF', 'data_type': 'VARCHAR'}, {'column_name': 'VariantAnnotation', 'data_type': 'VARCHAR'}, {'column_name': 'copy_number', 'data_type': 'VARCHAR'}]}. Try to answer the following question. Only the SQL should be returned within ''' SQL query '''. \"For the gene POLR3E, identify indications with significant differential expression in cancer vs normal tissue. Is it overexpressed or underexpressed in tumor tissue?\"\n",
      "Response is: Error: name 'log' is not defined\n",
      "You are an exceptional computational biologist and genomics expert and know everything about drug discovery.. Please write the appropriate SQL query using these three different dbs, and their schemas are as {'ccle_cn_gistic in files/db/CCLEGisticCNDB.db': [{'column_name': 'Hugo_Symbol', 'data_type': 'VARCHAR'}, {'column_name': 'CCLE_Name', 'data_type': 'VARCHAR'}, {'column_name': 'gistic_cn', 'data_type': 'VARCHAR'}, {'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}], 'ccle_mutation in files/db/CCLEMutDB.db': [{'column_name': 'Hugo_Symbol', 'data_type': 'VARCHAR'}, {'column_name': 'Entrez_Gene_Id', 'data_type': 'VARCHAR'}, {'column_name': 'NCBI_Build', 'data_type': 'VARCHAR'}, {'column_name': 'Chromosome', 'data_type': 'VARCHAR'}, {'column_name': 'Start_position', 'data_type': 'VARCHAR'}, {'column_name': 'End_position', 'data_type': 'VARCHAR'}, {'column_name': 'Strand', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_Classification', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_Type', 'data_type': 'VARCHAR'}, {'column_name': 'Reference_Allele', 'data_type': 'VARCHAR'}, {'column_name': 'Alternate_Allele', 'data_type': 'VARCHAR'}, {'column_name': 'dbSNP_RS', 'data_type': 'VARCHAR'}, {'column_name': 'dbSNP_Val_Status', 'data_type': 'VARCHAR'}, {'column_name': 'Genome_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Annotation_Transcript', 'data_type': 'VARCHAR'}, {'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}, {'column_name': 'cDNA_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Codon_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Protein_Change', 'data_type': 'VARCHAR'}, {'column_name': 'isDeleterious', 'data_type': 'VARCHAR'}, {'column_name': 'isTCGAhotspot', 'data_type': 'VARCHAR'}, {'column_name': 'TCGAhsCnt', 'data_type': 'VARCHAR'}, {'column_name': 'isCOSMIChotspot', 'data_type': 'VARCHAR'}, {'column_name': 'COSMIChsCnt', 'data_type': 'VARCHAR'}, {'column_name': 'ExAC_AF', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_annotation', 'data_type': 'VARCHAR'}, {'column_name': 'CGA_WES_AC', 'data_type': 'VARCHAR'}, {'column_name': 'HC_AC', 'data_type': 'VARCHAR'}, {'column_name': 'RD_AC', 'data_type': 'VARCHAR'}, {'column_name': 'RNAseq_AC', 'data_type': 'VARCHAR'}, {'column_name': 'SangerWES_AC', 'data_type': 'VARCHAR'}, {'column_name': 'WGS_AC', 'data_type': 'VARCHAR'}], 'ccle_variants in files/db/CCLEVarDB.db': [{'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}, {'column_name': 'Chrom', 'data_type': 'VARCHAR'}, {'column_name': 'Pos', 'data_type': 'VARCHAR'}, {'column_name': 'Ref', 'data_type': 'VARCHAR'}, {'column_name': 'Alt', 'data_type': 'VARCHAR'}, {'column_name': 'AF', 'data_type': 'VARCHAR'}, {'column_name': 'VariantType', 'data_type': 'VARCHAR'}, {'column_name': 'VariantInfo', 'data_type': 'VARCHAR'}, {'column_name': 'DNAChange', 'data_type': 'VARCHAR'}, {'column_name': 'ProteinChange', 'data_type': 'VARCHAR'}, {'column_name': 'gene_name', 'data_type': 'VARCHAR'}, {'column_name': 'Transcript', 'data_type': 'VARCHAR'}, {'column_name': 'CCLEDeleterious', 'data_type': 'VARCHAR'}, {'column_name': 'CosmicHotspot', 'data_type': 'VARCHAR'}, {'column_name': 'AssociatedWith', 'data_type': 'VARCHAR'}, {'column_name': 'LoF', 'data_type': 'VARCHAR'}, {'column_name': 'Driver', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyDriver', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyGoF', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyLoF', 'data_type': 'VARCHAR'}, {'column_name': 'VariantAnnotation', 'data_type': 'VARCHAR'}, {'column_name': 'copy_number', 'data_type': 'VARCHAR'}]}. Try to answer the following question. Only the SQL should be returned within ''' SQL query '''. \"Are there specific regions of the protein of the gene POLR3E where mutations occur at a higher frequency?\"\n",
      "Response is: Error: name 'log' is not defined\n",
      "You are an exceptional computational biologist and genomics expert and know everything about drug discovery.. Please write the appropriate SQL query using these three different dbs, and their schemas are as {'ccle_cn_gistic in files/db/CCLEGisticCNDB.db': [{'column_name': 'Hugo_Symbol', 'data_type': 'VARCHAR'}, {'column_name': 'CCLE_Name', 'data_type': 'VARCHAR'}, {'column_name': 'gistic_cn', 'data_type': 'VARCHAR'}, {'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}], 'ccle_mutation in files/db/CCLEMutDB.db': [{'column_name': 'Hugo_Symbol', 'data_type': 'VARCHAR'}, {'column_name': 'Entrez_Gene_Id', 'data_type': 'VARCHAR'}, {'column_name': 'NCBI_Build', 'data_type': 'VARCHAR'}, {'column_name': 'Chromosome', 'data_type': 'VARCHAR'}, {'column_name': 'Start_position', 'data_type': 'VARCHAR'}, {'column_name': 'End_position', 'data_type': 'VARCHAR'}, {'column_name': 'Strand', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_Classification', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_Type', 'data_type': 'VARCHAR'}, {'column_name': 'Reference_Allele', 'data_type': 'VARCHAR'}, {'column_name': 'Alternate_Allele', 'data_type': 'VARCHAR'}, {'column_name': 'dbSNP_RS', 'data_type': 'VARCHAR'}, {'column_name': 'dbSNP_Val_Status', 'data_type': 'VARCHAR'}, {'column_name': 'Genome_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Annotation_Transcript', 'data_type': 'VARCHAR'}, {'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}, {'column_name': 'cDNA_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Codon_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Protein_Change', 'data_type': 'VARCHAR'}, {'column_name': 'isDeleterious', 'data_type': 'VARCHAR'}, {'column_name': 'isTCGAhotspot', 'data_type': 'VARCHAR'}, {'column_name': 'TCGAhsCnt', 'data_type': 'VARCHAR'}, {'column_name': 'isCOSMIChotspot', 'data_type': 'VARCHAR'}, {'column_name': 'COSMIChsCnt', 'data_type': 'VARCHAR'}, {'column_name': 'ExAC_AF', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_annotation', 'data_type': 'VARCHAR'}, {'column_name': 'CGA_WES_AC', 'data_type': 'VARCHAR'}, {'column_name': 'HC_AC', 'data_type': 'VARCHAR'}, {'column_name': 'RD_AC', 'data_type': 'VARCHAR'}, {'column_name': 'RNAseq_AC', 'data_type': 'VARCHAR'}, {'column_name': 'SangerWES_AC', 'data_type': 'VARCHAR'}, {'column_name': 'WGS_AC', 'data_type': 'VARCHAR'}], 'ccle_variants in files/db/CCLEVarDB.db': [{'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}, {'column_name': 'Chrom', 'data_type': 'VARCHAR'}, {'column_name': 'Pos', 'data_type': 'VARCHAR'}, {'column_name': 'Ref', 'data_type': 'VARCHAR'}, {'column_name': 'Alt', 'data_type': 'VARCHAR'}, {'column_name': 'AF', 'data_type': 'VARCHAR'}, {'column_name': 'VariantType', 'data_type': 'VARCHAR'}, {'column_name': 'VariantInfo', 'data_type': 'VARCHAR'}, {'column_name': 'DNAChange', 'data_type': 'VARCHAR'}, {'column_name': 'ProteinChange', 'data_type': 'VARCHAR'}, {'column_name': 'gene_name', 'data_type': 'VARCHAR'}, {'column_name': 'Transcript', 'data_type': 'VARCHAR'}, {'column_name': 'CCLEDeleterious', 'data_type': 'VARCHAR'}, {'column_name': 'CosmicHotspot', 'data_type': 'VARCHAR'}, {'column_name': 'AssociatedWith', 'data_type': 'VARCHAR'}, {'column_name': 'LoF', 'data_type': 'VARCHAR'}, {'column_name': 'Driver', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyDriver', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyGoF', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyLoF', 'data_type': 'VARCHAR'}, {'column_name': 'VariantAnnotation', 'data_type': 'VARCHAR'}, {'column_name': 'copy_number', 'data_type': 'VARCHAR'}]}. Try to answer the following question. Only the SQL should be returned within ''' SQL query '''. \"For the gene POLR3E, determine how many of the mutations are pathogenic.\"\n",
      "Response is: Error: name 'log' is not defined\n",
      "You are an exceptional computational biologist and genomics expert and know everything about drug discovery.. Please write the appropriate SQL query using these three different dbs, and their schemas are as {'ccle_cn_gistic in files/db/CCLEGisticCNDB.db': [{'column_name': 'Hugo_Symbol', 'data_type': 'VARCHAR'}, {'column_name': 'CCLE_Name', 'data_type': 'VARCHAR'}, {'column_name': 'gistic_cn', 'data_type': 'VARCHAR'}, {'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}], 'ccle_mutation in files/db/CCLEMutDB.db': [{'column_name': 'Hugo_Symbol', 'data_type': 'VARCHAR'}, {'column_name': 'Entrez_Gene_Id', 'data_type': 'VARCHAR'}, {'column_name': 'NCBI_Build', 'data_type': 'VARCHAR'}, {'column_name': 'Chromosome', 'data_type': 'VARCHAR'}, {'column_name': 'Start_position', 'data_type': 'VARCHAR'}, {'column_name': 'End_position', 'data_type': 'VARCHAR'}, {'column_name': 'Strand', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_Classification', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_Type', 'data_type': 'VARCHAR'}, {'column_name': 'Reference_Allele', 'data_type': 'VARCHAR'}, {'column_name': 'Alternate_Allele', 'data_type': 'VARCHAR'}, {'column_name': 'dbSNP_RS', 'data_type': 'VARCHAR'}, {'column_name': 'dbSNP_Val_Status', 'data_type': 'VARCHAR'}, {'column_name': 'Genome_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Annotation_Transcript', 'data_type': 'VARCHAR'}, {'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}, {'column_name': 'cDNA_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Codon_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Protein_Change', 'data_type': 'VARCHAR'}, {'column_name': 'isDeleterious', 'data_type': 'VARCHAR'}, {'column_name': 'isTCGAhotspot', 'data_type': 'VARCHAR'}, {'column_name': 'TCGAhsCnt', 'data_type': 'VARCHAR'}, {'column_name': 'isCOSMIChotspot', 'data_type': 'VARCHAR'}, {'column_name': 'COSMIChsCnt', 'data_type': 'VARCHAR'}, {'column_name': 'ExAC_AF', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_annotation', 'data_type': 'VARCHAR'}, {'column_name': 'CGA_WES_AC', 'data_type': 'VARCHAR'}, {'column_name': 'HC_AC', 'data_type': 'VARCHAR'}, {'column_name': 'RD_AC', 'data_type': 'VARCHAR'}, {'column_name': 'RNAseq_AC', 'data_type': 'VARCHAR'}, {'column_name': 'SangerWES_AC', 'data_type': 'VARCHAR'}, {'column_name': 'WGS_AC', 'data_type': 'VARCHAR'}], 'ccle_variants in files/db/CCLEVarDB.db': [{'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}, {'column_name': 'Chrom', 'data_type': 'VARCHAR'}, {'column_name': 'Pos', 'data_type': 'VARCHAR'}, {'column_name': 'Ref', 'data_type': 'VARCHAR'}, {'column_name': 'Alt', 'data_type': 'VARCHAR'}, {'column_name': 'AF', 'data_type': 'VARCHAR'}, {'column_name': 'VariantType', 'data_type': 'VARCHAR'}, {'column_name': 'VariantInfo', 'data_type': 'VARCHAR'}, {'column_name': 'DNAChange', 'data_type': 'VARCHAR'}, {'column_name': 'ProteinChange', 'data_type': 'VARCHAR'}, {'column_name': 'gene_name', 'data_type': 'VARCHAR'}, {'column_name': 'Transcript', 'data_type': 'VARCHAR'}, {'column_name': 'CCLEDeleterious', 'data_type': 'VARCHAR'}, {'column_name': 'CosmicHotspot', 'data_type': 'VARCHAR'}, {'column_name': 'AssociatedWith', 'data_type': 'VARCHAR'}, {'column_name': 'LoF', 'data_type': 'VARCHAR'}, {'column_name': 'Driver', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyDriver', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyGoF', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyLoF', 'data_type': 'VARCHAR'}, {'column_name': 'VariantAnnotation', 'data_type': 'VARCHAR'}, {'column_name': 'copy_number', 'data_type': 'VARCHAR'}]}. Try to answer the following question. Only the SQL should be returned within ''' SQL query '''. \"Which mutations in PAS domain for the target protein exhibit high selectivity for the prostate cancer cell lines?\"\n",
      "Response is: Error: name 'log' is not defined\n",
      "You are an exceptional computational biologist and genomics expert and know everything about drug discovery.. Please write the appropriate SQL query using these three different dbs, and their schemas are as {'ccle_cn_gistic in files/db/CCLEGisticCNDB.db': [{'column_name': 'Hugo_Symbol', 'data_type': 'VARCHAR'}, {'column_name': 'CCLE_Name', 'data_type': 'VARCHAR'}, {'column_name': 'gistic_cn', 'data_type': 'VARCHAR'}, {'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}], 'ccle_mutation in files/db/CCLEMutDB.db': [{'column_name': 'Hugo_Symbol', 'data_type': 'VARCHAR'}, {'column_name': 'Entrez_Gene_Id', 'data_type': 'VARCHAR'}, {'column_name': 'NCBI_Build', 'data_type': 'VARCHAR'}, {'column_name': 'Chromosome', 'data_type': 'VARCHAR'}, {'column_name': 'Start_position', 'data_type': 'VARCHAR'}, {'column_name': 'End_position', 'data_type': 'VARCHAR'}, {'column_name': 'Strand', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_Classification', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_Type', 'data_type': 'VARCHAR'}, {'column_name': 'Reference_Allele', 'data_type': 'VARCHAR'}, {'column_name': 'Alternate_Allele', 'data_type': 'VARCHAR'}, {'column_name': 'dbSNP_RS', 'data_type': 'VARCHAR'}, {'column_name': 'dbSNP_Val_Status', 'data_type': 'VARCHAR'}, {'column_name': 'Genome_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Annotation_Transcript', 'data_type': 'VARCHAR'}, {'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}, {'column_name': 'cDNA_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Codon_Change', 'data_type': 'VARCHAR'}, {'column_name': 'Protein_Change', 'data_type': 'VARCHAR'}, {'column_name': 'isDeleterious', 'data_type': 'VARCHAR'}, {'column_name': 'isTCGAhotspot', 'data_type': 'VARCHAR'}, {'column_name': 'TCGAhsCnt', 'data_type': 'VARCHAR'}, {'column_name': 'isCOSMIChotspot', 'data_type': 'VARCHAR'}, {'column_name': 'COSMIChsCnt', 'data_type': 'VARCHAR'}, {'column_name': 'ExAC_AF', 'data_type': 'VARCHAR'}, {'column_name': 'Variant_annotation', 'data_type': 'VARCHAR'}, {'column_name': 'CGA_WES_AC', 'data_type': 'VARCHAR'}, {'column_name': 'HC_AC', 'data_type': 'VARCHAR'}, {'column_name': 'RD_AC', 'data_type': 'VARCHAR'}, {'column_name': 'RNAseq_AC', 'data_type': 'VARCHAR'}, {'column_name': 'SangerWES_AC', 'data_type': 'VARCHAR'}, {'column_name': 'WGS_AC', 'data_type': 'VARCHAR'}], 'ccle_variants in files/db/CCLEVarDB.db': [{'column_name': 'DepMap_ID', 'data_type': 'VARCHAR'}, {'column_name': 'Chrom', 'data_type': 'VARCHAR'}, {'column_name': 'Pos', 'data_type': 'VARCHAR'}, {'column_name': 'Ref', 'data_type': 'VARCHAR'}, {'column_name': 'Alt', 'data_type': 'VARCHAR'}, {'column_name': 'AF', 'data_type': 'VARCHAR'}, {'column_name': 'VariantType', 'data_type': 'VARCHAR'}, {'column_name': 'VariantInfo', 'data_type': 'VARCHAR'}, {'column_name': 'DNAChange', 'data_type': 'VARCHAR'}, {'column_name': 'ProteinChange', 'data_type': 'VARCHAR'}, {'column_name': 'gene_name', 'data_type': 'VARCHAR'}, {'column_name': 'Transcript', 'data_type': 'VARCHAR'}, {'column_name': 'CCLEDeleterious', 'data_type': 'VARCHAR'}, {'column_name': 'CosmicHotspot', 'data_type': 'VARCHAR'}, {'column_name': 'AssociatedWith', 'data_type': 'VARCHAR'}, {'column_name': 'LoF', 'data_type': 'VARCHAR'}, {'column_name': 'Driver', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyDriver', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyGoF', 'data_type': 'VARCHAR'}, {'column_name': 'LikelyLoF', 'data_type': 'VARCHAR'}, {'column_name': 'VariantAnnotation', 'data_type': 'VARCHAR'}, {'column_name': 'copy_number', 'data_type': 'VARCHAR'}]}. Try to answer the following question. Only the SQL should be returned within ''' SQL query '''. \"Relationship between dependency scores of JAK1 in DepMapDB.\"\n",
      "Response is: Error: name 'log' is not defined\n"
     ]
    }
   ],
   "source": [
    "# LLM checks to write a SQL query\n",
    "\n",
    "import replicate\n",
    "import pandas as pd\n",
    "import json\n",
    "import os\n",
    "import config\n",
    "import time\n",
    "import importlib\n",
    "importlib.reload(config)\n",
    "from config import config, reset_config\n",
    "from dotenv import load_dotenv\n",
    "load_dotenv()\n",
    "from config import config\n",
    "config.set_mode(\"dbs\")\n",
    "\n",
    "folder_path = 'files'\n",
    "if not os.path.exists(folder_path):\n",
    "    os.makedirs(folder_path)\n",
    "\n",
    "INSTRUCTION = config.INSTRUCTION\n",
    "F_NAME = config.F_NAME\n",
    "\n",
    "def load_file(file_path):\n",
    "    with open(file_path, 'r') as file:\n",
    "        return json.load(file)\n",
    "\n",
    "db_desc = load_file(config.db_layout)\n",
    "all_schemas = load_file(config.db_schema)\n",
    "\n",
    "df = pd.read_excel(config.questions)\n",
    "df.to_excel(config.q_original, index=False)\n",
    "\n",
    "df['Question'] = df['Question'].str.strip()  # Removes leading/trailing whitespace\n",
    "\n",
    "# Check for duplicate questions\n",
    "duplicates = df.duplicated(subset=['Question'], keep=False)\n",
    "if duplicates.any():\n",
    "    print(\"Duplicates found. Removing duplicates.\")\n",
    "    df = df.drop_duplicates(subset=['Question'], keep='first')\n",
    "    df.to_excel(config.q_db, index=False)\n",
    "else:\n",
    "    print(\"No duplicates found.\")\n",
    "\n",
    "# DataFrame to store the results\n",
    "results_df = pd.DataFrame(columns=['Model', 'Question', 'Response', 'Latency'])\n",
    "\n",
    "models = {\n",
    "    # \"qwen-14b\": \"nomagick/qwen-14b-chat:f9e1ed25e2073f72ff9a3f46545d909b1078e674da543e791dec79218072ae70\",\n",
    "    # \"falcon-40b\": \"joehoover/falcon-40b-instruct:7d58d6bddc53c23fa451c403b2b5373b1e0fa094e4e0d1b98c3d02931aa07173\",\n",
    "    # \"yi-34b\": \"01-ai/yi-34b-chat:914692bbe8a8e2b91a4e44203e70d170c9c5ccc1359b283c84b0ec8d47819a46\",\n",
    "    \"mistral-7b\": \"mistralai/mistral-7b-instruct-v0.2:f5701ad84de5715051cb99d550539719f8a7fbcf65e0e62a3d1eb3f94720764e\",\n",
    "    \"llama2-70b\": \"meta/llama-2-70b-chat\",\n",
    "    # \"openhermes2\": \"antoinelyset/openhermes-2.5-mistral-7b:d7ccd25700fb11c1787c25b580ac8d715d2b677202fe54b77f9b4a1eb7d73e2b\",\n",
    "    \"mixtral-instruct\": \"mistralai/mixtral-8x7b-instruct-v0.1:2b56576fcfbe32fa0526897d8385dd3fb3d36ba6fd0dbe033c72886b81ade93e\",\n",
    "    \"deepseek_33bq\": \"kcaverly/deepseek-coder-33b-instruct-gguf:ea964345066a8868e43aca432f314822660b72e29cab6b4b904b779014fe58fd\",\n",
    "    }\n",
    "\n",
    "prompt_for_qwen=\"\"\"<|im_start|>system\\n {INSTRUCTION}. Please write the appropriate SQL query using these three tables. The schemas are as {all_schemas}. Try to answer the following question. The SQL should be returned within ''' SQL query '''. <|im_end|>\\n<|im_start|>user\\n{question}<|im_end|>\\n<|im_start|>assistant\\n\"\"\"\n",
    "prompt_for_hermes = \"\"\"[\n",
    "{{\n",
    "  \"role\": \"system\",\n",
    "  \"content\": \"{INSTRUCTION}. Please write the appropriate SQL query using these three tables. The schemas are as {all_schemas}. Try to answer the following question. The SQL should be returned within ''' SQL query ''' \" \n",
    "}},\n",
    "{{\n",
    "  \"role\": \"user\",\n",
    "  \"content\": {question}\n",
    "}}\n",
    "]\"\"\"\n",
    "\n",
    "# Iterate through each model\n",
    "for model_key, model_value in models.items():\n",
    "    responses = []\n",
    "\n",
    "    for index, row in df.iterrows():\n",
    "        qn = row['Question']\n",
    "        question = json.dumps(qn)\n",
    "\n",
    "        if model_key == \"yi-34b\":  # Yi model\n",
    "            prompt = prompt_for_qwen.format(INSTRUCTION=INSTRUCTION, all_schemas=all_schemas, question=question)\n",
    "        if model_key == \"qwen-14b\":  # Qwen model\n",
    "            prompt = prompt_for_qwen.format(INSTRUCTION=INSTRUCTION, all_schemas=all_schemas,question=question)\n",
    "        elif model_key == \"openhermes2\":  # Hermes model\n",
    "            prompt = prompt_for_hermes.format(INSTRUCTION=INSTRUCTION, all_schemas=all_schemas,question=question)\n",
    "        else:\n",
    "            prompt = f\"{INSTRUCTION}. Please write the appropriate SQL query using these three different dbs, and their schemas are as {all_schemas}. Try to answer the following question. Only the SQL should be returned within ''' SQL query '''. {question}\"\n",
    "\n",
    "        start_time = time.time()  # Record the start time\n",
    "\n",
    "        try:\n",
    "            print(prompt)\n",
    "            output = replicate.run(\n",
    "                model_value,\n",
    "                input={\n",
    "                  \"debug\": False,\n",
    "                #   \"top_k\": 50,\n",
    "                  \"top_p\": 0.9,\n",
    "                  \"prompt\": prompt,\n",
    "                  \"temperature\": 0.7,\n",
    "                  \"max_new_tokens\": 500,\n",
    "                  \"min_new_tokens\": -1\n",
    "                }\n",
    "            )\n",
    "            response = \"\"\n",
    "            response_parts = []  # Initialize an empty list to collect string representations\n",
    "            for item in output:\n",
    "                item_str = str(item).strip()  # Convert item to string\n",
    "                response += item_str # if len(item_str) == 1 else f\" {item_str}\"\n",
    "\n",
    "            response = response.strip()\n",
    "            extracted_sql = extract_sql(response) # Get only the SQL query\n",
    "            valid = is_sql_valid(response) # Check if the SQL query is valid\n",
    "            response = {\n",
    "                \"response\": extracted_sql,\n",
    "                \"is_valid\": valid\n",
    "            }\n",
    "\n",
    "        except Exception as e:\n",
    "            response = f\"Error: {e}\"\n",
    "        print(f\"Response is: {response}\")\n",
    "        end_time = time.time()  # Record the end time\n",
    "        latency = end_time - start_time  # Calculate latency\n",
    "\n",
    "        new_row = pd.DataFrame({'Model': [model_key], 'Question': [qn], 'Response': [extracted_sql], 'Latency': [latency]})\n",
    "        results_df = pd.concat([results_df, new_row], ignore_index=True)\n",
    "\n",
    "        if index % 20 == 0:  # Save every 10 questions, adjust as needed\n",
    "            results_df.to_excel(config.llmresults_file_path, index=False, sheet_name='Sheet1')\n",
    "\n",
    "results_df.to_excel(config.llmresults_file_path, index=False, sheet_name='Sheet1')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'id': 'asst_GS4XshQo2cJznJ4g5sLVkPom', 'created_at': 1707784537, 'description': None, 'file_ids': [], 'instructions': 'You are an exceptional computational biologist and genomics expert and know everything about drug discovery.', 'metadata': {}, 'model': 'gpt-3.5-turbo-1106', 'name': 'galen AI Evaluator via reading DB', 'object': 'assistant', 'tools': []}\n",
      "{'response': \"SQL\\nSELECT ccle_cn_gistic.CCLE_Name\\nFROM ccle_cn_gistic\\nWHERE ccle_cn_gistic.Hugo_Symbol = 'POLR3E' AND ccle_cn_gistic.gistic_cn = 'high'\\n\", 'is_valid': True}\n",
      "{'response': \"\\nSELECT \\n    ccle_mutation.Hugo_Symbol,\\n    ccle_mutation.DepMap_ID,\\n    ccle_variants.DepMap_ID,\\n    ccle_variants.gene_name,\\n    ccle_variants.VariantType,\\n    ccle_variants.copy_number\\nFROM \\n    ccle_mutation \\nJOIN \\n    ccle_variants \\nON \\n    ccle_mutation.DepMap_ID = ccle_variants.DepMap_ID\\nWHERE \\n    ccle_mutation.Hugo_Symbol = 'POLR3E'\\n    AND \\n    ccle_variants.VariantType = 'Indication'\\n    AND \\n    (ccle_variants.copy_number = 'overexpressed' OR ccle_variants.copy_number = 'underexpressed')\\n\", 'is_valid': True}\n",
      "{'response': \"SELECT ProteinChange, COUNT(ProteinChange) AS MutationFrequency\\nFROM ccle_variants\\nWHERE gene_name = 'POLR3E'\\nGROUP BY ProteinChange\\nORDER BY MutationFrequency DESC;\\n\", 'is_valid': True}\n",
      "{'response': \"SELECT COUNT(*) AS PathogenicMutations\\nFROM ccle_mutation\\nWHERE Hugo_Symbol = 'POLR3E' AND isDeleterious = 'Yes';\\n\", 'is_valid': True}\n",
      "{'response': \"SELECT ccle_mutation.Hugo_Symbol, ccle_mutation.DepMap_ID\\nFROM ccle_mutation\\nJOIN ccle_variants ON ccle_mutation.DepMap_ID = ccle_variants.DepMap_ID\\nWHERE ccle_mutation.Hugo_Symbol = 'PAS_domain' \\nAND ccle_variants.VariantType = 'Missense_Mutation' \\nAND ccle_variants.AF > 0.5;\\n\", 'is_valid': True}\n",
      "{'response': \"SELECT cgle_cn_gistic.CCLE_Name, cgle_cn_gistic.gistic_cn, cgle_mutation.isDeleterious, cgle_variants.copy_number\\nFROM cgle_cn_gistic\\nJOIN cgle_mutation ON cgle_cn_gistic.DepMap_ID = cgle_mutation.DepMap_ID\\nJOIN cgle_variants ON cgle_cn_gistic.DepMap_ID = cgle_variants.DepMap_ID\\nWHERE cgle_cn_gistic.Hugo_Symbol = 'JAK1'\\n\", 'is_valid': True}\n"
     ]
    }
   ],
   "source": [
    "# GPT-4 writes a SQL query\n",
    "\n",
    "import pandas as pd\n",
    "import json\n",
    "import openai\n",
    "import requests\n",
    "from openai import OpenAI\n",
    "import time\n",
    "from dotenv import load_dotenv\n",
    "load_dotenv()\n",
    "import os\n",
    "from config import config\n",
    "config.set_mode(\"dbs\")\n",
    "\n",
    "def load_file(file_path):\n",
    "    with open(file_path, 'r') as file:\n",
    "        return json.load(file)\n",
    "\n",
    "db_desc = load_file(config.db_layout)\n",
    "all_schemas = load_file(config.db_schema)\n",
    "\n",
    "folder_path = 'files'\n",
    "if not os.path.exists(folder_path):\n",
    "    os.makedirs(folder_path)\n",
    "\n",
    "INSTRUCTION = config.INSTRUCTION\n",
    "F_NAME = config.F_NAME\n",
    "GPT_MODEL = config.GPT_MODEL\n",
    "INPUT_CSV_PATH = config.questions\n",
    "OUTPUT_CSV_PATH = config.gpt4results_csv_path\n",
    "\n",
    "client = OpenAI()\n",
    "def show_json(obj):\n",
    "    print(json.loads(obj.model_dump_json()))\n",
    "\n",
    "assistant = client.beta.assistants.create(\n",
    "    name=f\"{F_NAME} AI Evaluator via reading DB\",\n",
    "    instructions=INSTRUCTION,\n",
    "    model=GPT_MODEL,\n",
    ")\n",
    "show_json(assistant)\n",
    "\n",
    "# Utility functions\n",
    "def read_csv(file_path):\n",
    "    return pd.read_excel(file_path)\n",
    "\n",
    "def process_data_for_gpt(data):\n",
    "    prompts = []\n",
    "    for _, row in data.iterrows():\n",
    "        question = row['Question']\n",
    "        prompt = f\"Please write the appropriate SQL query using these three table schemas {all_schemas} to answer the following question. The SQL should be returned within ''' SQL query '''.:\\n\\n{question}\"\n",
    "        prompts.append(prompt)\n",
    "    return prompts\n",
    "\n",
    "def submit_message_and_create_run(assistant_id, prompt):\n",
    "    thread = client.beta.threads.create() # If you replace this globally it appends all answers to the one before.\n",
    "    client.beta.threads.messages.create(thread_id=thread.id, role=\"user\", content=prompt)\n",
    "    return client.beta.threads.runs.create(thread_id=thread.id, assistant_id=assistant_id), thread\n",
    "\n",
    "def wait_on_run_and_get_response(run, thread):\n",
    "    while run.status == \"queued\" or run.status == \"in_progress\":\n",
    "        run = client.beta.threads.runs.retrieve(thread_id=thread.id, run_id=run.id)\n",
    "        time.sleep(0.5)\n",
    "    messages = client.beta.threads.messages.list(thread_id=thread.id, order=\"asc\")\n",
    "    return [m.content[0].text.value for m in messages if m.role == 'assistant']\n",
    "\n",
    "def create_output_csv(data, responses, latencies, model_name, interim_csv_path):\n",
    "    new_rows = []\n",
    "    for question, response, latency in zip(data['Question'], responses, latencies):\n",
    "        new_rows.append({'Model': model_name, 'Question': question, 'Response': response, 'Latency': latency})\n",
    "    new_data = pd.DataFrame(new_rows)\n",
    "    new_data.to_excel(interim_csv_path, index=False)\n",
    "\n",
    "data = read_csv(INPUT_CSV_PATH)\n",
    "prompts = process_data_for_gpt(data)\n",
    "ASSISTANT_ID = assistant.id\n",
    "\n",
    "responses = []\n",
    "latencies = []  # Initialize a list to store latencies\n",
    "\n",
    "for prompt in prompts:\n",
    "    start_time = time.time()  # Capture start time\n",
    "    run, thread = submit_message_and_create_run(ASSISTANT_ID, prompt)\n",
    "    response = wait_on_run_and_get_response(run, thread)\n",
    "    if isinstance(response, list):\n",
    "        response = ' '.join(map(str, response))\n",
    "    response = response.replace(\"\\\\\\\\n\", \"\\\\n\")\n",
    "    response = response.strip()\n",
    "    extracted_sql = extract_sql(response) # Get only the SQL query\n",
    "    valid = is_sql_valid(response) # Check if the SQL query is valid\n",
    "    response = {\n",
    "        \"response\": extracted_sql,\n",
    "        \"is_valid\": valid\n",
    "    }\n",
    "    print(response)\n",
    "    responses.append(response)\n",
    "    end_time = time.time()  # Capture end time\n",
    "    latency = end_time - start_time  # Calculate latency\n",
    "    latencies.append(latency)  # Store latency\n",
    "\n",
    "create_output_csv(data, responses, latencies, GPT_MODEL, OUTPUT_CSV_PATH)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "import re\n",
    "import pandas as pd\n",
    "from difflib import SequenceMatcher\n",
    "from config import config\n",
    "\n",
    "config.set_mode('dbs')\n",
    "\n",
    "def clean_text(text):\n",
    "    \"\"\"\n",
    "    Remove non-ASCII characters from the text.\n",
    "    \"\"\"\n",
    "    return ''.join(char for char in text if char.isascii())\n",
    "\n",
    "def create_combined_csv(original_csv_path, interim_csv_path, combined_csv_path):\n",
    "    # Read the original and interim data\n",
    "    original_data = pd.read_excel(original_csv_path) #, encoding='utf-8-sig'\n",
    "    interim_data = pd.read_excel(interim_csv_path)\n",
    "\n",
    "    # Combine the data\n",
    "    combined_data = pd.concat([original_data, interim_data], ignore_index=True)\n",
    "\n",
    "    # Save the combined data to a new CSV file\n",
    "    combined_data.to_excel(combined_csv_path, index=False)\n",
    "\n",
    "def merge_on_contains(big_df, small_df, big_col, small_col):\n",
    "    # Lowercase and strip whitespace for more effective matching\n",
    "    big_df[big_col] = big_df[big_col].str.lower().str.strip()\n",
    "    small_df[small_col] = small_df[small_col].str.lower().str.strip()\n",
    "\n",
    "    # Check if 'category' column exists in small_df\n",
    "    if 'category' in small_df.columns:\n",
    "        # Create a new column for the merged category in big_df\n",
    "        big_df['category'] = ''\n",
    "\n",
    "        # Iterate over the small dataframe and update the category in the big dataframe\n",
    "        for _, row in small_df.iterrows():\n",
    "            contains_mask = big_df[big_col].str.contains(row[small_col])\n",
    "            big_df.loc[contains_mask, 'category'] = row['category']\n",
    "    else:\n",
    "        # Handle the case when 'category' column does not exist\n",
    "        # For example, you can set a default category or leave it as it is\n",
    "        big_df['category'] = 'default_category'  # or any other handling logic\n",
    "\n",
    "    return big_df\n",
    "\n",
    "# Load the files\n",
    "questions_file_path = config.questions\n",
    "llmresults_file_path = config.llmresults_file_path\n",
    "gpt4results_csv_path = config.gpt4results_csv_path\n",
    "results_file_path = config.results_file_path\n",
    "\n",
    "create_combined_csv(llmresults_file_path, gpt4results_csv_path, results_file_path)\n",
    "\n",
    "# Reading the files\n",
    "questions_df = pd.read_excel(questions_file_path)\n",
    "results_df = pd.read_excel(results_file_path)\n",
    "\n",
    "# Ensure the total number of questions in results_grouped_by_model.xlsx is a multiple of the number in questions.xlsx\n",
    "if len(results_df) % len(questions_df) != 0:\n",
    "    print(len(results_df))\n",
    "    print(len(questions_df))\n",
    "    raise ValueError(\"The total number of questions in results_grouped_by_model.xlsx must be a multiple of the number in questions.xlsx.\")\n",
    "\n",
    "# Replace questions in results_grouped_df with those from questions_df\n",
    "num_repetitions = len(results_df) // len(questions_df)\n",
    "repeated_questions = pd.concat([questions_df['Question']] * num_repetitions, ignore_index=True)\n",
    "results_df['Question'] = repeated_questions\n",
    "\n",
    "# Save the modified DataFrame to a new Excel file\n",
    "results_df.to_excel(results_file_path, index=False)  # Replace with your desired path\n",
    "\n",
    "# Applying the merge_on_contains function\n",
    "merged_df = merge_on_contains(results_df, questions_df, 'Question', 'Question')\n",
    "\n",
    "pivoted_responses = merged_df.pivot_table(index=['Question'], columns='Model', values='Response', aggfunc=lambda x: ' | '.join(x))\n",
    "pivoted_latencies = merged_df.pivot_table(index=['Question'], columns='Model', values='Latency', aggfunc='mean')\n",
    "\n",
    "# Rename columns for clarity\n",
    "pivoted_responses.columns = [f'{col} Response' for col in pivoted_responses.columns]\n",
    "pivoted_latencies.columns = [f'{col} Latency' for col in pivoted_latencies.columns]\n",
    "\n",
    "# Combine the pivoted DataFrames\n",
    "pivoted_combined = pd.concat([pivoted_responses, pivoted_latencies], axis=1)\n",
    "\n",
    "# Ensure the combined DataFrame maintains the original order of questions\n",
    "pivoted_combined.reset_index(inplace=True)\n",
    "\n",
    "# Save the combined DataFrame to an Excel file\n",
    "pivoted_combined.to_excel(config.combined_file_path, index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Archive the intermediate files\n",
    "import os\n",
    "import glob\n",
    "from config import config\n",
    "directory = 'files/'\n",
    "archive_directory = os.path.join(directory, '#Archive')\n",
    "\n",
    "# Create the #Archive directory if it doesn't exist\n",
    "if not os.path.exists(archive_directory):\n",
    "    os.makedirs(archive_directory)\n",
    "\n",
    "# List all files that start with F_NAME and exclude the specified files\n",
    "files_to_move = [f for f in glob.glob(f\"{directory}/{config.F_NAME}_*\") \n",
    "                 if '_model_rankings' not in f and '_llmeval_results' not in f and 'questions' not in f and '_results_grouped_by_question_' not in f and '_allresults_grouped_by_model_' not in f]\n",
    "\n",
    "# Move the files to the #Archive folder\n",
    "for file in files_to_move:\n",
    "    os.rename(file, os.path.join(archive_directory, os.path.basename(file)))\n",
    "    print(f\"Moved file: {file} to {archive_directory}\")\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
