{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The context does not provide information about the number of oncogenes or the specific impacts on them.\n",
      "[Document(id_='31573f5a-bc18-4433-830a-5f3bbe317dce', embedding=None, metadata={'page_label': '443', 'file_name': 'Emerging roles of pseudokinases.pdf', 'file_path': 'files/Papers_FullText/Emerging roles of pseudokinases.pdf', 'file_type': 'application/pdf', 'file_size': 3322435, 'creation_date': '2024-01-19', 'last_modified_date': '2024-01-10', 'last_accessed_date': '2024-01-19'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, hash='f6c0bf11cedafca4ec4027776ee1b827da25b301121459f5f924b182dfd11477', text='Emerging roles of pseudokinases\\nJe´roˆme Boudeau1, Diego Miranda-Saavedra2, Geoffrey J. Barton2and Dario R. Alessi1\\n1MRC Protein Phosphorylation Unit, School of Life Sciences, University of Dundee, Dundee, UK, DD1 5EH\\n2Division of Biological Chemistry and Molecular Microbiology, University of Dundee, Dundee, UK, DD1 5EH\\nKinases control virtually all aspects of biology.\\nForty-eight human proteins have a kinase-like domainthat lacks at least one of the conserved catalytic residues;these proteins are therefore predicted to be inactive and\\nhave been termed pseudokinases. Here, we describe\\nexciting work suggesting that pseudokinases, despitelacking the ability to phosphorylate substrates, are stillpivotal in regulating diverse cellular processes. Wereview evidence that the pseudokinase STRAD controlsthe function of the tumour suppressor kinase LKB1 andthat a single amino acid substitution within the pseudo-kinase domain of the tyrosine kinase JAK2 leads to sev-eral malignant myeloproliferative disorders. We alsodiscuss the emerging functions of other pseudokinases,including HER3 (also called ErbB3), EphB6, CCK4 (also\\ncalled PTK7), KSR, Trb3, GCN2, TRRAP, ILK and CASK.\\nIntroduction\\nProtein kinases regulate the function of a large fraction of\\ncellular proteins by catalyzing the covalent attachment of\\nphosphate onto Ser, Thr and Tyr residues in target pro-teins. In recent years, genome-wide analyses have led tothe complete description of the protein kinase complementof several eukaryotic organisms, including human [1],\\nmouse [2],Caenorhabditis elegans [3],Dictyostelium [4]\\nand yeast [5,6]. These studies have revealed that 2–3% of\\nall eukaryotic genes encode proteins containing a kinasedomain. Unexpectedly, /C2410% of these proteins lack one or\\nmore of the conserved amino acids in the kinase domainthat are required for catalytic activity; they are therefore\\npredicted to be catalytically inactive. Sequence analysis of\\nthese ‘pseudokinases’ indicates that they lack at least oneof three motifs in the catalytic domain that are essential forcatalysis. The three motifs are: the Val-Ala-Ile-Lys (VAIK)motif in subdomain II of the kinase domain, in which thelysine residue interacts with the aandbphosphates of\\nATP, anchoring and orienting the ATP; the His-Arg-Asp(HRD) motif in subdomain VIb, in which the aspartic acidis the catalytic residue, functioning as a base acceptor toachieve proton transfer; and the Asp-Phe-Gly (DFG) motifin subdomain VII, in which the aspartic acid binds the\\nMg\\n2+ions that coordinate the bandgphosphates of ATP in\\nthe ATP-binding cleft.\\nAn inventory of human pseudokinases\\nOut of 518 protein kinases encoded by the human genome(the ‘kinome’), 48 have been classiﬁed as pseudokinases [1].The location of these proteins on the phylogenetic tree of\\nkinases is illustrated in Figure 1 [1] . Pseudokinases are\\nscattered throughout the distinct protein kinase subfami-lies, suggesting that they have evolved from diverse active\\nkinases. Twenty-eight pseudokinases have homologues in\\nmouse, worms, ﬂies and yeast that lack the equivalentcatalytic residues [1,2]. We have classiﬁed the human\\npseudokinases into seven groups (A to G) according towhich of the three motifs in their pseudokinase domainthey lack ( Table 1 ). The amino acid sequence of each\\npseudokinase and a description of missing conserved resi-dues are reported in the landmark study by Manning andcolleagues [1].\\nFigure 2 shows a comparison of the domain structures of\\nall human pseudokinases. Several, such as isoforms of\\nSTRAD, Trb, NRBP, SgK495, Slob, VRK3 and SuRTK106\\n(see Glossary), have a simple structure consisting essen-tially of only a pseudokinase domain. Others, such asisoforms of ANP, CASK, CCK4 (also called PTK7), EphB6,EphA10 and TRRAP, are part of much larger multi-domainproteins and are likely to have functions independent oftheir pseudokinase domain ( Figure 2 ). Intriguingly, the\\nfour Janus tyrosine kinases (JAK1, JAK2, JAK3 and Tyk2)and the serine/threonine kinase GCN2 have a pseudoki-nase domain and a functional kinase domain within thesame polypeptide; the implications of this particular struc-\\nture are discussed in more detail below.\\nNot all kinase domains that lack essential catalytic\\nresidues are inactive. For example, the four isoforms ofthe kinase WNK were originally classiﬁed as ’unusual’ asthey lacked the invariant catalytic lysine residue in theVAIK motif in subdomain II of the catalytic domain [7].\\nHowever, WNK1 is catalytically active and structuralanalysis of its catalytic domain demonstrates that a lysineresidue in subdomain I substitutes for the missing lysineresidue in subdomain II [8]. Similarly, the kinase PRPK,\\ndespite lacking the lysine residue equivalent to that miss-\\ning in the WNK isoforms, is catalytically active [9].I ti s\\npossible that a lysine residue in a AVIK motif that isconserved in all eukaryotic and archaeal PRPK homolo-gues (Lys60 in human PRPK) substitutes for the canonicallysine residue (G. Manning, personal communication).\\nHaspin was originally reported to lack the conserved\\nmagnesium-binding DFG motif in subdomain VII of thecatalytic domain, but recently another motif, Asp-Tyr-Thr-Leu-Ser (DYTLS), has been suggested to substitute for thiskey motif [10] and there is biochemical evidence that\\nhaspin might function as an active kinase [11,12] . There\\nhas also been some debate as to whether titin (TTN), a\\nprotein that has over 26 000 residues, is catalytically\\nReview TRENDS in Cell Biology Vol.16 No.9\\nCorresponding author: Alessi, D.R. ( d.r.alessi@dundee.ac.uk ).\\nAvailable online 1 August 2006.\\nwww.sciencedirect.com 0962-8924/$ – see front matter /C2232006 Elsevier Ltd. All rights reserved. doi: 10.1016/j.tcb.2006.07.003', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), Document(id_='be23c80f-3840-4aeb-9386-1a2d816d3e85', embedding=None, metadata={'page_label': '444', 'file_name': 'Emerging roles of pseudokinases.pdf', 'file_path': 'files/Papers_FullText/Emerging roles of pseudokinases.pdf', 'file_type': 'application/pdf', 'file_size': 3322435, 'creation_date': '2024-01-19', 'last_modified_date': '2024-01-10', 'last_accessed_date': '2024-01-19'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, hash='ee6cbbc7e80751a31ff94512fc133efb9d9016776dfd2dd37a1ee391b5ca3e1d', text='active. All vertebrate homologues of titin contain a\\nglutamic acid residue instead of an aspartic acid at theDFG motif in subdomain VII, whereas this residue is anaspartic acid in invertebrate titin-like kinases. The 3Dstructure of the catalytic domain of titin reveals that it\\nis an active kinase that is maintained in an auto-inhibited\\nconformation and is activated by phosphorylation of atyrosine residue in the kinase domain and throughinteraction with Ca\\n2+and calmodulin [13,14] .\\nSimilarly, there is uncertainty about whether the Wnt\\nreceptor tyrosine kinase Ryk, which regulates axon gui-dance, is a pseudokinase. Mammalian Ryk and its Droso-\\nphila homologue Derailed (Drl), rather than having a DFG\\nmotif in subdomain VII, have Asp-Asn-Ala (DNA; human)or Asp-Ser-Ala (DSA; Drosophila ). Mutation of the DNA\\nmotif in Ryk to DFG reportedly restores catalytic activity\\n[15]. Furthermore, mutation of the invariant lysine residue\\nin subdomain II of Drl does not affect its ability to driveaxon guidance in Drosophila , indicating that Ryk/Drl\\nkinases do not need intrinsic kinase activity to regulateaxon guidance [16]. As Ryk and Drl still have the crucial\\nAsp residue in subdomain VII, and thus have the potentialto phosphorylate substrates, we and others [1]have not\\nclassiﬁed them as pseudokinases.\\nIt should be noted that protein kinases are well known\\nto have functional roles that are independent of theirability to phosphorylate substrates. The human EGF\\nreceptor tyrosine kinase (pseudokinase) HER3 (also called\\nErbB3) and its homologues in dog and zebraﬁsh lack thecatalytic aspartic acid residue in their HRD motifs and areinactive. Curiously, mouse HER3 has an aspartic acidresidue in its HRD motif but is still inactive, suggestingthat it lacks other residues that are required for kinase\\nactivity [2]. Attempts to convert human HER3 to an active\\nkinase by reinstating the HRD motif and other potentiallylacking residues have failed to restore activity [17].\\nIt is possible that some of the 48 pseudokinases listed in\\nFigures 1 and 2 , many of which have not been studied,\\nmight turn out to have catalytic activity when analyzed.Caution should be taken when expressing a pseudokinasein mammalian cells and assessing whether it has catalyticactivity. As discussed below, several pseudokinases formcomplexes with active protein kinases that might beresponsible for any activity measured with pseudokinasespuriﬁed from eukaryotic cells. Here, we review the roles ofsome of the better studied pseudokinases.\\nPseudokinases similar to receptor tyrosine kinases\\nReceptor protein-tyrosine kinases (RTKs) are transmem-\\nbrane proteins made up of an ext racellular ligand-binding\\ndomain and an intracellular tyrosine kinase domain.RTKs dimerize upon ligand binding, which results inthe intermolecular autophosp horylation and activation\\nof their cytoplasmic domains. This, in turn, leads to therecruitment of phosphotyrosi ne-binding signalling com-\\nponents to the plasma membrane and triggers the activa-tion of signalling cascades that regulate diverse cellularprocesses, including proliferation, survival, apoptosis andmigration [18]. Gain-of-function mutations of several\\nRTKs are frequently observed in cancer [19].F o u rR T K s\\nare classiﬁed as pseudokinases: the EGF receptor familymember HER3, the ephrin receptors EphA10 and EphB6and the orphan receptor CCK4, which is overexpressed inseveral cancers.Glossary\\nPseudokinase abbreviations\\nANP : atrionatriuretic peptide receptor (guanylate cyclase)\\nCAK1 : Cyclin-dependent kinase-activating kinase 1\\nCASK : calcium/calmodulin-dependent serine kinase\\nCCK4 : colon carcinoma kinase 4\\nCYGF : guanylate cyclase 2D (GUCY2D)\\nCYGF : guanylate cyclase 2F (GUCY2F)\\nDrl: derailed\\nEph: erythropoietin-producing hepatocyte (Eph); ephrin receptor\\nErbB : v-erb-b2 erythroblastic leukemia viral oncogene homolog\\nGCN2 : general control non-derepressible 2\\nHER: human epidermal growth factor receptor\\nHSER : heat stable enterotoxin receptor (GUCY2C; guanylate cyclase 2C)\\nILK: integrin-linked kinase\\nIRAK : interleukin-1 receptor-associated kinase\\nJAK: Janus kinase\\nJH: JAK homology domains\\nJNK: c-Jun N-terminal kinase\\nKSR: kinase suppressor of Ras\\nMLKL : mixed lineage kinase-like\\nNRBP : nuclear receptor binding protein\\nOTK : off-track kinase\\nPIKK : phosphatidylinositol 3-kinase-related kinase\\nPRPK : p53-related protein kinase\\nPSKH : protein serine kinase H\\nPTK7 : protein tyrosine kinase 7\\nRSKL : ribosomal protein S6 kinase-like\\nRyk: receptor-like tyrosine kinase\\nSgK: sugen kinase\\nSTRAD : STE20-related adaptor protein\\nSuRTK106 : serine threonine tyrosine kinase 1 (STYK1)\\nTBCK : tubulin-binding cofactor kinase\\nTRRAP : transactivation-transformation domain-associated protein\\nTrb: Tribbles homologTTN: giant muscle filament protein titin\\nULK: UNC (uncoordinated)-51-like kinase\\nVACAMKL : vesicle-associated CaM kinase-like\\nVRK : vaccinia-related kinase\\nOther abbreviations\\nATF: activating transcription factor\\nATM : ataxia-telangiectasia mutated kinase\\nCDC : cell division cycle\\nCHOP : C/EBP homologous protein\\nCOP1 : constitutive photomorphogenic protein-1\\nDNA-PK : DNA-dependent protein kinase\\nEGFR : epidermal growth factor receptor\\nERK: extracellular signal-regulated kinase\\nFERM : band F ezrin-radixin-moesin homology domains\\nHEAT : Huntington, elongation factor 3, PR65/A, TOR domain\\nIFN-g: interferon- g\\nIL: interleukin\\nLIM: Lin-11, Isl-1, Mec-3 homology domains\\nMAPK : mitogen-activated protein kinase\\nMEK : MAPK/ERK kinase\\nMO25 : mouse protein 25\\nmTOR : mammalian target of rapamycin\\nPDZ: PSD-95 (post-synaptic density protein 95), discs large protein, ZO-1 (zonula\\noccludens protein 1) homology domains\\nPI3-K : phosphatidylinositol 3-kinase\\nPKC: protein kinase C\\nPTB: phosphotyrosine binding domain\\nRTK: receptor protein-tyrosine kinase\\nSAM : sterile alpha motif\\nSH: Src homology\\nSTAT : signal transducers and activators of transcription\\nWNK : with-no-K(Lys) kinase\\nZAP-70 : zeta-chain-associated protein kinase-70444 Review TRENDS in Cell Biology Vol.16 No.9\\nwww.sciencedirect.com', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), Document(id_='163f5d06-274a-4ab2-882f-5d08acaaa438', embedding=None, metadata={'page_label': '445', 'file_name': 'Emerging roles of pseudokinases.pdf', 'file_path': 'files/Papers_FullText/Emerging roles of pseudokinases.pdf', 'file_type': 'application/pdf', 'file_size': 3322435, 'creation_date': '2024-01-19', 'last_modified_date': '2024-01-10', 'last_accessed_date': '2024-01-19'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, hash='1c377b1d5c1831077fd1ba84f9de1cf8deb2a160d72fd9565d6dfa39fb7ee36e', text='Figure 1 . The location of the pseudokinases in the phylogenetic tree of human kinases generated by Manning et al. [1](http://www.kinase.com ). Pseudokinases are in red.Review TRENDS in Cell Biology Vol.16 No.9 445\\nwww.sciencedirect.com', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), Document(id_='2ba41e32-9c93-4b1e-b5ce-e9de51dd6aaa', embedding=None, metadata={'page_label': '446', 'file_name': 'Emerging roles of pseudokinases.pdf', 'file_path': 'files/Papers_FullText/Emerging roles of pseudokinases.pdf', 'file_type': 'application/pdf', 'file_size': 3322435, 'creation_date': '2024-01-19', 'last_modified_date': '2024-01-10', 'last_accessed_date': '2024-01-19'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, hash='63215e6af7270fca1c3b9dacfd4616a1ef7cbf983246f0fc066bd2a40090a3aa', text='HER3\\nThe pseudokinase HER3 is one of the four members of theEGF receptor family, which also includes the active tyr-osine kinases HER1 (also called EGFR or ErbB1), HER2(also called Neu or ErbB2) and HER4 (also called ErbB4;Figure 1 ). The interaction of EGF family RTKs with their\\nligands induces their clustering and dimerization, trigger-ing the activation of signalling cascades that promote cellproliferation and survival. Overexpression and activatingmutations of all four EGF receptors are found in human\\ncancers [20], and HER1 and HER2 are targets of approved\\nanticancer therapeutics [21]. Upon binding to its ligand\\nneuregulin, HER3 forms a heterodimeric complex with theactive kinase HER2. This results in the inactive HER3pseudokinase domain interacting with the functional tyr-osine kinase domain of HER2 and stimulating the autopho-sphorylation and activation of the HER2 tyrosine kinasedomain [22,23] (Figure 3 a). The neuregulin-induced inter-\\naction of HER3 with HER2 stimulates signalling cascades,such as the PI3-kinase and ERK signalling pathways, thattumour cells rely upon for their development, proliferationand survival [22]. Interestingly, although HER2 has a\\nfunctional kinase domain, it cannot interact with the\\nneuregulin ligand and is therefore dependent upon hetero-dimerization with the neuregulin-binding HER3 to becomeactivated. A recent study has elegantly shown that, in theabsence of ligand, HER2 is maintained in an inhibitedconformation that is converted into an active conformationfollowing ligand-induced heterodimerization with HER3through an allosteric mechanism, as summarized inFigure 3 a[24].\\nEphB6\\nErythropoietin-producing hepatocyte (Eph) kinases are\\nthe largest family of receptor tyrosine kinases. Theirligands, called ephrins, are located on the surfaces ofadjacent cells. These kinases guide cell pattern formationduring tissue and organ development [25]. Of the 14 mem-\\nbers of the Eph family, two (EphA10 and EphB6) arepredicted pseudokinases. Although little is known aboutEphA10, EphB6 has been shown not to have intrinsictyrosine kinase activity and to be predominantly expressedin T cells [26]. Mice lacking EphB6 develop normally but\\nhave defects in T cell function, such as compromisedproliferation and secretion of lymphokines (IL-2, IL-4\\nand IFN- g), thus indicating a role for EphB6 in T cell\\ndevelopment in the immune system [27]. EphB6, despite\\nlacking tyrosine kinase activity, can still trigger activationof the tyrosine kinase ZAP-70 upon ephrin binding [27].\\nMechanistically, it is not known how this occurs, and it hasnot been tested whether EphB6-induced phosphorylationof ZAP-70 requires the pseudokinase domain. By analogywith HER3, it is possible that EphB6 exerts its effects byforming heterodimers with other Eph receptor kinases,\\nsuch as EphB1, which has been reported to interact with\\nEphB6 [28], and/or other signalling components.\\nCCK4\\nThe pseudokinase CCK4 was originally identiﬁed as aprotein overexpressed in several cancer cell lines, includ-ing melanoma [29]and colon cancer [30]lines. It is made\\nup of a signal peptide, seven immunoglobulin domains, asingle transmembrane region and a C-terminal domainwith homology to tyrosine kinases. The ligand for CCK4has not yet been identiﬁed. Mouse embryos expressing atruncated form of CCK4 consisting of only the ﬁrst 114\\nresidues (which does not include the pseudokinase\\ndomain) die perinatally, with profound defects in neuraltube closure and orientation of the stereociliary bundles inthe inner ear, indicating that this pseudokinase isinvolved in the regulation of polarity within planes ofepithelial cells [31]. The speciﬁc role of the pseudokinase\\ndomain of CCK4 in this process has not been established.TheDrosophila homologue of CCK4, OTK, has been impli-\\ncated in plexin signalling during axon guidance [32];i ti s\\nnot known whether the mammalian homologue has asimilar role. To learn more about CCK4, it will be impor-\\ntant to deﬁne whether it interacts with other receptor\\ntyrosine kinases and/or sig nalling components. Detailed\\nsequence analysis of the CCK4 pseudokinase domain hasalso shown that it is highly conserved throughout evolu-tion and is predicted to lack activity in all speciesexamined, including organisms such as Hydra ,w h i c h\\noccupies a near-basal position in the metazoan radiation[33]. This indicates that, despite lacking activity and a\\nknown role, the precise pseudokinase domain structureand the nature of the residues in this domain are never-t h e l e s sl i k e l yt oh a v ec r u c i a lr o l e s .Table 1. Classiﬁcation of pseudokinases into seven groups based on the catalytic motifs that they lack\\nGroup A B C D E F G\\nMotifs missing DFG HRD DFG, HRD VAIK DFG, VAIK HRD, VAIK DFG, HRD, VAIK\\nPseudokinases CASK ANP a ILK KSR1 ULK4 GCN2 EphA10\\nCCK4 ANP b IRAK2 KSR2 RSKL1 EphB6\\nSgK223 CYGD MLKL RSKL2 NRBP1\\nSgK269 CYGF SgK307 NRBP2\\nSgK495 HER3 STRAD b SCYL1\\nSuRTK106 HSER SCYL2\\nTrb1 JAK1 SCYL3\\nTrb2 JAK2 SgK196\\nTrb3 JAK3 SgK424\\nPSKH2 Slob\\nSgK071 STRAD a\\nSgK396 TBCK\\nTYK2 TRRAP\\nVACAMKL VRK3446 Review TRENDS in Cell Biology Vol.16 No.9\\nwww.sciencedirect.com', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), Document(id_='87041ed8-ebed-4ac0-a203-3ca4499cdd81', embedding=None, metadata={'page_label': '447', 'file_name': 'Emerging roles of pseudokinases.pdf', 'file_path': 'files/Papers_FullText/Emerging roles of pseudokinases.pdf', 'file_type': 'application/pdf', 'file_size': 3322435, 'creation_date': '2024-01-19', 'last_modified_date': '2024-01-10', 'last_accessed_date': '2024-01-19'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, hash='7b60f1fb516d2fd3044a26914ad0b5e54eb73e132212156eba18b7767ae71aa0', text='Figure 2 . Comparison of the domain structures of human pseudokinases. The figure summarizes the results obtained by interrogating InterPro using default pa rameters\\n[70] with the full-length pseudokinase protein sequences. InterPro integrates the major databases of protein families, domains and functional site sig natures, such as\\nSUPERFAMILY, Pfam, PRINTS and PROSITE. Results were manually inspected and drawn to scale with gff2ps [71]. The major groups of accessory domains include catalytic\\nactivity (boxes); lipid-binding domains (down-arrows); domains that link to GTPase (wedges); RNA-binding domains (diamonds); targeting domains (right-arrows); protein–\\nprotein interaction domains (stars); and intracellular sensor domains (pairs of triangles).Review TRENDS in Cell Biology Vol.16 No.9 447\\nwww.sciencedirect.com', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), Document(id_='f1d6eb87-216e-4bd2-91bb-048a70046f86', embedding=None, metadata={'page_label': '448', 'file_name': 'Emerging roles of pseudokinases.pdf', 'file_path': 'files/Papers_FullText/Emerging roles of pseudokinases.pdf', 'file_type': 'application/pdf', 'file_size': 3322435, 'creation_date': '2024-01-19', 'last_modified_date': '2024-01-10', 'last_accessed_date': '2024-01-19'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, hash='ccbd5f6a414c0d420299b3f99ea2ed17a1e9bc4551dab927b128ad2dcaee0f7b', text='The roles of the pseudokinases STRAD, JAK and GCN2\\nin the regulation of the activity of functional kinasesA picture is emerging in which several pseudokinasesfunction similarly to HER3, binding directly to the kinasedomain of functional protein kinases and thereby control-ling their activity.STRAD aand STRAD b\\nThe pseudokinases STRAD aand STRAD b, also called\\nSTLK5 and STLK6, share a high degree of sequence simi-larity to the STE20 family of protein kinases. Theseenzymes were originally thought to be active upstreamkinases that triggered the activation of the p38 and JNK\\nFigure 3 . Schematic models for the functions of HER3, STRAD and KSR. (a)In the absence of ligand, the catalytically active receptor HER2 is maintained in an\\ninactive auto-inhibited monomeric conformation. Binding of the ligand neuregulin to HER3 induces formation of a heterodimeric complex between HER 3 and HER2,\\nin which the HER3 pseudokinase domain allosterically activates the functional tyrosine kinase domain of HER2. This results in HER2 phosphorylating itself as well as\\nHER3, leading to recruitment of proteins containing SH2 and PTB domains to the receptor complexes, which are subsequently phosphorylated by the acti vated\\nreceptor. This leads to the activation of downstream signalling pathways that stimulate cell proliferation and survival. (b)The catalytic activity of the tumour\\nsuppressor kinase LKB1 is increased over 100-fold following the interaction of its catalytic domain with the pseudokinase domains of STRAD aand STRAD b.\\nThe scaffolding proteins MO25 aand MO25 bstabilize the LKB1–STRAD interaction by interacting with the C-terminal residues of STRAD proteins and an\\nas-yet unidentified region of LKB1. (c)In quiescent cells, inactive GDP-bound Ras is located at the plasma membrane, while inactive MEK associated to the\\npseudokinase domain of KSR and inactive Raf are located in the cytoplasm. In response to extracellular signals, Ras binds to GTP, which induces transl ocation of\\nRaf to the plasma membrane where it becomes activated. Upon Ras activation, KSR also translocates to the plasma membrane via its cysteine-rich zinc-f inger\\ndomain, bringing inactive MEK into close proximity with its upstream activator Raf. MEK is consequently phosphorylated and activated by Raf. The mem brane\\nlocalization of KSR exposes MAPK docking sites, resulting in MAPK binding to KSR and being phosphorylated by activated MEK. KSR also serves as a substr ate\\nfor MAP kinases.448 Review TRENDS in Cell Biology Vol.16 No.9\\nwww.sciencedirect.com', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), Document(id_='7f5add02-de89-4d70-8f09-e7b71ea3a7f8', embedding=None, metadata={'page_label': '449', 'file_name': 'Emerging roles of pseudokinases.pdf', 'file_path': 'files/Papers_FullText/Emerging roles of pseudokinases.pdf', 'file_type': 'application/pdf', 'file_size': 3322435, 'creation_date': '2024-01-19', 'last_modified_date': '2024-01-10', 'last_accessed_date': '2024-01-19'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, hash='82c60e598706c4e326e6b520286fbba340172a5550c54b8995a609f3f3dc1576', text='MAP kinase pathways [34], before it was realized that they\\nlack the DFG motif and the aspartic acid residue within theHRD motif, and they were shown experimentally to lackcatalytic activity [35]. Recent work has shown that\\nSTRAD aand STRAD bform heterotrimeric complexes\\nwith isoforms of the scaffolding protein MO25 and the\\ntumour suppressor protein kinase LKB1, which regulatescell proliferation, polarity and cellular energy levels [36].\\nUnlike most kinases, LKB1 is not activated by phosphor-ylation of its activation loop by an upstream kinase, but isinstead activated upon binding to STRAD proteins [35,37] .\\nThe pseudokinase domain of the STRAD proteins bindsdirectly to the kinase domain of LKB1, enhancing itscatalytic activity over 100-fold, and also functions toanchor LKB1 in the cytosol [35,36] . The molecular mechan-\\nism by which the STRAD proteins activate LKB1 has not\\nyet been elucidated, but it is possible that the LKB1–\\nSTRAD interaction leads to a conformational change ofLKB1 that stabilizes it in an active conformation(Figure 3 b). Isoforms of MO25 interact with the C-terminal\\nresidues of STRAD and stabilize the interaction betweenLKB1 and the STRAD proteins [38]. Interestingly, several\\nsingle amino acid substitution mutants of LKB1 isolatedfrom human cancers have lost the ability to interact withSTRAD aand are unable to induce a G1 cell cycle arrest\\nwhen overexpressed in cells, emphasizing the importanceof the STRAD–LKB1 interaction [35,37] .\\nCuriously, STRAD ainteracts with ATP and ADP with\\nhigh afﬁnity (K\\nd/C2430–100 mM)[37].T h ef u n c t i o n a l\\nsigniﬁcance of this is unclear, as mutations that preventSTRAD from binding ATP do not affect its ability toactivate LKB1 or induce its cytoplasmic localization[37]. It has been speculated that the STRAD proteins\\nevolved from an active protein kinase that perhaps oncecontrolled LKB1 [37]. However, there is no evidence as yet\\nto support this idea, as all vertebrate and sea urchinhomologues of STRAD also lack catalytic residues.Moreover, attempts at restoring the catalytic activity of\\nSTRAD aby mutating residues back to those found in\\nactive kinases failed to reactivate STRAD a, suggesting\\nthat other residues are lacking in STRAD athat prevent it\\nfrom phosphorylating substrates [37].\\nJAKs\\nThe Janus tyrosine kinases (JAKs) are associated with cellsurface cytokine receptors and are activated upon bindingof cytokines to these receptors. Activated JAKs phosphor-ylate the cytoplasmic domain of the receptor, creatingrecruitment sites for signalling proteins, such as STATs\\n(signal transducers and activators of transcription). STATs\\nare phosphorylated by JAKs, which results in their dimer-ization and translocation to the nucleus where they reg-ulate target gene expression. The JAK isoforms (JAK1,JAK2, JAK3 and Tyk2) have a C-terminal functional cat-alytic domain (JH1) that is preceded by a pseudokinase(JH2) domain as well as an N-terminus comprising aFERM domain ( Figure 2 ). The JH2 pseudokinase domain\\nbinds directly to the JH1 catalytic domain and regulates itsactivity [39,40] .\\nMutagenesis and molecular modelling analysis suggest\\nthat one role of the JH2 domain is to keep the kinasedomain inactive in the absence of cytokines [41,42] .I n\\nsupport of this idea, analysis of Drosophila JAK identi-\\nﬁed a point mutation (E696K) within the JH2 pseudoki-nase domain that activated the enzyme [39]. Striking\\nrecent ﬁndings in humans have identiﬁed an amino acid\\nsubstitution (V617F) in the JH2 pseudokinase domain of\\nJAK2 in most patients with polycythemia vera and in asigniﬁcant number of cases of essential thrombocythemiaas well as chronic idiopathic myeloﬁbrosis [43].T h e s e\\nmalignant myeloproliferative disorders are characterizedby the clonal overproduction of relatively normally dif-ferentiated hematopoietic lineages. The V617F substitu-tion leads to constitutive activation of JAK2 anddownstream effector signalling pathways including theSTAT5 transcription pathway and the PI3-kinase andERK signalling networks, which induce inappropriate\\ncytokine-independent proliferation of cells [43].I n t e r e s t -\\ningly, the homologous mutations in JAK1 (V658F) andTyk2 (V678F) also lead to the constitutive activation ofthese kinases, suggesting a common mechanism by which\\nthe JH2 pseudokinase domains control the catalyticactivity of these enzymes [44]. By contrast, the\\nhomologous M592F mutation in JAK3 does not lead toactivation, suggesting that JAK3 is regulated differentlyfrom other JAK isoforms [44]. The molecular mechanism\\nby which mutations within the JH2 pseudokinase domainof JAKs activate these enzymes is unknown. It is possible\\nthat these mutations disrupt an auto-inhibitory\\ninteraction between the JH2 and JH1 domains, or inducean activating JH2–JH1 interaction, or this might evenconvert the JH2 domain into an active tyrosine kinasedomain. Despite intensive effort, no crystallographicstructure of the JH1–JH2 domains fragment of anyJAK isoform has been reported; this would be aprerequisite to deﬁning how the JH2 pseudokinasedomain exerts its inﬂuence on JAK catalytic activity.\\nGCN2\\nPhosphorylation of translation initiation factor 2- a\\n(eIF2- a) is an important mechanism for down-regulating\\nprotein synthesis in response to nutrient starvationconditions. The kinase GCN2 is one of the eIF2- akinases\\nthat are activated by nutrient deprivation. Like JAKs,GCN2 has a pseudokinase domain N-terminal to itscatalytic domain ( Figure 2 ). The pseudokinase domain\\nof GCN2 interacts with the catalytic domain and inhibitsits catalytic activity under non-starvation conditions [45].\\nThis interaction has also been proposed to have a role inactivating GCN2 under starvation conditions [46].\\nFurther work is required to deﬁne the mechanism by\\nwhich the interaction between the pseudokinase andactive kinase domains of GCN2 controls its catalyticactivity.\\nPseudokinases that have scaffolding roles\\nAnother emerging picture is that many pseudokinaseshave scaffolding roles and participate in the assembly ofmulti-protein complexes, as many pseudokinases, such asKSR, TRRAP, the Trb isoforms, ILK and CASK, containprotein–protein interaction domains in addition to theirpseudokinase domain.Review TRENDS in Cell Biology Vol.16 No.9 449\\nwww.sciencedirect.com', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), Document(id_='7e183511-e15c-4424-bd33-f931ab7d2030', embedding=None, metadata={'page_label': '450', 'file_name': 'Emerging roles of pseudokinases.pdf', 'file_path': 'files/Papers_FullText/Emerging roles of pseudokinases.pdf', 'file_type': 'application/pdf', 'file_size': 3322435, 'creation_date': '2024-01-19', 'last_modified_date': '2024-01-10', 'last_accessed_date': '2024-01-19'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, hash='d83d7db0bf5d0740297fd8d190c8150f44a3788abd594a9ef551b26c93f919a4', text='KSR\\nKSR was originally identiﬁed as a suppressor of anactivated Ras phenotype in Drosophila and C. elegans .\\nAlthough there has been some debate as to whetherKSR1 or the related protein KSR2 have an active catalytic\\ndomain, the bulk of the evidence indicates that they are\\ndevoid of activity [47,48] . KSR1 and KSR2 function as\\nscaffolding proteins, interacting with the Raf, MEK andERK protein kinases to form a membrane-localized reg-ulatory complex that coordinates signal propagationthrough the ERK pathway, which is crucial in Ras signaltransduction [47,48] (Figure 3 c). The pseudokinase\\ndomains of KSR1 and KSR2 are related to the catalyticdomain of Raf-1, which interacts with KSR1 and KSR2, butthe functional signiﬁcance of this and how the KSR pseu-dokinase domain exerts its inﬂuence on recruitment and\\nactivation of components of the ERK signalling pathway is\\nnot known.\\nTRRAP\\nA single member of the atypical protein kinase PIKKsubfamily, TRRAP, which is homologous to the DNA-damage repair kinases ATM and DNA-PK, has an inactiveC-terminal pseudokinase domain that lacks the catalyticresidues in its orthologues in all eukaryotic organisms [49].\\nLike other PIKK subfamily members, TRRAP is a largeprotein ( /C24400 kDa). TRRAP and its yeast homologue Tra1\\nfunction as essential components of most histone acetyl-\\ntransferase complexes, which have crucial roles in regulat-ing chromatin remodelling and gene expression [50].\\nTRRAP also associates with several transcription factors,including c-Myc and E2F1 [50], and controls repair of\\ndouble-strand DNA breaks [51]. Genetic disruption of\\nTRRAP in mice resulted in early embryonic lethalityand defects in cell cycle progression [52]. The importance\\nof the pseudokinase domain of TRRAP in regulating itsfunctions has not been addressed. It would be important todeﬁne whether the pseudokinase domain of TRRAP is\\nrequired for the regulation of chromatin remodelling and\\ngene expression, or whether it has other roles.\\nTrb3\\nThe three isoforms of Trb are short proteins that are themammalian orthologues of Drosophila Tribbles, a protein\\nthat inhibits mitosis early in development. Tribbles bindsto the CDC25 homolog String, a key regulator of celldivision and DNA damage repair, and promotes its ubiqui-tination and proteasome-mediated degradation [53]. All\\nthree mammalian Trb isoforms as well as Tribbles lack\\ncatalytic residues. Trb3 is highly expressed in some human\\nlung, colon and breast tumours [54], and it has been\\nsuggested to interact with numerous proteins, includingthe transcription factors CHOP and ATF4, thereby inhi-biting their transcriptional activity [55,56] . A recent study\\nhas shown that the expression of Trb3 is increased underfasting conditions in the adipose tissue, where it functionsto inhibit lipid synthesis by stimulating the degradation ofthe rate-limiting enzyme of fatty acid synthesis, acetyl-coenzyme A, through an interaction with the COP1 E3ubiquitin ligase [57]. Trb3 has also been reported to sup-\\npress insulin signal transduction by inhibiting the activityof the protein kinase Akt [58,59] . However, a recent study\\nhas disputed the role of Trb3 in regulating Akt [60].\\nDespite this analysis, no clear molecular picture of howTrb isoforms work or function as scaffolds has emerged.\\nILK\\nThe Integrin-linked (pseudo)kinase (ILK) was ﬁrst identi-ﬁed in a yeast two-hybrid screen as a protein that inter-acted with the cytoplasmic tail of b1-integrin. Although\\nILK lacks the HRD and DFG motifs in its pseudokinasedomain, there has been considerable debate as to whetherit has catalytic activity, following a report that it couldphosphorylate Akt at one of its activating residues (Ser473)[61]. However, the phosphorylation of Akt at Ser473 is now\\ngenerally accepted to be regulated by the protein kinasemTOR [62], rather than ILK. Consistent with this, phos-\\nphorylation of Akt at Ser473 is unaffected in ILK-deﬁcient\\nﬁbroblasts [63] and chondrocytes [64]. Supporting the\\nnotion that ILK is a pseudokinase, we have also not beenable to demonstrate that ILK can phosphorylate any sub-strate tested, including Akt or itself (D.R.A., unpublished).The N-terminal domain of ILK contains ankyrin repeatsthat interact with LIM domain-containing adaptor pro-teins of the PINCH family. ILK also interacts throughits C-terminal region with parvin to form a ternary com-plex termed IPP. The assembly of the IPP complex hasbeen implicated in the control of many aspects of cell\\nbehaviour and morphology, including actin-cytoskeleton\\ndynamics, cell adhesion, spreading, migration, polarityand cell–cell contact (reviewed in [65]). IPP exerts its\\neffects by forming a scaffolding platform, linking integrinsat the surface of cells with the actin cytoskeleton and othersignalling pathways. The precise role of the pseudokinasedomain of ILK in regulating these functions has not beendeﬁned. Loss of ILK in mice results in early embryoniclethality (at embryonic day E5.5–E6.5) [63].\\nCASK\\nThe pseudokinase CASK has an N-terminal pseudokinase\\ndomain with similarity to the calcium/calmodulin-depen-dent protein kinase domains, an inactive C-terminal gua-nylate kinase domain, as well as several PDZ domains andtwo SH3 domains. CASK forms complexes with proteinsregulating presynaptic and postsynaptic processes [66,67] .\\nConsistent with this, genetic studies indicated thatDrosophila CAKI (or CMG), a homologue of human CASK,\\nhas an essential role in controlling the release of neuro-transmitter vesicles at synapses [68]. CASK also interacts\\nwith Tbr-1, a T-box transcription factor that is involved in\\nforebrain development [69].\\nConclusions and perspectives\\nAlthough much remains to be learnt about the intricatefunctions of pseudokinases, the initial studies outlinedhere suggest that, despite lacking catalytic activity, theseproteins are not uninteresting and are likely to haveimportant roles in regulating diverse fundamental pro-cesses relevant to understanding human disease. To ourknowledge, no 3D structure of any pseudokinase has beenreported, so in future it will be important to attemptstructural analysis of these proteins. This will be450 Review TRENDS in Cell Biology Vol.16 No.9\\nwww.sciencedirect.com', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), Document(id_='b55fffaa-207d-4b50-9095-a395cae800be', embedding=None, metadata={'page_label': '451', 'file_name': 'Emerging roles of pseudokinases.pdf', 'file_path': 'files/Papers_FullText/Emerging roles of pseudokinases.pdf', 'file_type': 'application/pdf', 'file_size': 3322435, 'creation_date': '2024-01-19', 'last_modified_date': '2024-01-10', 'last_accessed_date': '2024-01-19'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, hash='57a6928ba05395954f2048db66a5a33fd0a292ba3b7bb330491b16d94bae34f5', text='particularly crucial for deﬁning the mechanism by which\\nthe HER3, STRAD, JAKs and GCN2 pseudokinasedomains regulate the activity of functional protein kinasedomains. It will also be important to undertake systematicscreens to identify protein-binding partners of pseudoki-\\nnases, as a key role of some of these proteins is clearly to\\nparticipate in multi-protein complexes. It would be fasci-nating to understand why pseudokinases evolved andwhether it would be possible to introduce mutations thatrestore catalytic activity to pseudokinases to explore theirancestral roles. It would also be interesting to undertakemore detailed sequence analysis of homologues of pseudo-kinases in primitive eukaryotic species, to determinewhether any of these proteins evolved from active proteinkinases.\\nAcknowledgements\\nWe thank Gerard Manning for his thoughtful comments and helpful\\nsuggestions. D.M.S. is a recipient of a 4-year Wellcome TrustStudentship. We thank the Association for International CancerResearch, Diabetes UK, the Medical Research Council, the MoffatCharitable Trust and the pharmaceutical companies supporting theDivision of Signal Transduction Therapy Unit (AstraZeneca, Boehringer-Ingelheim, GlaxoSmithKline, Merck and Co. Inc, Merck KgaA and Pﬁzer)for ﬁnancial support.\\nReferences\\n1 Manning, G. et al. (2002) The protein kinase complement of the human\\ngenome. Science 298, 1912–1934\\n2 Caenepeel, S. et al. (2004) The mouse kinome: discovery and\\ncomparative genomics of all mouse protein kinases. Proc. Natl.\\nAcad. Sci. U. S. A. 101, 11707–11712\\n3 Plowman, G.D. et al. (1999) The protein kinases of Caenorhabditis\\nelegans : a model for signal transduction in multicellular organisms.\\nProc. Natl. Acad. Sci. U. S. A. 96, 13603–13610\\n4 Goldberg, J.M. et al. (2006) The dictyostelium kinome–analysis of the\\nprotein kinases from a simple model organism. PLoS Genet. 2, e38\\n5 Hunter, T. and Plowman, G.D. (1997) The protein kinases of budding\\nyeast: six score and more. Trends Biochem. Sci. 22, 18–22\\n6 Manning, G. et al. (2002) Evolution of protein kinase signaling from\\nyeast to man. Trends Biochem. Sci. 27, 514–520\\n7 Xu, B. et al. (2000) WNK1, a novel mammalian serine/threonine protein\\nkinase lacking the catalytic lysine in subdomain II. J. Biol. Chem. 275,\\n16795–16801\\n8 Min, X. et al. (2004) Crystal structure of the kinase domain of WNK1, a\\nkinase that causes a hereditary form of hypertension. Structure 12,\\n1303–1311\\n9 Abe, Y. et al. (2001) Cloning and characterization of a p53-related\\nprotein kinase expressed in interleukin-2-activated cytotoxic T-cells,\\nepithelial tumor cell lines, and the testes. J. Biol. Chem. 276, 44003–\\n44011\\n10 Higgins, J.M. (2001) Haspin-like proteins: a new family of\\nevolutionarily conserved putative eukaryotic protein kinases.Protein Sci. 10, 1677–1684\\n11 Tanaka, H. et al. (1999) Identiﬁcation and characterization of a haploid\\ngerm cell-speciﬁc nuclear protein kinase (Haspin) in spermatid nucleiand its effects on somatic cells. J. Biol. Chem. 274, 17049–17057\\n12 Tanaka, H. et al. (2001) Cloning and characterization of human haspin\\ngene encoding haploid germ cell-speciﬁc nuclear protein kinase. Mol.\\nHum. Reprod. 7, 211–218\\n13 Mayans, O. et al. (1998) Structural basis for activation of the titin\\nkinase domain during myoﬁbrillogenesis. Nature 395, 863–869\\n14 Scheeff, E.D. and Bourne, P.E. (2005) Structural evolution of the\\nprotein kinase-like superfamily. PLoS Comput. Biol. 1, e49\\n15 Katso, R.M. et al. (1999) Functional analysis of H-Ryk, an atypical\\nmember of the receptor tyrosine kinase family. Mol. Cell. Biol. 19,\\n6427–6440\\n16 Yoshikawa, S. et al. (2001) The derailed guidance receptor does not\\nrequire kinase activity in vivo .J. Neurosci. 21, RC11917 Prigent, S.A. and Gullick, W.J. (1994) Identiﬁcation of c-erbB-3 binding\\nsites for phosphatidylinositol 30-kinase and SHC using an EGF\\nreceptor/c-erbB-3 chimera. EMBO J. 13, 2831–2841\\n18 Schlessinger, J. (2000) Cell signaling by receptor tyrosine kinases. Cell\\n103, 211–225\\n19 van der Geer, P. et al. (1994) Receptor protein-tyrosine kinases and\\ntheir signal transduction pathways. Annu. Rev. Cell Biol. 10, 251–337\\n20 Yarden, Y. and Sliwkowski, M.X. (2001) Untangling the ErbB\\nsignalling network. Nat. Rev. Mol. Cell Biol. 2, 127–137\\n21 Hynes, N.E. and Lane, H.A. (2005) ERBB receptors and cancer: the\\ncomplexity of targeted inhibitors. Nat. Rev. Cancer 5, 341–354\\n22 Holbro, T. et al. (2003) The ErbB2/ErbB3 heterodimer functions as an\\noncogenic unit: ErbB2 requires ErbB3 to drive breast tumor cellproliferation. Proc. Natl. Acad. Sci. U. S. A. 100, 8933–8938\\n23 Berger, M.B. et al. (2004) ErbB3/HER3 does not homodimerize upon\\nneuregulin binding at the cell surface. FEBS Lett. 569, 332–336\\n24 Zhang, X. et al. (2006) An allosteric mechanism for activation of the\\nkinase domain of epidermal growth factor receptor. Cell125, 1137–1149\\n25 Gerlai, R. (2001) Eph receptors and neural plasticity. Nat. Rev.\\nNeurosci. 2, 205–209\\n26 Gurniak, C.B. and Berg, L.J. (1996) A new member of the Eph family of\\nreceptors that lacks protein tyrosine kinase activity. Oncogene 13, 777–\\n786\\n27 Luo, H. et al. (2004) EphB6-null mutation results in compromised T cell\\nfunction. J. Clin. Invest. 114, 1762–1773\\n28 Freywald, A. et al. (2002) The kinase-null EphB6 receptor undergoes\\ntransphosphorylation in a complex with EphB1. J. Biol. Chem. 277,\\n3823–3828\\n29 Easty, D.J. et al. (1997) Loss of expression of receptor tyrosine kinase\\nfamily genes PTK7 and SEK in metastatic melanoma. Int. J. Cancer 71,\\n1061–1065\\n30 Shaheen, R.M. et al. (2001) Inhibited growth of colon cancer\\ncarcinomatosis by antibodies to vascular endothelial and epidermalgrowth factor receptors. Br. J. Cancer 85, 584–589\\n31 Lu, X. et al. (2004) PTK7/CCK-4 is a novel regulator of planar cell\\npolarity in vertebrates. Nature 430, 93–98\\n32 Winberg, M.L. et al. (2001) The transmembrane protein Off-track\\nassociates with Plexins and functions downstream of Semaphorinsignaling during axon guidance. Neuron 32, 53–62\\n33 Kroiher, M. et al. (2001) Deceiving appearances: signaling by ‘‘dead’’\\nand ‘‘fractured’’ receptor protein-tyrosine kinases. Bioessays 23, 69–\\n76\\n34 Dan, I. et al. (2001) The Ste20 group kinases as regulators of MAP\\nkinase cascades. Trends Cell Biol. 11, 220–230\\n35 Baas, A.F. et al.\\n(2003) Activation of the tumour suppressor kinase\\nLKB1 by the STE20-like pseudokinase STRAD. EMBO J. 22, 3062–\\n3072\\n36 Alessi, D.R. et al. (2006) LKB1-dependent signaling pathways. Annu.\\nRev. Biochem. 75, 137–163\\n37 Boudeau, J. et al. (2004) Analysis of the LKB1-STRAD-MO25 complex.\\nJ. Cell Sci. 117, 6365–6375\\n38 Boudeau, J. et al. (2003) MO25alpha/beta interact with STRADalpha/\\nbeta enhancing their ability to bind, activate and localize LKB1 in thecytoplasm. EMBO J. 22, 5102–5114\\n39 Luo, H. et al. (1997) Mutation in the Jak kinase JH2 domain\\nhyperactivates Drosophila and mammalian Jak-Stat pathways. Mol.\\nCell. Biol. 17, 1562–1571\\n40 Saharinen, P. et al. (2003) Autoinhibition of Jak2 tyrosine kinase is\\ndependent on speciﬁc regions in its pseudokinase domain. Mol. Biol.\\nCell 14, 1448–1459\\n41 Yeh, T.C. et al. (2000) A dual role for the kinase-like domain of the\\ntyrosine kinase Tyk2 in interferon-alpha signaling. Proc. Natl. Acad.\\nSci. U. S. A. 97, 8991–8996\\n42 Lindauer, K. et al. (2001) Prediction of the structure of human Janus\\nkinase 2 (JAK2) comprising the two carboxy-terminal domains revealsa mechanism for autoregulation. Protein Eng. 14, 27–37\\n43 Shannon, K. and Van Etten, R.A. (2005) JAKing up hematopoietic\\nproliferation. Cancer Cell 7, 291–293\\n44 Staerk, J. et al. (2005) JAK1 and Tyk2 activation by the homologous\\npolycythemia vera JAK2 V617F mutation: cross-talk with IGF1receptor. J. Biol. Chem. 280, 41893–41899\\n45 Qiu, H. et al. (1998) Dimerization by translation initiation factor 2\\nkinase GCN2 is mediated by interactions in the C-terminal ribosome-Review TRENDS in Cell Biology Vol.16 No.9 451\\nwww.sciencedirect.com', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), Document(id_='24023b55-c1b7-4aab-af0b-2c2a9669a50c', embedding=None, metadata={'page_label': '452', 'file_name': 'Emerging roles of pseudokinases.pdf', 'file_path': 'files/Papers_FullText/Emerging roles of pseudokinases.pdf', 'file_type': 'application/pdf', 'file_size': 3322435, 'creation_date': '2024-01-19', 'last_modified_date': '2024-01-10', 'last_accessed_date': '2024-01-19'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, hash='fc04c7c0fd0267164541924b7cd4211f72481f88d1d3d4de1c8645f38ac7ca07', text='binding region and the protein kinase domain. Mol. Cell. Biol. 18,\\n2697–2711\\n46 Wek, R.C. et al. (1990) Identiﬁcation of positive-acting domains in\\nGCN2 protein kinase required for translational activation of GCN4expression. Mol. Cell. Biol. 10, 2820–2831\\n47 Kolch, W. (2005) Coordinating ERK/MAPK signalling through\\nscaffolds and inhibitors. Nat. Rev. Mol. Cell Biol. 6, 827–837\\n48 Morrison, D.K. (2001) KSR: a MAPK scaffold of the Ras pathway?\\nJ. Cell Sci. 114, 1609–1612\\n49 Templeton, G.W. and Moorhead, G.B. (2005) The phosphoinositide-\\n3-OH-kinase-related kinases of Arabidopsis thaliana .EMBO Rep. 6,\\n723–728\\n50 Herceg, Z. and Wang, Z.Q. (2005) Rendez-vous at mitosis: TRRAPed in\\nthe chromatin. Cell Cycle 4, 383–387\\n51 Murr, R. et al. (2006) Histone acetylation by Trrap-Tip60 modulates\\nloading of repair proteins and repair of DNA double-strand breaks.Nat. Cell Biol. 8, 91–99\\n52 Herceg, Z. et al. (2001) Disruption of Trrap causes early embryonic\\nlethality and defects in cell cycle progression. Nat. Genet. 29, 206–211\\n53 Rorth, P. et al. (2000) The level of C/EBP protein is critical for cell\\nmigration during Drosophila oogenesis and is tightly controlled by\\nregulated degradation. Mol. Cell 6, 23–30\\n54 Bowers, A.J. et al. (2003) SKIP3, a novel Drosophila tribbles ortholog, is\\noverexpressed in human tumors and is regulated by hypoxia. Oncogene\\n22, 2823–2835\\n55 Ohoka, N. et al. (2005) TRB3, a novel ER stress-inducible gene, is\\ninduced via ATF4-CHOP pathway and is involvedin cell death. EMBO J.\\n24, 1243–1255\\n56 Ord, D. and Ord, T. (2005) Characterization of human NIPK (TRB3,\\nSKIP3) gene activation in stressful conditions. Biochem. Biophys. Res.\\nCommun. 330, 210–218\\n57 Qi, L. et al. (2006) TRB3 links the E3 ubiquitin ligase COP1 to lipid\\nmetabolism. Science 312, 1763–1766\\n58 Du, K. et al. (2003) TRB3: a tribbles homolog that inhibits Akt/PKB\\nactivation by insulin in liver. Science 300, 1574–157759 Koo, S.H. et al. (2004) PGC-1 promotes insulin resistance in liver\\nthrough PPAR-alpha-dependent induction of TRB-3. Nat. Med. 10,\\n530–534\\n60 Iynedjian, P.B. (2005) Lack of evidence for a role of TRB3/NIPK as an\\ninhibitor of PKB-mediated insulin signalling in primary hepatocytes.Biochem. J. 386, 113–118\\n61 Delcommenne, M. et al. (1998) Phosphoinositide-3-OH kinase-\\ndependent regulation of glycogen synthase kinase 3 and proteinkinase B/AKT by the integrin-linked kinase. Proc. Natl. Acad. Sci.\\nU. S. A. 95, 11211–11216\\n62 Sarbassov, D.D. et al. (2004) Rictor, a novel binding partner of mTOR,\\ndeﬁnes a rapamycin-insensitive and raptor-independent pathway thatregulates the cytoskeleton. Curr. Biol.\\n14, 1296–1302\\n63 Sakai, T. et al. (2003) Integrin-linked kinase (ILK) is required for\\npolarizing the epiblast, cell adhesion, and controlling actinaccumulation. Genes Dev. 17, 926–940\\n64 Grashoff, C. et al. (2003) Integrin-linked kinase regulates chondrocyte\\nshape and proliferation. EMBO Rep. 4, 432–438\\n65 Legate, K.R. et al. (2006) ILK, PINCH and parvin: the tIPP of integrin\\nsignalling. Nat. Rev. Mol. Cell Biol. 7, 20–31\\n66 Li, Y. et al. (2002) Structural basis for nucleotide-dependent regulation\\nof membrane-associated guanylate kinase-like domains. J. Biol. Chem.\\n277, 4159–4165\\n67 Tabuchi, K. et al. (2002) CASK participates in alternative tripartite\\ncomplexes in which Mint 1 competes for binding with caskin 1, a novelCASK-binding protein. J. Neurosci. 22, 4264–4273\\n68 Zordan, M.A. et al. (2005) Drosophila CAKI/CMG protein, a homolog of\\nhuman CASK, is essential for regulation of neurotransmitter vesicle\\nrelease. J. Neurophysiol. 94, 1074–1083\\n69 Wang, T.F. et al. (2004) Identiﬁcation of Tbr-1/CASK complex target\\ngenes in neurons. J. Neurochem. 91, 1483–1492\\n70 Mulder, N.J. et al. (2005) InterPro, progress and status in 2005. Nucleic\\nAcids Res. 33, D201–D205\\n71 Abril, J.F. and Guigo, R. (2000) gff2ps: visualizing genomic\\nannotations. Bioinformatics 16, 743–744\\nAGORA initiative provides free agriculture journals to\\ndeveloping countries\\nThe Health Internetwork Access to Research Initiative (HINARI) of the WHO has launched a new\\ncommunity scheme with the UN Food and Agriculture Organization.\\nAs part of this enterprise, Elsevier has given hundreds of journals to Access to Global Online\\nResearch in Agriculture (AGORA). More than 100 institutions are now registered for the scheme,\\nwhich aims to provide developing countries with free access to vital research that will ultimately\\nhelp increase crop yields and encourage agricultural self-sufficiency.\\nAccording to the Africa University in Zimbabwe, AGORA has been welcomed by both students\\nand staff. ‘‘It has brought a wealth of information to our fingertips’’, says Vimbai Hungwe. ‘‘The\\ninformation made available goes a long way in helping the learning, teaching and research\\nactivities within the University. Given the economic hardships we are going through, it couldn’t\\nhave come at a better time.’’\\nFor more information, visit www.aginternetwork.org452 Review TRENDS in Cell Biology Vol.16 No.9\\nwww.sciencedirect.com', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), Document(id_='4c98d159-81c7-4204-ba07-76e855ccd2e4', embedding=None, metadata={'page_label': '1', 'file_name': 'Potential of the Synthetic Lethality Principle.pdf', 'file_path': 'files/Papers_FullText/Potential of the Synthetic Lethality Principle.pdf', 'file_type': 'application/pdf', 'file_size': 342006, 'creation_date': '2024-01-10', 'last_modified_date': '2024-01-10', 'last_accessed_date': '2024-01-19'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, hash='0679fe7bd3cffa3a2c8cf801c634edad5d858ffb09cd11c61356ee2b0600d179', text='www.sciencemag.org    SCIENCE    VOL 342    15 NOVEMBER 2013 809 PERSPECTIVES10.1126/science.1246519\\nover the East African continent. In a nearby de\\nep-sea sediment core off the Somalian coast, \\nHolocene aridiﬁ  cation over the course of\\n 2000 years was recorded from the radio-genic isotope \\nsignature of wind-transported dus\\nt ( 11). Tierney and deMenocal instead m\\neasured the deuterium-to-hydrogen ratio (D\\n/H) of the leaf waxes; this ratio is directly link\\ned to the isotopic composition of precip-itation \\nand hence to total rainfall.B\\ny comparing their Gulf of Aden r\\necord with published records, Tierney and deM\\nenocal conclude that the African Humid P\\neriod ended abruptly within a few centu-ries \\nand was synchronous in the western and easter\\nn part of Africa. By considering mod-e\\nrn climate observations and model simula-tions, \\nthey propose that East African rainfall responded \\nin a nonlinear way to surface tem-peratures in \\nthe Indian Ocean.T\\nierney and deMenocal’ s Gulf of Aden r\\necord provides key information for under-s\\ntanding North African climate during the H\\nolocene. However, much research is still ne\\neded to build a comprehensive view of h\\nydroclimatic changes during the African H\\numid Period. Most published records are ba\\nsed on proxies that are difficult to link unequivocally to rainfall changes. For exam-ple\\n, records of wind-transported material r\\neﬂ  ect both the decrease in vegetation cover a\\nnd the increase in the dust source area caused b\\ny widespread lake desiccation. Document-i\\nng these parameters requires information f\\nrom multiple proxies measured in the same c\\nontinental or marine archives. The time res-ol\\nution of records must also be improved by i\\nncreasing the sampling rate for these prox-i\\nes. Decadal-to-seasonal proﬁ  les measured w\\nith geochemical scanners ( 12) allow study-i\\nng African Humid Period transitions without r\\nelying on statistical corrections.M\\nore continuous records are needed f\\nrom north of the Gulf of Aden, along the N\\nile Valley and the Red Sea. Obtaining new data \\nfrom central North Africa is also crucial f\\nor backing up observations made in sedi-m\\nents of the small Lake Y oa ( 5,  12). Of par-t\\nicular importance is the Holocene history o\\nf Lake Chad, which was at least 10 times a\\ns large during the African Humid Period t\\nhan it is today ( 13). Pollen data from Lake C\\nhad ( 14) indicate that vegetation changes oc\\ncurred progressively over about two mil-l\\nennia, but that century-scale variability was s\\nuperimposed on the mid-Holocene dry-ing trend. Additional proxies remain to be m\\neasured and longer continuous cores to b\\ne collected and studied from Lake Chad t\\no advance understanding of the African Hu\\nmid Period.  Refer\\nences \\n1. J. E. Tierney, P. B. deMenocal, Science 342, 843 (2013); 10.1126/science.1240411\\n.   \\n2. P. B. deMenocal et al., Quat. Sci. Rev. 19, 347 (2000).   \\n3. D. McGee, P. B. deMenocal, G. Winckler, J. B. W. Stuut, L\\n. I. Bradtmiller, Earth Planet. Sci. Lett. 371-372, 163 (2013\\n).   \\n4. Y. Garcin, D. Melnick, M. R. Strecker, D. Olago, J.-J. Tier-celin\\n, Earth Planet. Sci. Lett. 331, 322 (2012).   \\n5. S. Kröpelin et al., Science 320, 765 (2008).   \\n6. D. Fleitmann et al., Quat. Sci. Rev. 26, 170 (2007).   \\n7. R. Kuper, S. Kröpelin, Science 313, 803 (2006).   \\n8. Intergovernmental Panel on Climate Change, Working G\\nroup I Contribution to the IPCC Fifth Assessment Report, T\\n. Stocker et al., Eds. (World Meteorological Organization and \\nUnited Nations Environment Programme, Geneva, Switzerland, 2013).\\n \\n9. V. Brovkin, M. Claussen, Science 322, 1326 (2008).   \\n10. M. Claussen, S. Bathiany, V. Brovkin, T. Kleinen, Nat. Geosci.\\n 10.1038/ngeo1962 (2013).  \\n11. S. Jung, G. R. Davies, G. M. Ganssen, D. Kroon, Earth Planet. Sci. \\nLett. 221, 27 (2004).   \\n12. P. Francus et al., Sedimentology 60, 911 (2013).   \\n13. M. Schuster et al., Quat. Sci. Rev. 24, 1821 (2005).   \\n14. P. G. C. Amaral et al., Clim. Past 9, 223 (2013).   \\n15. N. Roberts, The Holocene, An Environmental History  (Blac\\nkwell, Oxford, 1998).        M\\nost cancer mutations, including th\\nose causing a loss of function, a\\nre not directly “druggable” with c\\nonventional small-molecule drugs or bio-l\\nogicals, such as antibodies. Thus, despite our g\\nrowing knowledge of mutations that drive c\\nancer progression, there remains a frustrat-i\\nng gap in translating this information into the d\\nevelopment of targeted treatments that kill onl\\ny cancer cells. An approach that exploits a c\\noncept from genetics called “synthetic lethal-i\\nty” could provide a solution. But it has been o\\nver 15 years since that framework was pro-pos\\ned ( 1). Does the synthetic lethality princi-p\\nle still have the potential for treating cancer?Sy\\nnthetic lethality, ﬁ  rst observed in the f\\nruit ﬂ  y Drosophila melanogaster  almost a century ago, describes a phenomenon where onl\\ny the simultaneous perturbation of two g\\nenes results in a deadly combination. Thus, c\\nancer aberrations that are not readily targe-t\\nable [e.g., tumor suppressor proteins such a\\ns retinblastoma protein 1 (RB1) and p53 (\\nTP53); oncogenes such as RAS and c-MYC] c\\nould be indirectly exploited by inhibit-i\\nng the product of another gene ( 2– 4). The b\\nroader deﬁ  nition of synthetic lethality has a\\nlso been referred to as “nononcogene addic-t\\nion” or “induced essentiality” to distinguish i\\nt from its classical meaning in genetics. In t\\nhe budding yeast Saccharomyces cerevisiae , m\\nost genes display numerous synthetic lethal i\\nnteractions ( 5,  6), which may also apply to m\\nany human cancer genes. Furthermore, “\\npassenger” mutations, which do not directly c\\nontribute to tumorigenesis, and even rewir-i\\nng of cellular networks that give rise to a can\\ncerous state, may also be exploited with t\\nhe synthetic lethal principle. However, to date, only a single synthetic lethal interac-t\\nion has shown therapeutic promise. Why h\\nave synthetic lethal therapies largely failed t\\no deliver?T\\nhe proof of principle that the synthetic l\\nethality concept is clinically translatable i\\ns the efﬁ  cacy of drugs that target the sin-g\\nle-strand DNA repair enzyme poly(ADP-r\\nibose) polymerase (PARP) in tumors with mu\\ntations in the BRCA1 and BRCA2 genes ( \\n7). These genes encode tumor suppressor p\\nroteins that help repair damaged DNA. The r\\nemarkable ability of tumors to acquire resis-t\\nance to PARP inhibitors by regaining BRCA f\\nunction shows that PARP-targeting drugs a\\nct through a synthetic lethal mechanism ( 8). T\\nhis ﬁ  nding triggered an intensive search for s\\nynthetic lethal drug targets akin to PARP . I\\nn particular, large-scale RNA interference s\\ncreens (in which RNA molecules block t\\nhe expression of speciﬁ  c genes) have led t\\no a growing list of potential synthetic lethal Potential of the Synthetic Lethality Principle\\nCANCERSebastian \\nM. B. Nijman 1 and Stephen H. Friend 2  Elucidating the ﬁ  rst principles of synthetic lethality \\nin cancer, including biological context, w\\nill assist clinical translation.1\\nCeMM Research Center for Molecular Medicine of the Aus-tr\\nian Academy of Sciences, Vienna, Austria. 2Sage Bionet-w\\norks, 1100 Fairview Avenue North, Seattle, WA 98109, USA. E\\n-mail: snijman@cemm.oeaw.ac.at; friend@sagebase.orgP\\nublished by AAAS\\nDownloaded from https://www.science.org at Columbia University on January 10, 2024\\n', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), Document(id_='d0b3d543-f85d-4194-86d2-322bc4a5fcf2', embedding=None, metadata={'page_label': '2', 'file_name': 'Potential of the Synthetic Lethality Principle.pdf', 'file_path': 'files/Papers_FullText/Potential of the Synthetic Lethality Principle.pdf', 'file_type': 'application/pdf', 'file_size': 342006, 'creation_date': '2024-01-10', 'last_modified_date': '2024-01-10', 'last_accessed_date': '2024-01-19'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, hash='89aae66472cbb989c24f7ab7d78a1f85e1e03b15d0194af7ed143860f89ad22c', text='15 NOVEMBER 2013    VOL 342    SCIENCE    www.sciencemag.org 810 PERSPECTIVESge\\nne targets. It is too soon to know if any of the\\nse new drug targets can be translated to t\\nhe clinic. A major obstacle is that genetic a\\nnd pharmacological perturbations do not a\\nlways have the same functional outcome. T\\nhis means that a genetic synthetic lethal r\\nelationship may never be realized pharma-c\\nologically. Despite this understanding, there i\\ns a growing concern that many synthetic l\\nethal interactions are not easily transferred be\\nyond the models used for screening, chal-l\\nenging synthetic lethality as a broadly appli-cab\\nle therapeutic concept. Indeed, only a few s\\nynthetic lethal drug targets are in clinical d\\nevelopment, and none are as compelling as t\\nhe BRCA-PARP paradigm.A\\n major obstacle to achieving synthetic le\\nthal therapies is a lack of insight into the ﬁ\\n rst principles that govern the phenomenon in\\n cancer cells. For example, there is little und\\nerstanding of how variability in genet-ic\\ns, epigenetics, systemic signals, and the mic\\nroenvironment influences synthetic le\\nthal interactions (see the ﬁ  gure). Too often, a\\n line of investigation focuses on “what w\\norks” without considering the context or a\\nnalyzing unexpected or even contradic-t\\nory results. Phenotypes are rarely conserved a\\ncross a panel of biologically diverse repli-c\\nates (humans), indicating that context mat-t\\ners. This is especially important in cancer, w\\nhere molecular heterogeneity (contextual v\\nariability) is greater than for any other dis-e\\nase. Indeed, delving into biological context c\\nan be highly informative and provide thera-p\\neutic leads. For instance, colon cancer cells w\\nith activating and oncogenic mutations i\\nn the gene BRAF fail to respond to BRAF i\\nnhibitors. However, these cancer cells dis-play a feedback mechanism that induces s\\nignaling by the epidermal growth factor r\\neceptor, negating the effects of BRAF inhi-bi\\ntion. This “bigger picture” view yielded a r\\nationale to combine BRAF inhibitors with e\\npidermal growth factor inhibitors, a solu-t\\nion that blocked growth of seemingly “drug-r\\nesistant” cancer cells ( 9,  10).S\\nynthetic lethal interactions may be no l\\ness context dependent than other cellular p\\nhenotypes. The opposite actually may be t\\nrue. In yeast, single gene perturbations that le\\nad to ﬁ  tness defects (mutations in essen-t\\nial genes) are evolutionarily more conserved t\\nhan defects induced by two mutations ( 11). A\\ns clonal evolution during tumorigenesis r\\nesults in a divergence of cell states (through t\\nhe progressive accumulation of genetic and e\\npigenetic abnormalities in multiple and dif-f\\nerent genes), context dependencies may be e\\nven more pronounced for drugs that exploit s\\nynthetic lethality than for those that target “\\noncogene addiction,” in which only one g\\nene that is required for maintaining malig-n\\nancy is targeted.T\\nhe importance of context is also r\\nevealed in the comparison of genes that are di\\nfferentially essential in two yeast strains ( \\n12). It was predicted that some of these con-d\\nitional essential genes could be explained b\\ny mutations in synthetic lethal interactors in \\nthe affected strain. However, crossing the t\\nwo strains and analyzing the frequency of viab\\nle spores showed that most of the condi-tio\\nnal essential interactions are due to mul-t\\niple genes acting in concert. This implies t\\nhat synthetic interactions may also be sus-c\\neptible to genetic modiﬁ  ers and that quan-titati\\nve differences in gene activity (between cell types, tumors, and individuals) can a\\nffect synthetic lethality. Moreover, stochas-t\\nic variation in gene activity in the worm C\\naenorhabditis elegans  can affect the pen-etrance of \\nsynthetic lethal interactions ( 13). T\\nhis suggests that context will impact syn-t\\nhetic lethality in human cancer cells and a\\nrgues for a systematic study of this phenom-enon \\nto guide more successful clinical trans-l\\nation. Such studies should focus on iden-t\\nifying co-occurring mutations in tumors t\\nhat can modify the penetrance of synthetic lethal interactions. \\nThis can be achieved, for i\\nnstance, by testing the synthetic lethal inter-action \\nin a large panel of well-characterized c\\nancer cell lines or by using engineered is\\nogenic cell lines that model the variabil-ity \\nobserved in patients in a deﬁ  ned genetic c\\nontext. Although these cell culture stud-i\\nes would not directly address the role of th\\ne tumor microenviroment, any identiﬁ  ed genetic \\nmodiﬁ  ers could serve as biomarkers t\\no predict which tumors are most likely to d\\nisplay the synthetic lethal interaction. Ulti-m\\nately, performing a genome-wide screen f\\nor genetic modiﬁ  ers that enhance or sup-p\\nress a speciﬁ  c synthetic lethal interaction c\\nould reveal those that are least sensitive t\\no contextual variability, and thus would be m\\nore likely to be of general use in the clinic. R\\nesults of such suppressor-enhancer screens f\\nrom a variety of candidate synthetic lethal i\\nnteractions may reveal more fundamental r\\nules that govern context dependency.T\\nhere are several additional fundamen-t\\nal aspects of synthetic lethal interactions i\\nn human cells that have yet to be system-a\\ntically explored. Some of these relate to n\\network topology—the parameters that d\\nescribe the connectivity structure of syn-t\\nhetic lethal interaction. For example, quan-t\\nitative insights into the strength of synthetic le\\nthality in human cells are lacking; yet this m\\nay determine the eventual therapeutic i\\nndex—the concentration of a drug required f\\nor toxic effects divided by the concentration r\\nequired for therapeutic effects—that can be a\\nchieved in patients. Such knowledge will be in\\nstructive for selecting the most promising s\\nynthetic lethal interactions for drug discov-e\\nry and development ( 4). Indeed, compared t\\no normal cells, cells lacking BRCA1/2 are a\\nlmost three orders of magnitude more sen-s\\nitive to PARP inhibitors. By contrast, most ot\\nher described synthetic lethal interactions i\\nn human cells are much less striking, and s\\nome may not even conform to the strictest d\\neﬁ  nition requiring that each single gene per-t\\nurbation alone has no effect on cell viability.T\\nopological information may point to the be\\nst cellular processes for identifying syn-\\nContext and topology. Identifying synthetic lethal interactions (when simultaneous mutations in two genes a\\nre lethal to the cell) must consider the context of their interactions, such as genetic and epigenetic vari-a\\nbility, the microenvironment, and local systemic signals. Topological characteristics (strength, connectivity, d\\negree, and redundancy) of human synthetic lethal interaction networks are also an important consideration, b\\nut are largely unknown.CR\\nEDIT: C. BICKELL/SCIENCEP\\nublished by AAAS\\nDownloaded from https://www.science.org at Columbia University on January 10, 2024\\n', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), Document(id_='7e0989cd-fa9a-4bc1-a384-e02e262964f9', embedding=None, metadata={'page_label': '3', 'file_name': 'Potential of the Synthetic Lethality Principle.pdf', 'file_path': 'files/Papers_FullText/Potential of the Synthetic Lethality Principle.pdf', 'file_type': 'application/pdf', 'file_size': 342006, 'creation_date': '2024-01-10', 'last_modified_date': '2024-01-10', 'last_accessed_date': '2024-01-19'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, hash='f174574deae0a42140666968c08d05d83fd99804871b58d39e4ab113990cce89', text='www.sciencemag.org    SCIENCE    VOL 342    15 NOVEMBER 2013 811 PERSPECTIVESthe\\ntic lethal interactions. Which types of ge\\nnes or cellular processes tend to display th\\ne most interactions (interconnectivity) may in\\nstruct functional genomics efforts. Screens i\\nn model organisms such as yeast and worm indic\\nate that proteins involved in chromatin r\\negulation such as histone deacetylases dis-p\\nlay the most frequent genetic interactions ( \\n5,  14).One explanation is that transcrip-t\\nional regulation is well positioned for buff-e\\nring perturbations by tuning the expression o\\nf multiple genes simultaneously. However, t\\nhere are no experiments to corroborate this h\\nypothesis in human cancer cells. Even for D\\nNA synthesis and repair pathways that rep-r\\nesent highly conserved functional modules (\\nan obvious place to look for synthetic lethal-ity\\n), an inventory of synthetic lethal interac-tio\\nns in human cells has still not materialized.O\\nther largely uncharted topological areas c\\noncern the number of synthetic lethal con-nections between genes and their distribution a\\nnd redundancy. The complexity of a human cel\\nl compared to a yeast cell may suggest t\\nhat human cells display more redundancy, ma\\nking them more resilient to perturbations a\\nnd implying that synthetic lethal interac-t\\nions would be less frequent. Current and n\\next-generation genomics tools will help to a\\nnswer these questions.I\\nn the near term, the ability to perform p\\nersonalized screens for synthetic lethal i\\nnteractions on ex vivo tissue samples may p\\nrovide clinically useful knowledge until t\\nhe long-term goal of better understanding th\\ne biological rules can be achieved. Until a t\\nhorough understanding of synthetic interac-t\\nions and the ability to assess their promise is i\\nn hand, their validation and translation will r\\nemain hit-and-miss. Recognizing the chal-l\\nenges facing gene therapy and immune ther-a\\npy paved the way for moving from concept to clinical reality, and there is hope that learn-i\\nng the principles that govern synthetic lethal i\\nnteractions in cancer will do the same.R\\neferences \\n1. L. H. Hartwell, P. Szankasi, C. J. Roberts, A. W. Murray, S. H\\n. Friend, Science 278, 1064 (1997).   \\n2. W. G. Kaelin, Jr., Nat. Rev. Cancer 5, 689 (2005).   \\n3. J. Luo, N. L. Solimini, S. J. Elledge, Cell 136, 823 (2009).   \\n4. A. Ashworth, C. J. Lord, J. S. Reis-Filho, Cell 145, 30 (\\n2011).   \\n5. M. Costanzo et al., Science 327, 425 (2010).   \\n6. M. E. Hillenmeyer et al., Science 320, 362 (2008).   \\n7. P. C. Fong et al., N. Engl. J. Med. 361, 123 (2009).   \\n8. S. L. Edwards et al., Nature 451, 1111 (2008).   \\n9. A. Prahallad et al., Nature 483, 100 (2012).   \\n10. R. B. Corcoran et al., Cancer Discov. 2, 227 (2012).   \\n11. A. Roguev et al., Science 322, 405 (2008).   \\n12. R. D. Dowell et al., Science 328, 469 (2010).   \\n13. A. Burga, M. O. Casanueva, B. Lehner, Nature 480, 250 (\\n2011).   \\n14. B. Lehner, C. Crombie, J. Tischler, A. Fortunato, A. G. Fra-s\\ner, Nat. Genet. 38, 896 (2006).  10.1126/science.1244669\\n        T\\nhe large-scale generation of hydro-ge\\nn from water with sunlight could pr\\novide a sustainable source of this in\\ndustrially important gas, but could also pr\\novide fuel for vehicles and a storage m\\nedium for solar energy. The direct pho-t\\noelectrochemical (PEC) splitting of water into \\nhydrogen and oxygen, which combines a\\n photovoltaic cell and an electrolyzer into a\\n single device, remains an important goal ( \\n1). One problem is that some of the mate-r\\nials that work well for photovoltaics, such as \\nn-type silicon (Si), corrode in electrolyzer s\\nolutions. On page 836 of this issue, Ken-ne\\ny et al. ( 2) show that a 2-nm-thick nickel (N\\ni) ﬁ  lm on an n-type silicon semiconductor not onl\\ny provides some stability against cor-r\\nosion when used for oxygen evolution in a P\\nEC conﬁ  guration, but also generates a high v\\noltage via a metal-insulator-semiconductor (M\\nIS) conﬁ  guration.H\\nydrogen is used today primarily in the pe\\ntroleum reﬁ  ning industry and for ammo-n\\nia synthesis. More than 50 million tons of\\n hydrogen are produced worldwide every y\\near from fossil fuel feedstocks that gener-ate CO2 emissions. In a carbon-f\\nree energy system, however, h\\nydrogen must be produced f\\nrom water splitting by means of \\nrenewable resources such as w\\nind and solar energy. As solar e\\nnergy is our largest resource, c\\nombining photovoltaics with e\\nlectrolysis would seem to be t\\nhe clear choice for renew-a\\nble hydrogen production ( 3), b\\nut to date the only commer-c\\nially available pathway in this schem\\ne is the electrolysis step. U\\nnfortunately, hydrogen from e\\nlectrolysis with photovoltaic-ge\\nnerated electricity is far too e\\nxpensive to be commercially viab\\nle, so other pathways must be considered.\\nI\\nn a PEC water-splitting s\\nystem, a semiconductor elec-t\\nrode is immersed in an aqueous solution, a\\nnd when illuminated it splits water directly a\\nt the semiconductor’ s surface. For the use of n-type \\nsilicon, the study of Kenney et al. p\\nresents several critical results. The 2-nm N\\ni ﬁ  lm combined with the thin native sili-c\\non oxide layer (SiOx) that forms on silicon exposed to air, and the surface of the nickel o\\nxidized in the electrolyte to form nickel o\\nxide (NiOx). The resulting NiOx/Ni/SiOx/Si de\\nvice generated a voltage of 500 mV when e\\nxposed to light, with no need for the ther-m\\nally grown SiOx layer that has traditionally be\\nen required to achieve that voltage ( 4). A Nickel Finish Protects Silicon Photoanodes \\nfor Water SplittingCHEMISTRYJohn \\nA. TurnerUltrathin nickel coatings allow silicon to act as t\\nhe oxygen-generating electrode in the direct f\\normation of hydrogen from water with sunlight.Na\\ntional Renewable Energy Laboratory, 1617 Cole Boule-va\\nrd, Golden, CO 80401, USA. E-mail: john.turner@nrel.gov\\nSunlightElectrolyteH\\n2O2h\\nν < 1.7 eVe–T\\nransparent gas-separationme\\nmbranep-CGSN i/n-SiD\\noubling up for solar hydrogen production.  A design conﬁ  guration is \\nshown where two separate semiconductors with different band gaps a\\nre illuminated in series to form a tandem system for water splitting. S\\nunlight illuminates the p-type electrode, which absorbs the visible l\\night and transmits the red and near-infrared light that then illumi-na\\ntes the n-type electrode. The work of Kenney et al. shows that a thin ni\\nckel ﬁ  lm can protect n-type silicon from corrosion by the electrolyte.P\\nublished by AAAS\\nDownloaded from https://www.science.org at Columbia University on January 10, 2024\\n', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), Document(id_='fdc1d804-198a-41ec-9a65-69b9fe95c9fc', embedding=None, metadata={'page_label': '4', 'file_name': 'Potential of the Synthetic Lethality Principle.pdf', 'file_path': 'files/Papers_FullText/Potential of the Synthetic Lethality Principle.pdf', 'file_type': 'application/pdf', 'file_size': 342006, 'creation_date': '2024-01-10', 'last_modified_date': '2024-01-10', 'last_accessed_date': '2024-01-19'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, hash='0cc852ec31ec77652d5f6a53133274e57970f2236f14d780d31fc685d253a6f0', text='Use of this article is subject to the Terms of service\\nScience (ISSN 1095-9203) is published by the American Association for the Advancement of Science. 1200 New York Avenue NW,\\nWashington, DC 20005. The title Science is a registered trademark of AAAS. \\nCopyright © 2013, American Association for the Advancement of SciencePotential of the Synthetic Lethality Principle\\nSebastian M. B. Nijman and Stephen H. Friend\\nScience 342 (6160), .  DOI: 10.1126/science.1244669\\nView the article online\\nhttps://www.science.org/doi/10.1126/science.1244669\\nPermissions\\nhttps://www.science.org/help/reprints-and-permissions\\nDownloaded from https://www.science.org at Columbia University on January 10, 2024\\n', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), Document(id_='198c3d4e-ed97-4ec2-bfff-94d92268852c', embedding=None, metadata={'page_label': '34', 'file_name': 'Searching for synthetic lethality in cancer.pdf', 'file_path': 'files/Papers_FullText/Searching for synthetic lethality in cancer.pdf', 'file_type': 'application/pdf', 'file_size': 546151, 'creation_date': '2024-01-19', 'last_modified_date': '2024-01-10', 'last_accessed_date': '2024-01-19'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, hash='68b4b008c20e51a3ea648c4e04846cb7df3ca2d0a147722ff84fa19811fbcc78', text='Available online at www.sciencedirect.com\\nSearching for synthetic lethality in cancer\\nRachel Brough, Jessica R Frankum, Sara Costa-Cabral, Christopher J Lord\\nand Alan Ashworth\\nThe incentive to develop personalised therapy for cancer\\ntreatment is driven by the premise that it will increase\\ntherapeutic efﬁcacy and reduce toxicity. Understanding the\\nunderlying cellular and molecular basis of the disease has beenextremely important in the design of these novel therapies;however, identifying new drug targets for personalisedtherapies remains problematic. This review describes how thebiological concept of synthetic lethality has been successfullyimplemented to identify new therapeutic approaches andtargets in models from yeast through to human cells. We alsodiscuss how recent technical advances combined with anincreased understanding of the complexity of cellular networksmay facilitate therapeutic advances in the future.\\nAddress\\nThe Breakthrough Breast Cancer Research Centre, The Institute ofCancer Research, Fulham Road, London, SW3 6JB, UK\\nCorresponding author: Ashworth, Alan ( Alan.Ashworth@icr.ac.uk )\\nCurrent Opinion in Genetics & Development 2011, 21:34–41\\nThis review comes from a themed issue on\\nGenetic and cellular mechanisms of oncogenesisEdited by Chris Marshall and Karen Vousden\\n0959-437X/$ – see front matter\\n#2010 Elsevier Ltd. All rights reserved.\\nDOI 10.1016/j.gde.2010.10.009\\nIntroduction: targeting cancer with synthetic\\nlethality\\nCancer is fundamentally a genetic disease. During\\ntumour development, in a multistep mutational process,\\ncells acquire a set of genetic lesions that lead to the\\ncancerous state. For example, gain of function mutationsin oncogenes and loss of function mutations in tumour\\nsuppressor genes enable cells to acquire a series of\\ncharacteristics (‘hallmarks’) that facilitate the tumori-genic process [ 1]. While these mutations promote the\\ndisease process they also offer therapeutic opportunities.\\nThis is perhaps best exempliﬁed by the development of\\ndrugs that target gain of function oncogenic mutationssuch as the ERBB2 ampliﬁcation in breast cancer (with\\ntraztuzumab, lapatinib, etc.), the BCR-ABL translocation\\nin chronic myeloid leukaemia (with imatinib, etc.) and\\nmore recently BRAF mutations in metastatic melanoma\\n(with BRAF inhibitors) (reviewed in [ 2\\n/C15]). However,despite these successes, developing drugs that selectively\\nkill cancer cells without harming normal cells remains a\\nconsiderable challenge. This process is particularly pro-blematic when considering how to target a protein such as\\na dysfunctional tumour suppressor that is either largely\\ninactive or even completely absent ( Figure 1 a). In this\\ncase, the concept of synthetic sickness/lethality (SSL)\\nhas been advanced as a potential approach [ 3,4\\n/C15,5].\\nTwo genes have a SSL relationship when inhibition ormutation of either gene alone does not cause loss of\\nviability/sickness, but simultaneous inhibition of both\\ngenes results in reduced cell viability or an impairmentof cellular health/ﬁtness ( Figure 1 b). If one gene in an\\nSSL relationship is a tumour suppressor gene, then its\\nsynthetic lethal partner becomes a candidate therapeutictarget that could be used in tumour cells with a deﬁned\\ntumour suppressor gene dysfunction. SSL can occur\\nbetween genes acting in the same biochemical pathwayor in distinct but compensatory pathways [ 6] and com-\\nponents of the same pathway often share the same SSL\\npartners [ 7,8]. The concept of SSL was ﬁrst described by\\nBridges in 1922 (referenced in [ 9]), Dobzhansky in 1946\\n[10] and Sturtevant in 1956 [ 11] and was based primarily\\non studies in the fruit ﬂy, Drosophila melanogaster (D.\\nmelanogaster ). Since then numerous genetic screens have\\nbeen carried out in additional model organisms to identify\\nSSLs including studies in Caenorhabditis elegans (C. elegans\\n[12,13 ]),Saccharomyces cerevisiae (S. cerevisiae [14,15,16\\n/C15])\\nandEscherichia coli (E. coli [17]). New technology is now\\nenabling the rapid identiﬁcation of SSLs in human cells[18] that may allow the targeting of tumour suppressors or\\noncogenes that have limited pharmacological tractability\\n(Figure 1 c).\\nLessons learned from synthetic lethality in\\nmodel organisms\\nLower eukaryotic model organisms are often used for SSL\\nstudies on the basis that they are relatively amenable toexperimental manipulation; they have small genomes,can be easily mutated and mated to generate hybrid\\nstains and are relatively easy to use with gene silencing\\ntechniques such as RNA interference (RNAi). SSLs inmodel organisms have been identiﬁed using both low-\\nthroughput approaches, where two different mutations\\nare combined to assess SSL, or in larger screens wheremultiple mutations are combined. For example, Tong\\nand colleagues developed a synthetic genetic array (SGA)\\nassay in S. cerevisiae , where mutant yeast strains contain-\\ning a single disrupted gene of interest were mated with an\\narray of yeast strains harbouring /C244700 different deletions\\nCurrent Opinion in Genetics & Development 2011, 21:34–41 www.sciencedirect.com', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), Document(id_='ce836c45-b9c6-4c78-899b-ded30404c5b3', embedding=None, metadata={'page_label': '35', 'file_name': 'Searching for synthetic lethality in cancer.pdf', 'file_path': 'files/Papers_FullText/Searching for synthetic lethality in cancer.pdf', 'file_type': 'application/pdf', 'file_size': 546151, 'creation_date': '2024-01-19', 'last_modified_date': '2024-01-10', 'last_accessed_date': '2024-01-19'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, hash='1b166313845d42b6ae6dba2effdbbc36371e4ea5ce19afdd0cbe87779fefa48a', text='[16/C15]. Following haploid selection, SSL interactions were\\ndetected by the absence or poor growth of a particular\\nmutant combination.\\nOoiet al.[19/C15/C15] reﬁned this methodology by designing a\\nSLAM (synthetic lethality analysed by microarray)approach. SLAM combines the SGA method with a\\nunique bar code identiﬁcation system, which utilises\\nsequence ‘tags’ that ﬂank each deletion. Parallel poolsof 4700 mutant yeast strains were transformed with either\\na gene-targeting construct to disrupt the gene-of-interest\\nor a control DNA fragment. Following selection, DNAwas isolated from each yeast pool and PCR ampliﬁcation\\nwas performed using primers recognising the ‘tags’ that\\nﬂank each deletion in all 4700 strains. The PCR productswere then hybridised to oligonucleotide microarrays to\\nquantify the growth of each strain. Comparing the inten-\\nsity of the control and the targeted pool hybridisationsidentiﬁed SSL combinations [ 19\\n/C15/C15]. This technology has\\nmore recently been extended to the yeast, Schizosacchar-\\nomyces pombe (S. pombe ,[20]) and analogous approaches are\\nbeing developed in C. elegans [13] and E. coli [17].\\nAlthough yeast SSL studies should provide a useful\\nmodel for predicting fundamental biological interactions\\nconserved in higher eukaryotes, in practice they are often\\nnot conserved in human cells. It appears that the ‘wiring’of molecular networks from yeast to human cells may be\\nso different that a mutation in orthologous alleles often\\npredisposes cells from different organisms to differentfates. Nevertheless, the development of methods to\\nanalyse high-throughput methods such as SGA and\\nSLAM has provided invaluable lessons for the analysisof subsequent mammalian screens.\\nSynthetic lethality in human cells: from proof\\nof concept to the clinic\\nAlthough SSL relationships were proposed more than a\\ndecade ago as having potential applications in cancer\\ntherapeutics [ 4/C15], it is only recently that such approaches\\nhave reached the clinic. We, and others, have demon-strated that dysfunction of either of the tumour suppres-\\nsor genes, BRCA1 orBRCA2 , is synthetically lethal with\\ninhibition of the DNA repair enzyme Poly(ADP-Ribose)Polymerase 1 (PARP1) [ 21\\n/C15,22/C15]. BRCA1 or 2 deﬁcient\\ncells are not only inhibited by gene silencing of PARP1\\nbut are also profoundly sensitive to potent small moleculePARP inhibitors, when compared to isogenic wild-type\\ncells. These ﬁndings provided the rationale for clinical\\ntrials that are now assessing the potential of PARP inhibi-tors [ 23] and Phase 1 and 2 trial results suggest that PARP\\ninhibitors can elicit sustained anti-tumour responses inSearching for synthetic lethality in cancer Brough et al. 35\\nFigure 1\\nTargeting cancer with synthetic lethality. (a)Left: illustrates the difficulty\\nin developing an inhibitor targeting a cancer with a mutation in a tumoursuppressor gene since there is little or no protein with which to inhibit.Right: portrays that the development of small molecule inhibitorstargeting oncogenes is generally simpler since there is a product activityin which to target. (b)A schematic of the concept of synthetic lethality.\\nInhibition of either gene A or gene B alone has no effect on viability;however, when gene A and gene B are simultaneously inhibited it leadsto cell death. (c)Highlights the advantages of using synthetic lethality\\nwhen attempting to kill a cancer cell harbouring a ‘non-drugableoncogene’ or ‘absent tumour suppressor’.\\nwww.sciencedirect.com Current Opinion in Genetics & Development 2011, 21:34–41', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), Document(id_='48bc2d62-c248-473c-aee3-c0076736376a', embedding=None, metadata={'page_label': '36', 'file_name': 'Searching for synthetic lethality in cancer.pdf', 'file_path': 'files/Papers_FullText/Searching for synthetic lethality in cancer.pdf', 'file_type': 'application/pdf', 'file_size': 546151, 'creation_date': '2024-01-19', 'last_modified_date': '2024-01-10', 'last_accessed_date': '2024-01-19'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, hash='0e6d5e448b2ded0d3a215e87eecc7c8debb5914327d65dba193bbedc8e6e3fe6', text='patients carrying BRCA1 or-2mutations without causing\\nmany of the serious adverse side-effects associated with\\nconventional chemotherapy [ 24/C15,25,26 ]. Subsequent ‘hy-\\npothesis’ driven approaches for identifying SSL inter-\\nactions have involved predicting synthetic lethality based\\nupon already proven associations, as components of thesame pathway often share similar SSL partners. For\\nexample, we have recently demonstrated SSL between\\ndysfunction of the Phosphatase and Tensin Homolog\\n(PTEN ) gene (OMIM 601728 ) and PARP inhibition\\n[27]. This SSL relationship was predicted from the\\nprevious observation that PTEN mutant cells exhibit a\\nsimilar DNA repair defect as in BRCA1 orBRCA2 mutant\\ncells.\\nRecent examples of synthetic lethality by\\nfunctional compensation\\nFrom the earliest studies of cell biology, it has been\\nclear that cells readily adapt to a wide array of pertur-\\nbations, such as gene mutation. Underlying these obser-\\nvations is a complex system of functional buffering,such that when one molecular function, gene or protein\\nis lost, another can readily be used to mitigate any\\neffects that could limit either cell survival or ﬁtness.For example, from studies in S. cerevisiae andC. elegans,\\nit is proposed that only 10 –20% of genes are actually\\nessential for viability [ 28–30], suggesting an extensive\\nlevel of functional buffering.\\nThis concept of functional buffering provides us with a\\nway to think about, and hopefully identify, synthetic\\nlethal relationships [ 31\\n/C15/C15]. For example, it seems likely\\nthat some synthetic lethal partners could be elements offunctional buffering networks. This seems to be the case\\nwith the BRCA/PARP SSL. DNA is constantly bom-\\nbarded with damage, one of the most frequent formsbeing the formation of single strand DNA breaks (SSBs).\\nThese SSBs are normally repaired via a PARP-dependent\\nmechanism and in the presence of PARP inhibitors SSBsaccumulate and have the potential to limit cell viability if\\nnot repaired by another buffering process. In normal cells,\\nDNA repair via BRCA1 and BRCA2 provides this func-tional buffering. However, in BRCA1 or -2 deﬁcient cells,\\nfunctional compensation is lost and SSBs ultimately\\ncompromise the viability of cells. In keeping with thisfunctional buffering hypothesis, there is some evidence tosuggest that in the absence of BRCA1 or 2 function,\\nPARP1 is hyperactivated [ 32], possibly as a means to\\nlimit the impact of SSBs where a functional compensationmechanism is no longer present. Similarly, we have\\ndemonstrated a synthetic lethal interaction between\\nDNA mismatch repair (MMR) proteins and particularproofreading DNA polymerases [ 33]. In this case, it seems\\nlikely that MMR proteins and their synthetic lethal DNA\\npolymerase partners repair similar DNA lesions (oxi-dative DNA damage) but by differing mechanisms. This\\nfunctional buffering is also represented by the obser-vation that both MMR-deﬁcient cell models and tumours\\nexpress elevated expression of proofreading DNA poly-\\nmerases [ 33]. Extending these concepts of functional\\nbuffering, mechanistic redundancy should allow us to\\npredict additional SSL effects.\\nA variation of a compensatory SSL interaction was\\nrecently reported between the Epidermal Growth Factor\\nReceptor (EGFR) and Notch signalling pathways inbasal-like breast cancer (BLBC) ( Figure 2 a). EGFR\\npromotes cell proliferation in a number of tumour types\\nand small molecule inhibitors or antibodies targetingEGFR signalling have shown efﬁcacy in lung, colon\\nand head and neck cancers. The efﬁcacy of anti-EGFR\\ntherapies may be due in part to the signiﬁcant decrease inAKT activation that these agents elicit ( Figure 2 b).\\nHowever, anti-EGFR inhibitors have demonstrated lit-\\ntle clinical activity in breast cancer and it is believed thatthe hyperactivated Notch pathway seen in BLBC may be\\ncompensating for, or buffering, EGFR inhibition by\\nmaintaining levels of activated AKT ( Figure 2 c). Simul-\\ntaneous inhibition of the EGFR and Notch signalling\\npathway, however, results in SSL, most likely by limiting\\nthe Notch-mediated escape mechanism and enablingAKT suppression ( Figure 2 d) [34\\n/C15/C15]. We envisage that\\nthe development of more integrative screening\\napproaches will allow the prediction and validation ofmany more ‘buffering’ SSL interactions in the future\\n(Figure 2 e).\\nTechniques for the identiﬁcation of new\\nsynthetic lethal interactions\\nWith the advent of new high throughput genetic screen-\\ning technologies and the increased availability of\\ngenetically modiﬁed human cell lines, we have some-\\nwhat moved away from the ‘hypothesis’ driven methodstowards an unbiased screening approach to human SSL\\nidentiﬁcation. There are three common methods for\\nidentifying novel synthetic l ethal interactions in human\\ncells ( Figure 3 ). These include: (i) RNAi screens using\\nisogenic or functionally relevant cell models [ 18,35,36 ],\\n(ii) high throughput screens with chemical libraries toidentify compounds that kill cells in a genotype-speciﬁc\\nmanner [ 37] and (iii) synthetic lethal approaches using\\nRNAi libraries in combination with chemical inhibitor[38–40]. Each of these approaches can present consider-\\nable technical challenges [ 18]. For example, the\\ninterpretation of both RNAi and chemical inhibitor\\nscreens is somewhat confounded by the considerableoff-target effects observed using either approach [ 41].\\nThis, in part, can be mitigat ed by validating both RNAi\\nand chemical screen effects with multiple differentRNAi reagents and/or inhibitors or even performing\\nRNAi and chemical screens in parallel and integrating\\nthe resultant data [ 42]. However, the overlap between\\nchemical inhibitor sensiti vity and siRNA sensitivity\\nrequires careful interpretation [ 43] as many inhibitors36Genetic and cellular mechanisms of oncogenesis\\nCurrent Opinion in Genetics & Development 2011, 21:34–41 www.sciencedirect.com', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), Document(id_='683aae7b-2a1c-4e16-950d-5200b85562a9', embedding=None, metadata={'page_label': '37', 'file_name': 'Searching for synthetic lethality in cancer.pdf', 'file_path': 'files/Papers_FullText/Searching for synthetic lethality in cancer.pdf', 'file_type': 'application/pdf', 'file_size': 546151, 'creation_date': '2024-01-19', 'last_modified_date': '2024-01-10', 'last_accessed_date': '2024-01-19'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, hash='fe73a5adfcc615631c84c7a14565acf21ac80fddafa137d6f4d374bf0af5cc2c', text='Searching for synthetic lethality in cancer Brough et al. 37\\nFigure 2\\nSynthetic lethality by functional compensation. (a)The EGFR pathway is often active in cancer. One of the downstream consequences of EGFR\\nstimulation is the activation of AKT, which drives cellular proliferation. (b)EGFR inhibitor (gefitinib) sensitive cancers are unable to activate AKT and as\\na result die when treated with the inhibitor. (c)In basal-like breast cancer the notch pathway is hyperactivated. Notch activation compensates (or\\nbuffers) for EGFR inhibition by maintaining the levels of activated AKT. (d)Dual inhibition of EGFR and Notch, by inhibiting the essential Notch\\ncomponent g-secretase, results in synthetic lethality since the levels of activated AKT are depleted. (e)A potential route for predicting further synthetic\\nlethal interactions by functional compensation is illustrated. Firstly identifying genes that are up-regulated in mutant cell lines, when compare dt ot h e\\nwild-type counterpart, with the premise that this difference in expression may be compensating for the effect of the mutant allele. Targeting the up-\\nregulated protein product, or another in the same pathway, may be a potential route for therapy.\\nwww.sciencedirect.com Current Opinion in Genetics & Development 2011, 21:34–41', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), Document(id_='8e1ca5f3-e460-490b-9c96-94b3992ef860', embedding=None, metadata={'page_label': '38', 'file_name': 'Searching for synthetic lethality in cancer.pdf', 'file_path': 'files/Papers_FullText/Searching for synthetic lethality in cancer.pdf', 'file_type': 'application/pdf', 'file_size': 546151, 'creation_date': '2024-01-19', 'last_modified_date': '2024-01-10', 'last_accessed_date': '2024-01-19'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, hash='2e9a56e647379eff4dd75cd9bc962f7c1adb6be5d953ff92d296fee59d7c60a3', text='target catalytic activities whereas RNAi approaches\\nsuppress target protein exp ression and thus inhibit\\nboth catalytic and non-catalytic, such as scaffolding,\\nfunctions.We recently combined genetic and chemical approaches\\nin a high-throughput screen of a library of clinically used\\ndrugs to identify synthetic lethality effects with loss of the\\ntumour suppressor gene MSH2 [37]. Using isogenic cell38Genetic and cellular mechanisms of oncogenesis\\nFigure 3\\nTechniques for the identification of further synthetic lethal interactions and revalidation models. Three common approaches for identifying synt hetic\\nlethal interactions in human cells are illustrated. These include (i) high-throughput RNAi screens on isogenic or functionally relevant cell model s, (ii)\\nhigh-throughput screens with chemical libraries to identify compounds that kill cells in a genotype-specific manner and (iii) synthetic lethal app roaches\\nusing RNAi libraries in combination with single agent or dose of inhibitor. Following the screen and the identification of potential synthetic letha l\\ninteractions revalidation with numerous siRNA and inhibitors targeting the same gene on appropriate cell models. Preceding successful revalidati on\\nand xenograft studies could lead to a clinical trial.\\nCurrent Opinion in Genetics & Development 2011, 21:34–41 www.sciencedirect.com', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), Document(id_='90c0758d-25d5-4dd9-81d6-61a96655251d', embedding=None, metadata={'page_label': '39', 'file_name': 'Searching for synthetic lethality in cancer.pdf', 'file_path': 'files/Papers_FullText/Searching for synthetic lethality in cancer.pdf', 'file_type': 'application/pdf', 'file_size': 546151, 'creation_date': '2024-01-19', 'last_modified_date': '2024-01-10', 'last_accessed_date': '2024-01-19'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, hash='269fb65f75ddcf69b1c3cbc42958209acfe19b8aaacdf5da7c116d6aa910b7f6', text='lines, with wild type MSH2 expression or MSH2\\ndeﬁciency, we identiﬁed a number of compounds that\\nwere selectively lethal to the MSH2 deﬁcient cells,\\nincluding methotrexate. To establish the validity of theMSH2/methotrexate SSL, we used a panel of diverse\\nRNAi reagents that silenced the methotrexate target,\\nDHFR and also proﬁled multiple MSH2 models for\\nmethotrexate and DHFR siRNA sensitivity. Dissection\\nof the mechanism indicated that an accumulation of\\noxidative DNA lesions might explain the MSH2/metho-trexate SSL. Because our screen was of clinically used\\ndrugs, we were able to rapidly validate the screen effects\\ninin vivo models, and then establish a phase 2, non-\\nrandomised clinical trial to assess the performance of\\nmethotrexate in MSH2 -deﬁcient metastatic colorectal\\ncancer [ 37].\\nRecent SSL interaction studies highlight\\ncancer cell heterogeneity\\nIn addition to targeting tumour suppressor gene dysfunc-\\ntion, the concept of synthetic lethality has recently been\\nextended to target oncogenes that in themselves have\\nlimited pharmacological tractability. For example, theKRAS oncogene is commonly mutated in colorectal, pan-\\ncreatic and lung cancers but encodes a GTPase which has\\nnot yet proven to be ‘druggable’. One approach to target-ing KRAS may be to identify genes/proteins whose inhi-\\nbition is synthetically lethal with KRAS mutation.\\nTo date, four different KRAS SSL screens have been\\ndescribed. Luo and colleagues (2009) used a genome-wide RNAi screen in isogenic KRAS mutant and wild-\\ntype colorectal cell lines to identify synthetic lethality\\nwith inhibition of APC/C (anaphase-promoting complex/\\ncyclosome), an effect that was reproduced in non-smallcell lung cancer cell lines [ 44]. In an alternative approach,\\nScholl and colleagues [ 45] used an RNAi screen in a\\ndiverse panel of non-isogenic cancer cell lines to identifySSL between KRAS mutation and silencing of STK33.\\nSimilarly, Barbie and colleagues [ 46] used 19 tumour cells\\nlines to identify TBK1 as a synthetic lethal partner of\\noncogenic KRAS . Finally, Wang and colleagues screened\\na panel of isogenic colorectal cell lines to demonstrate\\nSNAIL2 (zinc ﬁnger transcriptional repressor) SSL withKRAS mutation [ 47]. What is striking from a comparison\\nof these papers ( Figure 4 ) is the lack of KRAS synthetic\\nlethal interactions that are common between studies and,more speciﬁcally, between genetically diverse tumour\\ncell lines. Variations in experimental procedures in the\\ndiffering studies (especially the use of differing RNAilibraries) will account for some of the differences in SSLs\\nidentiﬁed but it might also reﬂect the nature of KRAS\\nsynthetic lethality speciﬁcally and SSL relationshipsgenerally. It may be that some SSLs, such as those\\nassociated with KRAS, are much more affected by com-\\nmon modiﬁer or genetic background loci than others.\\nConclusions\\nTherapies derived from SSL interactions, such as thatbetween the BRCA1/2 and PARP, are now showingsuccess in the clinic. The application of genome-wide\\nhigh-throughput screens (from yeast to humans) is likelySearching for synthetic lethality in cancer Brough et al. 39\\nFigure 4\\nA summary of published KRAS RNAi screens is shown.\\nwww.sciencedirect.com Current Opinion in Genetics & Development 2011, 21:34–41', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), Document(id_='e6cfb310-d73c-4386-83cf-9def1db6d462', embedding=None, metadata={'page_label': '40', 'file_name': 'Searching for synthetic lethality in cancer.pdf', 'file_path': 'files/Papers_FullText/Searching for synthetic lethality in cancer.pdf', 'file_type': 'application/pdf', 'file_size': 546151, 'creation_date': '2024-01-19', 'last_modified_date': '2024-01-10', 'last_accessed_date': '2024-01-19'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, hash='ed8ea995d265afc7ab5d0af2e55d94a413258c5ce8ef5a586267d2ca2591d28c', text='to lead to the discovery of many new SSLs. Whether these\\nwill prove to be clinically applicable will likely depend on\\nthe magnitude of the effect as well as the prevalence ofgenetic modiﬁers mediating resistance.\\nAcknowledgements\\nWe thank Breakthrough Breast Cancer, Cancer Research UK, The\\nAmerican Association of Cancer Research, The Breast Cancer Campaign\\nand the Breast Cancer Research Foundation for funding the work in our\\nlaboratory.\\nReferences and recommended reading\\nPapers of particular interest, published within the period of review,\\nhave been highlighted as:\\n/C15of special interest\\n/C15/C15of outstanding interest\\n1. Hanahan D, Weinberg RA: The hallmarks of cancer .Cell 2000,\\n100:57-70.\\n2.\\n/C15Lord CJ, Ashworth A: Biology-driven cancer drug development:\\nback to the future .BMC Biol 2010, 8:38.\\nA comprehensive review summarising the history and development of\\ncancer therapy.\\n3. Stockwell BR, Haggarty SJ, Schreiber SL: High-throughput\\nscreening of small molecules in miniaturized mammalian cell-based assays involving post-translational modiﬁcations .\\nChem Biol 1999, 6:71-83.\\n4.\\n/C15Hartwell LH, Szankasi P, Roberts CJ, Murray AW, Friend SH:\\nIntegrating genetic approaches into the discovery of\\nanticancer drugs .Science 1997, 278:1064-1068.\\nA critical review outlining the use of genetic concepts in cancer\\ntherapy.\\n5. Simons AH, Dafni N, Dotan I, Oron Y, Canaani D: Genetic\\nsynthetic lethality screen at the single gene level in culturedhuman cells .Nucleic Acids Res 2001, 29:E100.\\n6. Hartman JLt, Garvik B, Hartwell L: Principles for the buffering of\\ngenetic variation .Science 2001, 291:1001-1004.\\n7. Ye P, Peyser BD, Pan X, Boeke JD, Spencer FA, Bader JS: Gene\\nfunction prediction from congruent synthetic lethal\\ninteractions in yeast .Mol Syst Biol 2005, 1:0026.\\n8. Tong AH, Lesage G, Bader GD, Ding H, Xu H, Xin X,\\nYoung J, Berriz GF, Brost RL, Chang M et al. :Global mapping of\\nthe yeast genetic interaction network .Science 2004,\\n303:808-813.\\n9. Lucchesi JC: Synthetic lethality and semi-lethality among\\nfunctionally related mutants of Drosophila melanogaster .\\nGenetics 1968, 59:37-44.\\n10. Dobzhansky T: Genetics of natural populations. Xiii.\\nRecombination and variability in populations of Drosophilapseudoobscura .Genetics 1946, 31:269-290.\\n11. Sturtevant AH: A highly speciﬁc complementary lethal\\nsystem in Drosophila melanogaster .\\nGenetics 1956,\\n41:118-123.\\n12. Lehner B, Crombie C, Tischler J, Fortunato A, Fraser AG:\\nSystematic mapping of genetic interactions in Caenorhabditis\\nelegans identiﬁes common modiﬁers of diverse signalingpathways .Nat Genet 2006, 38:896-903.\\n13. Byrne AB, Weirauch MT, Wong V, Koeva M, Dixon SJ, Stuart JM,\\nRoy PJ: A global analysis of genetic interactions in\\nCaenorhabditis elegans .J Biol 2007, 6:8.\\n14. Ooi SL, Pan X, Peyser BD, Ye P, Meluh PB, Yuan DS, Irizarry RA,\\nBader JS, Spencer FA, Boeke JD: Global synthetic-lethality\\nanalysis and yeast functional proﬁling .Trends Genet 2006,\\n22:56-63.\\n15. Simon JA, Szankasi P, Nguyen DK, Ludlow C, Dunstan HM,\\nRoberts CJ, Jensen EL, Hartwell LH, Friend SH: Differentialtoxicities of anticancer agents among DNA repair and\\ncheckpoint mutants of Saccharomyces cerevisiae .Cancer Res\\n2000, 60:328-333.\\n16.\\n/C15Tong AH, Evangelista M, Parsons AB, Xu H, Bader GD, Page N,\\nRobinson M, Raghibizadeh S, Hogue CW, Bussey H et al. :\\nSystematic genetic analysis with ordered arrays of yeastdeletion mutants .Science 2001, 294:2364-2368.\\nDescribes the development of the SGA assay.\\n17. Typas A, Nichols RJ, Siegele DA, Shales M, Collins SR, Lim B,\\nBraberg H, Yamamoto N, Takeuchi R, Wanner BL et al. :High-\\nthroughput, quantitative analyses of genetic interactions in E.\\ncoli.Nat Methods 2008, 5:781-787.\\n18. Iorns E, Lord CJ, Turner N, Ashworth A: Utilizing RNA\\ninterference to enhance cancer drug discovery .Nat Rev Drug\\nDiscov 2007, 6:556-568.\\n19.\\n/C15/C15Ooi SL, Shoemaker DD, Boeke JD: DNA helicase gene\\ninteraction network deﬁned using synthetic lethality analyzedby microarray .Nat Genet 2003, 35:277-286.\\nDescribes the development of the SLAM assay.\\n20. Roguev A, Wiren M, Weissman JS, Krogan NJ: High-throughput\\ngenetic interaction mapping in the ﬁssion yeast\\nSchizosaccharomyces pombe .Nat Methods 2007, 4:861-866.\\n21.\\n/C15Bryant HE, Schultz N, Thomas HD, Parker KM, Flower D, Lopez E,\\nKyle S, Meuth M, Curtin NJ, Helleday T: Speciﬁc killing of\\nBRCA2-deﬁcient tumours with inhibitors of poly(ADP-ribose)\\npolymerase\\n.Nature 2005, 434:913-917.\\nSee annotation to Ref. [ 22/C15].\\n22.\\n/C15Farmer H, McCabe N, Lord CJ, Tutt AN, Johnson DA,\\nRichardson TB, Santarosa M, Dillon KJ, Hickson I, Knights C et al. :\\nTargeting the DNA repair defect in BRCA mutant cells as a\\ntherapeutic strategy .Nature 2005, 434:917-921.\\nThis paper along with Ref. [21/C15] describes the PARP/BRCA2 synthetic\\nlethalities.\\n23. Ashworth A: A synthetic lethal therapeutic approach: poly(ADP)\\nribose polymerase inhibitors for the treatment of cancers\\ndeﬁcient in DNA double-strand break repair .J Clin Oncol 2008,\\n26:3785-3790.\\n24.\\n/C15Fong PC, Boss DS, Yap TA, Tutt A, Wu P, Mergui-Roelvink M,\\nMortimer P, Swaisland H, Lau A, O’Connor MJ et al. :Inhibition of\\npoly(ADP-ribose) polymerase in tumors from BRCA mutationcarriers .N Engl J Med 2009, 361:123-134.\\nThis paper describes the ﬁrst clinical demonstration of the potential of\\nPARP inhibitors in a synthetic lethal approach.\\n25. Audeh MW, Carmichael J, Penson RT, Friedlander M, Powell B,\\nBell-McGuinn KM, Scott C, Weitzel JN, Oaknin A, Loman N et al. :\\nOral poly(ADP-ribose) polymerase inhibitor olaparib inpatients with BRCA1 or BRCA2 mutations and recurrent\\novarian cancer: a proof-of-concept trial .Lancet 2010,\\n376:245-251.\\n26. Tutt A, Robson M, Garber JE, Domchek SM, Audeh MW,\\nWeitzel JN, Friedlander M, Arun B, Loman N, Schmutzler RK et al. :\\nOral poly(ADP-ribose) polymerase inhibitor olaparib inpatients with BRCA1 or BRCA2 mutations and advanced\\nbreast cancer: a proof-of-concept trial .Lancet 2010,\\n376:235-244.\\n27. Mendes-Pereira AM, Martin SA, Brough R, McCarthy A, Taylor JR,\\nKim JS, Waldman T, Lord CJ, Ashworth A: Synthetic lethal\\ntargeting of PTEN mutant cells with PARP inhibitors .EMBO\\nMol Med 2009, 1:315-322.\\n28. Giaever G, Chu AM, Ni L, Connelly C, Riles L, Veronneau S, Dow S,\\nLucau-Danila A, Anderson K, Andre B et al. :Functional proﬁling\\nof the Saccharomyces cerevisiae genome .Nature 2002,\\n418:387-391.\\n29. Winzeler EA, Shoemaker DD, Astromoff A, Liang H, Anderson K,\\nAndre B, Bangham R, Benito R, Boeke JD, Bussey H et al. :\\nFunctional characterization of the S. cerevisiae genome by\\ngene deletion and parallel analysis .Science 1999, 285:901-906.\\n30. Kamath RS, Fraser AG, Dong Y, Poulin G, Durbin R, Gotta M,\\nKanapin A, Le Bot N, Moreno S, Sohrmann M et al. :Systematic\\nfunctional analysis of the Caenorhabditis elegans genome\\nusing RNAi .Nature 2003, 421:231-237.40Genetic and cellular mechanisms of oncogenesis\\nCurrent Opinion in Genetics & Development 2011, 21:34–41 www.sciencedirect.com', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), Document(id_='458ee880-87ca-44f4-bc3f-65606e8ad374', embedding=None, metadata={'page_label': '41', 'file_name': 'Searching for synthetic lethality in cancer.pdf', 'file_path': 'files/Papers_FullText/Searching for synthetic lethality in cancer.pdf', 'file_type': 'application/pdf', 'file_size': 546151, 'creation_date': '2024-01-19', 'last_modified_date': '2024-01-10', 'last_accessed_date': '2024-01-19'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, hash='6faf1356f848c55da80739cb8016372b22ff14a726a6b8832133fd05a091279d', text='31.\\n/C15/C15Kaelin WG Jr: The concept of synthetic lethality in the context\\nof anticancer therapy .Nat Rev Cancer 2005, 5:689-698.\\nAn excellent review detailing the concepts of synthetic lethality. Along with\\nthe reviews by Hanahan and Weinberg and Hartwell and Friend, this review\\nfrom Bill Kaelin consolidates many of the genetic concepts such assynthetic lethality that are now being exploited in novel cancer therapeutics.\\n32. Gottipati P, Vischioni B, Schultz N, Solomons J, Bryant HE,\\nDjureinovic T, Issaeva N, Sleeth K, Sharma RA, Helleday T:\\nPoly(ADP-ribose) polymerase is hyperactivated in\\nhomologous recombination-defective cells .Cancer Res 2010,\\n70:5389-5398.\\n33. Martin SA, McCabe N, Mullarkey M, Cummins R, Burgess DJ,\\nNakabeppu Y, Oka S, Kay E, Lord CJ, Ashworth A: DNA\\npolymerases as potential therapeutic targets for cancersdeﬁcient in the DNA mismatch repair proteins MSH2 or MLH1 .\\nCancer Cell 2010, 17:235-248.\\n34.\\n/C15/C15Dong Y, Li A, Wang J, Weber JD, Michel LS: Synthetic lethality\\nthrough combined Notch-epidermal growth factor receptor\\npathway inhibition in basal-like breast cancer .Cancer Res\\n2010, 70:5465-5474.\\nOne of very few reports describing an example of compensatory synthetic\\nlethality.\\n35. Bernards R, Brummelkamp TR, Beijersbergen RL: shRNA\\nlibraries and their use in cancer genetics .Nat Methods 2006,\\n3:701-706.\\n36. Iorns E, Lord CJ, Grigoriadis A, McDonald S, Fenwick K,\\nMackay A, Mein CA, Natrajan R, Savage K, Tamber N et al. :\\nIntegrated functional, gene expression and genomic analysis\\nfor the identiﬁcation of cancer targets .PLoS One 2009,\\n4:e5120.\\n37. Martin SA, McCarthy A, Barber LJ, Burgess DJ, Parry S, Lord CJ,\\nAshworth A: Methotrexate induces oxidative DNA damage and\\nis selectively lethal to tumour cells with defects in theDNA mismatch repair gene MSH2 .EMBO Mol Med 2009,\\n1:323-337.\\n38. Turner NC, Lord CJ, Iorns E, Brough R, Swift S, Elliott R, Rayter S,\\nTutt AN, Ashworth A: A synthetic lethal siRNA screen identifying\\ngenes mediating sensitivity to a PARP inhibitor .EMBO J 2008,\\n27:1368-1377.39. Lord CJ, McDonald S, Swift S, Turner NC, Ashworth A: A high-\\nthroughput RNA interference screen for DNA repair\\ndeterminants of PARP inhibitor sensitivity .DNA Repair (Amst)\\n2008, 7:2010-2019.\\n40. Iorns E, Turner NC, Elliott R, Syed N, Garrone O, Gasco M, Tutt AN,\\nCrook T, Lord CJ, Ashworth A: Identiﬁcation of CDK10 as an\\nimportant determinant of resistance to endocrine therapy for\\nbreast cancer .Cancer Cell 2008, 13:91-104.\\n41. Echeverri CJ, Beachy PA, Baum B, Boutros M, Buchholz F,\\nChanda SK, Downward J, Ellenberg J, Fraser AG, Hacohen N\\net al. :Minimizing the risk of reporting false positives in\\nlarge-scale RNAi screens .Nat Methods 2006,\\n3:777-779.\\n42. Iorns E, Lord CJ, Ashworth A: Parallel RNAi and compound\\nscreens identify the PDK1 pathway as a target for tamoxifen\\nsensitization .Biochem J 2009, 417:361-370.\\n43. Weiss WA, Taylor SS, Shokat KM: Recognizing and exploiting\\ndifferences between RNAi and small-molecule inhibitors .\\nNat Chem Biol 2007, 3:739-744.\\n44. Luo J, Emanuele MJ, Li D, Creighton CJ, Schlabach MR,\\nWestbrook TF, Wong KK, Elledge SJ: A genome-wide RNAi\\nscreen identiﬁes multiple synthetic lethal interactions with theRas oncogene .Cell 2009, 137:835-848.\\n45. Scholl C, Frohling S, Dunn IF, Schinzel AC, Barbie DA, Kim SY,\\nSilver SJ, Tamayo P, Wadlow RC, Ramaswamy S et al. :Synthetic\\nlethal interaction between oncogenic KRAS dependency and\\nSTK33 suppression in human cancer cells .Cell 2009,\\n137:821-834.\\n46. Barbie DA, Tamayo P, Boehm JS, Kim SY, Moody SE,\\nDunn IF, Schinzel AC, Sandy P, Meylan E, Scholl C et al. :\\nSystematic RNA interference reveals that oncogenic\\nKRAS-driven cancers require TBK1 .Nature 2009,\\n462:108-112.\\n47. Wang Y, Ngo VN, Marani M, Yang Y, Wright G, Staudt LM,\\nDownward J: Critical role for transcriptional repressor Snail2 in\\ntransformation by oncogenic RAS in colorectal carcinoma\\ncells .Oncogene 2010, 29:4658-4670.Searching for synthetic lethality in cancer Brough et al. 41\\nwww.sciencedirect.com Current Opinion in Genetics & Development 2011, 21:34–41', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), Document(id_='c758461f-7a5b-4d13-a078-2c5eebb38e10', embedding=None, metadata={'page_label': '613', 'file_name': 'Synthetic lethality and cancer.pdf', 'file_path': 'files/Papers_FullText/Synthetic lethality and cancer.pdf', 'file_type': 'application/pdf', 'file_size': 2781945, 'creation_date': '2024-01-10', 'last_modified_date': '2024-01-10', 'last_accessed_date': '2024-01-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, hash='61823981bbb8cc99edd7371ab8083abcecd530524a3ba804e5e20d132e8f7123', text='Advances in genome sequencing are driving a paradigm \\nshift in cancer treatment1: it is now possible to rapidly \\nidentify genetic and epigenetic changes that differen -\\ntiate tumour cells from non-tumour cells in a patient. \\nTumour-specific genetic alterations reveal not only the \\nbiological changes that drive tumour progression but \\nalso the vulnerabilities that can be exploited to selec -\\ntively target the tumour with therapeutics. Personalized \\n(or precision) genotype-targeted cancer treatment \\nhas the potential to offer individualized, highly spe -\\ncific therapies with fewer adverse effects, as well as to  \\nreduce the overtreatment of tumours. Indeed, personalized  \\nonco  genomic approaches are currently being adopted \\non the front lines of cancer care and have had success in \\nthe treatment of patients with tumours that have failed \\nto respond to standard therapies2–4.\\nCurrently, most genotype-targeted cancer thera-\\npeutics exploit the phenomenon of ‘oncogene addiction’ , \\nthat is, where a tumour is dependent on an oncogene or \\non an oncogenic pathway for survival. However, although \\nsmall-molecule and antibody-based inhibitors of onco -\\ngenes have proved to be effective for some tumour geno-\\ntypes5, not all tumours have targetable gain-of-function \\noncogenes, and therapeutic resistance is a common out -\\ncome. In these tumours, it may be possible to leverage \\nboth oncogenic and non-oncogenic mutations by iden -\\ntifying and exploiting second-site targets that, when dis -\\nrupted in conjunction with a tumour-specific mutation, \\nresult in synthetic lethality  (FIG.\\xa01 ).\\nThe concept of synthetic lethality was developed from \\ngenetic studies in model organisms, such as fruit flies6,7 \\nand yeast8–10. In the context of cancer, a synthetic lethal \\ninteraction that involves a cancer-specific mutation is \\nsometimes referred to as ‘non-oncogene addiction’ ,  \\nas the mutant tumour cell requires (that is, it is addicted \\nto) the activity of the synthetic lethal partner gene for \\nviability11 (FIG.\\xa01a,b ). The protein product of a gene that has a\\xa0synthetic lethal interaction with a frequently occur -\\nring tumour-specific somatic mutation would be an \\nexcellent anticancer drug target, because a thera  peutic \\nthat exploits the synthetic lethal interaction should result \\nin favourable therapeutic indices, in which only tumour \\ncells that harbour the mutation would be sensitive to the \\ntherapeutic (FIG.\\xa01c ). Furthermore, synthetic lethal inter -\\nactions can expand the repertoire of anticancer thera -\\npeutic targets, as they facilitate the indirect targeting of \\nnon-druggable cancer mutations through the identifi -\\ncation of a second-site synthetic lethal target that may \\nbe druggable.\\nIt was first proposed 20\\xa0years ago that synthetic lethal \\ninteractions could be used to identify new anticancer \\ndrug targets12. Given the attractive concept of synthetic \\nlethality-based therapeutics, it may be surprising that \\nonly one has so far progressed to the clinic. A major \\nhurdle has been the identification of robust, clinically \\nrelevant synthetic lethal interactions. Screening for syn -\\nthetic lethal interactions poses three major challenges. \\nFirst, by definition, these genetic interactions result in \\nlethality, making mutant recovery and identification \\ndifficult. Second, many synthetic lethal interactions are \\ncondition-dependent interactions and may not be con -\\nserved in all genetic backgrounds or under different cel -\\nlular conditions. Third, synthetic lethal interactions are \\nrare, and large numbers of mutant gene-pair combina -\\ntions need to be queried to identify synthetic lethal inter -\\nactions. For these reasons, most large-scale synthetic \\nlethal genetic interaction screens have been carried out \\nin budding yeast or fission yeast, as technologies that \\nfacilitate the high-throughput generation and analysis of \\ndouble mutants under defined laboratory conditions are \\nreadily available. Advances in RNA interference (RNAi) \\nand, more recently, CRISPR technology have now made \\nit possible to carry out large-scale unbiased synthetic \\nlethality screening directly in human cell culture.Michael Smith Laboratories, \\nUniversity of British \\nColumbia, 2185\\xa0East\\xa0Mall, \\nVancouver, \\nBritish\\xa0Columbia\\xa0V6T\\xa01Z4, \\nCanada.\\nCorrespondence to P .H. \\nhieter@msl.ubc.ca\\ndoi:10.1038/nrg.2017.47\\nPublished online 26 Jun 2017Synthetic lethality\\nA synthetic lethal interaction \\noccurs between two genes \\nwhen a perturbation  \\n(a mutation, RNA interference \\nknockdown or inhibition) that \\naffects either gene alone is \\nviable but the perturbation of \\nboth genes simultaneously is \\nlethal.Synthetic lethality and cancer\\nNigel J.\\xa0O’Neil, Melanie\\xa0L.\\xa0Bailey and Philip\\xa0Hieter\\nAbstract | A synthetic lethal interaction occurs between two genes when the perturbation of either \\ngene alone is viable but the perturbation of both genes simultaneously results in the loss of viability. \\nKey to exploiting synthetic lethality in cancer treatment are the identification and the mechanistic \\ncharacterization of robust synthetic lethal genetic interactions. Advances in next-generation \\nsequencing technologies are enabling the identification of hundreds of tumour-specific mutations \\nand alterations in gene expression that could be targeted by a synthetic lethality approach.  \\nThe translation of synthetic lethality to therapy will be assisted by the synthesis of genetic \\ninteraction data from model organisms, tumour genomes and human cell lines.\\nREVIEWS\\nNATURE REVIEWS | GENETICS  VOLUME 18 | OCTOBER 2017 | 613\\n©\\n 2017\\n Macmillan\\n Publishers\\n Limited,\\n part\\n of\\n Springer\\n Nature.\\n All\\n rights\\n reserved.', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), Document(id_='d16091dc-fe84-4420-baf4-a9df091a6bb8', embedding=None, metadata={'page_label': '614', 'file_name': 'Synthetic lethality and cancer.pdf', 'file_path': 'files/Papers_FullText/Synthetic lethality and cancer.pdf', 'file_type': 'application/pdf', 'file_size': 2781945, 'creation_date': '2024-01-10', 'last_modified_date': '2024-01-10', 'last_accessed_date': '2024-01-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, hash='238ef97b9a6745f8e73ce38a5ed5cc6ad05221f0640e02043efcfef65a47cf5c', text='Nature Reviews | GeneticsGene A\\nGene BSingle mutanta\\nb c d\\nDouble mutant\\nGene A\\nGene BGene A\\nGene BGene A\\nGene B\\nSingle mutantNormal cell\\nSingle mutant\\nGene A\\nGene BGene A\\nGene BOverexpression\\nOverexpressionViable\\nLethal\\nGene A\\nGene B\\nInhibitor\\nNon-homologous \\nend-joining\\n(NHEJ). The repair of \\ndouble-strand DNA breaks by \\ndirect ligation without the use \\nof a homologous template.In this Review, we highlight some of the advances and \\nthe challenges of exploiting synthetic lethality to iden -\\ntify new anticancer therapeutic targets, including the \\nexpansion of the synthetic lethality paradigm to include \\nsynthetic lethal interactions with gene overexpression \\nand the implications of conditional synthetic lethality.  \\nWe also discuss how large-scale synthetic lethality \\nscreening in model organisms can directly complement \\ntargeted synthetic lethality screening in human cell lines. \\nFinally, we present recent innovations that are facilitating \\nhigh-throughput synthetic lethality screening in human \\ncell\\xa0lines.\\nExpanding synthetic lethality\\nSynthetic lethality-based approaches could, in theory, \\nbe expanded beyond the targeting of loss -of-function \\nor reduction-of -function mutations in tumours.\\nSynthetic dosage lethality.  Cancer cells often exhibit \\ngene overexpression, which can result either from \\nsomatic copy number alterations13, one of the most \\ncommon alterations observed in tumours, or from \\nepigenetic changes that increase gene transcription. \\nRegardless of origin, overexpressed genes in tumours \\ncan be targeted by identifying interaction partners that \\nresult in synthetic dosage lethality (SDL) (FIG.\\xa01d ). SDL is a genetic interaction in which the overexpression  \\nof one gene, combined with the reduction in function of \\na second gene, results in lethality. The SDL phenotype \\nwas first reported in yeast14, and the concept was codi -\\nfied in a screen for genes that, when mutated, resulted \\nin lethality when the gene CTF13 , which encodes a \\ncentromere-  binding protein,  or the replication origin \\ngene ORC6 were overexpressed15.\\nThree recent SDL screens in the budding yeast \\nSaccharomyces cerevisiae  using homologues of genes \\nfrequently overexpressed in tumours have identified \\nSDL interactions that are conserved in human cancer \\ncells. MAD2 , which is overexpressed in several tumour \\ntypes, including malignant lymphoma, liver cancer, lung \\ncancer and colorectal carcinoma, exhibits SDL with the \\nknockdown or inhibition of PP2A16. CKS1B , which \\nis frequently overexpressed in breast, lung and liver \\ntumours, shows SDL with the inhibition or knockdown \\nof PLK1 (REF.\\xa0 17). Additionally, TDP1 , which is over -\\nexpressed in rhabdomyosarcoma cell lines, displays SDL \\nwith the inhibition of histone deacetylases18. SDL has \\nthe potential to open new avenues of synthetic lethality-  \\nmediated cancer therapies by indirectly targeting these \\nor other commonly overexpressed genes in tumours.\\nConditional synthetic lethality.  Tumour cells are \\nhetero  geneous and are located in diverse environments, \\nboth of which can affect genetic interactions, as many \\nare condition-dependent genetic interactions (FIG.\\xa02A ). \\nThese conditions can be intrinsic, such as the genetic \\nbackground and metabolic state of individual cells, or \\ncan be extrinsic, such as the cellular microenvironment \\nand exposure to therapeutic agents. Therefore, certain \\ngenetic interactions may not be observed under stand -\\nard laboratory conditions, and others may be depend -\\nent on specific experimental parameters. Conditional \\nsynthetic lethality could explain why some genetic \\ninteractions are specific to a cell line; these interactions \\nare often referred to as ‘context-specific’ or ‘private’ syn -\\nthetic lethal interactions, whereas interactions that are \\ncommon to many cell lines are known as ‘core’ or ‘pan’ \\nsynthetic lethal interactions.\\nGenetic background can have both positive and \\nnegative effects on synthetic lethal interactions. Some \\ngenetic interactions require the disruption of three or \\nmore genes to generate a phenotype19. For example, \\ntriple  -mutant analysis in S.\\xa0cerevisiae uncovered trigenic \\ninteractions between the partially redundant histone \\nchaperones ASF1 and CAC1 and the SWI/SNF trans -\\nlocase RDH54, the cyclins CLB5 and CLB6, and genes \\ninvolved in chromosome segregation20. Conversely, \\nbackground mutations can suppress synthetic lethal \\ninteractions, resulting in synthetic viability (FIG.\\xa02B ). For \\nexample, the loss of 53BP1 or the loss of components \\nof the non-homologous end-joining  pathway can suppress \\nthe synthetic lethal interaction between poly(ADP-  \\nribose) polymerase\\xa01 (PARP1) and the breast cancer \\nsusceptibility gene BRCA1 (REFS\\xa0 21,22). Therefore, the \\ngenetic background of a tumour, such as the loss of p53  \\nor the activation of an oncogene, could either uncover or \\nsuppress synthetic lethal interactions.Figure 1 | The concept of synthetic lethality. The loss or the inhibition of either of the \\nprotein products of gene A or B alone or the overexpression of gene A is viable (part a). \\nMutation (part b) or pharmacological inhibition (part c) of the protein product of  \\ngene\\xa0B in cells with a mutation (parts b,c) or overexpression (part d) of gene A results in \\nsynthetic lethality. The thicker arrow denotes increased expression. The star shape \\ndenotes a mutation. The red crosses denote pharmacological inhibition. Viable cells are \\ndepicted as ovals, and inviable cells are depicted as random shapes.\\nREVIEWS\\n614 | OCTOBER 2017 | VOLUME 18  www.nature.com/nrg\\n©\\n 2017\\n Macmillan\\n Publishers\\n Limited,\\n part\\n of\\n Springer\\n Nature.\\n All\\n rights\\n reserved.', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), Document(id_='637b1ab4-68e5-4831-adcd-1a4df5ef2a21', embedding=None, metadata={'page_label': '615', 'file_name': 'Synthetic lethality and cancer.pdf', 'file_path': 'files/Papers_FullText/Synthetic lethality and cancer.pdf', 'file_type': 'application/pdf', 'file_size': 2781945, 'creation_date': '2024-01-10', 'last_modified_date': '2024-01-10', 'last_accessed_date': '2024-01-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, hash='cf6128ee0669afff62666dd578395a57cc1d048d1c0dcbb1668aee8bdc2d0cc7', text='Nature Reviews | GeneticsGene A\\nGene BGene A\\nGene BNormal cell A B Tumour cell\\nGene A Gene A\\nGene B Gene B\\nGene A Gene A\\nGene B Gene BSLP inhibitorGene A\\nGene BGene A\\nGene BNormal cell Tumour cell\\nGene A Gene A\\nGene B Gene BSLP inhibitor\\nGene A Gene A\\nGene B Gene B\\nSecondary mutationHigh ROS\\nDNA-damaging agentGene A Gene A\\nGene B Gene BHypoxia Metabolic changesSuppression of synthetic lethality Aa  Conditional synthetic lethality Ab  Synthetic cytotoxicity\\nHomologous recombination\\nThe exchange of nucleotide \\nsequences between identical \\n(or near identical) DNA \\nmolecules. Homologous \\nrecombination is the most \\ncommon form of \\nhomology-directed repair of \\ndouble-strand DNA breaks.\\nGenetic interaction network\\nA genetic interaction occurs \\nwhen the perturbation of one \\nor more genes affects the \\nphenotype of another gene \\nalteration. A genetic interaction \\nnetwork defines the functional \\nrelationship between many \\ngenes.\\nOrthologue\\nGenes in different species that \\nare originated from a single \\ngene of the last common \\nancestor.It may be possible to exploit conditional synthetic \\nlethality to leverage tumour-associated conditions, such \\nas hypoxia (FIG.\\xa02Aa ), increased mutational load, replica -\\ntive stress, altered metabolism and exposure to stand -\\nard-of -care antitumour therapeutics, to increase the range \\nof synthetic lethal interactions. For example, hypoxia \\nreduces the efficacy of homologous recombination23, result -\\ning in a DNA repair-compromised state that sensitizes \\ncells to the inhibition of the repair protein PARP1 (REF.\\xa0 24). \\nTherefore, screening for synthetic lethal inter  actions \\nin hypoxic or metabolic conditions that phenocopy  \\nthe tumour microenvironment may reveal new condi -\\ntional synthetic lethal interactions with an even greater \\nspecificity for tumour cell killing.\\nThese conditional synthetic lethal interactions could \\nbe leveraged to increase the cytotoxicity of existing \\ntherapeutic agents resulting in synthetic cytotoxicity \\n(FIG.\\xa02Ab ) and thereby increase the number of potential \\ngenetic interactions that can be therapeutically exploited \\nin tumours. Many tumours are exposed to therapeutic \\nagents, such as ionizing radiation or cytotoxic drugs, \\nthat change the tumour environment and that alter the \\ngenetic interaction network  compared with untreated cells. \\nLarge-scale studies in S.\\xa0cerevisiae  have uncovered dif -\\nferential genetic interactions in response to exposure \\nto DNA-damaging agents, demonstrating that genetic \\ninteraction networks can be dynamic and can ‘re -wire’ \\nin response to changing conditions25,26. A proof-    \\nof-principle synthetic cytotoxicity screen using a mutation in the S.\\xa0cerevisiae  ATM  orthologue  TEL1 , which \\nhas few synthetic lethal partners, and a sub-lethal con -\\ncentration of the topoisomerase inhibitor camptothecin \\nincreased the number of negative genetic interaction \\npartners with the TEL1 mutant27. PARP inhibitors have \\nbeen used as a synthetic lethality-based mono  therapy \\nin homologous recombination-deficient tumours, but \\nPARP inhibitors may also increase the sensitivity of \\ntumour cells to DNA-damaging agents28; several ongoing \\nclinical trials are assessing the efficacy of PARP inhibitors \\nin combination with cytotoxic agents. For this approach \\nto work, there must be a differential sensitivity between \\ntumour cells and non-tumour cells to the combination \\nof PARP inhibitors and the DNA-damaging agent. It is \\npossible that tumour-specific mutations are the cause of \\nthe differential killing of tumour cells through the inhi -\\nbition of PARP in the presence of DNA-damaging agents \\nby a synthetic cytotoxicity mechanism. For example, in \\nglioblastoma cell lines that contain mutations in the \\ncohesin component STAG2 , PARP inhibition resulted in  \\nsynthetic cytotoxicity in response to the DNA-damaging \\nchemotherapeutic temozolomide29.\\nSynthetic lethality in model organisms\\nGiven that there are hundreds of potential tumour query \\ngenes, thousands of potential synthetic lethal-partner \\ngenes and many different conditions, high-  throughput \\napproaches are needed to identify synthetic lethal \\ninteractions. Hartwell et\\xa0al.12 first proposed that model Figure 2 | The concept of conditional synthetic lethality. Synthetic lethal interactions may be dependent on certain intrinsic \\nconditions such as genetic background, hypoxia or metabolic changes (part Aa), or extrinsic conditions such as treatment \\nwith DNA-damaging agents, which results in synthetic cytotoxicity (part Ab). Conversely, synthetic lethal interactions may be \\nsuppressed by conditions such as genetic background mutations (part B). The star shape denotes a mutation. The red crosses \\ndenote pharmacological inhibition. The lightning bolt denotes DNA damage. Viable cells are depicted as ovals, and inviable \\ncells are depicted as random shapes. ROS, reactive oxygen species; SLP , synthetic lethal partner.\\nREVIEWS\\nNATURE REVIEWS | GENETICS   VOLUME 18 | OCTOBER 2017 | 615\\n©\\n 2017\\n Macmillan\\n Publishers\\n Limited,\\n part\\n of\\n Springer\\n Nature.\\n All\\n rights\\n reserved.', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), Document(id_='7897636a-06f0-4d4d-ae50-7d81dceb5027', embedding=None, metadata={'page_label': '616', 'file_name': 'Synthetic lethality and cancer.pdf', 'file_path': 'files/Papers_FullText/Synthetic lethality and cancer.pdf', 'file_type': 'application/pdf', 'file_size': 2781945, 'creation_date': '2024-01-10', 'last_modified_date': '2024-01-10', 'last_accessed_date': '2024-01-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, hash='4ccd2cdc041aa45b48409b18d5bc147994814c59e82959b1da275a39165edff2', text='Nature Reviews | GeneticsDirect screening\\nHuman cell culture\\nCRISPR and/or RNAi-mediated\\nSL interactions\\nCandidate\\nSL interactions\\nGenotyped human\\ncell line panelsDiscovery\\nValidationIn silico  tumour\\nsequence-based\\nSL interactionsModel organism\\nSL interactions\\nCas9\\nModules\\nGroups of genes or proteins \\nthat act together in a common \\ncellular function.\\nSynthetic sickness\\nA synthetic sickness interaction \\noccurs between two genes \\nwhen a perturbation  \\n(a mutation, RNA interference \\nknockdown or inhibition) that \\naffects either gene alone is \\nviable but the disruption of \\nboth genes simultaneously \\nresults in a reduction of \\nviability.organisms, such as yeast, could enable the identifica -\\ntion of synthetic lethality-based anticancer therapeutic \\ntargets12. Large-scale screens in model organisms can \\nsurvey a much larger interaction space than is currently \\nfeasible in human cell lines and can identify potential \\ncancer-relevant synthetic lethal interactions for direct \\ntesting in human cell lines, thereby reducing the num -\\nber of genetic interaction pairs that need to be tested \\nin human cell culture (FIG.\\xa03 ). This can be achieved by \\nmining existing model organism genetic interaction \\nnetworks for synthetic lethal interactions with cancer-  \\nrelevant homologues and by direct screening with muta -\\ntions that affect homologues of cancer-associated\\xa0genes.\\nThe success of a cross-species synthetic lethality \\napproach is predicated on the conservation of genetic \\ninteractions between model organisms and human \\ncells. Although many of the genes, pathways and cel -\\nlular processes that underlie the cancer phenotypes of genomic stability and proliferation control are evolution -\\narily conserved30–32, it is difficult to predict whether a \\ngenetic interaction will be conserved, as the conserva -\\ntion of genetic interactions is gene and pathway depend -\\nent33–36. Although specific digenic interactions may not  \\nbe conserved, genetic interaction networks often \\nretain conserved interactions between biological pro -\\ncesses34,37,38, suggesting that, if direct orthologues of can -\\ncer-mutated genes are not present in a model organism, \\nthen conserved biological processes can be targeted by \\ndisrupting another gene that is involved in the same cel -\\nlular process. For example, S.\\xa0cerevisiae lacks orthologues \\nof the tumour-associated homologous re   combination \\ngenes BRCA1  and BRCA2  but instead contains other \\ngenes encoding components of the homologous \\nrecombination pathway, such as RAD52 and MRE11. \\nThese genes can be mutated to disrupt homologous \\nrecombination, providing an opportunity to screen for \\nsynthetic lethal interactions in yeast cells that are defi -\\ncient in homologous recombination. Similarly, genetic \\ninteraction networks can identify interactions between \\nfunctional modules  or processes from which synthetic \\nlethal interactions could be inferred to certain genes \\nwithin that process in human cells even if they are not \\nconserved in the model organism. For example, yeast \\nsynthetic lethal interaction networks between cohesin \\nsubunits and replication fork mediators were used to \\npredict synthetic lethality between mutations in cohesin \\nand PARP inhibitors in human cell culture, even though \\nthe PARP family of genes is not conserved in yeast29,39.\\nLarge-scale synthetic lethality genetic networks  \\nin yeast.  High-throughput mating methodologies in \\nyeast, such as synthetic genetic array (SGA) analysis40 \\nand diploid synthetic lethality analysis with microarrays \\n(dSLAM)41, enable the large-scale construction of double \\nmutants and quantification of genetic interactions but \\nare not currently available for other model organisms. \\nRecently, long-term efforts to map digenic interactions \\ncomprehensively in S.\\xa0cerevisiae  culminated in an inter -\\naction network map of more than 23 million double \\nmutants, involving ~90% of all S.\\xa0cerevisiae  genes and \\ncovering most essential and non-essential gene com -\\nbinations. This global interaction network identified \\nmore than 500,000 synthetic lethal and synthetic sickness  \\ninteractions42. In addition to creating a global network \\nof genetic interactions that can be mined for cancer -\\nrelevant synthetic lethal interactions, the principles \\nof the comprehensive S.\\xa0cerevisiae  genetic interaction \\nmap provide a context for the elucidation of synthetic \\nlethality genetic networks in human cancer cell lines. \\nFor example, although only ~1,000 genes in S.\\xa0cerevisiae  \\nare individually essential for growth, hundreds of thou -\\nsands of genetic interactions result in growth defects \\nand ~10,000 digenic interactions between non-  essential \\nmutations result in synthetic lethality42. This suggests \\nthat the potential synthetic lethal interactions with \\ntumour mutations that can be mined for drug targets \\nis much larger than the space that can be targeted by \\nexploiting the inhibition of oncogenes alone, increasing \\nthe probability of finding druggable targets.Figure 3 | A cross-platform approach for discovering clinically relevant synthetic \\nlethal interactions. Synthetic lethal (SL) interaction data from model organisms, such as \\ndata from TheCellMap  or BioGRID , and tumour genomic sequence can be mined to \\ndiscover new candidate SL interactions for validation in human cell lines, or direct SL \\ninteraction screening can be undertaken in human cell lines. Candidate SL interactions \\ncan be validated in genotypically defined cell line panels to determine whether SL \\ninteractions are conserved across cell lines or are limited to specific cell lines. RNAi, RNA \\ninterference.\\nREVIEWS\\n616 | OCTOBER 2017 | VOLUME 18  www.nature.com/nrg\\n©\\n 2017\\n Macmillan\\n Publishers\\n Limited,\\n part\\n of\\n Springer\\n Nature.\\n All\\n rights\\n reserved.', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), Document(id_='7d92ec60-94be-4f0b-b7d7-7e802e1c0929', embedding=None, metadata={'page_label': '617', 'file_name': 'Synthetic lethality and cancer.pdf', 'file_path': 'files/Papers_FullText/Synthetic lethality and cancer.pdf', 'file_type': 'application/pdf', 'file_size': 2781945, 'creation_date': '2024-01-10', 'last_modified_date': '2024-01-10', 'last_accessed_date': '2024-01-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, hash='02fef989d07ffc9787751ebf80e5eef43af6bffb223196ff21bdace93c2eb7d4', text='Driver mutations\\nMutations that confer  \\na selective growth advantage \\nto a cancer or a pre-cancerous \\ncell.\\nIsogenic\\nOrganisms or cell lines that \\ncontain identical or nearly \\nidentical genotypes.\\nHeterogeneity\\nIn a cell line or tumour, the \\ndiversity of genotypes within \\nthe population.Query-specific screens in yeast have been successful \\nin identifying synthetic lethal interactions that are con -\\nserved in human cell culture17,39,43. The utility of synthetic \\nlethal interaction screens in yeast for identifying candi -\\ndates to be directly tested in human tumours was recently \\nshown by a multi-query study44. Srivas et\\xa0al.  used 111 \\nS.\\xa0cerevisiae  orthologues of human cancer driver muta -\\ntions  as query mutations with an array of 433 S.\\xa0cerevisiae  \\northologues of potentially druggable genes. This screen \\nidentified 1,420 synthetic lethal interactions. The yeast \\nsynthetic lethal interactions were used to guide a\\xa0tumour \\nsuppressor mutation–drug interaction screen in human \\ncell culture of 21 drugs by 112 tumour suppressor gene \\nRNAi knockdowns. The authors found that strong \\ngenetic interactions that were observed in S.\\xa0cerevisiae  \\nwere more likely to be conserved in human cells44, vali -\\ndating the idea that screening in yeast (or another model \\norganism) can narrow the number of candidate synthetic \\nlethal interactions for testing in human cells.\\nSynthetic lethality-based screening in human cells\\nIn the search for candidate synthetic lethal targets, \\nunbiased synthetic lethality screens in any organism \\nprovide mechanistic and therapeutic insights that are \\nnot always intuitive. Compared with lower organisms \\nsuch as yeast, the systematic interrogation of synthetic \\nlethal interactions in human cells is still in its infancy. \\nAs human cells cannot be manipulated through genetic \\nmating techniques, other technologies were required \\nbefore high-throughput screens became feasible. \\nRNAi-based gene targeting provided the first oppor -\\ntunity for easily scalable gene-specific manipulation \\nin human cells, although it has since been viewed by \\nsome as a cautionary tale of the challenges of screen -\\ning larger and more-complex genomes. Nevertheless, \\nthe goal of comprehensively mapping synthetic lethal \\ninteractions in human cells on a genome-wide scale \\nis not insurmountable, especially as the discovery \\nof CRISPR and its application to human synthetic \\nlethality-  based screening has transformed the field of \\nhuman genetics.\\nRNAi and the evolution of the large-scale screen.  \\nGenome-wide studies in human cells would arguably \\nnot have been possible without the development of \\nRNAi technology to specifically knockdown mRNAs \\nthrough the introduction of exogenous small interfer -\\ning RNA (siRNA) sequences45–47. Applied to large-scale \\nloss-of-function screening, this technology launched \\nthe era of synthetic lethality-based high-throughput \\nscreening in human cells, with studies in transformed \\nor immortalized human cell lines being the most com -\\nmon48–51. Although synthetic lethality screens using \\nRNAi have made an important contribution to human \\nfunctional genomics, the minimal overlap reported \\nbetween independent RNAi screens has raised the issue \\nof their reliability in annotating gene function52. Gene \\nspecificity of the RNAi library and the limitations of \\nthe human cell culture system in which the library is \\ntested are two general areas of concern in human RNAi \\nscreening.Considerable efforts have been made in recent \\nyears to improve the efficacy of RNAi libraries so that \\nscreens can be carried out with higher confidence. The \\nproblem of RNAi reagent specificity has been tackled \\nby increasing the on -target robustness and reducing \\nthe off-target effects of RNAi sequences53–56. Library \\ngeneration has also been improved through the incor -\\nporation of bio  informatics algorithms and tools53,57,58. \\nFurthermore, the analysis of large-scale RNAi screens has  \\nemphasized the importance of reducing false positives \\nby observing consistent phenotypes with multiple RNAi \\nreagents that target the same gene51,59,60.\\nUnfortunately, the generation of more-robust RNAi \\nlibraries does not address the problem of screening in \\ncell lines. Regardless of whether a synthetic lethality \\nscreen is carried out in a cell line panel or in isogenic  pairs \\n(FIG.\\xa04 ), cancer cell lines are still a model system. They \\noften do not reflect the genotype or the heterogeneity  \\nof actual tumours, and they can be highly adaptable to \\nthe conditions and practices of cell culture in individ -\\nual laboratories. Indeed, multiple screens that target the \\nsame cancer mutation have often reported highly varia -\\nble, non-  overlapping results52,61. These issues of irrepro -\\nducibility are not limited to RNAi screening but are also \\na\\xa0matter of debate for large-scale chemical screening62–64 \\nand underscore the need to confirm synthetic lethal \\ninteractions across multiple cell line contexts before  \\nproceeding with drug development.\\nNotably, the problems of both library specificity and \\ncell line context still exist for genome-wide knockout \\nscreens using the CRISPR system. Nonetheless, RNAi \\nscreens have undoubtedly paved the way for screens \\nusing CRISPR technology and have brought to the fore -\\nfront the challenges and limitations of human synthetic \\nlethality screening regardless of the gene-targeting strat -\\negy used. Although CRISPR has become the method \\nof choice for human genome-wide screening, RNAi \\nremains a powerful tool that can be used as a comple -\\nmentary approach in screens or during hit validation. \\nMoreover, this technology has the benefit of incomplete \\nand transient knockdown phenotypes that might differ \\nfrom CRISPR-generated complete knockout and may \\nmore closely parallel those that occur with chemical \\ninhibition during therapeutic application.\\nGenome-wide screens using CRISPR–Cas9.  In contrast to \\nRNAi knockdown, the CRISPR–Cas9 system targets DNA \\n(BOX\\xa01 ), which is the basis for several differences between \\nthe technologies65–67. Unlike RNAi, single-guide RNAs \\n(sgRNAs) can be designed for any DNA sequence in the \\nnuclear genome, including intragenic DNA. Furthermore, \\nCRISPR does not affect the endogenous RNAi pathway, \\nalthough cell repair pathways are needed to repair the \\ndouble-strand break (DSB). Finally, the knockout events \\nthat are produced by CRISPR are irreversible and do not \\ndepend on the long-term constitutive expression of trans -\\ngenes in the cell, a property that can greatly facilitate the \\nidentification of synthetic lethal targets.\\nEven though the technology has only recently been \\ndiscovered, researchers have learned from strategies \\nthat enhance RNAi library specificity and have made \\nREVIEWS\\nNATURE REVIEWS | GENETICS   VOLUME 18 | OCTOBER 2017 | 617\\n©\\n 2017\\n Macmillan\\n Publishers\\n Limited,\\n part\\n of\\n Springer\\n Nature.\\n All\\n rights\\n reserved.', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), Document(id_='e23e45b1-538c-47f7-a9a7-aff128a6cc5d', embedding=None, metadata={'page_label': '618', 'file_name': 'Synthetic lethality and cancer.pdf', 'file_path': 'files/Papers_FullText/Synthetic lethality and cancer.pdf', 'file_type': 'application/pdf', 'file_size': 2781945, 'creation_date': '2024-01-10', 'last_modified_date': '2024-01-10', 'last_accessed_date': '2024-01-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, hash='19cf456d42fdb2324df86e5ba8068a563630f3f5e336ffe6399922b3c8f4e22c', text='Nature Reviews | GeneticsRNAi or sgRNA lentiviral library\\nCell line outgrowth\\nIsolation of genomic DNA and PCR\\nampliﬁcation\\nDetermine common\\nessential genes\\nCandidate\\nSL targetsCandidate\\nSL targets\\nCell line outgrowth\\nIsolation of genomic DNA\\nand PCR ampliﬁcationWild-type cells Mutant or knockout cellsPanel of wild-type cell lines Panel of mutant or knockout cell lines\\nDetermine mutant or \\nknockout-speciﬁc essential genesComparison of RNAi or sgRNA\\nabundanceInfection and selection for \\nvirus-containing cellsInfection and selection\\nfor virus-containing cellsRNAi or sgRNA lentiviral library a b\\nextensive efforts to improve sgRNA targeting. Although \\nthe refinement of the features of on -target robustness \\nand an accounting of the full complement of off-target \\neffects are still ongoing (for a review, see REF.\\xa0 68), steps \\nare currently being taken to incorporate these results \\ninto library design and have, so far, evolved a second \\ngeneration of genome-wide CRISPR libraries that \\nhave exciting prospects for uncovering cancer-specific  \\nsynthetic lethal targets69–73.\\nLoss -of-function screens for essential genes using \\nsecond-generation sgRNA libraries have consistently \\noutperformed RNAi screens70–72. Furthermore, the \\nexpansion of these genome-wide screens across mul -\\ntiple cell lines has begun to uncover both core and cell line-specific fitness genes. Determining synthetic \\nlethal targets using this method will probably involve \\nextending these genome-wide screens to panels of cell \\nlines in which cancer mutations of interest are known74. \\nAlternatively, CRISPR–Cas9 synthetic lethality screens \\ncan be carried out on isogenic pairs, especially as the \\nCRISPR technology itself can generate collections of \\ncell knockouts either by targeting the same gene across \\nmultiple cell backgrounds or by producing robust \\ngenetic synthetic lethality networks for multiple genes \\nthat encode components of a protein complex or a sig -\\nnalling pathway of interest75. Such networks are likely \\nto become more common as collections of cell line  \\nknockouts become more accessible.Figure 4 | Strategy for large-scale synthetic lethality screens for a gene of interest in human cells. Human synthetic \\nlethality screens most commonly use either pairs of ‘matched’ or isogenic cell lines of the same background that have only \\nthe gene of interest mutated or inactivated (part a) or a panel of genetically diverse cell lines that are split into two groups \\ndepending on the status of the gene of interest (part b). After infection with a pool of lentivirus containing gene-specific \\nsmall interfering RNAs (siRNAs) or single-guide RNAs (sgRNAs), cell populations are grown and next-generation \\nsequencing technologies are used to identify sequences that are underrepresented (or ‘drop out’) solely in the cell line \\npopulations that are mutated or inactivated for the gene of interest. Genes targeted by multiple siRNAs or sgRNAs in this \\nsubset are candidate synthetic lethal (SL) partners for the gene of interest.\\nREVIEWS\\n618 | OCTOBER 2017 | VOLUME 18  www.nature.com/nrg\\n©\\n 2017\\n Macmillan\\n Publishers\\n Limited,\\n part\\n of\\n Springer\\n Nature.\\n All\\n rights\\n reserved.', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), Document(id_='ba36de67-1554-466e-a1c9-c9472a85c0b0', embedding=None, metadata={'page_label': '619', 'file_name': 'Synthetic lethality and cancer.pdf', 'file_path': 'files/Papers_FullText/Synthetic lethality and cancer.pdf', 'file_type': 'application/pdf', 'file_size': 2781945, 'creation_date': '2024-01-10', 'last_modified_date': '2024-01-10', 'last_accessed_date': '2024-01-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, hash='05bde4dd76fcf753f0c90b29a87079f69bc24806ad191e1fbeb64d176aaba5ae', text='Nature Reviews | GeneticsCas9\\nsgRNA\\nIndel mutation\\nIn-frame deletion Frameshift mutation\\nmRNA decayDSB3/uni20325/uni2032\\n5/uni20323/uni2032\\n3/uni20325/uni2032\\n3/uni20325/uni2032\\n5/uni20323/uni2032\\n3/uni20325/uni2032\\n5/uni20323/uni20325/uni20323/uni2032Normal\\nrepair \\nHR NHEJ\\nAlthough the era of synthetic lethality screening \\nusing CRISPR is only just beginning and holds great \\npromise for therapeutic discovery, this technology still \\nhas its limitations. First, as knockouts are generated \\nindependently by errors in DSB repair, individual cells \\nwithin a population may contain the same sgRNA but \\nhave different mutations in the target sequence. This \\ncan result in a mix of phenotypes that can add more \\nnoise to large-scale screens, depending on the gene or \\nthe sgRNA region that is targeted68. Studies have also \\nshown that CRISPR targeting and DSB generation are \\naffected by chromatin structure, a feature that can vary \\nbetween cell lines73,76,77. Similarly, amplified genomic regions also vary between cell lines, especially in those \\nthat are highly aneuploid, and recent experiments have \\ndemonstrated a high level of toxicity after genome edit -\\ning in these regions69,70. This problem of high toxicity \\ncan be addressed by using multiple cell lines that cover \\na range of genomic contexts, which is already a necessity \\nin human synthetic lethality screening. Alternatively, as \\nthe toxicity of\\xa0CRISPR–Cas9 in amplified regions seems \\nto be related to the DSBs generated by wild-type Cas9, \\nCRISPR interference (CRISPRi) can be used to validate \\nhits in these regions.\\nExploring genetic networks with CRISPR.  CRISPR \\ntechnology has been adapted for the large-scale map -\\nping of genetic networks in human cells. Both CRISPRi \\nand CRISPR activation (CRISPRa) take advantage \\nof epigenetic modifications and use a catalytically \\ndead mutant of Cas9 (dCas9) fused to either a tran -\\nscriptional repressor (CRISPRi) or a transcriptional \\nactivator (CRISPRa) domain78,79. When these Cas9–\\nchromatin modifier fusion proteins are recruited by \\nan sgRNA to bind near the promoter regions of target \\ngenes, they can regulate changes in gene expression \\nlevels. Improvements in sgRNA design and the addi -\\ntion of scaffolds to recruit more transcriptional modu -\\nlators have made CRISPRi and CRISPRa reliable tools \\nfor high-throughput screening73,80,81. Unlike genome-  \\nediting screens, CRISPRi and CRISPRa screens are \\nreversible and inducible, which allows the examina -\\ntion of the spatiotemporal dynamics of gene function. \\nCRISPRi and CRISPRa screens are also complemen -\\ntary and, when used in conjunction, may generate a \\nmore robust data set that explores biochemical and cell  \\nsignalling pathways73,80.\\nAs a method that investigates a larger genetic inter -\\naction space in a single experiment, combinatorial \\ngenetics en\\xa0masse  (CombiGEM)–CRISPR can gener -\\nate global synthetic lethality networks with great effi -\\nciency82. CombiGEM–CRISPR takes advantage of the \\nobservation that sgRNAs can be multiplexed, with up to \\nsix independent sgRNAs being combined for co -editing \\nevents in one cell83,84. Barcoded CombiGEM pairwise \\nlibraries are generated through a two-step cloning pro -\\ncess and consist of all possible pairwise combinations \\nof sgRNAs. This pool can then be screened in a cell \\nline for the pairs of sgRNAs that produce proliferation \\ndefects and analysed for negative genetic interactions82. \\nSmaller, targeted pairwise synthetic lethality screens \\nusing a multiplex approach have already investigated \\ninteractions that are relevant to human cancers85,86. \\nCombiGEM–CRISPR provides several other exciting \\nopportunities for synthetic lethality screening that \\nextend beyond pairwise genetic interaction screens. \\nSubsequent cloning steps with CombiGEM–CRISPR \\ncan generate trigenic and higher-order libraries that \\ncould be useful for uncovering interactions that consist \\nof multiple genes. Similarly, the screening of the same \\npairwise CombiGEM–CRISPR library under multiple \\nconditions (for example, the presence or the absence \\nof a drug) can also reveal context-dependent synthetic \\nlethal interactions87.Box 1 | CRISPR–Cas9 gene editing in synthetic lethality screening\\nThe CRISPR–Cas system was discovered as a form of immunity in bacteria and adapted \\nfor use in genome editing in other organisms ranging from yeast to human cells.  \\nAs a genome-editing technique, CRISPR is relatively new, but it has rapidly become an \\nimportant tool for synthetic lethality screening in mammalian cells. The CRISPR–Cas9 \\nribonucleoprotein complex is composed of a single-guide RNA (sgRNA) that enables \\ntargeting of the Cas9 endonuclease to specific sequences in the genome, where Cas9 \\nintroduces a blunt-ended double-strand break (DSB) that then needs to be repaired65,67 \\n(see the figure). Repair can occur through either the homologous recombination (HR) \\nrepair pathway or end-joining pathways such as non-homologous end-joining (NHEJ) and \\nalternative end-joining (Alt -EJ). HR can be carried out using either the sister chromatid or \\nexogenous donor DNA if it is available, although repair using donor DNA occurs at a very \\nlow rate and is not currently useful for screening.\\nEnd-joining pathways result in small insertions and/or deletions (indels). These small \\nindels are normally selected for in CRISPR screens, as normal error-free repair will \\nre-establish the endogenous sequence, which can then be re -cut by the sgRNA-guided \\nCas9 to restart the cycle. Libraries of sgRNAs for screening are designed to target the open \\nreading frames (ORFs) of the genome69–73. Indels in the ORF can result in either a frameshift \\nthat creates a knockout of the allele through nonsense-mediated decay of the mRNA or \\ntruncation of the protein or an in -frame mutation that may or may not exert a phenotypic \\neffect depending on the structural or functional importance of the altered region70.\\nREVIEWS\\nNATURE REVIEWS | GENETICS   VOLUME 18 | OCTOBER 2017 | 619\\n©\\n 2017\\n Macmillan\\n Publishers\\n Limited,\\n part\\n of\\n Springer\\n Nature.\\n All\\n rights\\n reserved.', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), Document(id_='2af73021-7942-4e30-bd23-d167413d5d23', embedding=None, metadata={'page_label': '620', 'file_name': 'Synthetic lethality and cancer.pdf', 'file_path': 'files/Papers_FullText/Synthetic lethality and cancer.pdf', 'file_type': 'application/pdf', 'file_size': 2781945, 'creation_date': '2024-01-10', 'last_modified_date': '2024-01-10', 'last_accessed_date': '2024-01-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, hash='fd7a1bf4fd4cfa30c0993289b9fc8b32efa295d079a74b94b131768386eeed49', text='Gene essentiality profiles\\nSets of genes required for \\nproliferation or viability in the \\ncontext of a single cell line or \\ntumour type.Until recently, synthetic lethality screens using \\nCRISPR have only looked at the effects of fitness on cell \\npopulations. However, the method of Perturb-seq (also \\nknown as CRISP-seq and CRISPR droplet sequencing \\n(CROP-seq)) combines the effects of CRISPR altera -\\ntions with RNA-sequencing at the single-cell level88–91. \\nThis approach has the advantage of providing a tran -\\nscriptome-wide view of the state of each cell after the \\nknockout or the knockdown of a CRISPR target and \\navoids averaging phenotypic effects over the cell pop -\\nulation. Perturb-seq can also look at the effects of com -\\nbinations of sgRNAs, which is a useful technique for \\nsynthetic lethality research89. Furthermore, unlike tra -\\nditional screening methods, Perturb-seq can go beyond \\nmeasurements of cell fitness and can provide extensive \\nmolecular and pathway information on a single-cell \\nbasis, which could make the approach invaluable for the \\nelucidation of synthetic lethal interaction mechanisms.\\nScreening using statistical genetics\\nThe use of population statistics to predict genetic inter -\\nactions is not a new technique, but current efforts have \\nbenefited from taking advantage of the rapidly accu -\\nmulating cancer genome data that have become avail -\\nable from next-generation sequencing. Researchers \\ncan mine databases, such as The Cancer Genome Atlas  \\n(TCGA)  or the Catalogue of Somatic Mutations in \\nCancer (COSMIC),  to search for patterns of mutations \\nthat are mutually exclusive. These mutual exclusivity \\nanalyses have the advantage of using actual patient data, \\nbut they have a bias towards cancer mutations that have \\na\\xa0higher frequency, an issue that has been addressed in \\nmore-recent statistical algorithms92,93. Furthermore, in \\ncomputational analyses, the accuracy and the cover -\\nage of predicted networks improve as the size and the \\nquality of the data improve. Efforts to improve mutual \\nexclusivity analysis include using a hybrid model with \\nknowledge of cell signalling pathways and analysing not \\nonly copy number alteration data but also gene mutation \\nand expression data92,94,95.\\nWhether mutual exclusivity occurs because the muta -\\ntion of both members of the gene pair has no selective \\nadvantage or because there is a selective disadvantage is \\ncurrently impossible to distinguish. In fact, it is possible \\nthat different gene pairs are mutually exclusive for dif -\\nferent reasons depending on gene function96. Knowing \\nthe reason behind the mutual exclusivity of two genes \\nis crucial for synthetic lethality prediction, as double \\nmutation in a synthetic lethal gene pair should result in \\na\\xa0selective disadvantage for cell\\xa0growth.\\nAnother approach to predict synthetic lethal inter -\\nactions using bioinformatics is to combine population \\ndata with data from experimental studies. Jerby-Arnon \\net\\xa0al.97 developed DAISY , a computational pipeline that \\nintegrates three data sources: first, somatic copy number \\nalterations and mutation data from clinical samples and \\ncell lines; second, gene essentiality profiles  from cell line \\nRNAi screens; and third, cell line gene expression data. \\nA combined interaction score using all three approaches \\npredicts candidate synthetic lethality (as well as SDL) \\ngene pairs. A subset of candidate synthetic lethal pairs predicted by the DAISY method has been validated in \\ncells using RNAi and drugs, whereas a DAISY-generated \\ngenome-wide synthetic lethality network has shown \\npromise in predicting clinical prognosis in breast can -\\ncer97. This integrative method suggests that successful \\nin\\xa0silico  approaches will combine genomic and expres -\\nsion data from patient samples with experimental data \\nfrom human cell lines. This study also highlights the \\nrole of statistical genetics in prioritizing target synthetic \\nlethal pairs for experimental validation when a data set \\nis too large to investigate practically.\\nTranslating synthetic lethality into therapeutics\\nIn theory, there are three benefits to using a synthetic \\nlethality-based therapeutic strategy. First, synthetic lethal -\\nity is selective for a cancer cell-specific genetic mutation, \\nwhich should make it easy to identify patient respond -\\ners. Second, when there is a large therapeutic window, \\nchemotherapeutic treatment should have limited adverse \\neffects, and lower drug doses should be efficacious. Third, \\nthe strategy can be applied to any type of cancer muta -\\ntion, including tumour suppressors and those deemed \\nundruggable. In practice, however, only one synthetic \\nlethal interaction — that between PARP and BRCA1 \\nand/or BRCA2 — has made the journey from discovery \\nto clinical approval, although it has not done so without \\novercoming considerable obstacles (BOX\\xa02 ). Synthetic \\nlethality databases, such as SynLethDB98, contain tens of \\nthousands of interactions from the published literature; so \\nwhy are there not more synthetic lethal interactions that \\nshare the success story of PARP?\\nIt is difficult, if not impossible, for basic research -\\ners to account for all the variables of a patient setting. \\nHowever, clearly, additional parameters need to be deter -\\nmined for a synthetic lethal interaction before it can be \\nconsidered for therapeutic application, even when a drug \\nis available. With the recent publication of a comprehen -\\nsive, single-condition yeast negative genetic interaction \\nmap and the addition of CRISPR screening approaches \\nin human cells42,69,72–74, the future is bright for synthetic \\nlethality candidate screening, and perhaps now is the \\ntime to re -evaluate how we can best prioritize candidates \\nfor therapeutic testing.\\nSynthetic lethal candidates for cancer therapeutics \\nare available from three sources: conservation from \\nmodel organism genetic networks, human synthetic \\nlethality screening and in\\xa0silico  predictions (FIG.\\xa03 ). One \\nstraightforward method to increase the probability \\nof finding true synthetic lethal interactions for thera -\\npeutics is to integrate data across these three platforms. \\nIn addition, to address the problems of irre  producibility \\nin cell culture, synthetic lethal interactions need to be \\nconfirmed across multiple cell culture contexts and \\nin\\xa0vivo . Fortunately, cell culture techniques for cancer \\ncells have advanced greatly in the past few years, with \\nexperiments using 3D cell cultures, patient-derived \\norganoids, and larger cell line and biobank repositories \\nbecoming more common99–101. Notably, although the \\nuse of animals can limit the scale of the experiment, \\nloss-of-function screens can also be carried out in\\xa0vivo , \\nwith micro  environmental, immune and metabolic \\nREVIEWS\\n620 | OCTOBER 2017 | VOLUME 18  www.nature.com/nrg\\n©\\n 2017\\n Macmillan\\n Publishers\\n Limited,\\n part\\n of\\n Springer\\n Nature.\\n All\\n rights\\n reserved.', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), Document(id_='26bbae4e-4f62-441c-ad5c-d3a4b3352dae', embedding=None, metadata={'page_label': '621', 'file_name': 'Synthetic lethality and cancer.pdf', 'file_path': 'files/Papers_FullText/Synthetic lethality and cancer.pdf', 'file_type': 'application/pdf', 'file_size': 2781945, 'creation_date': '2024-01-10', 'last_modified_date': '2024-01-10', 'last_accessed_date': '2024-01-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, hash='f295ebfd0098fd90ba4b2ae4dbf088d1499cd888060a7f3fbc434bf249a25a9d', text='Nature Reviews | GeneticsHR+HR–Normal cell\\nDNA repaired\\nCell survivalDNA repaired\\nCell survival Cell deathBRCA -mutated cell Resistance mutation\\n(e.g. TP53BP1 loss)\\nPartial restoration of\\nHRDouble-strand breaksPARPPARP inhibition\\nand trapping\\nBox 2 | PARP inhibitors in synthetic lethality-based therapeutics: from discovery to the clinic\\nSynthetic lethality is a useful approach to uncover biological mechanisms in both normal and cancerous cells,  \\nbut the ultimate therapeutic goal is to add another weapon to the arsenal of personalized medicine. The synthetic \\nlethal interaction of poly(ADP-ribose) polymerase (PARP) with the breast cancer susceptibility genes BRCA1  and \\nBRCA2  was discovered by hypothesis-driven direct testing in 2005 (REFS\\xa0 106,107). PARP inhibitors were first used \\nin clinical trials as a synthetic lethal therapy in BRCA1/2  germline-mutated tumours in 2009 (REF.\\xa0 108) and \\ncurrently represent the only synthetic lethality strategy that has been approved for use in patients with cancer.  \\nAs such, it is useful to review its journey from discovery to the clinic and to examine the challenges that may need \\nto be addressed for other synthetic lethal targets that enter the clinic.\\nIdentification of an appropriate biomarker\\nAlthough early clinical trials focused on the efficacy of PARP inhibitors in germline-mutated BRCA1/2  tumours108,  \\nit was later discovered that the response was not necessarily limited to those with carrier mutations109. The \\nsensitivity of BRCA1/2 -mutated tumours to PARP inhibitors has been broadly linked to their defects in homologous \\nrecombination (HR) repair (see the figure). In an effort to increase the number of patients who might benefit from \\nthis synthetic lethality strategy, there have been efforts to identify other gene mutations in the HR pathway that are \\nsensitive to PARP inhibitors110,111. However, not all mutations in a gene are equal in their effect on function. Similarly, \\nthe function of each gene in the pathway is unlikely to contribute equally to HR repair efficiency. This has led to \\ncurrent efforts to determine a ‘HR repair defective’ molecular signature using global profiling analyses that might \\nbe a better predictor of drug response112.\\nInvestigation of resistance mechanisms\\nThe possibility of resistance to PARP inhibitors was raised when it was discovered that not all BRCA1/2 -mutated \\ntumours were responsive to synthetic lethality intervention108,109. Known resistance mechanisms primarily involve \\nthe narrowing of the original synthetic lethality therapeutic window through the restoration of HR (see the figure). \\nThis can occur either genetically, through secondary mutations that partially restore the HR function of the BRCA1  \\nand/or BRCA2  genes113,114, or functionally, through mutation and the downregulation of compensatory repair \\npathways22,115. It is also important to consider how PARP inhibitors can indirectly modulate their toxicity. This \\nincludes their drug-specific off-target effects116,117, as well as the downregulation of PARP-mediated non-DNA \\nrepair functions such as the regulation of transcription or cellular stress118,119.\\nElucidation of mechanism\\nAlthough PARP inhibitors have shown promise as a synthetic lethal therapeutic in the clinic, knowledge of the \\nbiological mechanism behind this lethality is still incomplete. One matter that remains under investigation is \\nwhether PARP inhibitors work mainly through catalytic inhibition or through the trapping of PARP proteins on the \\nDNA120,121. Researchers are also more deeply pursuing the mechanism of lethality specific to HR-deficient cells, an \\nissue that is partly complicated by the fact that both PARP and HR functions intersect at double-strand break \\nrepair and replication fork restart112,122.\\nUltimately, these three hurdles in the PARP synthetic lethality journey remind us that, although synthetic \\nlethality-based cancer therapy is derived from a genetic interaction, the lethality still occurs at the phenotypic \\nlevel. Therefore, the determination of the biological mechanism or mechanisms of synthetic lethality is not merely \\nincidental but is fundamental to the understanding of therapeutic parameters such as dose, patient selection and \\nthe possibility of tumour resistance.\\nREVIEWS\\nNATURE REVIEWS | GENETICS   VOLUME 18 | OCTOBER 2017 | 621\\n©\\n 2017\\n Macmillan\\n Publishers\\n Limited,\\n part\\n of\\n Springer\\n Nature.\\n All\\n rights\\n reserved.', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), Document(id_='80d5059b-e3d4-49ec-a95e-e1b499d1dce5', embedding=None, metadata={'page_label': '622', 'file_name': 'Synthetic lethality and cancer.pdf', 'file_path': 'files/Papers_FullText/Synthetic lethality and cancer.pdf', 'file_type': 'application/pdf', 'file_size': 2781945, 'creation_date': '2024-01-10', 'last_modified_date': '2024-01-10', 'last_accessed_date': '2024-01-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, hash='05c891bf9b3df7c84648b2fb8f8cd61b9967373b65ba0bfb447a067e3048088a', text='conditions aligning more closely with those found in \\ntumours102,103. Finally, we propose extending preclinical \\nstudies to include several other criteria for the potential \\nsynthetic lethal therapeutic interaction, including the \\neffect of standard chemotherapeutics, the prevalence \\nand heterogeneity of the biomarker, possible resistance \\nmechanisms and off-target factors, and a thorough elu -\\ncidation of the biological mechanism. These practices \\ncan all help to determine the context dependency and \\ntherapeutic applicability of a synthetic lethal interaction.\\nIt may also be necessary to evaluate potential  \\nsynthetic lethal interactions beyond their potential to \\ntreat cancer in order to include those that can be applied \\nto cancer prevention104. This could involve, for example, \\nusing PARP inhibitors to prevent breast cancer in women \\nwith BRCA1  or BRCA2  mutations before cancers become \\ndetectable. Evidence for this synthetic lethality-based \\nchemopreventive strategy is still in its early stages \\nand is mainly being gathered using animal models104.  In theory, a synthetic lethality-based strategy may be more \\neffective before pre-malignant lesions become highly \\nhetero  geneous. It also needs to be determined whether \\nlower or intermittent drug doses will be effective for this  \\nstrategy to avoid the toxicity that is associated with high \\ncontinuous doses of chemotherapeutics105.\\nConclusions\\nSynthetic lethality is a simple genetic concept that contin -\\nues to have a major impact on cancer research. Synthetic \\nlethal interaction data from model organisms, bio -\\ninformatic predictions using tumour genome-  sequencing \\ndata and direct screening of human cancer lines have \\nidentified potential drug targets. However, these data have \\nalso highlighted the mechanistic complexities underlying \\ncancer phenotypes that will need to be unravelled to suc -\\ncessfully identify the specific targets that can be used for \\nthe selective killing of tumour cells and to realize the full \\npotential of personalized anticancer therapeutics.\\n1. Hyman,\\xa0D.\\xa0M., T aylor,\\xa0B.\\xa0S. & Baselga,\\xa0J. \\nImplementing genome-driven oncology. Cell 168, \\n584–599 (2017).\\n2. Laskin,\\xa0J.  et\\xa0al.  Lessons learned from the application \\nof whole-genome analysis to the treatment of patients \\nwith advanced cancers. Cold Spring Harb. Mol. Case \\nStud. 1, a000570 (2015).\\n3. Stockley,\\xa0T .\\xa0L.  et\\xa0al.  Molecular profiling of advanced \\nsolid tumors and patient outcomes with genotype-\\nmatched clinical trials: the Princess Margaret IMPACT/\\nCOMPACT trial. Genome Med. 8, 109 (2016).\\n4. Swanton,\\xa0C.  et\\xa0al.  Consensus on precision medicine for \\nmetastatic cancers: a report from the MAP conference. \\nAnn. Oncol. 27, 1443–1448 (2016).\\n5. Pagliarini,\\xa0R., Shao,\\xa0W. & Sellers,\\xa0W.\\xa0R. Oncogene \\naddiction: pathways of therapeutic response, \\nresistance, and road maps toward a cure. EMBO Rep. \\n16, 280–296 (2015).\\n6. Dobzhansky,\\xa0T . Genetics of natural populations; \\nrecombination and variability in populations of Drosophila \\npseudoobscura . Genetics 31, 269–290 (1946).\\n7. Lucchesi,\\xa0J.\\xa0C. Synthetic lethality and semi-lethality \\namong functionally related mutants of Drosophila \\nmelanogaster . Genetics 59, 37–44 (1968).\\n8. Kaiser,\\xa0C.\\xa0A. & Schekman,\\xa0R. Distinct sets of SEC genes \\ngovern transport vesicle formation and fusion early in \\nthe secretory pathway. Cell 61, 723–733 (1990).\\n9. Hennessy,\\xa0K.\\xa0M., Lee,\\xa0A., Chen,\\xa0E. & Botstein,\\xa0D.  \\nA group of interacting yeast DNA replication genes. \\nGenes Dev. 5, 958–969 (1991).\\n10. Bender,\\xa0A. & Pringle,\\xa0J.\\xa0R. Use of a screen for synthetic \\nlethal and multicopy suppressee mutants to identify \\ntwo new genes involved in morphogenesis in \\nSaccharomyces cerevisiae. Mol. Cell. Biol. 11,  \\n1295–1305 (1991).\\n11. Nagel,\\xa0R., Semenova,\\xa0E.\\xa0A. & Berns,\\xa0A. Drugging the \\naddict: non-oncogene addiction as a target for cancer \\ntherapy. EMBO Rep. 17, 1516–1531 (2016).\\n12. Hartwell,\\xa0L.\\xa0H., Szankasi,\\xa0P ., Roberts,\\xa0C.\\xa0J., \\nMurray,\\xa0A.\\xa0W. & Friend,\\xa0S.\\xa0H. Integrating genetic \\napproaches into the discovery of anticancer drugs. \\nScience 278, 1064–1068 (1997).\\nThis study suggests that model organism genetics can \\nbe used to identify drug targets in human cancer and \\nis the first to propose synthetic lethality screening as \\na strategy to develop anticancer therapeutics.\\n13. Zack,\\xa0T .\\xa0I.  et\\xa0al.  Pan-cancer patterns of somatic copy \\nnumber alteration. Nat. Genet. 45, 1134–1140 (2013).\\n14. Yan,\\xa0H., Gibson,\\xa0S. & T ye,\\xa0B.\\xa0K. Mcm2 and Mcm3, two \\nproteins important for ARS activity, are related in \\nstructure and function. Genes Dev. 5, 944–957 (1991).\\n15. Kroll,\\xa0E.\\xa0S., Hyland,\\xa0K.\\xa0M., Hieter,\\xa0P . & Li,\\xa0J.\\xa0J. \\nEstablishing genetic interactions by a synthetic dosage \\nlethality phenotype. Genetics 143, 95–102 (1996).\\n16. Bian,\\xa0Y.  et\\xa0al.  Synthetic genetic array screen identifies \\nPP2A as a therapeutic target in Mad2 -overexpressing \\ntumors. Proc. Natl Acad. Sci. USA 111, 1628–1633 \\n(2014).\\n17. Reid,\\xa0R.\\xa0J.  et\\xa0al.  A synthetic dosage lethal genetic \\ninteraction between CKS1B and PLK1 is conserved in yeast and human cancer cells. Genetics 204,  \\n807–819 (2016).\\n18. Duffy,\\xa0S.  et\\xa0al.  Overexpression screens identify \\nconserved dosage chromosome instability genes in \\nyeast and human cancer. Proc. Natl Acad. Sci. USA \\n113, 9967–9976 (2016).\\n19. T ong,\\xa0A.\\xa0H.  et\\xa0al.  Global mapping of the yeast genetic \\ninteraction network. Science 303, 808–813 (2004).\\n20. Haber,\\xa0J.\\xa0E.  et\\xa0al.  Systematic triple-mutant analysis \\nuncovers functional connectivity between pathways \\ninvolved in chromosome regulation. Cell Rep. 3, \\n2168–2178 (2013).\\n21. Bunting,\\xa0S.\\xa0F.  et\\xa0al.  53BP1 inhibits homologous \\nrecombination in Brca1 -deficient cells by blocking \\nresection of DNA breaks. Cell 141, 243–254 \\n(2010).\\n22. Jaspers,\\xa0J.\\xa0E.  et\\xa0al.  Loss of 53BP1 causes PARP \\ninhibitor resistance in Brca1 -mutated mouse \\nmammary tumors. Cancer Discov. 3, 68–81 (2013).\\n23. Bindra,\\xa0R.\\xa0S.  et\\xa0al.  Down-regulation of Rad51 and \\ndecreased homologous recombination in hypoxic \\ncancer cells. Mol. Cell. Biol. 24, 8504–8518 (2004).\\n24. Chan,\\xa0N.  et\\xa0al.  Contextual synthetic lethality of cancer \\ncell kill based on the tumor microenvironment. Cancer \\nRes. 70, 8045–8054 (2010).\\n25. Bandyopadhyay,\\xa0S.  et\\xa0al.  Rewiring of genetic networks \\nin response to DNA damage. Science 330,  \\n1385–1389 (2010).\\nThis study describes the creation of condition-  \\ndependent genetic interaction maps in yeast and \\ndemonstrates that genetic interactions can change \\nin response to DNA damage.\\n26. Guenole,\\xa0A.  et\\xa0al.  Dissection of DNA damage \\nresponses using multiconditional genetic interaction \\nmaps. Mol. Cell 49, 346–358 (2013).\\n27. Li,\\xa0X., O’Neil,\\xa0N.\\xa0J., Moshgabadi,\\xa0N. & Hieter,\\xa0P. \\nSynthetic cytotoxicity: digenic interactions with TEL1/\\nATM mutations reveal sensitivity to low doses of \\ncamptothecin. Genetics  197, 611–623 (2014).\\n28. Rouleau,\\xa0M., Patel,\\xa0A., Hendzel,\\xa0M.\\xa0J., Kaufmann,\\xa0S.\\xa0H. \\n& Poirier,\\xa0G.\\xa0G. PARP inhibition: PARP1 and beyond. \\nNat. Rev. Cancer 10, 293–301 (2010).\\n29. Bailey,\\xa0M.\\xa0L.  et\\xa0al.  Glioblastoma cells containing \\nmutations in the cohesin component STAG2 are \\nsensitive to PARP inhibition. Mol. Cancer Ther. 13, \\n724–732 (2014).\\n30. Hartwell,\\xa0L.\\xa0H. & Kastan,\\xa0M.\\xa0B. Cell cycle control and \\ncancer. Science 266, 1821–1828 (1994).\\n31. Rudrapatna,\\xa0V.\\xa0A., Cagan,\\xa0R.\\xa0L. & Das,\\xa0T .\\xa0K. Drosophila  \\ncancer models. Dev. Dyn. 241, 107–118 (2012).\\n32. Kirienko,\\xa0N.\\xa0V., Mani,\\xa0K. & Fay,\\xa0D.\\xa0S. Cancer models in \\nCaenorhabditis elegans. Dev. Dyn. 239, 1413–1448 \\n(2010).\\n33. T arailo,\\xa0M., T arailo,\\xa0S. & Rose,\\xa0A.\\xa0M. Synthetic lethal \\ninteractions identify phenotypic “interologs” of the \\nspindle assembly checkpoint components. Genetics \\n177, 2525–2530 (2007).\\n34. Dixon,\\xa0S.\\xa0J., Andrews,\\xa0B.\\xa0J. & Boone,\\xa0C. Exploring the \\nconservation of synthetic lethal genetic interaction \\nnetworks. Commun. Integr. Biol. 2, 78–81 (2009).35. Boucher,\\xa0B. & Jenna,\\xa0S. Genetic interaction networks: \\nbetter understand to better predict. Front. Genet. 4, \\n290 (2013).\\n36. Baryshnikova,\\xa0A., Costanzo,\\xa0M., Myers,\\xa0C.\\xa0L., \\nAndrews,\\xa0B. & Boone,\\xa0C. Genetic interaction networks: \\ntoward an understanding of heritability. Annu. Rev. \\nGenomics Hum. Genet. 14, 111–133 (2013).\\n37. Roguev,\\xa0A.  et\\xa0al.  Conservation and rewiring of \\nfunctional modules revealed by an epistasis map in \\nfission yeast. Science 322, 405–410 (2008).\\n38. Ryan,\\xa0C.\\xa0J.  et\\xa0al.  Hierarchical modularity and the \\nevolution of genetic interactomes across species. Mol. \\nCell 46, 691–704 (2012).\\n39. McLellan,\\xa0J.\\xa0L.  et\\xa0al.  Synthetic lethality of cohesins \\nwith PARPs and replication fork mediators. PLoS \\nGenet. 8, e1002574 (2012).\\n40. T ong,\\xa0A.\\xa0H.  et\\xa0al.  Systematic genetic analysis with \\nordered arrays of yeast deletion mutants. Science \\n294, 2364–2368 (2001).\\n41. Pan,\\xa0X.  et\\xa0al.  A robust toolkit for functional profiling \\nof the yeast genome. Mol. Cell 16, 487–496 (2004).\\n42. Costanzo,\\xa0M.  et\\xa0al.  A global genetic interaction \\nnetwork maps a wiring diagram of cellular function. \\nScience  353, aaf1420 (2016).\\nThis study describes a global genetic interaction \\nnetwork derived from the analysis of pairwise \\ninteractions of nearly all the 6,000 genes in \\nS.\\xa0cerevisiae.\\n43. McManus,\\xa0K.\\xa0J., Barrett,\\xa0I.\\xa0J., Nouhi,\\xa0Y. & Hieter,\\xa0P . \\nSpecific synthetic lethal killing of RAD54B -deficient \\nhuman colorectal cancer cells by FEN1 silencing. \\nProc. Natl Acad. Sci. USA 106, 3276–3281 (2009).\\n44. Srivas,\\xa0R.  et\\xa0al.  A network of conserved synthetic \\nlethal interactions for exploration of precision cancer \\ntherapy. Mol. Cell 63, 514–525 (2016).\\nThis study describes a large-scale cross-species \\napproach to identify clinically relevant genetic \\ninteractions with genes mutated in cancer.\\n45. Fire,\\xa0A.  et\\xa0al.  Potent and specific genetic interference \\nby double-stranded RNA in Caenorhabditis elegans. \\nNature 391, 806–811 (1998).\\n46. Elbashir,\\xa0S.\\xa0M.  et\\xa0al.  Duplexes of 21 -nucleotide RNAs \\nmediate RNA interference in cultured mammalian \\ncells. Nature 411, 494–498 (2001).\\n47. Paddison,\\xa0P .\\xa0J., Caudy,\\xa0A.\\xa0A., Bernstein,\\xa0E., \\nHannon,\\xa0G.\\xa0J. & Conklin,\\xa0D.\\xa0S. Short hairpin RNAs \\n(shRNAs) induce sequence-specific silencing in \\nmammalian cells. Genes Dev. 16, 948–958 \\n(2002).\\n48. Berns,\\xa0K.  et\\xa0al.  A large-scale RNAi screen in human \\ncells identifies new components of the p53 pathway. \\nNature 428, 431–437 (2004).  \\n49. Paddison,\\xa0P .\\xa0J.  et\\xa0al.  A resource for large-scale RNA-\\ninterference-based screens in mammals. Nature 428, \\n427–431 (2004).\\n50. Cheung,\\xa0H.\\xa0W.  et\\xa0al.  Systematic investigation of \\ngenetic vulnerabilities across cancer cell lines reveals \\nlineage-specific dependencies in ovarian cancer. \\nProc. Natl Acad. Sci. USA 108, 12372–12377 \\n(2011).\\nREVIEWS\\n622 | OCTOBER 2017 | VOLUME 18  www.nature.com/nrg\\n©\\n 2017\\n Macmillan\\n Publishers\\n Limited,\\n part\\n of\\n Springer\\n Nature.\\n All\\n rights\\n reserved.', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), Document(id_='8b691ad3-df07-4e05-98fe-63cfc62edfe1', embedding=None, metadata={'page_label': '623', 'file_name': 'Synthetic lethality and cancer.pdf', 'file_path': 'files/Papers_FullText/Synthetic lethality and cancer.pdf', 'file_type': 'application/pdf', 'file_size': 2781945, 'creation_date': '2024-01-10', 'last_modified_date': '2024-01-10', 'last_accessed_date': '2024-01-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, hash='f9f1bd467a72296d513ee4e840f7f199e1d55a62f882081a311794fd911f8a0a', text='51. Vizeacoumar,\\xa0F.\\xa0J.  et\\xa0al.  A negative genetic interaction \\nmap in isogenic cancer cell lines reveals cancer cell \\nvulnerabilities. Mol. Syst. Biol. 9, 696 (2013).\\n52. Bhinder,\\xa0B. & Djaballah,\\xa0H. Systematic analysis of \\nRNAi reports identifies dismal commonality at gene-\\nlevel and reveals an unprecedented enrichment in \\npooled shRNA screens. Comb. Chem. High Throughput \\nScreen. 16, 665–681 (2013).\\n53. Reynolds,\\xa0A.  et\\xa0al.  Rational siRNA design for RNA \\ninterference. Nat. Biotechnol. 22, 326–330 (2004).\\n54. Fellmann,\\xa0C.  et\\xa0al.  Functional identification of \\noptimized RNAi triggers using a massively parallel \\nsensor assay. Mol. Cell 41, 733–746 (2011).\\n55. Jackson,\\xa0A.\\xa0L.  et\\xa0al.  Expression profiling reveals off-\\ntarget gene regulation by RNAi. Nat. Biotechnol. 21, \\n635–637 (2003).\\n56. Birmingham,\\xa0A.  et\\xa0al.  3ʹ UTR seed matches, but not \\noverall identity, are associated with RNAi off-targets. \\nNat. Methods 3, 199–204 (2006).\\n57. Sigoillot,\\xa0F.\\xa0D.  et\\xa0al.  A bioinformatics method identifies \\nprominent off-targeted transcripts in RNAi screens. \\nNat. Methods 9, 363–366 (2012).\\n58. Buehler,\\xa0E.  et\\xa0al.  siRNA off-target effects in genome-\\nwide screens identify signaling pathway members. Sci. \\nRep. 2, 428 (2012).\\n59. Hoffman,\\xa0G.\\xa0R.  et\\xa0al.  Functional epigenetics approach \\nidentifies BRM/SMARCA2 as a critical synthetic lethal \\ntarget in BRG1 -deficient cancers. Proc. Natl Acad. Sci. \\nUSA 111, 3128–3133 (2014).\\n60. Bassik,\\xa0M.\\xa0C.  et\\xa0al.  Rapid creation and quantitative \\nmonitoring of high coverage shRNA libraries. Nat. \\nMethods 6, 443–445 (2009).\\n61. Downward,\\xa0J. RAS synthetic lethal screens revisited: \\nstill seeking the elusive prize? Clin. Cancer Res. 21, \\n1802–1809 (2015).\\n62. Haibe-Kains,\\xa0B.  et\\xa0al.  Inconsistency in large \\npharmacogenomic studies. Nature 504, 389–393 (2013).\\n63. Haverty,\\xa0P .\\xa0M.  et\\xa0al.  Reproducible pharmacogenomic \\nprofiling of cancer cell line panels. Nature 533,  \\n333–337 (2016).\\n64. Iorio,\\xa0F.  et\\xa0al.  A landscape of pharmacogenomic \\ninteractions in cancer. Cell 166, 740–754 (2016).\\n65. Cong,\\xa0L.  et\\xa0al.  Multiplex genome engineering using \\nCRISPR/Cas systems. Science 339, 819–823 (2013).\\n66. Jinek,\\xa0M.  et\\xa0al.  RNA-programmed genome editing in \\nhuman cells. eLife 2, e00471 (2013).\\n67. Mali,\\xa0P .  et\\xa0al.  RNA-guided human genome engineering \\nvia Cas9. Science 339, 823–826 (2013).\\n68. Shalem,\\xa0O., Sanjana,\\xa0N.\\xa0E. & Zhang,\\xa0F. High-\\nthroughput functional genomics using CRISPR -Cas9. \\nNat. Rev. Genet. 16, 299–311 (2015).\\n69. Aguirre,\\xa0A.\\xa0J.  et\\xa0al.  Genomic copy number dictates  \\na gene-independent cell response to CRISPR/Cas9 \\ntargeting. Cancer Discov. 6, 914–929 (2016).\\n70. Munoz,\\xa0D.\\xa0M.  et\\xa0al.  CRISPR screens provide  \\na comprehensive assessment of cancer vulnerabilities \\nbut generate false-positive hits for highly amplified \\ngenomic regions. Cancer Discov. 6, 900–913 (2016).\\n71. Wang,\\xa0T .  et\\xa0al.  Identification and characterization of \\nessential genes in the human genome. Science 350, \\n1096–1101 (2015).\\n72. Hart,\\xa0T .  et\\xa0al.  High-resolution CRISPR screens reveal \\nfitness genes and genotype-specific cancer liabilities. \\nCell 163, 1515–1526 (2015).\\n73. Horlbeck,\\xa0M.\\xa0A.  et\\xa0al.  Compact and highly active next-\\ngeneration libraries for CRISPR-mediated gene \\nrepression and activation. eLife 5, e19760 (2016).\\nThis study introduces the second generation of \\ngenome-wide CRISPRi and CRISPRa libraries.\\n74. Wang,\\xa0T .  et\\xa0al.  Gene essentiality profiling reveals gene \\nnetworks and synthetic lethal interactions with \\noncogenic Ras. Cell 168, 890–903 (2017).\\nThis study is one of several recent papers that use a new \\ngeneration of genome-wide CRISPR knockout libraries \\nto search for cell line-dependent genetic vulnerabilities \\nand potential synthetic lethality candidates.\\n75. Blomen,\\xa0V.\\xa0A.  et\\xa0al.  Gene essentiality and synthetic \\nlethality in haploid human cells. Science 350,  \\n1092–1096 (2015).\\n76. Horlbeck,\\xa0M.\\xa0A.  et\\xa0al.  Nucleosomes impede Cas9 access \\nto DNA in\\xa0vivo  and in\\xa0vitro. eLife 5, e12677 (2016).\\n77. Kuscu,\\xa0C., Arslan,\\xa0S., Singh,\\xa0R., Thorpe,\\xa0J. & Adli,\\xa0M. \\nGenome-wide analysis reveals characteristics of off-\\ntarget sites bound by the Cas9 endonuclease. Nat. \\nBiotechnol. 32, 677–683 (2014).\\n78. Qi,\\xa0L.\\xa0S.  et\\xa0al.  Repurposing CRISPR as an RNA-guided \\nplatform for sequence-specific control of gene \\nexpression. Cell 152, 1173–1183 (2013).\\n79. Gilbert,\\xa0L.\\xa0A.  et\\xa0al.  CRISPR-mediated modular RNA-\\nguided regulation of transcription in eukaryotes. Cell \\n154, 442–451 (2013).80. Gilbert,\\xa0L.\\xa0A.  et\\xa0al.  Genome-scale CRISPR-mediated \\ncontrol of gene repression and activation. Cell 159, \\n647–661 (2014).\\n81. Konermann,\\xa0S.  et\\xa0al.  Genome-scale transcriptional \\nactivation by an engineered CRISPR -Cas9 complex. \\nNature 517, 583–588 (2015).\\n82. Wong,\\xa0A.\\xa0S.  et\\xa0al.  Multiplexed barcoded CRISPR -Cas9 \\nscreening enabled by CombiGEM. Proc. Natl Acad. \\nSci. USA 113, 2544–2549 (2016).\\nThis study combines the method CombiGEM with \\nthe CRISPR–Cas9 system to enable multiplexed \\nsynthetic lethality screening strategies.\\n83. Kabadi,\\xa0A.\\xa0M., Ousterout,\\xa0D.\\xa0G., Hilton,\\xa0I.\\xa0B. & \\nGersbach,\\xa0C.\\xa0A. Multiplex CRISPR/Cas9 -based \\ngenome engineering from a single lentiviral vector. \\nNucleic Acids Res. 42, e147 (2014).\\n84. Sakuma,\\xa0T ., Nishikawa,\\xa0A., Kume,\\xa0S., Chayama,\\xa0K. & \\nYamamoto,\\xa0T . Multiplex genome engineering in human \\ncells using all -in-one CRISPR/Cas9 vector system. Sci. \\nRep. 4, 5400 (2014).\\n85. Han,\\xa0K.  et\\xa0al.  Synergistic drug combinations for cancer \\nidentified in a CRISPR screen for pairwise genetic \\ninteractions. Nat. Biotechnol.  35, 463–474 (2017).\\n86. Shen,\\xa0J.\\xa0P .  et\\xa0al.  Combinatorial CRISPR -Cas9 screens for \\nde\\xa0novo  mapping of genetic interactions. Nat. Methods  \\n14, 573–576 (2017).\\n87. Wong,\\xa0A.\\xa0S., Choi,\\xa0G.\\xa0C., Cheng,\\xa0A.\\xa0A., Purcell,\\xa0O. & \\nLu,\\xa0T .\\xa0K. Massively parallel high-order combinatorial \\ngenetics in human cells. Nat. Biotechnol. 33,  \\n952–961 (2015).\\n88. Dixit,\\xa0A.  et\\xa0al.  Perturb-seq: dissecting molecular \\ncircuits with scalable single-cell RNA profiling of pooled \\ngenetic screens. Cell 167, 1853–1866 (2016).\\n89. Adamson,\\xa0B.  et\\xa0al.  A multiplexed single-cell CRISPR \\nscreening platform enables systematic dissection of the \\nunfolded protein response. Cell 167, 1867–1882 (2016).\\nReferences 88 and 89 introduce the method of \\nPerturb-seq, which combines CRISPR-based \\nscreening with single-cell transcriptional readout \\nfor a highly integrated phenotypic analysis of \\ngenetic perturbations.\\n90. Jaitin,\\xa0D.\\xa0A.  et\\xa0al.  Dissecting immune circuits by \\nlinking CRISPR-pooled screens with single-cell RNA-\\nseq. Cell 167, 1883–1896 (2016).\\n91. Datlinger,\\xa0P .  et\\xa0al.  Pooled CRISPR screening with \\nsingle-cell transcriptome readout. Nat. Methods  14, \\n297–301 (2017).\\n92. Leiserson,\\xa0M.\\xa0D., Wu,\\xa0H.\\xa0T ., Vandin,\\xa0F. & Raphael,\\xa0B.\\xa0J. \\nCoMEt: a statistical approach to identify combinations \\nof mutually exclusive alterations in cancer. Genome \\nBiol. 16, 160 (2015).\\n93. Babur,\\xa0O.  et\\xa0al.  Systematic identification of cancer \\ndriving signaling pathways based on mutual \\nexclusivity of genomic alterations. Genome Biol. 16, \\n45 (2015).\\n94. Zhang,\\xa0F.  et\\xa0al.  Predicting essential genes and \\nsynthetic lethality via influence propagation in \\nsignaling pathways of cancer cell fates. J.\\xa0Bioinform. \\nComput. Biol. 13, 1541002 (2015).\\n95. Wappett,\\xa0M.  et\\xa0al.  Multi-omic measurement of mutually \\nexclusive loss -of-function enriches for candidate \\nsynthetic lethal gene pairs. BMC Genomics 17, 65 \\n(2016).\\n96. Jackson,\\xa0R.\\xa0A. & Chen,\\xa0E.\\xa0S. Synthetic lethal approaches \\nfor assessing combinatorial efficacy of chemotherapeutic \\ndrugs. Pharmacol. Ther. 162, 69–85 (2016).\\n97. Jerby-Arnon,\\xa0L.  et\\xa0al.  Predicting cancer-specific \\nvulnerability via data-driven detection of synthetic \\nlethality. Cell 158, 1199–1209 (2014).\\nThe authors developed and applied the statistical \\nmodel DAISY, which analyses and integrates cancer \\ngenomic data and experimental genetic screens to \\npredict synthetic lethal candidates.\\n98. Guo,\\xa0J., Liu,\\xa0H. & Zheng,\\xa0J. SynLethDB: synthetic \\nlethality database toward discovery of selective and \\nsensitive anticancer drug targets. Nucleic Acids Res. \\n44, D1011–D1017 (2016).\\n99. Thoma,\\xa0C.\\xa0R.  et\\xa0al.  A high-throughput-compatible 3D \\nmicrotissue co -culture system for phenotypic RNAi \\nscreening applications. J.\\xa0Biomol. Screen. 18,  \\n1330–1337 (2013).\\n100. Matano,\\xa0M.  et\\xa0al.  Modeling colorectal cancer using \\nCRISPR -Cas9 -mediated engineering of human \\nintestinal organoids. Nat. Med. 21, 256–262 (2015).\\n101. Bruna,\\xa0A.  et\\xa0al.  A biobank of breast cancer explants \\nwith preserved intra-tumor heterogeneity to screen \\nanticancer compounds. Cell 167, 260–274 (2016).\\n102. Braun,\\xa0C.\\xa0J.  et\\xa0al.  Versatile in\\xa0vivo  regulation of tumor \\nphenotypes by dCas9 -mediated transcriptional \\nperturbation. Proc. Natl Acad. Sci. USA 113,  \\nE3892–E3900 (2016).103. Chen,\\xa0S.  et\\xa0al.  Genome-wide CRISPR screen in \\na\\xa0mouse model of tumor growth and metastasis.  \\nCell 160, 1246–1260 (2015).\\n104. Walcott,\\xa0F.\\xa0L., Patel,\\xa0J., Lubet,\\xa0R., Rodriguez,\\xa0L. & \\nCalzone,\\xa0K.\\xa0A. Hereditary cancer syndromes as model \\nsystems for chemopreventive agent development. \\nSemin. Oncol. 43, 134–145 (2016).\\n105. Wu,\\xa0X. & Lippman,\\xa0S.\\xa0M. An intermittent approach for \\ncancer chemoprevention. Nat. Rev. Cancer 11,  \\n879–885 (2011).\\n106. Bryant,\\xa0H.\\xa0E.  et\\xa0al.  Specific killing of BRCA2 -deficient \\ntumours with inhibitors of poly(ADP-ribose) \\npolymerase. Nature 434, 913–917 (2005).\\n107. Farmer,\\xa0H.  et\\xa0al.  T argeting the DNA repair defect in \\nBRCA mutant cells as a therapeutic strategy. Nature \\n434, 917–921 (2005).\\nReferences 106 and 107 show the first evidence of \\na synthetic lethality-based therapeutic strategy \\nbetween the BRCA genes, which are commonly \\nmutated in breast and other cancers, and PARP1.\\n108. Fong,\\xa0P .\\xa0C.  et\\xa0al.  Inhibition of poly(ADP-ribose) \\npolymerase in tumors from BRCA mutation carriers. \\nN.\\xa0Engl. J.\\xa0Med. 361, 123–134 (2009).\\n109. Gelmon,\\xa0K.\\xa0A.  et\\xa0al.  Olaparib in patients with \\nrecurrent high-grade serous or poorly differentiated \\novarian carcinoma or triple-negative breast cancer: \\na\\xa0phase\\xa02, multicentre, open-label, non-randomised \\nstudy. Lancet Oncol. 12, 852–861 (2011).\\n110. Mateo,\\xa0J.  et\\xa0al.  DNA-repair defects and olaparib in \\nmetastatic prostate cancer. N.\\xa0Engl. J.\\xa0Med. 373, \\n1697–1708 (2015).\\n111. McCabe,\\xa0N.  et\\xa0al.  Deficiency in the repair of DNA \\ndamage by homologous recombination and sensitivity \\nto poly(ADP-ribose) polymerase inhibition. Cancer \\nRes. 66, 8109–8115 (2006).\\n112. Lord,\\xa0C.\\xa0J. & Ashworth,\\xa0A. BRCAness revisited. Nat. \\nRev. Cancer 16, 110–120 (2016).\\n113. Barber,\\xa0L.\\xa0J.  et\\xa0al.  Secondary mutations in BRCA2 \\nassociated with clinical resistance to a PARP inhibitor. \\nJ.\\xa0Pathol. 229, 422–429 (2013).\\n114. T er Brugge,\\xa0P .  et\\xa0al.  Mechanisms of therapy resistance \\nin patient-derived xenograft models of \\nBRCA1 -deficient breast cancer. J.\\xa0Natl Cancer Inst. \\n108, djw148 (2016).\\n115. Xu,\\xa0G.  et\\xa0al.  REV7 counteracts DNA double-strand \\nbreak resection and affects PARP inhibition. Nature \\n521, 541–544 (2015).\\n116. Rottenberg,\\xa0S.  et\\xa0al.  High sensitivity of BRCA1 -deficient \\nmammary tumors to the PARP inhibitor AZD2281 \\nalone and in combination with platinum drugs. Proc. \\nNatl Acad. Sci. USA 105, 17079–17084 (2008).\\n117. Knezevic,\\xa0C.\\xa0E.  et\\xa0al.  Proteome-wide profiling of clinical \\nPARP inhibitors reveals compound-specific secondary \\ntargets. Cell Chem. Biol. 23, 1490–1503 (2016).\\n118. Schiewer,\\xa0M.\\xa0J. & Knudsen,\\xa0K.\\xa0E. T ranscriptional roles \\nof PARP1 in cancer. Mol. Cancer Res. 12, 1069–1080 \\n(2014).\\n119. Burkle,\\xa0A. & Virag,\\xa0L. Poly(ADP-ribose): PARadigms and \\nPARadoxes. Mol. Aspects Med. 34, 1046–1065 (2013).\\n120. Murai,\\xa0J.  et\\xa0al.  Stereospecific PARP trapping by BMN \\n673 and comparison with olaparib and rucaparib. \\nMol. Cancer Ther. 13, 433–443 (2014).\\n121. Hopkins,\\xa0T .\\xa0A.  et\\xa0al.  Mechanistic dissection of PARP1 \\ntrapping and the impact on in\\xa0vivo  tolerability and \\nefficacy of PARP inhibitors. Mol. Cancer Res. 13, \\n1465–1477 (2015).\\n122. Helleday,\\xa0T . The underlying mechanism for the PARP \\nand BRCA synthetic lethality: clearing up the \\nmisunderstandings. Mol. Oncol. 5, 387–393 (2011).\\nAcknowledgements\\nP .H. is a senior fellow in the Genetic Networks program at the \\nCanadian Institute for Advanced Research.\\nCompeting interests statement\\nThe authors declare no competing interests.\\nPublisher’s note\\nSpringer Nature remains neutral with regard to jurisdictional \\nclaims in published maps and institutional affiliations.\\nDATABASES\\nBiological General Repository for Interaction Datasets \\n(BioGRID):  https://thebiogrid.org/\\nCatalogue of Somatic Mutations in Cancer: http://cancer.\\nsanger.ac.uk/cosmic\\nSynLethDB:  http://histone.sce.ntu.edu.sg/SynLethDB/\\nThe Cancer Genome Atlas:  https://cancergenome.nih.gov  \\nTheCellMap:  http://thecellmap.org/\\nALL LINKS ARE ACTIVE IN THE ONLINE PDF\\nREVIEWS\\nNATURE REVIEWS | GENETICS   VOLUME 18 | OCTOBER 2017 | 623\\n©\\n 2017\\n Macmillan\\n Publishers\\n Limited,\\n part\\n of\\n Springer\\n Nature.\\n All\\n rights\\n reserved.', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), Document(id_='e3d54861-2e24-424a-94fb-a0ea7ab0acfb', embedding=None, metadata={'page_label': '23', 'file_name': 'Synthetic lethality as an engine for cancer drug target discovery.pdf', 'file_path': 'files/Papers_FullText/Synthetic lethality as an engine for cancer drug target discovery.pdf', 'file_type': 'application/pdf', 'file_size': 1996992, 'creation_date': '2024-01-10', 'last_modified_date': '2024-01-10', 'last_accessed_date': '2024-01-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, hash='b9bfa81e541aa56c3ec1c45c63598d9c0ec910fe878795c6a0d17050aca82a91', text='Cancer is a disease of the genome, and almost all can \\xad\\ncers have multiple genetic lesions that must be addressed  \\nto develop curative combination therapies1. A reduc\\xad\\ntionist view of the hallmarks of cancer suggests that tar \\xad\\ngeting oncogenic drivers, tumour suppressor gene loss \\nand the underlying mechanisms by which cancer cells evade immune destruction are the minimum that will be required for cures\\n1.\\nCompletion of the Human Genome Project in 2003 \\n(refs2,3) followed by advances in sequencing technol\\xad\\nogy and the analysis of thousands of human tumours4–7 \\nenabled discovery of the first generation of genetically \\ntargeted cancer therapies, which, even as single agents, \\nchanged the lives of many people with cancer. Imatinib, which targets the BCR–ABL fusion tyrosine kinase, has extended the median survival of patients with chronic \\nmyelogenous leukaemia to more than 10 years\\n8–11. \\nImatinib also inhibits KIT and is effective in treating KIT\\xad mutant gastrointestinal stromal tumours, with \\nresponse rates of up to 50% and a median progression\\xad \\nfree survival of approximately 18 months\\n12–15. The \\nBRAF inhibitor vemurafenib16, followed a few years \\nlater by dabrafenib17 and encorafenib18, transformed \\nBRAF\\xad mutant melanoma from an untreatable, rapidly \\nprogressive malignancy to a disease in which more than 50% of patients have meaningful clinical responses\\n19,20 \\nand when combined with a MEK inhibitor, have a \\nmedian progression\\xad free survival of approximately \\n12\\xa0months with limited toxicity21–23. There are multiple \\nother examples of clinical successes, including drugs that target amplified ERBB2 (encoding HER2) in breast cancer\\n24,25, EGFR mutations and ALK translocations in \\nnon\\xad small\\xadcell lung cancer (NSCLC)26–35 and numerous \\nothers36. However, our ability to make further progress \\nwith genetically targeted cancer therapy has been limited by two main issues. First, although partial responses to \\ntargeted therapies in selected patient populations are common, converting those partial responses to dura\\xadble complete responses, which is needed for cures, will \\nrequire combination regimens that have been challeng\\xad\\ning to define. Additionally, although DNA sequencing technology has enabled the identification of most, if not all, oncogenes that arise as a consequence of genetic \\nalterations\\n37–39, they represent a relatively small percent\\xad\\nage of genes that are relevant in cancer, not all of which are druggable with conventional approaches. Proteolysis\\xad \\ntargeting chimaeras (PROTACS) and other protein  \\ndegradation approaches are likely to change the definition of ‘druggable’ in the coming years\\n40,41, but independent of  \\nadvances that may redefine druggability, identification  \\nof the next wave of cancer drug targets requires more than deep sequencing of multiple tumours. CRISPR \\xad enabled \\nfunctional genomic screening platforms are powerful tools  \\nfor this application.\\nAs enthusiasm for targeted cancer therapy waned \\nsomewhat in the face of its limitations, progress harness \\xad\\ning the immune system to treat cancer allowed another wave of clinically meaningful responses, and, in some \\ncases, cures. Many patients with melanoma, regardless \\nof BRAF mutation status\\n42–44, renal cell carcinoma45, \\nNSCLC46–49 and a number of other cancers50 have clin \\xad\\nically meaningful responses to checkpoint inhibitors Synthetic lethality as an engine  \\nfor cancer drug target discovery\\nAlan\\xa0Huang1, Levi\\xa0A.\\xa0Garraway2,4, Alan\\xa0Ashworth3 and Barbara\\xa0Weber  1*\\nAbstract | The first wave of genetically targeted therapies for cancer focused on drugging gene \\nproducts that are recurrently mutated in specific cancer types. However, mutational analysis  \\nof tumours has largely been exhausted as a strategy for the identification of new cancer targets \\nthat are druggable with conventional approaches. Furthermore, some known genetic drivers  \\nof cancer have not been directly targeted yet owing to their molecular structure (undruggable oncogenes) or because they result in functional loss (tumour suppressor genes). Functional \\ngenomic screening based on the genetic concept of synthetic lethality provides an avenue to \\ndiscover drug targets in all these areas. Although synthetic lethality is not a new idea, recent advances, including CRISPR- based gene editing, have made possible systematic screens for \\nsynthetic lethal drug targets in human cancers. Such approaches have broad potential to drive the discovery of the next wave of genetic cancer targets and ultimately the introduction of \\neffective medicines that are still needed for most cancers.\\n1T ango Therapeutics, \\nCambridge, MA, USA.\\n2Eli Lilly and Company, \\nIndianapolis, IN, USA.\\n3UCSF Helen Diller Family \\nComprehensive Cancer \\nCenter, San Francisco, CA, \\nUSA.\\n4Present address:  \\nRoche/Genentech, South  \\nSan Francisco, CA, USA.\\n*e- mail: bweber@tangotx.com\\nhttps://doi.org/10.1038/ \\ns41573-019-0046-z\\nREvIEwS\\nnature reviews | DRug DISCoveRy\\n  volume 19 | January 2020 | 23\\n', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), Document(id_='62716356-9e03-4874-b47f-46af1cbebccf', embedding=None, metadata={'page_label': '24', 'file_name': 'Synthetic lethality as an engine for cancer drug target discovery.pdf', 'file_path': 'files/Papers_FullText/Synthetic lethality as an engine for cancer drug target discovery.pdf', 'file_type': 'application/pdf', 'file_size': 1996992, 'creation_date': '2024-01-10', 'last_modified_date': '2024-01-10', 'last_accessed_date': '2024-01-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, hash='3571caca915ba8bfae794270ee15d9398c069f685a7bdecba321a0f4c09d0eeb', text='such as anti\\xad programmed cell death 1 (PD1) or anti\\xad \\nPD1 ligand 1 (PD\\xad L1), as well as in some cases antibod\\xad\\nies against cytotoxic T lymphocyte antigen 4 (CTLA4), \\nalone or in combination with another checkpoint inhibitor\\n51,52. These important successes further damp \\xad\\nened enthusiasm for genetically targeted therapies and \\nspurred a wave of clinical investigation in a broad range \\nof tumour types with many putative immune cell tar \\xad\\ngets. This next wave of immunotherapy agents has yet to prove as effective as anti\\xad PD1 and anti\\xad PD\\xadL1 agents\\n53, \\nand it remains to be determined whether immunother \\xad\\napies will do what the initial targeted therapies failed to do: to produce durable complete responses and cures in large numbers of patients with cancer.\\nTherefore, although advances in both targeted therapy \\nand immunotherapy have improved cancer treatment for many people in remarkable ways, barriers to target\\xading tumour suppressor gene loss, identifying context\\xad dependent driver genes that are not marked by genetic \\nalterations (which result in non\\xad oncogene addiction\\n54,55, \\ncalled here ‘unmarked oncogenes’), designing the next generation of novel combinations and exploring the \\nlargely unknown genetics of tumour \\xad intrinsic immune \\nevasion have stalled progress. However, the genetic prin\\xadciple of synthetic lethality coupled with the power of CRISPR \\xad based functional genomic screening technology  \\noffers a path forward.\\nSynthetic lethality, initially described in Drosophila  \\nas recessive lethality\\n56, is classically defined as the setting \\nin which inactivation of either of two genes individually \\nhas little effect on cell viability but loss of function of \\nboth genes simultaneously leads to cell death. In can\\xadcer, the concept of synthetic lethality has been extended to pairs of genes, in which inactivation of one by dele\\xad\\ntion or mutation and pharmacological inhibition of the \\nother leads to death of cancer cells whereas normal cells (which lack the fixed genetic alteration) are spared the effect of the drug. In the most straightforward applica\\xad\\ntion, this means identifying targeted therapies that kill \\ncancer cells that lack a specific tumour suppressor gene but spare normal cells. In addition to this (conceptu\\xadally) simple application, the tools needed to discover \\nsynthetic lethal interactions in human cancer cells can \\nbe applied in multiple ways to identify a number of other types of cancer drug targets.\\nIn this Review, we discuss how the genetic con\\xad\\ncept of synthetic lethality paired with CRISPR \\xad based \\nfunctional genomic screening can be applied to iden\\xadtify the next generation of effective cancer drugs and \\ncombinations. Although much has been written about \\nsynthetic lethality in cancer since Hartwell, Friend and colleagues raised the idea in 1997 \\n(refs57–59), application \\nof the concept to cancer drug discovery has been largely \\naspirational. This is largely because of the limitations \\nof using yeast and Drosophila  as model organisms for \\nhuman disease and the limitations of previous genera \\xad\\ntions of genetic tools, such as RNA interference (RNAi), \\nfor mammalian studies. The widespread availability of \\nCRISPR \\xad based tools and the increasingly varied ways \\nthey can be used has created an inflexion point that has the potential to dramatically alter cancer target discov\\xad\\nery and create a wave of new therapeutics in the next decade. Here we analyse the evidence that many cancer \\ntargets remain to be discovered using a CRISPR \\xad based \\nfunctional genomic screening and how this approach \\ncan lead to the discovery of additional targets based on \\nloss of tumour suppressor genes. We also highlight the importance of \\ngenetic context  in designing target discov\\xad\\nery strategies, and analyse the technical considerations \\nfor scalable synthetic lethal target discovery, including \\nthe relative benefits of various CRISPR \\xad based tools \\nand libraries, discussing inhibition of protein arginine  \\nN\\xad methyltransferase 5 (PRMT5) in cancers with dele\\xad\\ntion of S \\xad methyl\\xad5′ \\xadthioadenosine phosphorylase \\n(MTAP) as an example of targeting synthetic lethality beyond poly(ADP \\xad ribose) polymerase (PARP) inhibitors \\nin BRCA1 \\xadmutant and BRCA2 \\xadmutant contexts. Finally, \\nwe discuss the applications of CRISPR \\xad based screening \\nwith targeted drugs for novel combination discovery and in\\xa0vivo screening for non\\xad cell\\xadautonomous mechanisms.\\nSynthetic lethality\\nThe genetic concept of synthetic lethality was first described in Drosophila  in 1922 \\n(refs56,60–62). Fruit flies \\nwith individual abnormal eye phenotypes (phenotypes that were attributed to either Bar  or glass mutations) \\ncould survive and reproduce but viable offspring with a \\ncombination phenotype were never observed\\n56. We now \\nknow that Bar  and glass encode transcription factors \\nexpressed in the eye, an extension of the central nervous \\nsystem, and are involved in embryonal development. We can therefore postulate that loss of both genes simultane\\xadously results in neural development defects not compat\\xad\\nible with life, although the specific lethality mechanism \\nof this synthetic lethal pair has not been studied directly. The concept was subsequently also shown to be relevant in yeast\\n63, and eventually was proposed as a basis for \\ndrug discovery for human disease by Hartwell, Friend \\nand colleagues 20 years ago57. Hartwell, and subsequently \\nKaelin, proposed that novel targets for cancer therapeu\\xadtics could be discovered by exploiting the concept of syn\\xad\\nthetic lethality, in which one member of a synthetic lethal pair is a gene product with a cancer \\xad specific mutation \\nand the second gene product is the drug target\\n57,58.\\nThe systematic identification of synthetic lethal pairs \\nrelevant to human disease was initially limited to loss\\xad of\\xadfunction screens in model organisms. For example, genetic interaction maps in yeast were readily gener \\xad\\nated by genetic screening and crossing of knockout strains\\n57,63–65, but the utility of these findings for cancer \\ndrug discovery relies on the existence of relevant homo\\xad\\nlogues in human cells. The advent of RNAi technology \\nallowed the broader application of these concepts to human cell line systems, and a number of subsequent screening efforts in cancer cells were undertaken\\n66. \\nProof\\xad of\\xadconcept classical synthetic lethal screening \\nusing small interfering RNA (siRNA) to identify depen\\xaddencies conferred by loss of the tumour suppressor gene VHL (encoding von Hippel–Lindau disease tumour suppressor) in clear cell renal carcinoma was described \\nin 2008 \\n(ref.67). Further application of systematic RNAi \\nscreening led to the discovery of additional synthetic lethal pairs, including members of the SWI/SNF chro\\xad\\nmatin remodelling complexes such as SMARCA2 and Genetic context\\nHistology and genetic \\narchitecture that define a \\nspecific set of cancer patients \\n(for example, patients with \\nBRCA1-mutant ovarian cancer).\\nwww.nature.com/nrdReviews\\n24 | January 2020 | volume 19 \\n', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), Document(id_='c2518c3c-98e6-4171-9655-098cf24877e2', embedding=None, metadata={'page_label': '25', 'file_name': 'Synthetic lethality as an engine for cancer drug target discovery.pdf', 'file_path': 'files/Papers_FullText/Synthetic lethality as an engine for cancer drug target discovery.pdf', 'file_type': 'application/pdf', 'file_size': 1996992, 'creation_date': '2024-01-10', 'last_modified_date': '2024-01-10', 'last_accessed_date': '2024-01-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, hash='749e4f60d954475129ca39ec914a7ee5fdcadfc1c4e548f8f892f1766dd74e7b', text='SMARCA4 (refs68–70) and ARID1A and ARID1B (ref.71), \\nin addition to the PRMT5–MTAP interaction72–74  \\ndiscussed in detail later in this Review.\\nThe recent success of PARP inhibitors in BRCA \\xad \\nmutant ovarian cancers is the first clinical example of \\nusing synthetic lethality to target tumour suppressor gene loss\\n75–79 (fig.\\xa0 1). The basis for this finding is that both \\nPARP and BRCA1 and BRCA2 are components of effi\\xad\\ncient DNA repair. This interaction makes tumour cells \\nwith mutations in BRCA1 or BRCA2 sensitive to PARP inhibition, driving efficacy. Normal cells, which have at least one copy of BRCA1 or BRCA2, are largely spared, \\nwhich limits toxicity. It now has been well described \\nthat all PARP inhibitors that have reached clinical stages have both catalytic inhibitory and DNA trapping activ\\xadity\\n80,81, which has led to some controversy regarding the \\nmechanism of lethality. Whereas the DNA trapping \\nactivity of PARP inhibitors clearly enhances the effect of \\ngenetic knockdown of PARP , the activity of PARP inhib\\xaditors in tumours with BRCA1 or BRCA2 mutations is \\ntightly linked to loss of function of the gene products. Furthermore, it has been shown that PARP inhibitors with markedly different DNA trapping potencies have comparable activity as measured by growth inhibition at  \\nmaximum tolerated doses in xenograft models of BRCA1 \\xad \\nmutant triple\\xad negative breast cancer\\n82. As PARP inhi  b\\xad\\nitors are used at doses that maximize both PARP cata\\xad\\nlytic inhibition and DNA trapping activity, the relative \\nimportance of those functions in clinical response remains unknown. Therefore, although the exact mecha\\xadnism underlying PARP–BRCA1 and PARP–BRCA2 \\nsynthetic lethality remains unclear, DNA damage \\nrepair as the basis for the interaction remains undis\\xadputed. The FDA has approved four PARP inhibitors for use in patients with BRCA \\xad mutant cancers (olaparib,  rucaparib, niraparib and talazoparib)\\n76. Of note, PARP \\ninhibitors seem more effective in BRCA1\\xad mutant and \\nBRCA2\\xad mutant ovarian cancers than in breast cancers \\nwith these mutations, which raises the consideration of additional ‘genetic context’ , a critical factor in design\\xading functional genomic target discovery screens, as  \\ndiscussed later.\\nMany cancer targets to be discovered\\nLarge\\xad scale gene\\xad knockout studies across many genetic \\ncontexts are now being used to map synthetic lethal \\ninteractions in human cancer cells — first using short \\nhairpin RNA (shRNA)\\xadbased approaches and more recently using CRISPR technology, which has elim\\xadinated many of the technical hurdles of RNAi\\xad based \\nfunctional genomic screens\\n83–85 (Box\\xa0 1). Project DRIVE \\n(conducted at Novartis) and Project Achilles (con\\xad\\nducted at the Broad Institute) use t he Cancer Cell Line \\nEncyclopedia — a large panel of human cancer cell lines \\nthat represent multiple cancer types86 — to create\\xa0a \\ncatalogue of essential genes and synthetic lethal inter \\xad\\nactions87–91. The Sanger Institute has taken a similar \\napproach in 324 human cancer cell lines from 30 cancer types (P roject Score)\\n92. The number of novel druggable \\ntargets that have been nominated and fully validated by \\nProject Achilles and Project DRIVE has thus far been \\nlimited, with PRMT5 (refs72–74) and Werner syndrome \\nATP \\xad dependent helicase (WRN)92–95 being the best \\nexamples. However, Project Score focused specifically \\non identification of additional novel drug targets, and \\nhas provided convincing evidence that many undiscov\\xadered targets indeed exist, that most are context depen\\xaddent, and that they can be discovered with a functional \\ngenomics approach\\n92.\\nThe Sanger Institute effort, recently described by \\nGarrett and colleagues, includes 941 CRISPR–Cas9 \\nscreens in 339 cell lines from the Cell Model Passport collection using a genome\\xad scale library targeting \\n~18,000 genes\\n92. The final analysis included 324 of those \\ncell lines. Genes required for cell fitness (called ‘core fit\\xad\\nness genes’ or ‘essential genes’) across the majority of cell \\nlines were deemed unlikely to be good drug targets due to a narrow therapeutic index. In total, 7,470 targeted genes (41%) altered cell viability in at least one cell line \\n(Box\\xa0 2). Good drug targets, likely to be represented by \\nfitness genes restricted to specific molecular contexts \\nor histologies, were defined as a subset of those genes \\nwith fitness effects in 12 or fewer of the 13 cancer types included in the screen. With a defined median of 1,459 \\nfitness genes per cell line, a median of 553 genes were \\nconsidered pan\\xad cancer fitness genes and 866 were nom\\xad\\ninated as cancer type\\xad specific genes. To further refine \\npriority drug targets, genes were scored for various \\nmeasures of the effect of gene ablation, including extent \\nof the effect, levels of expression of the target gene and mutational status (70% of the priority score), as well as evidence of a genetic biomarker associated with the tar \\xad\\nget dependency and the somatic alteration frequency of the target gene in human cancers (the remaining 30% of the priority score). With this paradigm, 617 priority cancer type\\xad specific targets and 92 priority pan\\xad cancer \\ntargets were defined. Most of the cancer type\\xad specific Progression-fr ee survival (%)100\\n75\\n50\\n25\\n0\\nMonths since r andomization0 2 4 6 8 10 12 14 16 18 20 22 24\\n138 125 107 98 89 79 63 44 28 26 16 31\\n65 52 34 21 12 86222110Cell death No eﬀectPARPi+ PARPi+BRCA-mut\\ncancer cellab\\nNormal\\ncell\\nPlacebo\\nPlaceboNira parib\\nNiraparibNumber at riskHazard  ratio, 0.27 (95% Cl, 0.17–0.41)\\nP<0.001\\nFig. 1 | Synthetic lethality: a genetic concept reduced to clinical practice. Schematic \\n(part a) and phase III clinical data197 (part b) showing a clinical example of the utility of \\nsynthetic lethality for drug development. Poly(ADP- ribose) polymerase inhibition (PARPi), \\nwhich has minimal effect in normal cells with wild- type BRCA gene function, causes \\ntumour cell- specific death in BRCA gene- mutated tumour cells, significantly extending \\nthe progression- free survival of patients with germline mutations in BRCA genes.  \\nBRCA- mut, BRCA gene mutant. Part b from ref.187, N. Engl. J. Med., Mirza et\\xa0al., Niraparib \\nmaintenance therapy in platinum- sensitive, recurrent ovarian cancer, 375, 2154–2164. \\nCopyright © 2016, Massachusetts Medical Society. Reprinted with permission from \\nMassachusetts Medical Society.\\nnature reviews | DRug DISCoveRyReviews\\n  volume 19 | January 2020 | 25\\n', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), Document(id_='c647d528-5d7c-425a-a467-d104a03f6690', embedding=None, metadata={'page_label': '26', 'file_name': 'Synthetic lethality as an engine for cancer drug target discovery.pdf', 'file_path': 'files/Papers_FullText/Synthetic lethality as an engine for cancer drug target discovery.pdf', 'file_type': 'application/pdf', 'file_size': 1996992, 'creation_date': '2024-01-10', 'last_modified_date': '2024-01-10', 'last_accessed_date': '2024-01-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, hash='c83ca22b2e73ab32cee6df4f19979335ddfdb95d4151c44a0dd46c99f6104089', text='priority targets were identified in two or fewer cancer \\ntypes (n  = 457; 74%), emphasizing the importance of \\ncontext. Finally, a tractability filter for the development of small molecules and therapeutic antibodies was applied. Tractability group 1, which included 40 genes (6%), included all targets of drugs approved or already \\nin development. A large number of priority targets, \\n277 genes (44%), fell into tractability group 2, defined as targets with no known development efforts but with evidence supporting target tractability. Finally, the lar\\xad\\ngest number of putative targets, 311 (48%), were deemed \\nuntractable by conventional means. Although these  \\ndata do not suggest that all genes with a fitness effect\\xa0are good drug targets, or that all undiscovered targets  \\nare represented in this analysis, they provide strong evi\\xaddence for the presence of a large target space remaining to be explored and many paths to discover effective new treatments for patients with cancer.\\nGiven the evidence from the Sanger analysis that a \\nlarge number of druggable cancer targets remain to be discovered, why have so few emerged from other large\\xad scale experiments thus far? The answer likely lies in \\nboth the primary aim of Project Achilles and similar \\nprojects\\n87,92 and their design. Although novel drug tar \\xad\\nget discovery is a potential benefit of Project Achilles, \\nthe overarching goal is to increase our understanding of \\ncancer biology. The data obtained are extremely valuable for many applications, but efficient and comprehensive drug target and combination discovery requires some \\nrefinements tailored to specific applications.\\nFirst, although Project Achilles and Project DRIVE \\nhave included more than 300 cancer cell lines so far, \\nthere are many cancer subtypes insufficiently repre\\xad\\nsented for context\\xad specific analyses and some histologies \\nare not represented at all. For example, in the collection \\nthere are no cell lines for human papillomavirus\\xad positive \\nhead and neck cancer, which represents more than half \\nof all newly diagnosed head and neck cancer cases in the  United\\xa0States\\n96 and, whereas there are more than 90 \\nNSCLC cell lines, there is only one cell line each rep\\xadresenting ALK translocation or MET  amplification, \\nand there are no cell lines with MET  exon 14 skipping \\nmutations\\n87,90. ALK translocations are present in less \\nthan 5% of lung cancers overall97, but they are dispro \\xad\\nportionally represented (40%) in younger non\\xad smokers \\n(<40 years) with lung adenocarcinoma98. As in the case of \\nALK mutations, genetic alterations in MET  are similarly \\nuncommon in lung cancer overall but represent a very \\nimportant therapeutic target in a similar proportion of patients\\n97. Second, the intentional heterogeneity of the \\nCancer Cell Line Encyclopedia collection introduces a \\nmathematical challenge to identifying even strong signals \\nwhen they are present in a small subset of the collection. This is the same problem encountered in clinical trials of targeted agents conducted without patient selection: if \\nthe number of patients in the trial with the genetic con\\xad\\ntext necessary for response is small, even a large signal in individual patients will be diluted by non\\xad responders, \\nand the trial will fail. An analogous problem occurs when \\none is analysing large heterogeneous panels of cell lines: \\na strong signal from a small number of cells will be lost. This problem was addressed analytically in Project Score, revealing the strength of the effect: of the 628 priority \\ntargets identified in the analysis, 56% had a fitness effect \\nin only one cancer type and an additional 18% had a fitness effect in only two cancer types. Third, the readout for Project DRIVE and Project Achilles screens is guide \\nRNA (gRNA) ratios, which reflect growth rate changes \\nand/or cell death from in\\xa0vitro monocultures, eliminat\\xading the ability to discover non\\xad cell\\xadautonomous mecha\\xad\\nnisms such as immune evasion. Finally, the screens are \\nconducted in\\xa0vitro with CRISPR or shRNA libraries as \\nthe sole perturbation, and thus the data cannot be used to identify novel combination therapies because only one gene is perturbed in each cell and no drugs are used in \\ncombination with the genetic perturbation. These last \\ntwo issues cannot be addressed analytically in any of the large datasets now available and require experimental designs specific to those types of targets.\\nFollowing the successful application of loss\\xadof\\xad  \\nfunction screening across large cell line panels and the discovery of CRISPR, a number of biotechnology com\\xadpanies have been founded in recent years to identify and \\nprosecute novel synthetic lethal drug targets, and it is \\nlikely that a number of pharmaceutical companies have started to use this approach for target discovery when \\nrelevant model systems are available. The first wave of \\ntargeted therapies from these efforts is emerging, and it seems likely that the initial clinical trials from these efforts will start to enrol patients within the next few years.\\nThe importance of genetic context\\nAs noted already, considering genetic context is impor \\xad\\ntant to avoid signal dilution due to heterogeneity in both \\nclinical trial design and experimental design. The two \\nare tightly linked: the genetic context selected for target discovery should form the basis for a patient selection strategy in clinical development. In addition, many syn\\xad\\nthetic lethal pairs are likely to be context dependent, with \\nalterations in other genes in specific contexts altering Box 1 | Big data approaches to synthetic lethal drug target discovery\\nProject Achilles is a functional genomic screening initiative from the Broad Institute\\xa0with  \\nthe goal of creating a genome- wide catalogue of tumour vulnerabilities associated  \\nwith genetic and epigenetic alterations. Short hairpin RNA (shRNA) screening and now \\nCRISPR- based screening have been applied to cancer cell lines at the genome scale  \\nto interrogate gene essentiality, providing the foundation for a cancer dependency map\\n88–91. Project DRIVE (deep RNA interference (RNAi) interrogation of viability effects \\nin cancer) is a large- scale RNAi screen of almost 400 cancer cell lines led by Novartis to \\ndefine cancer dependency genes. Project DRIVE was designed to overcome the inability \\nof RNAi screens to distinguish between on- target and off- target effects when using low \\nnumbers of shRNAs per gene and to increase the statistical power to describe molecular \\ncorrelates of knockdown effects. In a large- scale robotics approach, a lentiviral library \\ntargeting 7,837 human genes was produced with a median of 20 shRNAs per gene \\n(compared with the standard use of three to five shRNAs per gene) and was used to \\nscreen 398 cancer cell lines in a pooled format87.\\nProject Score is an initiative from the Sanger Institute to profile the genetic \\ndependencies of 324 cancer cell lines across 19 tissues and 30 cancer types using a \\nwhole- genome CRISPR library targeting approximately 18,000 genes92. These data \\nwere specifically analysed to determine the number of novel, tractable drug targets \\nthat remain to be discovered, convincingly demonstrating that the number is likely in \\nthe hundreds. This dataset can be used to identify potential drug targets that are  \\nactive in a predictable subset of cancer cell lines, as exemplified by discussion in the \\npublication of Werner syndrome ATP- dependent helicase (WRN) as a target in \\nmicrosatellite instable tumours.\\nwww.nature.com/nrdReviews\\n26 | January 2020 | volume 19 \\n', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), Document(id_='bb027bad-66c9-466f-b2f1-ff5adb2dc166', embedding=None, metadata={'page_label': '27', 'file_name': 'Synthetic lethality as an engine for cancer drug target discovery.pdf', 'file_path': 'files/Papers_FullText/Synthetic lethality as an engine for cancer drug target discovery.pdf', 'file_type': 'application/pdf', 'file_size': 1996992, 'creation_date': '2024-01-10', 'last_modified_date': '2024-01-10', 'last_accessed_date': '2024-01-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, hash='4de2ee9d5d9b33642d49d29f14aeaeb7e7a596667a0c5a9c323bbeee6331e02a', text='the functional interaction of the synthetic lethal pair. We \\ndefine ‘genetic context’ as the driver mutation, tissue of \\norigin (histology) and other functional genetic lesions \\nthat make up subgroups of cancers. The importance of genetic context is exemplified by the discovery of the selective dependency of ovarian clear cell carcinoma \\nand melanoma cell lines on the oxygen sensor EGLN1 \\n(ref.99). EGLN1 is a member of the EgIN family of prolyl \\nhydroxylases that regulates levels of hypoxia\\xad inducible \\nfactor 1α (HIF1α) via hydroxylation and subsequent \\ndegradation by VHL100. Hahn and colleagues linked a \\ncancer dependency to HIF1α upregulation, consistent \\nwith the role of EGLN1 in regulating cellular HIF1α \\nlevels. They postulated that pharmacologic inhibition of EGLN1, which stabilizes HIF1α, reduces cellular fit\\xadness, leading reduced proliferation and cell death\\n99. In a \\nmore focused analysis of just the ovarian carcinoma cell \\nlines that were included in Project Achilles, Briggs et\\xa0al.  \\nidentified ARID1A as an additional dependency101.  \\nIn this analysis, it is not possible to separate the effects of HIF1α upregulation and ARID1A mutation as all \\ncell lines with ENGL1 dependency are characterized by HIF1α upregulation, and most of those have ARID1A  \\nmutations. ARID1A  is mutationally inactivated in ~60% \\nof ovarian clear cell carcinomas\\n102,103 and is frequently \\nmutated in other histologies but, of note, analysis of the \\nProject Achilles dataset in its entirety does not identify \\nany druggable ARID1A synthetic lethal interactions (ARID1B, a paralogue of ARID1A, is a strong synthetic lethal partner with ARID1A\\n71 but is not considered con \\xad\\nventionally druggable owing to lack of enzymatic activ\\xad\\nity or previously targeted domains). Briggs et\\xa0al. showed \\nthat pharmacologic inhibition of EGLN1 selectively kills ARID1A- mutant ovarian cancer cells; thus, it is plausi\\xad\\nble that the combination of these two abnormalities is \\nresponsible for the dependency\\n101. The data from both reports99,101 suggest that EGLN inhibitors — currently \\nused clinically to treat anaemia — may be effective in treating women with ovarian clear cell carcinoma. The \\ntherapeutic effect may be most effective in an ARID1A\\xad mutant subset, but that hypothesis has not been tested in clinical trials.\\nAnother recent example of a synthetic lethal drug tar \\xad\\nget discoverable only through analysis of cell lines with a specific genetic context is DNA polymerase θ (POLQ) in BRCA1\\xadmutant and BRCA2-mutant cancers. POLQ \\nis a low\\xad fidelity DNA polymerase that participates in \\nalternative non\\xad homologous end joining\\n104– 107, a critical \\npathway for the repair of DNA double\\xad strand breaks in \\ntumours with defective homologous recombination108,109. \\nPOLQ knockdown reduces cellular survival both on \\nits own and in combination with PARP inhibitors in \\na\\xa0BRCA1-mutant and BRCA2\\xadmutant context but not a \\nwild\\xad type context108,109. This target was not identified by \\nthe original Achilles and Project DRIVE analyses87,89, but \\nis mentioned in the Sanger analysis92. This is probably \\ndue to the fact that there was an under \\xad representation of \\nBRCA1\\xadmutant and BRCA2\\xadmutant cell lines, but then this target became easily discoverable in a small panel of curated BRCA1\\xadmutant cancer cell lines paired with wild\\xad type BRCA1 isogenic derivatives\\n110.\\nKRAS  mutations provide another example of the \\ncomplexity of context: cell lines and tumour models with G12C or G12A mutations in KRAS  are very sensitive \\nto tyrosine\\xad protein phosphatase non\\xad receptor type 11  \\n(PTPN11) inhibitors, whereas those with mutations at \\nG13 and Q61 are not\\n111, 112. This occurs because onco\\xad\\ngenic G12 variants, but not G13 and Q61 variants, are dependent on PTPN11\\xadmediated GTP loading to pro\\xad\\nmote downstream signalling\\n111. Given that G12C and \\nG12A mutations are enriched in NSCLC but are\\xa0rare in \\ncolorectal and pancreatic cancer, there may be differ \\xad\\nences in sensitivity to PTPN11 inhibition on the basis \\nof histology. Moreover, DRIVE data analysis suggests that cells with KRAS  mutations that co\\xad occur with \\nSMARCA4 or KEAP1 loss of function are less depen\\xaddent on KRAS than those with wild\\xad type SMARCA4 \\nor KEAP1 function\\n87. These data again highlight the \\nimportance of considering additional genetic context \\nin target discovery, and further suggest that in some \\ncases mutation subtyping may be needed for patient selection.\\nLarge- scale target discovery approaches\\nAs noted already, functional genomic screens, specif\\xadically high\\xad throughput loss\\xad of\\xadfunction screens that \\nidentify the pairwise effects of synthetic lethal gene pairs, have been envisioned as a path to identifying novel targets and combinations for many years\\n58,59; however, \\npre\\xad CRISPR technology was not sufficiently robust for \\nthese applications. Before the discovery of CRISPR113– 116, \\nRNAi was the standard tool for loss\\xad of\\xadfunction genetic \\nscreens in mammalian cells117, 118. Although initially \\nvery promising, it rapidly became clear that RNAi lacks \\nthe specificity for high\\xad throughput applications. This \\nproblem is inherent to the technology, as stretches of \\nshRNA and siRNA sequences (called ‘seed sequences’) can bind to and downregulate mRNAs unrelated to the Box 2 | CRISPR screen in a large cancer cell line panel suggests many more \\ndrug targets to be discovered\\nA panel of 324 cell lines from \\nthe Sanger Project Score was screened using a genome-  \\nscale CRISPR library and then analysed to nominate those genes important for cellular fitness. Of the genes tested, \\n41% impacted growth of at \\nleast one cell line (7,470 fitness genes represented in the figure), with most of these genes impacting \\n \\na minority of the cell lines (falling below\\xa0the 50% of cell lines mark), suggesting context \\nspecificity of the \\ndependency\\n92. Strong drug \\ntargets would be those that selectively kill a subset of cell \\nlines with a predictable \\ngenetic feature \\n \\nand harbour a conventionally druggable protein domain.\\nFigure reproduced with permission from ref.92, Springer Nature Limited.\\nNumber of dependent cell lines350\\n300\\n250\\n200\\n150\\n100\\n50\\n0350\\n300\\n250\\n200\\n150\\n100\\n50\\n0\\n10210310\\nNumber of genes1 0 20 40 60 80\\nGenes (%)50% cell lines\\nnature reviews | DRug DISCoveRyReviews\\n  volume 19 | January 2020 | 27\\n', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), Document(id_='f54c5c2d-b089-4cab-92c8-3d4e59b524f0', embedding=None, metadata={'page_label': '28', 'file_name': 'Synthetic lethality as an engine for cancer drug target discovery.pdf', 'file_path': 'files/Papers_FullText/Synthetic lethality as an engine for cancer drug target discovery.pdf', 'file_type': 'application/pdf', 'file_size': 1996992, 'creation_date': '2024-01-10', 'last_modified_date': '2024-01-10', 'last_accessed_date': '2024-01-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, hash='cddcb5f5f0deddb1954cdb7caf48706d854664e5cdaa9c1a0f5c23e6c2ca1063', text='gene of interest, resulting in multiple, largely unavoid\\xad\\nable, false positive hits in genetic screens119– 121. These \\noff\\xad target effects can now be predicted and computa\\xad\\ntionally filtered to improve interpretation of existing shRNA screening data\\n89; however, unless 15–20 shRNAs \\nper gene are used for screening, identifying true hits \\nfrom shRNA screens remains challenging and largely \\nimpractical without large\\xad format robotics87. By con\\xad\\ntrast, CRISPR is a highly specific, efficient and scalable \\ngenome\\xad editing technology markedly outperforming \\nRNAi\\xad based reagents83–85,122 and can be applied in high\\xad \\nthroughput screens to discover novel drug targets123,124 \\nusing multiple related approaches.\\nThe initial discovery of CRISPR in microorgan\\xad\\nisms was based on the native bacterial enzyme Cas9 \\nwhich excises segments of foreign DNA from the host genome\\n125– 127. The utility of this powerful system in \\nhuman cells was quickly realized with the development \\nof gene\\xad editing tools that pair stable expression of bac\\xad\\nterial Cas9 with sequence\\xad specific gRNAs that guide \\nthe enzyme to excise precise DNA fragments from the \\nhuman genome and at genome scale if desired114– 116,  now sometimes called ‘CRISPR cutting’ . These remarkable advances opened the door to a variety of modifications, as well as the discovery of additional editing enzymes such as the endonuclease Cpf1 \\n(refs128,129). Many of these \\nemerging genomic engineering approaches have a role \\nin cancer target discovery, as described in more detail in  \\nBox\\xa0 3. As additional genetic tools and modifications \\nemerge, target discovery approaches will continue to be enhanced.\\nStrategies for CRISPR library selection. When one \\nis considering functional genomic target discovery \\nscreens using CRISPR \\xad based tools, the primary deter \\xad\\nminants of size, cost and time are the number of cell \\nlines to be screened and the library size. The determi\\xadnation of the number of cell lines largely depends on \\nthe purpose of the screen: useful information can be \\nobtained with just one cell line or one isogenic pair, particularly when CRISPR screening is combined with a pharmacologic inhibitor, or hundreds of cell lines, \\nas with Project DRIVE, Project Achilles and Project \\nScore\\n87,90,92,130. isogenic cell line pairs  have the advantage Isogenic cell line pairs\\nCultured cell lines genetically \\nengineered to have only a \\nsingle genetic difference \\nbetween them.\\nBox 3 | CRISPR technologies that enable synthetic lethal drug target discovery\\nCRISPR\\nCRISPR technology uses a guide RNA with a target sequence of 20 base pairs in length to direct Cas9 to sequence-  \\nspecific regions of the genome, resulting in the DNA cuts that lead to gene product loss of function113– 115. In human cells, \\nsuch an approach can be leveraged to systematically study the functional effect of loss of each gene in the genome123, 124. \\nIn addition to Cas9, Cpf1 is a class 2 CRISPR system that relies on a single RNA- guided nuclease effector128, 129.\\nCRISPR interference\\nCRISPR interference (CRISPRi) uses catalytically inactive Cas9 (‘dead Cas9’, dCas9) fused to a Krüppel- associated box \\n(KRAB) protein domain that interferes with transcription, suppressing gene expression rather than inducing double-  \\nstrand DNA breaks172– 174. CRISPRi has the advantage of more closely mimicking the effect of a pharmacologic inhibitor, \\nwhich incompletely suppresses activity. As a result, CRISPRi may allow differentiation between the enzymatic effect of a \\ngene product versus a potential scaffolding effect, the latter not being amenable to pharmacologic inhibition. However, \\nthe number of poorly characterized transcription start sites in the genome limits this application and has the potential to \\nintroduce false negative results in a large genomic screen.\\nCombinatorial CRISPR\\nCRISPR systems that allow interrogation of more than one gene per cell will be particularly valuable in defining novel drug combination regimens. The Cas9 and Cpf1 systems may both be used to this end. The Cpf1 system allows one- step \\ndirect cloning of concatenated gDNA and seems to have a substantially lower rate of recombination in excising the guide \\nRNAs than Cas9 owing to the vector design\\n175– 180. Combinatorial CRISPR also can be applied for high- throughput isogenic \\ntesting, with the first position of the vector targeting a tumour suppressor gene and the second position targeting the \\ndruggable genome. This approach has the potential to dramatically scale the identification of synthetic lethal targets \\nusing well- controlled isogenic systems.\\nBase editing\\nBase editing also uses dCas9; however, the transcriptional repressor is replaced by a DNA deaminase, which results in \\nbase pair modifications that are precisely targeted181– 184. This technology is particularly useful for rapid generation of \\nisogenic pairs of cell lines, for example, changing the endogenous KRAS allele from mutant to wild type, or vice versa. \\nSuch a system is well controlled in comparison with traditional methods of overexpressing an exogenous expression \\nconstruct and could therefore be very powerful for synthetic lethal- based target discovery.\\nRNA targeting\\nC2c2, Cas13b and Cas13d are the latest CRISPR systems reported185– 191. These enzymes recognize and cut mRNA, making \\npossible both reversible and scalable target gene modulation. Such an approach mimics small- molecule or antibody  \\ndrug modes of action better than DNA editing, as pharmacologic inhibitors rarely fully inhibit the target gene product, \\nsuggesting these will be powerful tools for drug target discovery.\\nSingle- cell sequencing of pooled CRISPR screens\\nThe complexity of the readout of pooled CRISPR- based screens can be greatly expanded by the application of single- cell \\nsequencing to determine transcriptional activity and genetic perturbation in each individual cell192– 196. Such an approach \\nallows understanding of heterogeneity in a cell population as well as the elucidation of more complex mechanisms, beyond the viability readout that has historically been applied. This additional information is likely to be very useful for \\nthe prioritization of potential drug targets.\\nwww.nature.com/nrdReviews\\n28 | January 2020 | volume 19 \\n', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), Document(id_='35156527-0aaf-4627-b504-65446e0c614b', embedding=None, metadata={'page_label': '29', 'file_name': 'Synthetic lethality as an engine for cancer drug target discovery.pdf', 'file_path': 'files/Papers_FullText/Synthetic lethality as an engine for cancer drug target discovery.pdf', 'file_type': 'application/pdf', 'file_size': 1996992, 'creation_date': '2024-01-10', 'last_modified_date': '2024-01-10', 'last_accessed_date': '2024-01-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, hash='bd24e7248d4ffd23db15a3ab776a6c2419f846e26a5ea648dccf1b31d89c05bd', text='of being well controlled as the only difference between \\nthem is the removal of a single gene; however, this is a \\nhighly engineered system that may not recapitulate gene \\nfunction observed in tumours from patients. With a cell panel approach for novel target discovery in the context of a tumour suppressor loss, five to ten well\\xad matched \\nlines will often usually provide sufficient confidence to support subsequent validation. Dependency surveys across the genetic spectrum of human cancer, such as Project Achilles, Project DRIVE and Project Score, will, \\nhowever, require very large numbers of cell lines and are \\npossible only with significant resources.\\nOnce the number of cell lines to be screened has been \\ndetermined, gRNA library size is the primary determi\\xad\\nnant of the resources required. The potential advantage \\nof reducing library size without limiting actionable find\\xadings can be appreciated when one considers the method used for most large\\xad scale CRISPR \\xad based screens. These \\nscreens are usually conducted in a pooled format to allow maximum throughput with multiple internal controls. A library of gRNA constructs, usually pack\\xadaged in lentivirus, is used to infect target cell lines and \\nthe effect of each gene knockout on cell growth can be \\nindividually assessed using next\\xad generation sequencing. \\nAfter an appropriate interval (usually 1–3 weeks) the abundance of each gRNA is measured: guide sequences \\nassociated with loss of cell viability will be depleted in \\nthe postinfection pool compared with their abundance in the preinfection pool. As library size is not a technical limitation for pooled screening, exome\\xad wide interro\\xad\\ngation is feasible for a limited number of cell lines with\\xadout access to large\\xad format robotics. By contrast, for \\nplate \\xad based screens, a tenfold increase in gRNA library \\nsize, such as that between a 500\\xadgene kinome and a \\n5000 \\xadgene druggable genome, requires sophisticated \\nautomation equipment for a robust scale\\xad up. However, \\neven for pooled screening, library size does define the total number of cells required for each screen and can \\nresult in rate\\xad limiting amounts of tissue culture when \\nmultiple cell line panels are screened with a large library for pooled screens. Larger cell line panels are important \\nto adequately power a comparison between wild\\xad type \\nand mutant cell lines and account for secondary genetic \\nchanges present in all cancer cells. Therefore, limiting the size of the gRNA library size does not limit target \\ndiscovery if the number of genes interrogated in a single \\nscreen is also limited at the same time. Moreover, in case of fixed resourcing, limiting the size of the gRNA library allows analysis of a larger number of cell lines with the \\nsame amount of tissue culture resource, which in turn \\nadds statistical power to target discovery efforts.\\nWhen the experimental goal is drug target discov\\xad\\nery, efficiency and productivity are important consider \\xad\\nations, and several options may be considered. Kinome and phosphatome libraries have been used effectively for this purpose, but although they are manageable in size (518 and 298 genes respectively) many novel targets \\nwill fall outside these target classes. Project Score effec\\xad\\ntively used a genome\\xad scale library to identify almost 300 \\nputative novel druggable targets, but used a postscreen filter to eliminate genes that are considered undruggable \\nby conventional methods\\n92. Although this approach is analytically straightforward, a more efficient approach could be to use a ‘druggable genome’ library, effectively \\ninserting that filter before the screen, and therefore markedly reducing the resources and time needed to generate and analyse the data. On the basis of the his\\xad\\ntorical success of target classes, sequence\\xad based and \\nstructure\\xad based prediction, catalytic activity, and capac\\xad\\nity of binding of an endogenous ligand, some studies \\nhave estimated that no more than a quarter of the human \\ngenome is druggable by conventional means (~5,000 \\ngenes)\\n131– 133. These analyses suggest that using a drug\\xad\\ngable genome CRISPR library will eliminate more than \\n70% of screen hits that are not conventionally druggable \\nwhen a whole\\xad genome library is used, allowing time and \\nresources to remain focused on other critical activities relevant to target discovery, including screening in a \\nbroader panel of context \\xad specific cell lines and subse\\xad\\nquent target validation strategies. Some studies have \\nsuccessfully adopted such an approach by using a set of druggable genes defined as desired by the user\\n132,134,135. \\nOne important consideration when one is constructing \\na druggable genome library is the value of adding a set \\nof additional, conventionally undruggable genes relevant to cancer biology, such as the MAPK signalling path\\xadway and all known recurrently mutated genes in cancer. \\nThe undruggable genes add great value by placing novel \\ntargets identified by screening into known biological pathways. They also provide both validation for hits and biological rationale when one is selecting poten\\xad\\ntial drug targets for validation. Regardless of the defi\\xad\\nnition, druggability is in the eyes of the beholder, new drug discovery paradigms such as induced proteolysis \\xad \\ntargeting chimaeras are emerging\\n40,41 and the scope of \\nany druggable genome library will need to be revisited as  \\ntechnology develops.\\nTargeting tumour suppressor gene loss\\nTumour suppressor gene loss is a central mechanism by which normal cells undergo malignant transformation, \\noften by causing or allowing genome instability. Many \\ntumour suppressor genes have been well characterized, such as TP53, RB1 and BRCA1, but tumour suppressor \\ngene loss is by definition undruggable, as the function \\nand often the genes themselves are lost. Identification of \\ndruggable synthetic lethal partners is currently the only way of targeting the functional loss of tumour suppressor genes in cancer.\\nAs noted earlier, the discovery of PARP inhibition \\nand BRCA1 or BRCA2 mutation as a synthetic lethal \\ninteraction by both Ashworth and colleagues and Helleday and colleagues in 2005 \\n(refs75,79) is the first to be \\ntranslated into clinical benefit for patients. However, this \\ndiscovery was not made by genomic screening, rather it \\nwas hypothesis\\xad driven76,79. It was also blessed by a bit of \\ngood luck: a small\\xad molecule PARP inhibitor had already \\nbeen developed with plans for use as a cytotoxic agent, the role of BRCA1 and BRCA2 in DNA damage response  \\nwas known, and the synthetic lethal effect of PARP inhib\\xadition with BRCA loss is very strong\\n74. However, the ini \\xad\\ntial enthusiasm that additional hypothesis \\xad driven efforts \\nwould identify druggable synthetic lethal partners for other DNA damage\\xad related genes has not yet been \\nnature reviews | DRug DISCoveRyReviews\\n  volume 19 | January 2020 | 29\\n', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), Document(id_='2285c6dc-3f68-4fa9-b2ff-5a6695a319ed', embedding=None, metadata={'page_label': '30', 'file_name': 'Synthetic lethality as an engine for cancer drug target discovery.pdf', 'file_path': 'files/Papers_FullText/Synthetic lethality as an engine for cancer drug target discovery.pdf', 'file_type': 'application/pdf', 'file_size': 1996992, 'creation_date': '2024-01-10', 'last_modified_date': '2024-01-10', 'last_accessed_date': '2024-01-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, hash='b39b877602a91dfe9fe6dbc7b115a0fc8dc9a057eae834d6d2ccae024a549a6a', text='fulfilled with the lack of success of these efforts. This \\nis at least partly due to the lack of selective and potent \\npharma cologic reagents136. Furthermore, hypothesis\\xad \\ndriven discovery of synthetic lethal DNA repair path\\xadways is limited by the sheer volume of hypotheses to test \\nwith seven well \\xad described unique DNA repair pathways \\nand multiple genes within each.\\nGiven the potentially large number of synthetic lethal \\ninteractions in the human genome, it is clear that dis\\xadcovery of additional synthetic lethal pairs amenable to drug discovery requires a functional genomic approach. To this end, a CRISPR library can be used to identify \\nsynthetic lethal ‘hits’ in cell line panels that share loss of \\na specific tumour suppressor gene matched as closely as possibly to cell lines that retain wild\\xad type function of \\nthe gene of interest. Hits from these screens are poten\\xad\\ntial drug targets, and the patient population expected to \\nbenefit from inhibiting the novel targets is defined by the genetic context being interrogated.\\nMTAP deletion and PRMT5 inhibition: synthetic lethal-\\nity beyond PARP inhibitors. One of the strongest and most prevalent synthetic lethal interactions discov\\xadered by Project Achilles and Project DRIVE is PRMT5 \\ndependence in cells with MTAP deletions\\n72–74. This \\ndependency represents a subset of synthetic lethal \\xad\\nity termed ‘collateral lethality’137. Collateral lethality occurs when a ‘passenger’ gene adjacent to a tumour suppressor gene is lost along with the ‘driver’ gene. In \\nthis case, MTAP is the passenger gene and is frequently  \\nco\\xad deleted with the driver cyclin\\xad dependent kinase \\ninhibitor ( CDKN2A, encoding p16\\xadINK4)\\n72–74. Several \\ngroups discovered that MTAP\\xad null cancer cells have a \\nmarked dependency on PRMT5, an essential methyl\\xadtransferase, making these cells much more suscepti \\xad\\nble to PRMT5 knockdown than those without MTAP  \\ndeletion\\n72–74 (fig.\\xa0 2). This dependency occurs because \\nMTAP\\xad null cells accumulate high levels of the PRMT5 \\ninhibitory cofactor S \\xad methyl\\xad5′ \\xadthioadenosine (MTA). \\nAs a result, PRMT5 is partially inhibited at the baseline in MTAP\\xad null cells, and they are profoundly sensitive to  \\nfurther reduction of PRMT5 activity (for example, by \\ngenetic knockdown). This dependency provides the \\npotential for a large therapeutic window for PRMT5 inhibitors in patients with MTAP \\xad deleted tumours given \\nthat normal cells (without MTAP  deletion) would be \\nlargely spared, limiting toxicity.\\nMTAP is deleted in approximately 15% of all human \\ncancers, including more than 50% of glioblastomas and 25% of pancreatic cancers; thus, development of \\nan effective therapy for MTAP\\xad deleted tumours could \\nhave high patient impact. However, the effect of PRMT5 \\nknockdown has not been recapitulated with existing PRMT5 inhibitors\\n72–74. The lack of concordance between No eﬀectMTA Met\\nMTAPNormal cell\\nPRMT5\\nSubstr ate Substr ateMe\\nMeMTAMTA\\nMTA\\nMTA MTA\\nMTAP\\nMMM\\nT\\nAPP\\nT\\nTTumour cell\\nCell deathCell death Cell death\\nSAM SAH\\nMTA-coopera tive\\nPRMT5 inhibitorsMTA MTA\\nMTAP\\nPRRRMTTT5\\nSAMAM\\n SAHSAHSAM-competitive\\nPRMT5 inhibitors\\nFig. 2 | PRMT5 and MTAP are a synthetic lethal pair. Methylthioadenosine phosphorylase (MTAP) is frequently deleted \\nin human tumours, causing its substrate methyl-5′ -thioadenosine (MTA) to accumulate. MTA functions as a competitive \\ninhibitor for the S- methyl-5′ -thioadenosine phosphorylase (PRMT5)-activating cofactor S- adenosylmethionine (SAM), \\nwhich provides a methyl donor group for client proteins. Therefore, MTA accumulation reduces, but does not eliminate, \\nPRMT5 activity. PRMT5 inhibitors that leverage the accumulation of MTA (MTA- cooperative inhibitors) should drive \\nselectivity for the MTAP- deleted tumours while sparing the wild- type normal cells. Existing PRMT5 inhibitors that are  \\nSAM cooperative do not leverage MTA accumulation, inhibit PRMT5 regardless of the genetic context and therefore  \\nare not selective for MTAP- deleted tumours. SAH, S- adenosyl-l - homocysteine.\\nwww.nature.com/nrdReviews\\n30 | January 2020 | volume 19 \\n', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), Document(id_='5d81030f-74a4-4760-a1cd-c08af1f0b405', embedding=None, metadata={'page_label': '31', 'file_name': 'Synthetic lethality as an engine for cancer drug target discovery.pdf', 'file_path': 'files/Papers_FullText/Synthetic lethality as an engine for cancer drug target discovery.pdf', 'file_type': 'application/pdf', 'file_size': 1996992, 'creation_date': '2024-01-10', 'last_modified_date': '2024-01-10', 'last_accessed_date': '2024-01-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, hash='ce0a8ca04495d61bf56f747dd5845670ec799671c73f5084b6c869b7137558cf', text='genetic knockdown and pharmacologic inhibition in \\nthis setting could suggest that functional genomics \\nmay not be a good drug target discovery tool, when it \\nis in fact due to the mechanism of action of existing inhibitors. PRMT5 has two cofactors: the activating cofactor S\\xad adenosylmethionine (SAM) and the inhib\\xad\\nitory cofactor MTA. All known PRMT5 inhibitors are either SAM\\xad cooperative inhibitors (GlaxoSmithKline) \\nor MTA \\xad \\ncompetitive inhibitors  (Eli Lilly and Company, \\nJohnson & Johnson). Because MTA accumulates  \\nin MTAP\\xad deleted cancers to much higher levels than in \\nnormal cells, an MTA \\xad cooperative and SAM\\xad competitive \\nPRMT5 inhibitor will be required to recapitulate the effect of PRMT5 knockdown\\n73. Such an inhibitor would \\nincrease the amount of inactive PRMT5 (bound to \\nMTA) relative to active PRMT5 (bound to SAM), which \\nwould result in death of MTAP\\xad deficient cells but not \\nwild\\xad type cells. Existing PRMT5 inhibitors do not act by \\nthis mechanism and therefore kill both MTAP \\xad deleted \\ncells and wild\\xad type cells at similar exposures, resulting in \\ntoxicity with limited activity73. MTA \\xad cooperative, SAM\\xad \\ncompetitive PRMT5 inhibitors are likely in development \\nbut not currently available. Instead, the first effort to \\nexploit this synthetic lethal pair has been with MAT2A inhibitors, another component of the metabolic path\\xadway that acts by reducing SAM levels. The first MAT2A \\ninhibitor is now in clinical trials but limited efficacy data \\nhave been released\\n138.\\nBeyond synthetic lethal targets\\nCombining CRISPR- based screening with known tar -\\ngeted drugs. DNA sequencing provided the means to \\nidentify the first wave of genetic drug targets for spe\\xad\\ncific cancer subtypes because they are ‘marked’ with fixed genetic alterations, and many of these genetically altered oncogenes have now been successfully drugged. \\nAs noted earlier, BRAF inhibitors and HER2 inhibitors \\nare very active drugs in BRAF\\xad mutant melanoma and \\nERBB2-amplified breast cancer, respectively. However, these inhibitors have limited activity, despite the pres \\xad\\nence of the same genetic alterations, in colon and gastric cancer. Thus, BRAF and ERBB2 are context\\xad dependent \\n‘marked’ oncogenes. We define ‘unmarked oncogenes’ as genes that are not genetically activated through \\nmutation, amplification or translocation but are sim \\xad\\nilarly important oncogenic drivers in specific genetic \\ncontexts. On the basis of the well\\xad described concept of \\nnon\\xad oncogene addition, which is driven by ‘unmarked \\noncogenes’ , we postulate that there are many such onco\\xadgenes relevant to specific genetic contexts Many of these genes will be good drug targets but will require a functional genomic screening approach for discovery, \\nas hypothesis\\xad driven approaches are hampered by our \\nlimited understanding of the biology of complex systems \\nand the reality that empiric approaches are not scalable. A modified synthetic lethal screening platform can be \\nused to identify novel context\\xad dependent unmarked \\noncogenes that may be good drug targets for single\\xad agent  \\nor combination therapy.\\nThis concept is exemplified by the finding that breast \\ncancers that express oestrogen receptor (ER positive) \\nare sensitive to a combination of inhibition of ER and inhibition of CDK4 and CDK6 (CDK4/6)\\n139– 142, an active \\ncombination because of the crosstalk between ER and \\ncell cycle signalling143. The discovery of this interaction \\ncame from profiling of a CDK4/6 inhibitor in a panel of 47 breast cancer cell lines and finding of significant \\ngrowth inhibition clearly limited to the ER \\xad positive cell \\nlines in the panel. CDK4/6 inhibition was therefore tested in combination with several modulators of oes\\xad\\ntrogen blockade, including tamoxifen, and was found to \\nbe synergistic both preclinically and in patients\\n139,142. As \\nsingle agents, CDK4/6 inhibitors are not active in ER \\xad \\npositive breast cancer, and the combination of CDK4/6 inhibition and oestrogen blockade is not active in ER \\xad \\nnegative breast cancer subtypes. Thus, CDK4/ CDK6 is \\na context\\xad dependent ‘unmarked’ oncogene in the ER \\xad \\npositive breast cancer context. This hypothesis\\xad driven \\napproach led to an important clinical advance, but it is not scalable. Empiric approaches to combination discov\\xadery have been largely unsuccessful, and are prohibitively \\nexpensive if performed in clinical trials as is common\\xad\\nplace at present with immunologic agents, highlighting the potential value of a functional genomics strategy.\\nNovel drug combination discovery. One approach to dis\\xad\\ncovering drug combinations is to uncover novel context\\xad dependent unmarked oncogenes using a CRISPR library in combination with a targeted drug that is relevant to \\na specific genetic context. We define this as an ‘anchor’ \\nscreen, in which a targeted drug is the ‘anchor’ \\n(fig.\\xa0 3). \\nThis approach was used by Bernards and colleagues in \\n2012 using shRNA \\xad based screening (before CRISPR sys\\xad\\ntems were widely available) to identify combination drug \\ntargets that would enhance the efficacy of the BRAF inhibitor vemurafenib in BRAF\\xad mutant colon cancer\\n144. \\nVemurafenib has a 50–60% overall response rate in mela\\xad\\nnoma19,20 but only a 5% overall response rate in colon \\ncancers145,146 with activating BRAF mutations. With use \\nof a focused library of shRNAs representing 518 kinases (the ‘kinome’), a total of six colon cancer cell lines with \\nor without BRAF\\nV600E activating mutations were screened \\nfor genetic dependencies in the presence of vemurafenib. \\nThis drug anchor screen identified EGFR as a combi\\xad\\nnation target with vemurafenib in this context144. This \\ncombination has since been shown to be clinically active, \\nwith an EGFR inhibitor doubling the response rate of \\nBRAF inhibition alone in BRAF\\xad mutant colon can\\xad\\ncer, yet unfortunately remaining at a very modest 10% overall response rate\\n147. The Bernards group also used a  \\n298\\xadgene ‘phosphatome’ shRNA library to identify \\nPTPN11 inhibitors as another way to enhance the effect \\nof BRAF inhibitors in this setting148, and clinical trials to \\ntest this hypothesis are ongoing149,150. Garraway and col\\xad\\nleagues also conducted a BRAF inhibitor anchor screen \\nin a BRAF\\xad mutant colon cancer cell line but instead used \\na larger, genome\\xad scale shRNA library and discovered \\nadditional combination targets, although the conven\\xad\\ntionally druggable targets identified by this much larger \\nscreen were included in the smaller, more manageable \\nkinome and phosphatome libraries144, 148, 151. Finally, a \\nsimilar approach has been taken to identify combina\\xad\\ntion partners for MEK inhibitors as several potent and \\nselective MEK inhibitors are in clinical use, but they Competitive inhibitors\\nsmall molecules that compete \\nwith the substrates or cofactors \\nwhen binding to the target \\nenzyme, resulting in functional \\ninhibition. By contrast, an \\nuncompetitive inhibitor binds \\nto an enzyme–substrate \\ncomplex more tightly than to \\nthe enzyme alone, also \\nresulting in functional \\ninhibition.\\nnature reviews | DRug DISCoveRyReviews\\n  volume 19 | January 2020 | 31\\n', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), Document(id_='350826b8-527d-4b03-a48a-b0b76079807f', embedding=None, metadata={'page_label': '32', 'file_name': 'Synthetic lethality as an engine for cancer drug target discovery.pdf', 'file_path': 'files/Papers_FullText/Synthetic lethality as an engine for cancer drug target discovery.pdf', 'file_type': 'application/pdf', 'file_size': 1996992, 'creation_date': '2024-01-10', 'last_modified_date': '2024-01-10', 'last_accessed_date': '2024-01-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, hash='aa776c7c4572cc9abbb7f7de528e9369fa47a2ceebf22d02dac44d2140d40507', text='are minimally active in KRAS \\xad mutant NSCLC152,153. An \\nshRNA \\xad based kinome anchor screen identified ERBB3 \\nblockade as an approach for enhancing the effect of \\nMEK inhibition in this setting, a hypothesis that remains to be tested in clinical studies\\n130. These anchor screens, \\nalthough informative, can now be enhanced by use of \\nCRISPR technologies as the genetic screening tool. As a \\nresult, anchor screens can be widely applied to identify drug combinations with either known or novel targets that will be required to start seeing sustained, complete \\nresponses in patients with metastatic solid tumours.\\nScreening for non- cell-autonomous targets: immune eva-\\nsion context. The classic definition of synthetic lethal\\xad\\nity is cell autonomous, but a synthetic lethal approach \\ncan be adapted to identify druggable targets that do not kill cancer cells directly, but rather attract immune cells to destroy them. As has been well described, the accumulation of genetic alterations in cancer cells cre\\xadates neoantigens, which should be recognized by the immune system as ‘foreign’ and trigger immune destruc\\xad\\ntion of nascent cancers\\n154,155. However, all cancers that \\nbecome clinically relevant have, by definition, escaped immune destruction. A wide variety of immune evasion \\nmechanisms have been postulated, including immune editing, T\\xa0cell exhaustion and an inhibitory microenvi\\xadronment\\n156, but important drivers of immune evasion \\nmust emanate at least in part from the cancer cell itself. \\nHowever, the genetics of tumour \\xad intrinsic immune eva\\xad\\nsion have only recently begun to be described155, 157, 158 \\nand no drug targets with the potential to reverse this hallmark of cancer have yet been discovered.\\nIn 2016, Ribas and colleagues reported that loss\\xad \\nof\\xadfunction mutations in JAK1 enhance immune evasion \\nand conferred anti\\xad PD1 resistance in a patient treated \\nwith a checkpoint inhibitor\\n158. This study provided the 0\\n0.40.30.20.1\\n0.5\\n0.6\\n0.7\\n0.8\\n0.9\\n1\\n–0.8 –0.6 –0.4 –0.2 0 0.2 0.4 0.6 0.8 1False discov ery r ate\\nEffect size\\nPotential\\ndrug targetsPotential\\nresistance marker sa\\nbCRISPR dr uggable\\ngenome libr aryDrug\\nNo dr ug\\nGenes that sensitize\\ncells to anchor dr ug \\nare depletedNGS\\nNGSLog2 fold change dr ug versus plasmid\\nLog2 fold change untr eated versus plasmid\\nNon-tar geting sgRNA\\nDropout gene sgRNAEssential gene sgRNA\\nRescue gene sgRNA\\nFig. 3 | Identifying novel combination targets using CRISPR screening. a | Relevant cancer cell lines with or without  \\na mutation of interest are infected with a CRISPR library and then passaged in the presence or absence of selective  \\nsmall- molecule inhibitors at their clinically relevant doses. At the end of the treatment period, cells are harvested, and  \\nthe abundance of each guide RNA (gRNA) is evaluated using next- generation sequencing (NGS). Levels of gRNA are \\ncompared between drug treatment samples and control (no drug) samples to identify those gRNAs that are selectively \\ndepleted in the presence of the drug (green gRNAs). b | gRNA counts from NGS are quantified, then normalized and \\ncompared with the plasmid library. Genes that sensitize (which are potential drug targets in combination with the anchor drug) or rescue (which are markers of resistance to the anchor drug) cell viability to the anchor drug would selectively  \\nbe depleted or enriched in the drug- treated group but not the control group (left panel). The results of this effect are \\nstatistically quantified using MAGeCK\\n198 to identify both potential drug targets (dropouts) and resistance marker (rescue) \\ngenes (right panel). sgRNA , single- guide RNA.\\nwww.nature.com/nrdReviews\\n32 | January 2020 | volume 19 \\n', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), Document(id_='885274d6-6c26-4a3a-8262-e4336d47e1ae', embedding=None, metadata={'page_label': '33', 'file_name': 'Synthetic lethality as an engine for cancer drug target discovery.pdf', 'file_path': 'files/Papers_FullText/Synthetic lethality as an engine for cancer drug target discovery.pdf', 'file_type': 'application/pdf', 'file_size': 1996992, 'creation_date': '2024-01-10', 'last_modified_date': '2024-01-10', 'last_accessed_date': '2024-01-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, hash='f59383eb2f6c9eece283ca512ba6eb1ee5149d5335fcf2c5c0e8e5ad81f732e3', text='first evidence that a tumour cell can evade immune pres\\xad\\nsure by evolving intrinsic genetic changes. Additional \\nrecurring loss\\xad of\\xadfunction mutations were subsequently \\nidentified in genes encoding other components of the \\nantigen presentation machinery, such as components of the class I major histocompatibility complex HLA \\nand B2M, further supporting this observation\\n159–161. \\nEmerging data also indicate that many well\\xad studied \\noncogenic drivers and tumour suppressor genes may \\nplay prominent parts in immune evasion. One exam\\xadple of a putative tumour \\xad intrinsic immune evasion \\ngene is the protein kinase gene LKB1 (also known as \\nSTK11), a tumour suppressor gene inactivated in ~20% \\nof NSCLC\\n97. Loss of function of LKB1 has been postu\\xad\\nlated to drive tumorigenesis through activation of the \\nmechanistic target of rapamycin pathway in a cancer \\ncell\\xad autonomous manner162. However, LKB1  loss also \\nresults in accumulation of neutrophils with T\\xa0cell sup\\xad\\npressive effects and an increase in the levels of tumour \\xad \\npromoting cytokines163. A retrospective analysis of \\ntumours from patients who did not respond to treatment with PD1 inhibitors correlated LKB1 loss with reduced \\nPD\\xad L1 expression\\n164, further suggesting that LKB1 is a \\nbona fide suppressor of immune evasion, which may be \\none of its primary functions. Other genetic alterations \\nlinked to immune evasion include MYCN amplifica\\xad\\ntion, which limits T\\xa0cell infiltration in neuroblastoma through downregulation of the interferon response\\n165, \\nCASP8 loss of function, which rescues cancer cells \\nfrom T\\xa0cell \\xad mediated lethality by blocking the tumour \\nnecrosis factor pathway166 and PTEN loss of function, \\nwhich promotes resistance to T\\xa0cell killing by increasing production of immunosuppressive cytokines\\n167. Thus, a \\nsystematic functional genomic evaluation of all known cancer genes is warranted to then allow identification of \\ndrug targets that reverse immune evasion signalling. In this setting, the immune evasion gene (such as LKB1 ) \\nprovides the context in which a novel drug target (to be \\ndetermined) functions as the patient selection biomarker \\nfor clinical trial development.\\nApproved drugs that target the immune cell check\\xad\\npoints PD1, PD\\xad L1 and CTLA4 are now widely used \\nas anticancer therapy, but these medicines are directed towards and activate T\\xa0cells as opposed to targeting tumour \\xad intrinsic immune evasion mechanisms. Building \\non the success of the current treatment para digm, the \\ndiscovery of novel immuno\\xad oncology targets and devel\\xad\\nopment efforts are almost exclusively focused on host immunity, such as modulating T\\xa0cells, natural killer cells, macrophages and the tumour micro  environment. \\nTherefore, a novel approach that targets tumour \\xad intrinsic \\nimmune evasion mechanisms (likely in combination \\nwith checkpoint inhibitors) will have several advan\\xadtages. First, it will open a largely unexplored target \\nspace. Second, targeting tumour \\xad intrinsic mechanisms \\nwill direct immune cells to the tumour specifically and \\nmay limit the systemic autoimmune toxicity that is the primary toxic effect of checkpoint inhibitors\\n168. Finally, \\nthe genetic context in which specific immune evasion \\ntargets are active will define patient selection biomarkers \\nthat have been elusive for checkpoint inhibitor therapy. Such drugs could be used in combination with reagents stimulating host immunity, such as checkpoint inhibi\\xad\\ntors. In addition, single\\xad agent activity may be possible \\nin cases in which active T\\xa0cell infiltration has already \\noccurred in the tumour environment and in which the  \\ntumour cell \\xad driven immune evasion mechanism is  \\nthe key gatekeeping event for preventing immune  \\neradication. This approach has the potential to discover  \\nnovel medicines for cancers that are insensitive to immune checkpoint inhibition and have the advantage \\nof defining a genetic patient selection strategy for clinical \\ndevelopment.\\nDiscovering tumour \\xad intrinsic immune evasion tar \\xad\\ngets requires a two\\xad step process: first, a genetic context \\nthat confers immune evasion is identified and, second, the drug targets that reverse such a phenotype are iden\\xadtified \\n(fig.\\xa0 4). In the first step, a synthetic lethal\\xad based \\nCRISPR screen can be applied by changing the readout \\nfrom growth rate alterations or cell death to immune \\ncell\\xad mediated lethality and using in\\xa0vivo screening \\napproaches. Providing a strong proof of principle, an in\\xa0vivo CRISPR screen that evaluated 2,700 candidate \\ngenes was recently reported in B16 syngeneic mice \\ntreated with PD1 checkpoint blockade\\n169. When the \\ntumours were collected and sequenced, genes driv\\xad\\ning immune evasion and immune sensitization were \\nidentified by comparison of their relative abundance in tumours grown under increasing immune pressure. Known immune evasion genetic contexts, such as JAK1  \\nand B2M loss of function, scored highly in this study. \\nHowever, to expand these findings, studies that com\\xadprehensively identify the genetic contexts that drive immune evasion are needed. These context discovery \\nscreens will define the genetic contexts in which to \\nconduct target discovery screens, the second step in  \\ntumour \\xad intrinsic immune evasion target discovery.\\nOnce an immune evasion genetic context has been \\ndiscovered, target discovery screens can be performed either in\\xa0vivo or in\\xa0vitro. In\\xa0vivo models are more reflec\\xadtive of the human immune system, whereas in\\xa0vitro screens are more reflective of histology\\xad specific human \\ncancer genetics. In\\xa0vivo screens have the advantage of an intact immune system, but require special consideration. There are limited numbers of syngeneic mouse models, and those that exist typically have chemically induced \\ntumours and therefore a mutation spectrum that is not \\nreflective of corresponding human cancers. In addition, the throughput of in\\xa0vivo CRISPR screening is limited by the library size (generally fewer than 4,000 sgRNAs) \\nand the number of cells that can be implanted in each \\nmouse. By contrast, there are many genetically engi\\xadneered murine models designed to study human cancer genetics, but they are mostly non\\xad immunogenic and  \\nare therefore less useful for immune evasion context  \\nand target discovery.\\nWhen the mechanism of gene\\xad specific immune eva\\xad\\nsion is known or postulated, in\\xa0vitro screening with a \\nrelevant immune readout such as PD\\xad L1 expression \\nmay be the most efficient approach for identifying drug \\ntargets. For example, JAK1 loss of function mediates \\nimmune evasion through suppression of interferon \\nsignalling\\n170,171. Therefore, a target discovery screen in \\nhuman cancer cell lines that harbour a JAK1 inactivating \\nnature reviews | DRug DISCoveRyReviews\\n  volume 19 | January 2020 | 33\\n', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), Document(id_='04416af3-e763-4cd0-b0e8-336810d544a0', embedding=None, metadata={'page_label': '34', 'file_name': 'Synthetic lethality as an engine for cancer drug target discovery.pdf', 'file_path': 'files/Papers_FullText/Synthetic lethality as an engine for cancer drug target discovery.pdf', 'file_type': 'application/pdf', 'file_size': 1996992, 'creation_date': '2024-01-10', 'last_modified_date': '2024-01-10', 'last_accessed_date': '2024-01-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, hash='c1b7148b8c00ba9f9820724fff10c535482a1be18bf51bd78fc59debacbf9cb8', text='mutation could measure PD \\xad L1 induction following \\ninterferon stimulation. Additional in\\xa0vitro screen\\xad\\ning approaches may include tumour cell and T\\xa0cell  \\nco\\xad cultures, such as the ovalbumin mouse model, in \\nwhich T\\xa0cells are engineered to express the relevant \\nT\\xa0cell receptor for ovalbumin recognition, and the \\ntumour cells are modified to express chicken ovalbumin \\nantigen166. Such systems can be used to interrogate the \\nphenotype of T\\xa0cell\\xad mediated killing directly and could \\nbridge in\\xa0vivo and in\\xa0vitro approaches.\\nAs the underlying cancer genetics that drive immune \\nevasion become better understood, the tools of synthetic lethal\\xad based target discovery can be used to discover novel  \\ntumour \\xad intrinsic drug targets that reverse immune eva\\xad\\nsion. Context\\xad dependent, tumour \\xad intrinsic, immune \\nevasion target discovery screens can be conducted using paired isogenic models engineered with the immune eva\\xad\\nsion gene loss (such as JAK1 loss of function) and a wild\\xad \\ntype control. Potential drug targets are those genes that reverse the immune evasion phenotype when knocked out. These targets have the potential to overcome immune \\ncheckpoint inhibitor resistance and thereby address a  \\nsignificant unmet medical need.\\nFuture directions\\nWe are reaching the limits of sequence\\xad based genetic \\ntarget discovery for cancer but there are compelling reasons to believe that many cancer drug targets are \\nstill to be identified. Recent analysis of a genome\\xad scale CRISPR \\xad based screen in a large panel of cancer cell \\nlines provides direct evidence that this number is likely in the hundreds, but the large majority will be context specific\\n92. Applying the genetic concept of synthetic \\nlethality with both its classic definition and with some \\nconceptual modifications to identify these drug tar \\xad\\ngets, and the genetic contexts (and therefore patients) \\nin which they will be effective therapies, represents a transformative opportunity for patients, and the tools \\nwith which to do this are becoming ever more powerful. \\nContinuously evolving CRISPR technology, including single\\xad cell techniques, holds the promise of addressing \\nthe tumour heterogeneity involved in primary drug \\nresistance and the genetic evolution that drives sec\\xad\\nondary resistance. Combination CRISPR vectors will simplify the search for effective, context\\xad specific drug \\ncombinations, and base\\xad editing techniques are stream\\xad\\nlining the construction of the tools needed for target validation. Finally, the continued refinement of technol\\xadogies that will make in\\xa0vivo CRISPR screening as scal\\xadable as in\\xa0vitro approaches is enabling the integration of  \\ncancer genetics and immuno \\xad oncology and driving our \\nunderstanding of the genetic basis of immune evasion. Enhanced by these new tools, CRISPR \\xadbased functional \\ngenomic screening can be applied in increasingly unique \\nways to address the loss of tumour suppressor genes, \\nunmarked oncogenes and non \\xadcell\\xadautonomous path \\xad\\nways, and will yield the next wave of effective targeted  \\ntherapies.a  Context disco very scr een\\nCRISPR context libr ary (such\\nas tumour suppr essor genes) Immuno-\\ncompetent+ anti-PD1\\nb  Target disco very scr een\\nCRISPR dr uggable\\ngenome libr ary\\nSyngenic cells w ith\\ndeﬁned immune ev asion\\ncontext mutation Syngenic tumour cells\\nGenes that re verse context-\\ndependent immune ev asion\\nare depleted in mutant cells\\nbut not w ild-ty pe cellsGenes that confer immune\\nevasion ar e enr iched in\\nimmunocompetent mice \\nMutant\\nWild-ty peImmuno-\\ncompetent\\nImmuno-\\ncompr omisedIncr easing\\nimmune\\npressur e\\nFig. 4 | Identifying synthetic lethal drug targets that reverse tumour- intrinsic immune evasion. a | Syngeneic mouse \\nmodels with known responsiveness to immune therapies, such as MC38 and CT26, can be used to screen mice for  \\ngenes that drive immune evasion when inactivated (immune evasion genes). A ‘context discovery’ CRISPR library that \\nconsists of candidate immune evasion genes can then be introduced into the tumour models as a pool using lentiviral infection. Infected cells are subsequently implanted into different mouse strains that are either immunocompromised or immunocompetent. Immune pressure in the immunocompetent mice can be further enhanced by activating T\\xa0cells with an anti- programmed cell death 1 (anti- PD1) antibody. Tumours are then allowed to grow , and at the time of tumour harvesting, \\ncells with immune evasion gene loss will survive best in models with the highest immune pressure. Those genes can be identified using next- generation sequencing to measure the relative abundance of guide RNA. b | For the discovery of drug \\ntargets that reverse immune evasion driven in a specific genetic context, genetically engineered cell lines expressing genes that confer an immune evasion genetic context (mutant) and wild- type cells are infected with a druggable genome CRISPR \\nlibrary using the schema described above to supply immune pressure. Novel drug targets can be identified by identifying \\nthose guide RNAs depleted in the presence of immune pressure in mutant cells but not wild- type cells.\\nwww.nature.com/nrdReviews\\n34 | January 2020 | volume 19 \\n', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), Document(id_='98526920-8afe-4dd1-8ed2-cdeda06ee9a4', embedding=None, metadata={'page_label': '35', 'file_name': 'Synthetic lethality as an engine for cancer drug target discovery.pdf', 'file_path': 'files/Papers_FullText/Synthetic lethality as an engine for cancer drug target discovery.pdf', 'file_type': 'application/pdf', 'file_size': 1996992, 'creation_date': '2024-01-10', 'last_modified_date': '2024-01-10', 'last_accessed_date': '2024-01-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, hash='d00612f7c0c04dcd4b8fa1075070613ec7eb53e2b7844f051671263eff9ce604', text='New technologies that expand the number of tar \\xad\\ngets that are considered druggable have the potential to \\nhave a great impact on drug discovery. Targeted protein \\ndegradation is one such example that could markedly expand the number of druggable targets. Despite the great promise of these approaches, there remain many \\nchallenges in moving from the novel target discovery \\nthat is now within our grasp to medicines that are clin\\xadically effective. The large majority of novel targets will be intracellular proteins that require a small\\xad molecule \\ninhibitor for clinical activity. Under the best of circum\\xadstances, 5 years from target discovery to a clinical proof of concept is considered ‘fast’ , so time alone is a hurdle that is difficult to avoid with current drug discovery \\ntechnology. In addition to the frustratingly long time\\xad\\nlines for drug discovery, the impact of context depen\\xaddency means that focused functional genomic discovery approaches need to be applied to many cancer subsets, further compounding the time needed to bring truly \\ntransformative combinations to a wide spectrum of \\npatients. Finally, evolving drug resistance is inevitable until complete tumour ablation is achieved, likely requir \\xad\\ning a combination of targeted therapy and immunother \\xad\\napy. Thus, understanding resistance mechanisms and the application of this approach in those settings will also be essential. Nonetheless, CRISPR, with all of its current and future applications, is a tool that has opened the \\ndoor to an enormous range of possibilities. Many people \\nwith cancer have been cured in our lifetimes, an achieve \\xad\\nment not thought possible when synthetic lethality was first discovered in fruit flies, and many more people \\nwill be cured as a result of the work now made possi \\xad\\nble by applying the concept of synthetic lethality with  \\nthe genome editing power of CRISPR technology.\\nPublished online 11 November 2019\\n1. Hanahan, D. & Weinberg, R. A. Hallmarks of cancer: \\nthe next generation. Cell 144, 646–674 (2011).\\n2. Lander, E. S. et\\xa0al. Initial sequencing and analysis  \\nof the human genome. Nature 409, 860–921  \\n(2001).\\n3. Venter, J. C. et\\xa0al. The sequence of the human \\ngenome. Science 291, 1304–1351 (2001).\\n4. Hoadley, K. A. et\\xa0al. Cell- of-origin patterns dominate \\nthe molecular classification of 10,000 tumors from  \\n33 types of cancer. Cell 173, 291–304.e6 (2018).\\n5. T ate, J. G. et\\xa0al. COSMIC: the catalogue of somatic mutations in cancer. Nucleic Acids Res. 47, \\nD941–D947 (2019).\\n6. International Cancer Genome Consortium. International network of cancer genome projects. \\nNature 464, 993–998 (2010).\\n7. Cancer Genome Atlas Research Network et\\xa0al.  \\nThe Cancer Genome Atlas Pan- Cancer analysis project. \\nNat. Genet. 45, 1113–1120 (2013).\\n8. Hochhaus, A. et\\xa0al. Long- term outcomes of imatinib \\ntreatment for chronic myeloid leukemia. N. Engl. J. Med.  \\n376, 917–927 (2017).\\n9. Druker, B. J. et\\xa0al. Five- year follow- up of patients \\nreceiving imatinib for chronic myeloid leukemia.  \\nN. Engl. J. Med. 355, 2408–2417 (2006).\\n10. Kantarjian, H. et\\xa0al. Improved survival in chronic \\nmyeloid leukemia since the introduction of imatinib therapy: a single- institution historical experience. \\nBlood 119, 1981–1987 (2012).\\n11. Roy, L. et\\xa0al. Survival advantage from imatinib compared with the combination interferon- α plus \\ncytarabine in chronic- phase chronic myelogenous \\nleukemia: historical comparison between two phase 3 trials. Blood 108, 1478–1484 (2006).\\n12. Verweij, J. et\\xa0al. Progression- free survival in \\ngastrointestinal stromal tumours with high- dose \\nimatinib: randomised trial. Lancet 364, 1127–1134 \\n(2004).\\n13. Blanke, C. D. et\\xa0al. Phase III randomized, intergroup trial assessing imatinib mesylate at two dose levels  \\nin patients with unresectable or metastatic \\ngastrointestinal stromal tumors expressing the kit \\nreceptor tyrosine kinase: S0033. J. Clin. Oncol. 26, \\n626–632 (2008).\\n14. Blanke, C. D. et\\xa0al. Long- term results from a \\nrandomized phase II trial of standard- versus higher-  \\ndose imatinib mesylate for patients with unresectable \\nor metastatic gastrointestinal stromal tumors \\nexpressing KIT . J. Clin. Oncol. 26, 620–625 (2008).\\n15. Demetri, G. D. et\\xa0al. Efficacy and safety of imatinib \\nmesylate in advanced gastrointestinal stromal tumors. \\nN. Engl. J. Med. 347, 472–480 (2002).\\n16. Bollag, G. et\\xa0al. Clinical efficacy of a RAF inhibitor \\nneeds broad target blockade in BRAF- mutant \\nmelanoma. Nature 467, 596–599 (2010).\\n17. Rheault, T . R. et\\xa0al. Discovery of dabrafenib:  \\na selective inhibitor of Raf kinases with antitumor \\nactivity against B- Raf-driven tumors. ACS Med.  \\nChem. Lett. 4, 358–362 (2013).\\n18. Delord, J. P. et\\xa0al. Phase I dose- escalation and \\n-expansion study of the BRAF inhibitor encorafenib (LGX818) in metastatic BRAF- mutant melanoma.  \\nClin. Cancer Res. 23, 5339–5348 (2017).19. Chapman, P . B. et\\xa0al. Improved survival with vemurafenib in melanoma with BRAF V600E \\nmutation. N. Engl. J. Med. 364, 2507–2516  \\n(2011).\\n20. Hauschild, A. et\\xa0al. Dabrafenib in BRAF- mutated \\nmetastatic melanoma: a multicentre, open- label, \\nphase 3 randomised controlled trial. Lancet 380, \\n358–365 (2012).\\n21. Larkin, J. et\\xa0al. Combined vemurafenib and \\ncobimetinib in BRAF- mutated melanoma. N. Engl.  \\nJ. Med. 371, 1867–1876 (2014).\\n22. Robert, C. et\\xa0al. Improved overall survival in  \\nmelanoma with combined dabrafenib and trametinib. N. Engl. J. Med. 372, 30–39 (2015).\\n23. Dummer, R. et\\xa0al. Overall survival in patients with BRAF- mutant melanoma receiving encorafenib  \\nplus binimetinib versus vemurafenib or encorafenib \\n(COLUMBUS): a multicentre, open- label, randomised, \\nphase 3 trial. Lancet Oncol. 19, 1315–1327  \\n(2018).\\n24. Slamon, D. J. et\\xa0al. Use of chemotherapy plus a monoclonal antibody against HER2 for metastatic \\nbreast cancer that overexpresses HER2. N. Engl. J. Med.  \\n344, 783–792 (2001).\\n25. Baselga, J. et\\xa0al. Pertuzumab plus trastuzumab  \\nplus docetaxel for metastatic breast cancer. N. Engl.  \\nJ. Med. 366, 109–119 (2012).\\n26. Mok, T . S. et\\xa0al. Gefitinib or carboplatin- paclitaxel in \\npulmonary adenocarcinoma. N. Engl. J. Med. 361, \\n947–957 (2009).\\n27. Maemondo, M. et\\xa0al. Gefitinib or chemotherapy  \\nfor non- small-cell lung cancer with mutated EGFR.  \\nN. Engl. J. Med. 362, 2380–2388 (2010).\\n28. Mitsudomi, T . et\\xa0al. Gefitinib versus cisplatin plus docetaxel in patients with non- small-cell lung cancer \\nharbouring mutations of the epidermal growth factor receptor (WJTOG3405): an open label, randomised phase 3 trial. Lancet Oncol. 11, 121–128 (2010).\\n29. Zhou, C. et\\xa0al. Erlotinib versus chemotherapy as  \\nfirst- line treatment for patients with advanced EGFR \\nmutation- positive non- small-cell lung cancer \\n(OPTIMAL, CTONG-0802): a multicentre, open- label, \\nrandomised, phase 3 study. Lancet Oncol. 12, \\n735–742 (2011).\\n30. Rosell, R. et\\xa0al. Erlotinib versus standard \\nchemotherapy as first- line treatment for European \\npatients with advanced EGFR mutation- positive non-  \\nsmall-cell lung cancer (EURTAC): a multicentre, open-  \\nlabel, randomised phase 3 trial. Lancet Oncol. 13, \\n239–246 (2012).\\n31. Sequist, L. V. et\\xa0al. Phase III study of afatinib or \\ncisplatin plus pemetrexed in patients with metastatic lung adenocarcinoma with EGFR mutations. J. Clin. \\nOncol. 31, 3327–3334 (2013).\\n32. Soria, J. C. et\\xa0al. Osimertinib in untreated EGFR-  \\nmutated advanced non- small-cell lung cancer. N. Engl. \\nJ. Med. 378, 113–125 (2018).\\n33. Solomon, B. J. et\\xa0al. First- line crizotinib versus \\nchemotherapy in ALK- positive lung cancer. N. Engl.  \\nJ. Med. 371, 2167–2177 (2014).\\n34. Shaw, A. T . et\\xa0al. Crizotinib versus chemotherapy in \\nadvanced ALK- positive lung cancer. N. Engl. J. Med.  \\n368, 2385–2394 (2013).35. Shaw, A. T . & Engelman, J. A. Ceritinib in ALK-  \\nrearranged non- small-cell lung cancer. N. Engl. J. Med.  \\n370, 2537–2539 (2014).\\n36. Sawyers, C. T argeted cancer therapy. Nature 432, \\n294–297 (2004).\\n37. Lawrence, M. S. et\\xa0al. Discovery and saturation \\nanalysis of cancer genes across 21 tumour types. Nature 505, 495–501 (2014).  \\nThis landmark study integrates large- scale next-  \\ngeneration whole- exome sequencing data from \\nhuman tumours to define the landscape of tumour \\nsuppressor genes and oncogenes across 21 \\ndifferent cancer types.\\n38. Bailey, M. H. et\\xa0al. Comprehensive characterization  \\nof cancer driver genes and mutations. Cell 173, \\n371–385 e318 (2018).\\n39. Beroukhim, R. et\\xa0al. The landscape of somatic copy-  \\nnumber alteration across human cancers. Nature  \\n463, 899–905 (2010).\\n40. Churcher, I. Protac- induced protein degradation in \\ndrug discovery: breaking the rules or just making new \\nones? J. Med. Chem.\\n 61, 444–452 (2018).\\n41. Lai, A. C. & Crews, C. M. Induced protein degradation: an emerging drug discovery paradigm. Nat. Rev. Drug \\nDiscov. 16, 101–114 (2017).\\n42. Robert, C. et\\xa0al. Nivolumab in previously untreated melanoma without BRAF mutation. N. Engl. J. Med.  \\n372, 320–330 (2015).\\n43. Schachter, J. et\\xa0al. Pembrolizumab versus ipilimumab for advanced melanoma: final overall survival results of a multicentre, randomised, open- label phase 3 study \\n(KEYNOTE-006). Lancet 390, 1853–1862 (2017).\\n44. Weber, J. S. et\\xa0al. Nivolumab versus chemotherapy in patients with advanced melanoma who progressed after anti- CTLA-4 treatment (CheckMate 037):  \\na randomised, controlled, open- label, phase 3 trial. \\nLancet Oncol. 16, 375–384 (2015).\\n45. Motzer, R. J. et\\xa0al. Nivolumab versus everolimus in \\nadvanced renal- cell carcinoma. N. Engl. J. Med. 373, \\n1803–1813 (2015).\\n46. Brahmer, J. et\\xa0al. Nivolumab versus docetaxel in \\nadvanced squamous- cell non- small-cell lung cancer.  \\nN. Engl. J. Med. 373, 123–135 (2015).\\n47. Borghaei, H. et\\xa0al. Nivolumab versus docetaxel in \\nadvanced nonsquamous non- small-cell lung cancer.  \\nN. Engl. J. Med. 373, 1627–1639 (2015).\\n48. Reck, M. et\\xa0al. Pembrolizumab versus chemotherapy for PD- L1-positive non- small-cell lung cancer. N. Engl. \\nJ. Med. 375, 1823–1833 (2016).\\n49. Langer, C. J. et\\xa0al. Carboplatin and pemetrexed with \\nor without pembrolizumab for advanced, non-  \\nsquamous non- small-cell lung cancer: a randomised, \\nphase 2 cohort of the open- label KEYNOTE-021 study. \\nLancet Oncol. 17, 1497–1508 (2016).\\n50. Sun, C., Mezzadra, R. & Schumacher, T . N. Regulation \\nand function of the PD- L1 checkpoint. Immunity 48, \\n434–452 (2018).\\n51. Hodi, F. S. et\\xa0al. Improved survival with ipilimumab in patients with metastatic melanoma. N. Engl. J. Med.  \\n363, 711–723 (2010).\\n52. Larkin, J. et\\xa0al. Combined nivolumab and ipilimumab \\nor monotherapy in untreated melanoma. N. Engl.  \\nJ. Med. 373, 23–34 (2015).\\nnature reviews | DRug DISCoveRyReviews\\n  volume 19 | January 2020 | 35\\n', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), Document(id_='f49cb475-4a6c-48e9-b828-20ccdd8c489e', embedding=None, metadata={'page_label': '36', 'file_name': 'Synthetic lethality as an engine for cancer drug target discovery.pdf', 'file_path': 'files/Papers_FullText/Synthetic lethality as an engine for cancer drug target discovery.pdf', 'file_type': 'application/pdf', 'file_size': 1996992, 'creation_date': '2024-01-10', 'last_modified_date': '2024-01-10', 'last_accessed_date': '2024-01-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, hash='ca2ee7a8479284ec3ba7b877e0ec9099680ab728b4a8595ad9e8a086095e0943', text='53. T ang, J. et\\xa0al. The clinical trial landscape for PD1/\\nPDL1 immune checkpoint inhibitors. Nat. Rev. Drug \\nDiscov. 17, 854–855 (2018).\\n54. Weinstein, I. B. Addiction to oncogenes–the Achilles heal of cancer. Science 297, 63–64 (2002).\\n55. Solimini, N. L., Luo, J. & Elledge, S. J. Non- oncogene \\naddiction and the stress phenotype of cancer cells.  \\nCell 130, 986–988 (2007).\\n56. Dobzhansky, T . Genetics of natural populations; recombination and variability in populations of \\nDrosophila pseudoobscura. Genetics 31, 269–290 \\n(1946).\\n57. Hartwell, L. H., Szankasi, P ., Roberts, C. J., Murray, A. W. & Friend, S. H. Integrating genetic approaches into \\nthe discovery of anticancer drugs. Science 278, \\n1064–1068 (1997).  \\nThese authors proposed, for the first time, applying the genetic principle of synthetic lethality \\nusing yeast genetic screens to identify novel \\nvulnerabilities in cancer cells based on defined \\ngenetic defects.\\n58. Kaelin, W. G. Jr Choosing anticancer drug targets in \\nthe postgenomic era. J. Clin. Invest. 104, 1503–1506 \\n(1999).\\n59. Kaelin, W. G. Jr The concept of synthetic lethality in \\nthe context of anticancer therapy. Nat. Rev. Cancer 5, \\n689–698 (2005).\\n60. Sturtevant, A. H. A highly specific complementary lethal system in Drosophila melanogaster. Genetics  \\n41, 118–123 (1956).\\n61. Lucchesi, J. C. Synthetic lethality and semi- lethality \\namong functionally related mutants of Drosophila melanogaster. Genet. 59, 37–44 (1968).\\n62. Bridges, C. B. The origin of variations in sexual and sex- limited characters. Am. Nat. 56, 51–63 (1922).\\n63. Bender, A. & Pringle, J. R. Use of a screen for synthetic lethal and multicopy suppressee mutants to identify \\ntwo new genes involved in morphogenesis in \\nSaccharomyces cerevisiae. Mol. Cell Biol. 11, \\n1295–1305 (1991).\\n64. Kaiser, C. A. & Schekman, R. Distinct sets of SEC genes govern transport vesicle formation and fusion \\nearly in the secretory pathway. Cell 61, 723–733 \\n(1990).\\n65. Hennessy, K. M., Lee, A., Chen, E. & Botstein, D.  \\nA group of interacting yeast DNA replication genes. Genes Dev. 5, 958–969 (1991).\\n66. Mullenders, J. & Bernards, R. Loss- of-function genetic \\nscreens as a tool to improve the diagnosis and treatment \\nof cancer. Oncogene 28, 4409–4420 (2009).\\n67. Bommi- Reddy, A. et\\xa0al. Kinase requirements in human \\ncells: III. Altered kinase requirements in VHL− /− cancer \\ncells detected in a pilot synthetic lethal screen. Proc. Natl Acad. Sci. USA 105, 16484–16489 (2008).\\n68. Hoffman, G. R. et\\xa0al. Functional epigenetics approach \\nidentifies BRM/SMARCA2 as a critical synthetic lethal target in BRG1-deficient cancers. Proc. Natl Acad. Sci. \\nUSA 111, 3128–3133 (2014).  \\nThis is the first report to demonstrate the synthetic \\nlethal interaction between SMARCA2 and SMARCA4.\\n69. Wilson, B. G. et\\xa0al. Residual complexes containing \\nSMARCA2 (BRM) underlie the oncogenic drive of \\nSMARCA4 (BRG1) mutation. Mol. Cell Biol. 34, \\n1136–1144 (2014).\\n70. Oike, T . et\\xa0al. A synthetic lethality- based strategy to \\ntreat cancers harboring a genetic deficiency in the \\nchromatin remodeling factor BRG1. Cancer Res. 73, \\n5508–5518 (2013).\\n71. Helming, K. C. et\\xa0al. ARID1B is a specific vulnerability \\nin ARID1A- mutant cancers. Nat. Med. 20, 251–254 \\n(2014).\\n72. Kryukov, G. V. et\\xa0al. MTAP deletion confers enhanced \\ndependency on the PRMT5 arginine methyltransferase \\nin cancer cells. Science 351, 1214–1218 (2016).\\n73. Mavrakis, K. J. et\\xa0al. Disordered methionine \\nmetabolism in MTAP/CDKN2A- deleted cancers leads \\nto dependence on PRMT5. Science 351, 1208–1213 \\n(2016).  \\nReferences 72 and 73 are the first to demonstrate \\nthe synthetic lethal interaction between PRMT5 \\nand MTAP deletion. MTAP is lost as a passenger \\nowing to proximity to CDKN2A; therefore, this is \\none of the first examples of collateral lethality.\\n74. Marjon, K. et\\xa0al. MTAP deletions in cancer create \\nvulnerability to targeting of the MAT2A/PRMT5/\\nRIOK1 axis. Cell Rep. 15, 574–587 (2016).\\n75. Farmer, H. et\\xa0al. T argeting the DNA repair defect in BRCA mutant cells as a therapeutic strategy. Nature  \\n434, 917–921 (2005).\\n76. Lord, C. J. & Ashworth, A. PARP inhibitors: synthetic lethality in the clinic. Science 355, 1152–1158 \\n(2017).77. Ashworth, A. A synthetic lethal therapeutic approach: \\npoly(ADP) ribose polymerase inhibitors for the \\ntreatment of cancers deficient in DNA double- strand \\nbreak repair. J. Clin. Oncol. 26, 3785–3790 (2008).\\n78. Fong, P . C. et\\xa0al. Inhibition of poly(ADP- ribose) \\npolymerase in tumors from BRCA mutation carriers. \\nN. Engl. J. Med. 361, 123–134 (2009).\\n79. Bryant, H. E. et\\xa0al. Specific killing of BRCA2-deficient tumours with inhibitors of poly(ADP- ribose) \\npolymerase. Nature 434, 913–917 (2005).  \\nThis article describes the discovery of the PARP–BRCA2 synthetic lethal relationship using a \\npharmacologic PARP inhibitor and isogenic mouse \\nembryo fibrobast cell lines.\\n80. Murai, J. et\\xa0al. T rapping of PARP1 and PARP2 by clinical PARP inhibitors. Cancer Res. 72, 5588–5599 \\n(2012).\\n81. Hopkins, T . A. et\\xa0al. Mechanistic dissection of PARP1 \\ntrapping and the impact on in\\xa0vivo tolerability and \\nefficacy of PARP inhibitors. Mol. Cancer Res. 13, \\n1465–1477 (2015).\\n82. Hopkins, T . A. et\\xa0al. PARP1 T rapping by PARP inhibitors drives cytotoxicity in both cancer cells and healthy \\nbone marrow. Mol. Cancer Res. 17, 409–419 (2019).\\n83. Smith, I. et\\xa0al. Evaluation of RNAi and CRISPR technologies by large- scale gene expression profiling \\nin the connectivity map. PLOS Biol. 15, e2003213 \\n(2017).\\n84. Morgens, D. W., Deans, R. M., Li, A. & Bassik, M. C. \\nSystematic comparison of CRISPR/Cas9 and RNAi \\nscreens for essential genes. Nat. Biotechnol. 4, \\n634–636 (2016).\\n85. Evers, B. et\\xa0al. CRISPR knockout screening outperforms \\nshRNA and CRISPRi in identifying essential genes.  \\nNat. Biotechnol. 34, 631–633 (2016).\\n86. Barretina, J. et\\xa0al. The Cancer Cell Line Encyclopedia \\nenables predictive modelling of anticancer drug \\nsensitivity. Nature 483, 603–607 (2012).\\n87. McDonald, E. R. 3rd et\\xa0al. Project DRIVE: a compendium of cancer dependencies and synthetic \\nlethal relationships uncovered by large- scale, deep \\nRNAi screening. Cell 170, 577–592.e10 (2017).  \\nThis article from Novartis describes the results of a large- scale shRNA- based functional genomic screen \\nin an extensive panel of human cancer cell lines. This group used the integrated genomic and \\npathway data to systemically interrogate synthetic \\nlethal and other dependency relationships across \\nmultiple histologies.\\n88. Cowley, G. S. et\\xa0al. Parallel genome- scale loss of \\nfunction screens in 216 cancer cell lines for the \\nidentification of context- specific genetic dependencies. \\nSci. Data 1, 140035 (2014).\\n89. T sherniak, A. et\\xa0al. Defining a cancer dependency map. Cell 170, 564–576.e16 (2017).  \\nThis article reports the results of a second shRNA-based functional genomic screen designed \\nto identify genetic dependencies in approximately \\n500 human cancer cell lines. An informatics tool developed to eliminate false positive data from seed effects — and therefore to reduce the impact of  \\noff- target gene silencing — is included in this report.\\n90. Meyers, R. M. et\\xa0al. Computational correction of copy \\nnumber effect improves specificity of CRISPR- Cas9 \\nessentiality screens in cancer cells. Nat. Genet. 49, \\n1779–1784 (2017).\\n91. Aguirre, A. J. et\\xa0al. Genomic copy number dictates  \\na gene- independent cell response to CRISPR/Cas9 \\ntargeting. Cancer Discov. 6, 914–929 (2016).\\n92. Behan, F. M. et\\xa0al. Prioritization of cancer therapeutic targets using CRISPR- Cas9 screens. Nature 568, \\n511–516 (2019).  \\nThis report from the Sanger Institute describes  \\nthe results of a genome- wide CRISPR- based screen \\nin a diverse human cancer cell panel. The analysis of the data was specifically designed to identify \\npotential therapeutic targets and associated \\ngenetic alterations that could be used for patient \\nselection in clinical studies.\\n93. Chan, E. M. et\\xa0al. WRN helicase is a synthetic lethal \\ntarget in microsatellite unstable cancers. Nature 568, \\n551–556 (2019).  \\nThis article describes the discovery that the \\nhelicase WRN has a synthetic lethal interaction with microsatellite instability in colon cancer. \\nAdditional data suggest this effect will be histology \\nagnostic.\\n94. Lieb, S. et\\xa0al. Werner syndrome helicase is a selective vulnerability of microsatellite instability- high tumor \\ncells. eLife 8, e43333 (2019).\\n95. Kategaya, L., Perumal, S. K., Hager, J. H. &  \\nBelmont, L. D. Werner syndrome helicase is required for the survival of cancer cells with microsatellite instability. iScience 13, 488–497 (2019).\\n96. Gillison, M. L., Chaturvedi, A. K., Anderson, W. F.  \\n& Fakhry, C. Epidemiology of human papillomavirus-  \\npositive head and neck squamous cell carcinoma.  \\nJ. Clin. Oncol. 33, 3235–3242 (2015).\\n97. Cancer Genome Atlas Research Network. \\nComprehensive molecular profiling of lung \\nadenocarcinoma. Nature 511, 543–550 (2014).\\n98. Kwak, E. L. et\\xa0al. Anaplastic lymphoma kinase \\ninhibition in non- small-cell lung cancer. N. Engl. J. Med.  \\n363, 1693–1703 (2010).\\n99. Price, C. et\\xa0al. Genome- wide interrogation of human \\ncancers identifies EGLN1 dependency in clear cell \\novarian cancers. Cancer Res. 79, 2564–2579 (2019).\\n100. Kaelin, W. G. Jr & Ratcliffe, P . J. Oxygen sensing by \\nmetazoans: the central role of the HIF hydroxylase \\npathway. Mol. Cell 30, 393–402 (2008).\\n101. Briggs, K. J., Min, C., Zhang, H. & Huang, A. Abstract \\n2892: EGLN1 is a synthetic lethal target in ARID1A- \\nmutant ovarian cancer. J. Cancer Res. 78, 2892–2892 \\n(2018).\\n102. Jones, S. et\\xa0al. Frequent mutations of chromatin \\nremodeling gene ARID1A in ovarian clear cell \\ncarcinoma. Science 330, 228–231 (2010).\\n103. Wiegand, K. C. et\\xa0al. ARID1A mutations in \\nendometriosis- associated ovarian carcinomas.  \\nN. Engl. J. Med. 363, 1532–1543 (2010).\\n104. Arana, M. E. et\\xa0al. synthesis by human DNA polymerase theta. Nucleic Acids Res. 36, 3847–3856 (2008).\\n105. Chan, S. H., Yu, A. M. & McVey, M. Dual roles for DNA \\npolymerase theta in alternative end- joining repair of \\ndouble- strand breaks in Drosophila. PLOS Genet. 6, \\ne1001005 (2010).\\n106. Roerink, S. F., van Schendel, R. & Tijsterman, M. \\nPolymerase theta- mediated end joining of replication-  \\nassociated DNA breaks in C. elegans. Genome Res.  \\n24, 954–962 (2014).\\n107. Kent, T ., Chandramouly, G., McDevitt, S. M.,  \\nOzdemir, A. Y. & Pomerantz, R. T . Mechanism of microhomology- mediated end- joining promoted by \\nhuman DNA polymerase theta. Nat. Struct. Mol. Biol.  \\n22, 230–237 (2015).\\n108. Mateos- Gomez, P . A. et\\xa0al. Mammalian polymerase \\ntheta promotes alternative NHEJ and suppresses \\nrecombination. Nature 518, 254–257 (2015).\\n109. Ceccaldi, R. et\\xa0al. Homologous- recombination-\\ndeficient tumours are dependent on Poltheta-  \\nmediated repair. Nature 518, 258–262 (2015).\\n110. Mengwasser, K. E. et\\xa0al. Genetic screens reveal  \\nFEN1 and APEX2 as BRCA2 synthetic lethal targets. Mol. Cell 73, 885–899.e6 (2019).\\n111. Nichols, R. J. et\\xa0al. RAS nucleotide cycling underlies \\nthe SHP2 phosphatase dependence of mutant BRAF-, NF1- and RAS- driven cancers. Nat. Cell Biol. 20, \\n1064–1073 (2018).\\n112. Chen, Y. N. et\\xa0al. Allosteric inhibition of SHP2 \\nphosphatase inhibits cancers driven by receptor tyrosine kinases. Nature 535, 148–152 (2016).\\n113. Jinek, M. et\\xa0al. A programmable dual- RNA-guided \\nDNA endonuclease in adaptive bacterial immunity. \\nScience 337, 816–821 (2012).\\n114. Cong, L. et\\xa0al. Multiplex genome engineering using \\nCRISPR/Cas systems. Science 339, 819–823 (2013).\\n115. Mali, P . et\\xa0al. RNA- guided human genome engineering \\nvia Cas9. Science 339, 823–826 (2013).\\n116. Jinek, M. et\\xa0al. RNA- programmed genome editing in \\nhuman cells. eLife 2, e00471 (2013).\\n117. Grimm, S. The art and design of genetic screens: \\nmammalian culture cells. Nat. Rev. Genet. 5, 179–189 \\n(2004).\\n118. Boutros, M. & Ahringer, J. The art and design of \\ngenetic screens: RNA interference. Nat. Rev. Genet. 9, \\n554–566 (2008).\\n119. Birmingham, A. et\\xa0al. 3’ UTR seed matches, but not \\noverall identity, are associated with RNAi off- targets. \\nNat. Methods 3, 199–204 (2006).\\n120. Buehler, E. et\\xa0al. siRNA off- target effects in genome-  \\nwide screens identify signaling pathway members.  \\nSci. Rep. 2, 428 (2012).\\n121. Jackson, A. L. & Linsley, P . S. Noise amidst the silence: \\noff- target effects of siRNAs? T rends Genet. 20, 521–524 \\n(2004).\\n122. Kampmann, M. CRISPRi and CRISPRa screens in mammalian cells for precision biology and medicine. \\nACS Chem. Biol. 13, 406–416 (2018).\\n123. Shalem, O. et\\xa0al. Genome- scale CRISPR- Cas9 \\nknockout screening in human cells. Science 343, \\n84–87 (2014).\\n124. Wang, T ., Wei, J. J., Sabatini, D. M. & Lander, E. S. \\nGenetic screens in human cells using the CRISPR- Cas9 \\nsystem. Science 343, 80–84 (2014).\\nwww.nature.com/nrdReviews\\n36 | January 2020 | volume 19 ', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), Document(id_='5d032074-be48-41c0-8e75-19ec44a02649', embedding=None, metadata={'page_label': '37', 'file_name': 'Synthetic lethality as an engine for cancer drug target discovery.pdf', 'file_path': 'files/Papers_FullText/Synthetic lethality as an engine for cancer drug target discovery.pdf', 'file_type': 'application/pdf', 'file_size': 1996992, 'creation_date': '2024-01-10', 'last_modified_date': '2024-01-10', 'last_accessed_date': '2024-01-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, hash='07f37faa63c89b8ef0d922ae2958298e8264ec8828eb708fd615b88e03c4765f', text='125. Koonin, E. V., Makarova, K. S. & Zhang, F. Diversity, \\nclassification and evolution of CRISPR- Cas systems. \\nCurr. Opin. Microbiol. 37, 67–78 (2017).\\n126. Shmakov, S. et\\xa0al. Diversity and evolution of class 2 \\nCRISPR- Cas systems. Nat. Rev. Microbiol. 15, \\n169–182 (2017).\\n127. Ishino, Y., Krupovic, M. & Forterre, P . History of CRISPR- Cas from encounter with a mysterious \\nrepeated sequence to genome editing technology.  \\nJ. Bacteriol. 200, e00580-17 (2018).\\n128. Zetsche, B. et\\xa0al. Cpf1 is a single RNA- guided \\nendonuclease of a class 2 CRISPR- Cas system.  \\nCell 163, 759–771 (2015).  \\nThese authors discover and develop the Cpf1 system as an alternative gene- editing approach  \\nto CRISPR–Cas9 and demonstrate the enhanced \\nfunctionality of this system in combo- CRISPR \\nscreens using this system due to a markedly \\nreduced recombination frequency.\\n129. Fonfara, I., Richter, H., Bratovic, M., Le Rhun, A. & \\nCharpentier, E. The CRISPR- associated DNA- cleaving \\nenzyme Cpf1 also processes precursor CRISPR RNA. \\nNature 532, 517–521 (2016).\\n130. Sun, C. et\\xa0al. Intrinsic resistance to MEK inhibition  \\nin KRAS mutant lung and colon cancer through \\ntranscriptional induction of ERBB3. Cell Rep. 7, \\n86–93 (2014).\\n131. Hopkins, A. L. & Groom, C. R. The druggable genome. Nat. Rev. Drug Discov. 1, 727–730 (2002).\\n132. Griffith, M. et\\xa0al. DGIdb: mining the druggable \\ngenome. Nat. Methods 10, 1209–1210 (2013).\\n133. Santos, R. et\\xa0al. A comprehensive map of molecular \\ndrug targets. Nat. Rev. Drug Discov. 16, 19–34 \\n(2016).\\n134. Duggan, S. P . et\\xa0al. siRNA library screening identifies  \\na druggable immune- signature driving esophageal \\nadenocarcinoma cell growth. Cell Mol. Gastroenterol. \\nHepatol. 5, 569–590 (2018).\\n135. Baratta, M. G. et\\xa0al. An in- tumor genetic screen \\nreveals that the BET bromodomain protein, BRD4, is  \\na potential therapeutic target in ovarian carcinoma. \\nProc. Natl Acad. Sci. USA 112, 232–237 (2015).\\n136. Lord, C. J. & Ashworth, A. The DNA damage response and cancer therapy. Nature 481, 287–294 (2012).\\n137. Muller, F. L. et\\xa0al. Passenger deletions generate therapeutic vulnerabilities in cancer. Nature 488, \\n337–342 (2012).  \\nThis computational analysis of the The Cancer \\nGenome Atlas glioblastoma dataset provides the \\nfirst examples of the phenomenon of ‘collateral damage’, a passenger gene deletion that results in a novel dependency, and describes ENO1– ENO2  \\nin glioblastoma as a putative example.\\n138. US National Library of Medicine. ClinicalT rials.gov  \\nhttps://clinicaltrials.gov/ct2/show/NCT03435250 (2019).\\n139. Finn, R. S. et\\xa0al. The cyclin- dependent kinase 4/6 \\ninhibitor palbociclib in combination with letrozole \\nversus letrozole alone as first- line treatment of \\noestrogen receptor- positive, HER2-negative, advanced \\nbreast cancer (PALOMA-1/TRIO-18): a randomised \\nphase 2 study. Lancet Oncol. 16, 25–35 (2015).\\n140. Finn, R. S. et\\xa0al. PD 0332991, a selective cyclin D kinase 4/6 inhibitor, preferentially inhibits proliferation \\nof luminal estrogen receptor- positive human breast \\ncancer cell lines in\\xa0vitro. Breast Cancer Res. 11, R77 \\n(2009).\\n141. DeMichele, A. et\\xa0al. CDK 4/6 inhibitor palbociclib \\n(PD0332991) in Rb+  advanced breast cancer:  \\nphase II activity, safety, and predictive biomarker assessment. Clin. Cancer Res. 21, 995–1001 (2015).\\n142. T urner, N. C. et\\xa0al. Palbociclib in hormone- receptor-\\npositive advanced breast cancer. N. Engl. J. Med.  \\n373, 209–219 (2015).\\n143. Finn, R. S., Aleshin, A. & Slamon, D. J. T argeting  \\nthe cyclin- dependent kinases (CDK) 4/6 in estrogen \\nreceptor- positive breast cancers. Breast Cancer Res.  \\n18, 17 (2016).\\n144. Prahallad, A. et\\xa0al. Unresponsiveness of colon  \\ncancer to BRAF(V600E) inhibition through feedback \\nactivation of EGFR. Nature 483, 100–103 (2012).\\n145. Kopetz, S. et\\xa0al. Phase II pilot study of vemurafenib  \\nin patients with metastatic BRAF- mutated colorectal \\ncancer. J. Clin. Oncol. 33, 4032–4038 (2015).\\n146. Hyman, D. M. et\\xa0al. Vemurafenib in multiple \\nnonmelanoma cancers with BRAF V600 mutations.  \\nN. Engl. J. Med. 373, 726–736 (2015).\\n147. Corcoran, R. B. et\\xa0al. Combined BRAF, EGFR, and  \\nmek inhibition in patients with BRAF(V600E)-mutant colorectal cancer. Cancer Discov. 8, 428–443 (2018).\\n148. Prahallad, A. et\\xa0al. PTPN11 Is a central node in \\nintrinsic and acquired resistance to targeted cancer \\ndrugs. Cell Rep. 12, 1978–1985 (2015).149. US National Library of Medicine. ClinicalT rials.gov  \\nhttps://clinicaltrials.gov/ct2/show/NCT03114319 \\n(2019).\\n150. US National Library of Medicine. ClinicalT rials.gov  \\nhttps://clinicaltrials.gov/ct2/show/NCT03634982 (2019).\\n151. Whittaker, S. R. et\\xa0al. Combined pan- RAF and MEK \\ninhibition overcomes multiple resistance mechanisms to selective RAF inhibitors. Mol. Cancer Ther. 14, \\n2700–2711 (2015).\\n152. Blumenschein, G. R. Jr et\\xa0al. A randomized phase II study of the MEK1/MEK2 inhibitor trametinib \\n(GSK1120212) compared with docetaxel in KRAS-  \\nmutant advanced non- small-cell lung cancer (NSCLC). \\nAnn. Oncol. 26, 894–901 (2015).\\n153. Janne, P . A. et\\xa0al. Selumetinib plus docetaxel for \\nKRAS- mutant advanced non- small-cell lung cancer:  \\na randomised, multicentre, placebo- controlled,  \\nphase 2 study. Lancet Oncol. 14, 38–47 (2013).\\n154. Hacohen, N., Fritsch, E. F., Carter, T . A., Lander, E. S. & Wu, C. J. Getting personal with neoantigen- based \\ntherapeutic cancer vaccines. Cancer Immunol. Res. 1, \\n11–15 (2013).\\n155. Schumacher, T . N. & Schreiber, R. D. Neoantigens  \\nin cancer immunotherapy. Science 348, 69–74 \\n(2015).\\n156. Vinay, D. S. et\\xa0al. Immune evasion in cancer: \\nmechanistic basis and therapeutic strategies.  \\nSemin. Cancer Biol. 35, S185–S198 (2015).\\n157. Wellenstein, M. D. & de Visser, K. E. Cancer- cell-\\nintrinsic mechanisms shaping the tumor immune landscape. Immunity 48, 399–416 (2018).\\n158. Zaretsky, J. M. et\\xa0al. Mutations associated with acquired resistance to PD-1 blockade in melanoma.  \\nN. Engl. J. Med. 375, 819–829 (2016).  \\nThis group discovered and described an  \\nacquired JAK1 loss- of-function mutation in  \\ndriving therapeutic resistance in a patient being \\ntreated with an anti- PD1 checkpoint inhibitor,  \\nthe first example that tumour- intrinsic genetic \\nmutations play a part in mediating immune  \\ncell killing.\\n159. Sade- Feldman, M. et\\xa0al. Resistance to checkpoint \\nblockade therapy through inactivation of antigen presentation. Nat. Commun. 8, 1136 (2017).\\n160. Ozcan, M., Janikovits, J., von Knebel Doeberitz, M.  \\n& Kloor, M. Complex pattern of immune evasion in MSI colorectal cancer. Oncoimmunology 7, e1445453 \\n(2018).\\n161. Grasso, C. S. et\\xa0al. Genetic mechanisms of immune evasion in colorectal cancer. Cancer Discov. 8, \\n730–749 (2018).\\n162. Dupuy, F. et\\xa0al. LKB1 is a central regulator of  \\ntumor initiation and pro- growth metabolism in  \\nErbB2-mediated breast cancer. Cancer Metab. 1, 18 \\n(2013).\\n163. Koyama, S. et\\xa0al. STK11/LKB1 Deficiency promotes neutrophil recruitment and proinflammatory cytokine production to suppress T\\xa0cell activity in the lung tumor microenvironment. Cancer Res. 76, 999–1008 \\n(2016).  \\nThis study is the first to associate loss of  \\nfunction of a known tumour suppressor gene \\n(LKB1) with a suppressive immune phenotype  \\nin an engineered mouse tumour model. This  \\nfinding was subsequently clinically validated in  \\nthe retrospective analysis described in reference \\n164.\\n164. Skoulidis, F. et\\xa0al. STK11/LKB1 mutations and PD-1 \\ninhibitor resistance in KRAS- mutant lung \\nadenocarcinoma. Cancer Discov. 8, 822–835 (2018).  \\nThis analysis of a clinical study of patients with  \\nlung cancer treated with the anti- PD1 checkpoint \\ninhibitor pembrolizumab is the first to provide clinical evidence that LKB1 loss of function has a \\nsignificant negative impact on the response to checkpoint inhibitors.\\n165. Layer, J. P . et\\xa0al. Amplification of N- Myc is associated \\nwith a T\\xa0cell- poor microenvironment in metastatic \\nneuroblastoma restraining interferon pathway activity \\nand chemokine expression. Oncoimmunology 6, \\ne1320626 (2017).\\n166. Kearney, C. J. et\\xa0al. Tumor immune evasion arises through loss of TNF sensitivity. Sci. Immunol. 3, \\neaar3451 (2018).\\n167. Peng, W. et\\xa0al. Loss of PTEN promotes resistance to \\nT\\xa0cell- mediated immunotherapy. Cancer Discov. 6, \\n202–216 (2016).\\n168. Hassel, J. C. et\\xa0al. Combined immune checkpoint blockade (anti- PD-1/anti- CTLA-4): evaluation and \\nmanagement of adverse drug reactions. Cancer T reat Rev. 57, 36–49 (2017).169. Manguso, R. T . et\\xa0al. In\\xa0vivo CRISPR screening identifies Ptpn2 as a cancer immunotherapy target. \\nNature 547, 413–418 (2017).  \\nThis study demonstrates the utility of in\\xa0vivo  \\nCRISPR genomic screening in a syngeneic mouse model to identify tumour- intrinsic critical nodes of \\nimmune modulation in solid tumours, and describes several specific genes that could be therapeutic \\ntargets designed to reverse the immune evasion \\nphenotype.\\n170. Gao, J. et\\xa0al. Loss of IFN- gamma pathway genes  \\nin tumor cells as a mechanism of resistance to  \\nanti- CTLA-4 therapy. Cell 167, 397–404 e399 \\n(2016).\\n171. Garcia- Diaz, A. et\\xa0al. Interferon receptor signaling \\npathways regulating PD- L1 and PD- L2 expression.  \\nCell Rep. 19, 1189–1201 (2017).\\n172. Gilbert, L. A. et\\xa0al. CRISPR- mediated modular  \\nRNA- guided regulation of transcription in eukaryotes.  \\nCell 154, 442–451 (2013).\\n173. Gilbert, L. A. et\\xa0al. Genome- scale CRISPR- mediated \\ncontrol of gene repression and activation. Cell 159, \\n647–661 (2014).  \\nThese authors discover and describe the utility of \\nCRISPR- based gene silencing as an alternative to \\ngene editing (‘cutting’) in genome- wide functional \\nscreens.\\n174. Qi, L. S. et\\xa0al. Repurposing CRISPR as an RNA- guided \\nplatform for sequence- specific control of gene \\nexpression. Cell 152, 1173–1183 (2013).\\n175. Zetsche, B. et\\xa0al. Multiplex gene editing by CRISPR-  \\nCpf1 using a single crRNA array. Nat. Biotechnol. 35, \\n31–34 (2017).\\n176. Horlbeck, M. A. et\\xa0al. Mapping the genetic landscape \\nof human cells. Cell 174, 953–967 e922 (2018).\\n177. Du, D. et\\xa0al. Genetic interaction mapping in mammalian cells using CRISPR interference.  \\nNat. Methods 14, 577–580 (2017).\\n178. Han, K. et\\xa0al. Synergistic drug combinations for  \\ncancer identified in a CRISPR screen for pairwise genetic interactions. Nat. Biotechnol. 35, 463–474 \\n(2017).\\n179. Wong, A. S. et\\xa0al. Multiplexed barcoded CRISPR- Cas9 \\nscreening enabled by CombiGEM. Proc. Natl Acad. Sci. USA 113, 2544–2549 (2016).  \\nThis is the first study to explore the interdependency of two individual genes using a multiplexed  \\ncombo- CRISPR system.\\n180. Najm, F. J. et\\xa0al. Orthologous CRISPR- Cas9 enzymes \\nfor combinatorial genetic screens. Nat. Biotechnol. 36, \\n179–189 (2018).\\n181. Komor, A. C., Kim, Y. B., Packer, M. S., Zuris, J. A. & \\nLiu, D. R. Programmable editing of a target base in genomic DNA without double- stranded DNA cleavage. \\nNature 533, 420–424 (2016).\\n182. Gao, X. et\\xa0al. T reatment of autosomal dominant \\nhearing loss by in\\xa0vivo delivery of genome editing agents. Nature 553, 217–221 (2018).\\n183. Nishida, K. et\\xa0al. T argeted nucleotide editing using hybrid prokaryotic and vertebrate adaptive immune \\nsystems. Science 353, aaf8729 (2016).\\n184. Gaudelli, N. M. et\\xa0al. Programmable base editing of \\nA*T to G*C in genomic DNA without DNA cleavage. \\nNature 551, 464–471 (2017).\\n185. Abudayyeh, O. O. et\\xa0al. C2c2 is a single- component \\nprogrammable RNA- guided RNA- targeting CRISPR \\neffector. Science 353, aaf5573 (2016).\\n186. Abudayyeh, O. O. et\\xa0al. RNA targeting with CRISPR-  \\nCas13. Nature 550, 280–284 (2017).\\n187. Cox, D. B. T . et\\xa0al. RNA editing with CRISPR- Cas13. \\nScience 358, 1019–1027 (2017).\\n188. Konermann, S. et\\xa0al. T ranscriptome engineering with \\nRNA- targeting type VI- D CRISPR effectors. Cell 173, \\n665–676 e614 (2018).\\n189. Yan, W. X. et\\xa0al. Cas13d is a compact RNA- targeting \\ntype VI CRISPR effector positively modulated by a \\nWYL- domain-containing accessory protein. Mol. Cell  \\n70, 327–339.e5 (2018).\\n190. Gootenberg, J. S. et\\xa0al. Multiplexed and portable \\nnucleic acid detection platform with Cas13, Cas12a, \\nand Csm6. Science 360, 439–444 (2018).\\n191. Myhrvold, C. et\\xa0al. Field- deployable viral diagnostics \\nusing CRISPR- Cas13. Science 360, 444–448  \\n(2018).\\n192. Dixit, A. et\\xa0al. Perturb- Seq: dissecting molecular \\ncircuits with scalable single- cell RNA profiling of \\npooled genetic screens. Cell 167, 1853–1866.e17 \\n(2016).  \\nThis article describes Perturb- Seq, a method of \\nusing single- cell transcriptional output as the \\nreadout for CRISPR- based functional genomic \\nscreens.\\nnature reviews | DRug DISCoveRyReviews\\n  volume 19 | January 2020 | 37', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), Document(id_='cad51f3a-65a4-408a-91c9-c14ca1d75a37', embedding=None, metadata={'page_label': '38', 'file_name': 'Synthetic lethality as an engine for cancer drug target discovery.pdf', 'file_path': 'files/Papers_FullText/Synthetic lethality as an engine for cancer drug target discovery.pdf', 'file_type': 'application/pdf', 'file_size': 1996992, 'creation_date': '2024-01-10', 'last_modified_date': '2024-01-10', 'last_accessed_date': '2024-01-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, hash='beb945b2a8a546e2dab330bd2f2d6564f5c0f4ab52c65bfcc411c7948b5675ba', text='193. Adamson, B. et\\xa0al. A multiplexed single- cell CRISPR \\nscreening platform enables systematic dissection of the \\nunfolded protein response. Cell 167, 1867–1882.e21 \\n(2016).\\n194. Jaitin, D. A. et\\xa0al. Dissecting immune circuits by linking \\nCRISPR- pooled screens with single- cell RNA- Seq. Cell  \\n167, 1883–1896.e15 (2016).\\n195. Xie, S., Duan, J., Li, B., Zhou, P . & Hon, G. C. Multiplexed engineering and analysis of combinatorial enhancer \\nactivity in single cells. Mol. Cell 66, 285–299.e5 \\n(2017).\\n196. Datlinger, P . et\\xa0al. Pooled CRISPR screening with single- cell transcriptome readout. Nat. Methods 14, \\n297–301 (2017).\\n197. Mirza, M. R. et\\xa0al. Niraparib maintenance therapy in \\nplatinum- sensitive, recurrent ovarian cancer. N. Engl. \\nJ. Med. 375, 2154–2164 (2016).\\n198. Li, W. et\\xa0al. MAGeCK enables robust identification  \\nof essential genes from genome- scale CRISPR/Cas9 \\nknockout screens. Genome Biol. 15, 554 (2014).Acknowledgements\\nThe authors thank C.P . Johnson and K. Briggs for helpful dis -\\ncussions and preparation of the manuscript and F. Li and  \\nT. Teng for curation and analysis of the Project Achilles data-set. All are full- time employees of and shareholders in T ango \\nTherapeutics.\\nAuthor contributions\\nB.W. and A.H. researched data for article and wrote the arti-cle. A.A. contributed to writing the article. All authors contrib-\\nuted substantially to discussion of the content and reviewed \\nand edited the manuscript before submission:\\nCompeting interests\\nA.H. and B.W. are employees of and shareholders in Tango Therapeutics. L.A.G. is an employee of and shareholder in Eli\\xa0Lilly \\nand Company and is a shareholder in T ango Therapeutics. A.A. \\nis a shareholder in T ango Therapeutics, a consultant for AtlasMDX, Third Rock Ventures, Pfizer, ProLynx and Bluestar and a Genentech scientific advisory board member and receives grant support from Sun Pharma and AstraZeneca. Patents on the use of PARP inhibitors held jointly with AstraZeneca that A.A. \\nhas benefited from financially (and may do so in the future) \\nthrough the ICR Rewards to Inventors Scheme include WO2014013231 (A1) — 2014-01-23, US2012135983 (A1) \\n— 2012-05-31, US2012010204 (A1) — 2012-01-12, \\nUS2006142231 (A1) — 2006-06-29, WO2008020180 (A2) \\n— 2008-02-21.\\nPublisher’s note\\nSpringer Nature remains neutral with regard to jurisdictional \\nclaims in published maps and institutional affiliations.\\nReLa Ted LinkS\\nCancer Cell Line encyclopedia: https://portals.\\nbroadinstitute.org/ccle\\nProject Achilles: https://depmap.org/portal/achilles/\\nProject DRive: https://oncologynibr.shinyapps.io/drive/\\nProject score: https://score.depmap.sanger.ac.uk/\\nwww.nature.com/nrdReviews\\n38 | January 2020 | volume 19 ', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), Document(id_='9b796969-4b97-47ee-911c-cbfd0f0e327d', embedding=None, metadata={'page_label': '689', 'file_name': 'THE CONCEPT OF SYNTHETIC LETHALITY IN THE CONTEXT OF ANTICANCER THERAPY.pdf', 'file_path': 'files/Papers_FullText/THE CONCEPT OF SYNTHETIC LETHALITY IN THE CONTEXT OF ANTICANCER THERAPY.pdf', 'file_type': 'application/pdf', 'file_size': 600362, 'creation_date': '2024-01-10', 'last_modified_date': '2024-01-10', 'last_accessed_date': '2024-01-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, hash='fe2c2f1f724c1d9a80b33b443308776582a032eeac50bd2cda59af0670f965de', text='© 2005 Nature Publi shing Group Howard Hughes Medical \\nInstitute, 44 Binney Street, \\nMayer 457, Boston, \\nMassachusetts 02115, USA.\\ne-mail:  william_kaelin@\\ndfci.harvard.edu\\ndoi:10.1038/nrc1691\\nPublished online 19 August 2005THERAPEUTIC INDEX\\nThe therapeutic index for a \\ndrug is defined as the dose \\n(concentration) required for \\ntoxic effects divided by the dose \\n(concentration) required for \\ntherapeutic effects.\\nTHERAPEUTIC WINDOW\\nThe therapeutic window for a \\ndrug refers to the concentration \\nrange over which therapeutic \\neffects can be expected.THE CONCEPT OF SYNTHETIC \\nLETHALITY IN THE CONTEXT \\nOF ANTICANCER THERAPY\\nWilliam G. Kaelin Jr\\nAbstract | Two genes are synthetic lethal if mutation of either alone is compatible with viability \\nbut mutation of both leads to death. So, targeting a gene that is synthetic lethal to a cancer-\\nrelevant mutation should kill only cancer cells and spare normal cells. Synthetic lethality therefore \\nprovides a conceptual framework for the development of cancer-specific cytotoxic agents. \\nThis paradigm has not been exploited in the past because there were no robust methods for \\nsystematically identifying synthetic lethal genes. This is changing as a result of the increased \\navailability of chemical and genetic tools for perturbing gene function in somatic cells.\\nThe bottleneck to the development of safe and effective \\nanticancer drugs does not lie in an inability to identify \\nchemicals that will kill cancer cells. In fact, thousands \\nof compounds have been identified over the past 50 \\nyears that will accomplish this feat. Instead, the bottle-\\nneck lies in our inability to identify chemicals that will \\nkill cancer cells at concentrations that do not harm \\npatients. Most of the chemotherapeutic agents used \\ntoday have remarkably low THERAPEUTIC INDICES  and \\nnarrow THERAPEUTIC WINDOWS . The therapeutic window \\nis influenced by a number of factors, including the \\nshape of the curve that relates the intended biologi-\\ncal effect of the drug to changes in the activity of its \\nintended target (‘on-target’), and the propensity of \\nthe drug to affect unintended targets (‘off-targets’) \\nat higher doses. Off-target effects can cause toxicity \\nand, in some cases, antagonize on-target biological \\neffects. Most anticancer drugs in use today were dis-\\ncovered based on their ability to kill rapidly dividing \\ncancer cells in vitro . Predictably, when administered to \\npatients, many of these drugs also injure rapidly divid-\\ning normal cells, such as bone-marrow haematopoietic \\nprecursors and gastrointestinal mucosal epithelial cells. \\nIn addition, many of these drugs are toxic to normal \\ncells that are not rapidly dividing. Examples include \\ndoxorubicin (toxic to the heart), bleomycin (toxic to \\nthe lung) and cytarabine (toxic to the cerebellum). These other forms of organ damage become particu-\\nlarly important (dose-limiting) in settings in which \\ntoxicity to rapidly dividing cells can be partially amel-\\niorated through supportive-care measures (such as \\nbone-marrow transplantation). For these reasons, it is \\nimperative that anticancer drugs be developed that can \\nkill cancer cells at clinically achievable concentrations, \\nwith therapeutic indices that are higher than those of \\nclassic cytotoxic agents.\\nTherapeutic index\\nMany factors influence the therapeutic index of a drug. \\nSome relate to the quality of the drug itself — for exam-\\nple, its ability to distinguish between intended and unin-\\ntended targets. Others relate to the nature of its target \\n— for example, its distribution, its normal function(s), \\nand the degree to which those functions must be altered \\nto achieve the desired effect. Most antibacterial agents \\nare remarkably safe because their targets are present in \\nthe organisms they are designed to kill but not in normal \\nhost cells. However, many other relatively ‘safe’ drugs \\n— such as anti-hypertensives, anti-anxiety drugs and \\ncholesterol-lowering agents — inhibit normal cellular \\nproteins. These drugs are clinically useful because their \\neffects are titratable (through changes in dose and sched-\\nule), and quantitative changes in the activities of their \\ntargets lead to the desired changes in host physiology.\\nNATURE REVIEWS | CANCER   VOLUME 5 | SEPTEMBER 2005 | 689REVIEWS', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), Document(id_='5cba57b7-19ad-4a30-ab27-d9f343585d03', embedding=None, metadata={'page_label': '690', 'file_name': 'THE CONCEPT OF SYNTHETIC LETHALITY IN THE CONTEXT OF ANTICANCER THERAPY.pdf', 'file_path': 'files/Papers_FullText/THE CONCEPT OF SYNTHETIC LETHALITY IN THE CONTEXT OF ANTICANCER THERAPY.pdf', 'file_type': 'application/pdf', 'file_size': 600362, 'creation_date': '2024-01-10', 'last_modified_date': '2024-01-10', 'last_accessed_date': '2024-01-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, hash='6e8e2818eb79fbd39ee62f0643c753ccaf53956d1847d318ffab009316bd6e43', text='© 2005 Nature Publi shing Group a  Target-driven therapeutic index\\nNormal cell Cancer cell\\nb  Context-driven therapeutic index\\nNormal cell Cancer cell\\nc  Context-driven therapeutic index\\nNormal cell Cancer cellTarget\\nCell intrinsic \\ndifferences\\nCell extrinsic \\ndifferencesGenetic\\nEpigenetic\\nAltered\\nmicro-\\nenvironmentTwo paths can be envisioned to arrive at an anti-\\ncancer drug that would selectively kill cancer cells. \\nThe first, which is modelled on the development \\nof anti-infectious agents, would be to identify drug \\ntargets that are essential for the viability of cancer \\ncells but are not present in normal cells (the so-called \\n‘target-driven therapeutic index’)1,2 \\uf6aeFIG. 1\\uf6af . The fusion \\nproteins generated by cancer-associated chromosomal \\ntranslocations might, at first glance, seem to be ideal \\nin this regard. However, this presumes that drugs can \\nbe developed that will discriminate between a particu-\\nlar protein (or functional subdomain) in its normal \\ncontext and in its pathogenic, fused state. This might \\nbe difficult. For example, it is fallacious to argue that \\nthe efficacy and safety of imatinib mesylate (Glivec)  \\nfor the treatment of chronic myelogenous  leukaemia  \\n(CML) stems from the fact that its target, breakpoint \\ncluster region ( BCR )–Abelson murine leukaemia viral \\noncogene homologue ( ABL ), is unique to CML cells \\nbecause imatinib mesylate inhibits the kinase activi-\\nties of both BCR–ABL and ABL (in addition to several \\nother cellular kinases)3. So, the relatively high thera-\\npeutic index of imatinib mesylate cannot be explained \\nby the restriction of its target(s) to CML cells (see \\nbelow for potential alternative explanations). Similarly, \\nit might be difficult to develop drugs that directly \\ninhibit oncoproteins that result from point mutations \\nwithout affecting their normal counterparts.\\nA second way to achieve enhanced cancer-cell selec-\\ntivity, however, would be to identify situations where \\nthe requirement for a particular target was enhanced \\nin the context of a cancer cell compared with normal \\ncells (the so-called ‘context-driven therapeutic index’)1,2 \\n\\uf6aeFIG. 1\\uf6af . The requirement for a particular target might \\nbe increased because of changes that are intrinsic to the \\ncancer cell (for example, through epigenetic or genetic \\nchanges), extrinsic to the cancer cell (for example, as a \\nresult of microenvironmental changes leading to altered \\ncell–matrix and cell–cell interactions), or both.\\nAll of the anticancer drugs in use today affect tar-\\ngets that are shared between normal cells and cancer \\ncells, including enzymes involved in fundamental \\nprocesses such as DNA replication. The fact that their \\ntherapeutic indices, however small, exceed unity, \\ncoupled with the observation that they can, in certain \\nsettings, induce striking remissions and occasionally \\ncures (for example, cisplatinum-based regimens for \\ntesticular cancer), indicates that contextual differences \\nbetween normal cells and cancer cells are therapeuti-\\ncally exploitable. So, can our growing knowledge of \\ncancer genetics, coupled with a more sophisticated \\nunderstanding of gene–gene interactions, be used to \\nidentify drug targets that have enhanced therapeu-\\ntic indices by virtue of such contextual differences? \\nStudies of gene–gene interactions in model organisms \\nhave provided a conceptual framework for this task.\\nSynthetic lethality\\nTwo genes (‘ A’ and ‘ B’) are said to be ‘synthetic lethal’ \\nif mutation of either gene alone is compatible with \\nviability but simultaneous mutation of both genes Summary\\n• Many chemicals kill cancer cells but their toxicity to normal cells limits their \\nusefulness as anticancer drugs.\\n• Epigenetic and genetic alterations within ca ncer cells, as well as changes in their \\nmicroenvironment, might increase their requirement for a particular molecular \\ntarget (or targets) relative to normal cells, creating an opportunity for selectivity.\\n• Two genes are synthetic lethal if mutation of either gene alone is compatible with \\nviability but mutation of both leads to death. Inhibiting the products of genes that \\nare synthetic lethal to cancer-causing mutations should, by definition, kill cells that \\nharbour such mutations, while sparing normal cells.\\n• Most drugs induce a loss-of-function phenotype. High-throughput screens \\nusing matched cell-line pairs and chemical libraries allow the identification of \\nchemicals that inhibit or kill cells in a genotype-specific manner. The challenge \\nin this setting is to identify the relevant target (or targets) of compounds that \\nscore positively.\\n• Genome-wide RNA-interference screens can now be used to identify synthetic \\nlethal interactions in cells that are derived from higher eukaryotes, including \\nhumans.\\n• Gene–gene interactions, including synthetic lethal interactions that are \\ndiscovered in cell-culture experiments, will ultimately need to be validated \\nin vivo . It seems likely that some gene–gene interactions will be highly robust, \\nwhereas others might be valid only in specific cells or under specific \\nexperimental conditions. \\nFigure 1 |  Framework for developing anticancer drugs \\nwith a high therapeutic index.  An anticancer drug might \\nhave a high therapeutic index because its target is uniquely \\npresent in cancer cells ( a), or because the requirement for \\nits target is quantitatively or qualitatively different in cancer \\ncells than in normal cells ( b and c). This differential \\nrequirement might be because of intrinsic differences in the \\ncells ( b), such as genetic (red) and epigenetic (blue) \\ndifferences, or extrinsic differences in the cells ( c), such as \\nloss of survival signals provided by normal cell–cell and \\ncell–matrix interactions. Modified with permission from \\nREF. 2  © (2002) Elsevier Science.\\n690 | SEPTEMBER 2005 | VOLUME 5  www.nature.com/reviews/cancerREVIEWS', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), Document(id_='ca3c156d-fbc1-47f5-b4e7-9a70a031ec09', embedding=None, metadata={'page_label': '691', 'file_name': 'THE CONCEPT OF SYNTHETIC LETHALITY IN THE CONTEXT OF ANTICANCER THERAPY.pdf', 'file_path': 'files/Papers_FullText/THE CONCEPT OF SYNTHETIC LETHALITY IN THE CONTEXT OF ANTICANCER THERAPY.pdf', 'file_type': 'application/pdf', 'file_size': 600362, 'creation_date': '2024-01-10', 'last_modified_date': '2024-01-10', 'last_accessed_date': '2024-01-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, hash='8c5e04855572763b833a5a93e83142b30eddbf40535133eceb462fa25afbbab3', text='© 2005 Nature Publi shing Group a  Synthetic lethality\\nGene AGene Bb  Extragenic suppression\\nGene AGene B\\nAB Viable\\nViable\\nViable\\nLethalAb\\naB\\nabAB Viable\\nViable\\nLethal\\nViableAb\\naB\\nab% fitness\\n0\\n0100\\n1000100\\n0100% fitness\\n% fitness\\n% B activity0 100\\n% B activity0 100\\n% B activity\\nWild type\\nA–/–PARALOGUES\\nParalogues are genes that share \\nsignificant homology within a \\nparticular species. Such genes \\nare paralogous to each other.\\ncauses death4–9 (FIG. 2) . This concept can be extended to \\nsituations in which simultaneous mutation of two genes \\nimpairs cellular fitness more than mutation of either \\ngene alone (‘synthetic sick’). In either of these two situa-\\ntions, A buffers the effect of changes in B and vice-versa, \\nbut this buffering is lost when both A and B are mutated \\nat the same time4,6,10. Synthetic lethal interactions have \\nmost commonly been described for loss-of-function \\nalleles, but can also involve gain-of-function alleles. For \\nexample, gene B might become essential for survival \\nwhen a particular gene A is overexpressed (known as \\nsynthetic dosage lethality)11–13. Approximately 20% of \\ngenes in the budding yeast Saccharomyces cerevisiae  \\nare individually essential, but genetic screens in this \\norganism suggest that synthetic lethal interactions are \\ncommon among the remaining 80% (perhaps on the \\norder of 10 interactions per gene)10,14,15.\\nLoss-of-function alleles having a synthetic lethal (or \\nsynthetic sick) relationship can often, but not always, \\nbe easily rationalized based on the functions of their \\nprotein products. They might, for example, be uniquely \\nredundant with respect to an essential function (as \\noccurs in some PARALOGUES ), be two subunits of an \\nessential multiprotein complex, be two interconnected \\ncomponents in an essential linear pathway (with each mutation decreasing the flux through the pathway), \\nor participate in parallel pathways that are together \\nessential for survival (for example, a crucial metabolic \\npathway and an alternative or salvage pathway). The \\nconcept of synthetic lethality can be further extended \\nto embrace the situation where mutation of A is lethal \\nonly in combination with mutations that affect several \\nnon-essential genes B, C, D and so on2,6.\\nIt has been suggested that the concept of synthetic \\nlethality could be used to choose anticancer drug tar-\\ngets1,7,16. In particular, the protein products of genes that \\nare synthetic lethal to known cancer-causing mutations, \\nif amenable to pharmacological attack (for example, if \\nthey encode an enzyme), should theoretically represent \\nexcellent targets for anticancer therapy. This approach \\nsimultaneously tackles two vexing problems in cancer \\npharmacology. The first relates to the fact that many \\ncancer-associated mutations, like most drugs, induce a \\nloss of function1,2. Therefore, it is not immediately obvi-\\nous how to pharmacologically approach cancer cells \\nin which, for example, a particular tumour-suppressor \\nprotein is crippled (or worse yet, absent). Targeting a \\nprotein that is synthetic lethal to such a lost or crippled \\nprotein provides an elegant solution to this problem. The \\nsecond problem relates to whether it is possible to achieve \\nselectivity by inhibiting proteins that are also important \\nfor cellular homestasis. If A and B are synthetic lethal \\n(or synthetic sick), then inhibitors of B should selectively \\nkill (or inhibit) cancer cells with mutant A. In the ideal \\nsituation, complete neutralization of B, genetically or \\npharmacologically, would have no effect on normal cells, \\nand even partial inhibition of B in cancer cells would \\ncause death (because of mutant A; FIG. 3, left panel). \\nHowever, B inhibitors might display a significant thera-\\npeutic index even when these ideal conditions are not \\nmet. This would require that the A mutation shifts or \\nalters the fitness dose–response curve of the B inhibitor \\nsuch that keeping B activity below a certain threshold \\nselectively impairs cells with mutant A (FIG. 3, middle and \\nright panels). \\nIt could be argued that some (and perhaps most) \\nanticancer drugs in use today are, at least in hind-\\nsight, exploiting synthetic lethal, or synthetic sick, \\ninteractions. For example, synthetic lethal relation-\\nships between DNA-replication genes (such as certain \\nDNA polymerases) and DNA-repair genes (such as \\nmismatch-repair genes) are well documented in model \\norganisms7,16. It seems likely that the efficacy of the \\nmany anticancer drugs that interfere with DNA syn-\\nthesis is due, at least in some cases, to the presence of \\ntumour-associated mutations that affect DNA repair \\nor the response to DNA damage. Another example of \\nsynthetic interactions is provided by certain chemo-\\ntherapeutic agents and mutations that directly or \\nindirectly compromise the function of the retinoblas-\\ntoma protein ( pRB, encoded by the RB1 gene) tumour \\nsuppressor. Inactivation of pRB has been documented \\nin many cancers and leads to an increase in E2F activ-\\nity, which, in turn, activates various genes involved in \\nS-phase entry17. One of these, topoisomerase II, causes \\nDNA strand breaks and apoptosis when bound to Figure 2 | Gene–gene interactions: synthetic lethal \\nand suppressive interactions for two genes. Two \\ngenes (‘ A’ and ‘ B’) are said to be ‘synthetic lethal’ if \\nmutation of either gene alone is compatible with viability \\nbut simultaneous mutation of both genes causes death. \\nB is an extragenic suppressor of A if mutation of B \\nsuppresses the phenotype observed when A is mutated. \\nA lowercase letter denotes a mutant.\\nFigure 3 | Theoretical fitness curves for wild-type and A–/– cells in response to a drug \\nthat inhibits the B gene product. A reading of  0% fitness denotes death, whereas 100% \\nfitness denotes the wild-type state (for simplicity, fitness >100% is not considered in these \\nexamples). In the middle panel, a therapeutic window is created by a shift in the fitness curve \\nwhen gene A is absent. In the left and right panels the therapeutic window is created by \\nchanges in the shapes of the fitness curves when gene A is absent.\\nNATURE REVIEWS | CANCER   VOLUME 5 | SEPTEMBER 2005 | 691REVIEWS', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), Document(id_='090b35e5-a1af-4641-9b82-82d0e9956453', embedding=None, metadata={'page_label': '692', 'file_name': 'THE CONCEPT OF SYNTHETIC LETHALITY IN THE CONTEXT OF ANTICANCER THERAPY.pdf', 'file_path': 'files/Papers_FullText/THE CONCEPT OF SYNTHETIC LETHALITY IN THE CONTEXT OF ANTICANCER THERAPY.pdf', 'file_type': 'application/pdf', 'file_size': 600362, 'creation_date': '2024-01-10', 'last_modified_date': '2024-01-10', 'last_accessed_date': '2024-01-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, hash='2e1f36dd555967af48b9ad87c88a3f6f1e8d33c7a188750f8ed8aacd6ebe9206', text='© 2005 Nature Publi shing Group topoisomerase inhibitors such as etoposide. As would \\nbe predicted, pRB-pathway mutations sensitize cells to \\ndrugs that inhibit topoisomerase II \\uf6aeREFS 18\\uf6bb21\\uf6af . In addi-\\ntion, E2F1 , like the oncoprotein MYC , increases the \\nexpression of many pro-apoptotic genes, including the \\np53 paralogue p73, which might sensitize pRB-defec-\\ntive cells to drugs that elicit  additional apoptotic signals \\n(such as DNA-damaging agents)22–25.\\nTwo newer anticancer agents also exploit con-\\ntextual differences between cancer cells and normal \\ncells. Studies in model organisms suggest that muta-\\ntions affecting chaperones that are involved in protein \\nfolding can unmask the deleterious consequences of \\nvarious mutations26. Preclinical data indicate that \\nHSP90  (heat-shock protein of 90kDa) inhibitors \\nhave anticancer activity, and that certain mutant \\noncoproteins, such as mutant BRAF and mutant \\nEGFR  (epidermal growth factor receptor), have an \\nincreased requirement for HSP90 function27–29. One \\nHSP90 inhibitor, 17AAG, has completed phase I test-\\ning and is entering phase II studies. The accumula-\\ntion of mutated and/or misfolded proteins might \\nalso alter the requirement of a cell for proteasomal \\nfunction30. The proteasomal inhibitor bortezomib is \\nwell tolerated in humans and was recently approved \\nfor the treatment of multiple myeloma31.\\nDiscovery of human synthetic lethal interactions\\nOur knowledge of the molecular networks that are \\nestablished in normal cells and cancer cells is too \\nrudimentary to allow reliable predictions of the genes \\nthat will be synthetic lethal to a given cancer gene. \\nNonetheless, a few ideas have been put forward for \\nhow synthetic lethal combinations might be achieved, \\nbased on first principles. Many oncoproteins, includ-\\ning E2F1 and MYC, represent a double-edged \\nsword for cancer cells because they deliver both \\npro-mitogenic and pro-apoptotic signals. A counter-\\nintuitive approach to treating cancer cells that have \\nhyperactive oncoproteins such as these would be treat-\\ning them with drugs that enhance their action further, \\nin the hope of crossing an apoptotic threshold. For \\nexample, E2F1 is negatively regulated by both pRB and \\ncyclin A32–35 . Loss of the pRB pathway establishes a \\npositive-feedback loop in which E2F1 activates its own \\npromoter36, and blocking the remaining interaction of \\ncyclin A with E2F1 kills transformed cells but not their \\nnormal counterparts37–39. Unfortunately, inhibiting the \\nactivity of the cyclin-A partner CDK2 (cyclin-depend-\\nent kinase 2) does not have the same effect40, possibly \\nbecause another catalytic partner can substitute for \\nCDK2 in its absence41,42. Synthetic lethal interactions \\nmight also be predicted based on the loss of particular \\ncell-cycle checkpoints16. For example, S-phase cells, in \\ncontrast to G1 cells, can be  induced to undergo prema-\\nture chromosomal condensation under certain con-\\nditions, such as treatment with caffeine at doses that \\ninhibit ATR (ataxia telangiectasia and RAD3-related \\nprotein)43–46. Cells that lack p53, which has a role in \\nG1 control, are more susceptible to caffeine than their \\nwild-type counterparts47.There are now multiple examples of cancers that \\nseem to be dependent on or ‘addicted’ to certain acti-\\nvated oncogenes (gene-replacement experiments sug-\\ngest that tumour cells can also become addicted to the \\ninactivation of tumour-suppressor genes). Oncogene \\naddiction might underlie the success of the kinase \\ninhibitor imatinib mesylate for CML (in which the \\noncogene is BCR–ABL ) and gastrointe stinal stromal \\ntumours (in which the oncogene is KIT)3 and of the \\nEGFR inhibitor gefitinib for EGFR-mutated non-small-\\ncell lung cancer48–51. Bernard Weinstein, who coined the \\nterm ‘oncogene addiction’ , initially envisioned that this \\nphenomena was related to the ability of such oncogenes, \\nwhich can be viewed as nodes in complex molecular \\nnetworks, to simultaneously deliver proliferative and \\nantiproliferative signals52 (FIG. 4a) . As long as the onco-\\ngene signal is sustained, the proliferative signal — which \\nmight promote mitogenesis, survival, or both — would \\ndominate. However, if the oncogene is acutely silenced, \\nthe antiproliferative signal dominates, leading to cessa-\\ntion of growth or cell death (in this scenario it must be \\ninvoked that the antiproliferative signal ‘decays’ more \\nslowly than the proliferative signal when the oncogene \\nis inhibited)2.\\nSuperimposed on the network abnormalities that \\nare induced by activated oncogenes are network abnor-\\nmalities that are induced by mutations at other loci. The \\nresulting abnormalities in molecular circuitry create \\nadditional opportunities for oncogene addiction1,2,53,54, \\nincluding those that arise as a result of gene–gene \\ninteractions, such as synthetic lethality and extragenic \\nsuppression. Cancers arise through sequential genetic \\nchanges that ultimately convert a normal cell to a fully \\ntransformed one. These mutations are under selective \\npressure to be adaptive or neutral, from the point of \\nview of the cancer, in the context of the mutations that \\npreceded them (FIG. 4b) . It seems likely, a priori , that \\nsome of the mutations that occur late in the evolution \\nof a cancer cell might only be advantageous, or indeed \\neven tolerated, because of the mutations that preceded \\nthem (or put another way, these mutations would be \\ndeleterious if not for the mutations that had preceded \\nthem). In the extreme case, an early A mutation might \\nbe an extragenic suppressor of the lethality that would \\notherwise be caused by a late B mutation ( FIG. 2 , right \\npanel). If this is true, correcting the A mutation should \\ncause death because of the acquisition of the B mutation. \\nFor example, RB1 inactivation, as described above, leads \\nto increased E2F activity, which can stimulate S-phase \\nentry but can also promote p53-dependent apoptosis55,56. \\nSo, a tumour in which TP53  was already mutated might \\nderive an additional benefit from mutating RB1 but at \\nthe price of becoming addicted to p53 loss (in the sense \\nthat restoring p53 function would lead to apoptosis).\\nSimilarly, Mills and colleagues have suggested that \\noncogene addiction might arise because of the loss of \\ncollateral signalling pathways. This is due to genomic \\ninstability coupled with the loss of selection pressure \\nto maintain the collateral signalling pathways57, a \\nprocess referred to as ‘genetic streamlining’58 (FIG. 4c) . \\nCollectively, these ideas suggest that the pathways that \\n692 | SEPTEMBER 2005 | VOLUME 5  www.nature.com/reviews/cancerREVIEWS', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), Document(id_='d945b688-9018-43c9-b878-ce07a3c6a367', embedding=None, metadata={'page_label': '693', 'file_name': 'THE CONCEPT OF SYNTHETIC LETHALITY IN THE CONTEXT OF ANTICANCER THERAPY.pdf', 'file_path': 'files/Papers_FullText/THE CONCEPT OF SYNTHETIC LETHALITY IN THE CONTEXT OF ANTICANCER THERAPY.pdf', 'file_type': 'application/pdf', 'file_size': 600362, 'creation_date': '2024-01-10', 'last_modified_date': '2024-01-10', 'last_accessed_date': '2024-01-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, hash='da52ef4284a04aafe5866a4715c6829245a1fea1e0bd2dd795c4d4ec490255ca', text='© 2005 Nature Publi shing Group Proto-oncogene Normal growth\\nOncogene Growth stimulation\\nOncogene Diminished growthInhibitor\\nInhibitor\\nInhibitora\\ncb\\nMutation ‘n’, neutral or\\nbeneficial in the context \\nof mutations 1 to n – 1.\\nOtherwise deleterious\\nMutation 3, neutral or\\nbeneficial in the context\\nof mutations 1 and 2\\nMutation 2, neutral or\\nbeneficial in the context\\nof mutation 1\\nMutation 1\\nNormalTransformed\\nTreatment Collapse\\nParallel\\nProliferation\\nPathways+ Oncoprotein + Time + InhibitorNormal cell Die or fail to proliferate\\nare activated early in the course of tumour progression \\n(owing to oncogene activation or tumour-suppressor-\\ngene inactivation) are likely to be excellent therapeu-\\ntic targets because of synthetic interactions with the \\nmutational changes that followed them. Silencing these \\npathways should reveal the deleterious consequences \\nof these subsequent changes, whether these changes \\ndid or did not contribute to tumour progression. The \\npotential interrelationship between oncogene addiction \\nand synthetic lethality is illustrated by the phosphatase \\nand tensin homologue ( PTEN ) tumour-suppressor \\nprotein, which negatively regulates the phosphatidyli-\\nnositol 3-kinase (PI3K) pathway, and mTOR (mamma-\\nlian target of rapamycin). PTEN–/– cells are reported be \\nmore sensitive to the antiproliferative effects of mTOR \\ninhibitors than their wild-type counterparts59. This \\nobservation indicates that PTEN–/– cells are ‘addicted’ \\nto PI3K–mTOR signalling, and that PTEN and mTOR \\nhave a synthetic sick relationship.\\nChromosomal deletions in cancer cells lead to the \\nloss of one or both copies of many genes. Frei suggested \\nthat cancer-cell vulnerabilities to pharmacological attack might also be gleaned by examining the functions of \\ncontiguous genes that are homozygously deleted along \\nwith tumour-suppressor genes60. For example, the gene \\nencoding methylthioadenosine phosphorylase (MTAP) \\n— which has a role in a salvage pathway for adenosine \\nbiosynthesis — is often co-deleted with the adjacent \\nCDKN2A  locus, which encodes the tumour-suppressor \\nproteins INK4A and ARF on 9p21 \\uf6aeREF. 61\\uf6af . As would \\nbe predicted, cells that lack MTAP have increased sen-\\nsitivity to l-alanosine — a potent inhibitor of de novo  \\nAMP synthesis — and to an inhibitor of de novo  purine-\\nnucleotide synthesis, 6-methylmercaptopurine riboside \\n(MMPR)62.\\nKamb suggested that expression databases be mined \\nfor paralogous genes in which one or more members \\nwere underexpressed in cancer cells relative to normal \\ncells (for example, as a result of haploinsufficiency or \\nhomozygous deletion)58. A drug that inhibited the \\nremaining paralogue(s), but not the differentially \\nexpressed paralogue, would, theoretically, be cancer-cell \\nselective. This approach, however, presumes that it is \\npossible to develop drugs that can discriminate between Figure 4 |  Models of oncogene addiction. a  | Many oncogenes paradoxically induce pro-mitogenic signals as well as anti-\\nmitogenic (or pro-apoptotic) signals. Growth stimulation results from oncogene activation presumably because the former is \\ndominant to the latter. However, acute inactivation of the oncogene might cause growth cessation or death if the anti-mitogenic /\\npro-apoptotic signals decay more slowly than the mitogenic signals (for example, because of differences in mRNA and protein hal f-\\nlife). Adapted from REF. 53 . b | Oncogene dependency due to gene–gene interactions. Cancer cells accumulate mutations (arrows) \\nover time that cumulatively lead to a transformed phenotype. Selection favours acquisition of mutations that are neutral or ben eficial \\n(adaptive) in the context of the mutations that preceded them. However, some of these changes might be deleterious (red arrow) \\nwere it not for the changes that preceded them. If true, correcting early genetic changes (yellow arrow) will unmask these \\ndeleterious effects. In this model, cancer cells behave like a molecular ‘house of cards’. c | Activation (indicated by bold arrow) of an \\noncogenic pathway diminishes selection pressure to maintain collateral signalling pathways. Silencing of these collateral pathw ays \\nover time, because of genetic or epigenetic changes, leads to oncogene dependency. Adapted from REF. 57 .\\nNATURE REVIEWS | CANCER   VOLUME 5 | SEPTEMBER 2005 | 693REVIEWS', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), Document(id_='b4780a62-3fe4-4b7f-87b7-167969edd3ab', embedding=None, metadata={'page_label': '694', 'file_name': 'THE CONCEPT OF SYNTHETIC LETHALITY IN THE CONTEXT OF ANTICANCER THERAPY.pdf', 'file_path': 'files/Papers_FullText/THE CONCEPT OF SYNTHETIC LETHALITY IN THE CONTEXT OF ANTICANCER THERAPY.pdf', 'file_type': 'application/pdf', 'file_size': 600362, 'creation_date': '2024-01-10', 'last_modified_date': '2024-01-10', 'last_accessed_date': '2024-01-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, hash='ebdf6d19e50811bb1b19ccdd9a71998580d46947cbdb13e052a4a7e0fbb39829', text='© 2005 Nature Publi shing Group Chemical or RNAi library\\nWild type TSG–/–\\nWT Cancer gene mutated TSG–/– + GFPλ1\\nYFP positive BFP positive\\nCo-culture in multiwell plates\\ncontaining test compoundsCo-culture in multiwell plates\\ncontaining test compoundsEpisomal plasmid \\nencoding TSG \\nand GFP λ2abBROMODEOXYURIDINE \\uf6aeBRDU\\uf6af \\nINCORPORATION\\nIncorporation of the thymidine \\nnucleotide analogue \\nbromodeoxyuridine into DNA \\ncan be used to measure the rate \\nof DNA synthesis. \\nISOGENIC\\nTwo cell lines are isogenic if \\nthey are derived from the same \\nparental cell line, or from one \\nanother such that they are \\ngenetically identical.\\nparalogous proteins. Moreover, synthetic lethal screens \\nin yeast indicate that paralogous pairs represent a \\nminority of the potential synthetic lethal combinations \\nin a cell10,15,63. Therefore, unbiased chemical and genetic \\nscreens are likely to be the most fruitful methods for \\nidentifying novel synthetic lethal relationships on which \\nto base new cancer treatments.\\nScreens for synthetic lethal interactors\\nThe example of topisomerase II inhibitors, as cited \\nabove, demonstrates that proteins bound to drugs \\nmight have effects that are very different from those \\npredicted by true null mutations, or by techniques such \\nas RNA interference (RNAi) that cause quantitative reductions in protein abundance. For example, a drug \\nmight interfere with one function of a multifunctional \\nprotein, or cause a protein to act in a dominant-negative \\nor dominant-positive manner. For this reason, screens \\nfor synthetic lethality that are carried out using libraries \\nof chemical compounds are likely to be complementary \\nto screens that are carried out using genetic tools (such \\nas RNAi or short interfering RNA; siRNA).\\nChemical screens. Hartwell and Friend pioneered the \\nidea of screening for drug-like chemicals that specifi-\\ncally kill yeast deletion mutants with defects in cell-\\ncycle checkpoints or DNA repair16,64. This paradigm \\ncan be extended to human cells. A number of groups \\nhave identified chemicals from collections of pure \\ncompounds, or that are present in complex mixtures \\n(for example, extracts or broths), that selectively inhibit \\ncells with cancer-relevant genetic alterations using iso-\\ngenic human cell-line pairs grown in multiwell plates \\n(FIG. 5) . Schreiber and co-workers identified marine \\nsponge extracts that preferentially inhibited the pro-\\nliferation of Trp53–/– mouse embryonic fibroblasts, as \\ndetermined by BROMODEOXYURIDINE \\uf6aeBRDU\\uf6af INCORPORATION , \\nrelative to wild-type mouse embryonic fibroblasts65. \\nHowever, the chemical entities responsible for these \\neffects were not identified. Kinzler and co-workers \\nco-cultured KRAS -mutated colon cancer cells (engi-\\nneered to produce blue fluorescent protein) with a \\nsubclone in which the mutant KRAS  allele was elimi-\\nnated by homologous recombination (and engineered \\nto produce yellow fluorescent protein), and monitored \\ndifferential killing using the ratio of blue/yellow fluo-\\nrescence66 (FIG. 6a) . Several chemical entities, including \\na novel cytidine nucleoside, were found that selectively \\nkilled cells containing mutant KRAS . A fluorescence-\\nbased mammalian synthetic lethal assay, which was \\nmodelled after earlier yeast assays67, was also devel-\\noped by Canaani and colleagues68,69 (FIG. 6b) . Leder and \\nco-workers discovered a small molecule called F16, \\nwhich selectively kills ERBB2  (also known as HER2/\\nNEU)-overexpressing mammary epithelial cells, com-\\npared with their normal counterparts70,71. The toxicity \\nof F16 correlates with its selective uptake in, and dis-\\nruption of, mitochondria of cells that are transformed \\nwith ERBB2. Stockwell and co-workers identified a \\nnumber of compounds that preferentially killed pri-\\nmary human cells that were transformed in vitro  with \\nhuman telomerase reverse transcriptase (TERT), RAS, \\nand oncoproteins that affect pRB, p53 and/or protein \\nphosphatase 2A (PP2A)21. Included among these were \\nclinically useful inhibitors of topoisomerase I and II. \\nIn a focused screen of pro-apoptotic agents Quon and \\ncolleagues discovered that human cells overexpress-\\ning MYC displayed increased sensitivity to the death \\nreceptor DR5 agonist tumour-necrosis-factor-related \\napoptosis-inducing ligand (TRAIL) in vitro  and in vivo , \\nand linked this to p53-independent induction of DR5 \\nby MYC72. Recent studies suggest that it is possible to \\nscreen pairwise combinations of drugs against ISOGENIC  \\ncell lines to uncover novel drug–gene and drug–drug \\ninteractions73,74.Figure 5 |  Synthetic lethal screening with chemical or \\ninterfering RNA libraries. Isogenic cell-line pairs that do or \\ndo not harbour a cancer-relevant mutation (in the case \\nillustrated, the cell-line pair differs only with respect to a \\nparticular tumour-suppressor gene ( TSG)) are grown in \\nmultiwell plates to which different chemical or genetic (short \\ninterfering RNAs, short hairpin RNAs or other interfering \\nRNAs) perturbants are added. In time, such assays might be \\ncarried out using microarrays spotted with chemicals or \\nsiRNA species104,105. A ‘hit’ is a perturbant that is cytostatic or \\ncytotoxic to the cell with the cancer-relevant mutation (arrow). \\nIt should be noted that the interpretation of such assays \\nneeds to consider potentially confounding effects, such as \\ndifferences in proliferation rate and cell-cycle distribution.\\nFigure 6 |  Fluorescence-based mammalian synthetic lethal assay. a  | The Kinzler \\nmethod66. Isogenic cell-line pairs that do/do not harbour a cancer-relevant mutation are \\nengineered to produce blue fluorescent protein (BFP) and yellow fluorescent protein (YFP), \\nrespectively, and are co-cultured in multiwell plates to which different chemicals are \\nadded. Selective killing of blue cells is indicative of a synthetic lethal interaction (yellow well). \\nb | The Canaani method68,69. Cells lacking a tumour-suppressor gene ( TSG) are engineered to \\nstably produce a green fluorescent protein (GFP) with an emission wavelength of ‘1’. These \\ncells are transfected with an unstable episomal plasmid encoding the TSG along with a GFP \\nthat has a different emission wavelength (‘2’). Retention of the episomal plasmid after exposure \\nto chemical or genetic perturbants is indicative of a synthetic lethal relationship. WT, wild type.\\n694 | SEPTEMBER 2005 | VOLUME 5  www.nature.com/reviews/cancerREVIEWS', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), Document(id_='569b78c7-aa99-41e2-8eda-1d79818fd8ce', embedding=None, metadata={'page_label': '695', 'file_name': 'THE CONCEPT OF SYNTHETIC LETHALITY IN THE CONTEXT OF ANTICANCER THERAPY.pdf', 'file_path': 'files/Papers_FullText/THE CONCEPT OF SYNTHETIC LETHALITY IN THE CONTEXT OF ANTICANCER THERAPY.pdf', 'file_type': 'application/pdf', 'file_size': 600362, 'creation_date': '2024-01-10', 'last_modified_date': '2024-01-10', 'last_accessed_date': '2024-01-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, hash='f52e20d1d3a907a4af593260d216cfa102c805e62b283418212e6dd9d247d797', text='© 2005 Nature Publi shing Group LIPINSKI’S RULES\\nLipinski noted that the \\nfollowing properties predict \\nthat a chemical will have poor \\nbioavailability after oral \\nadministration: molecular mass \\ngreater than 500 Da, high \\nlipophilicity (calculated LogP \\ngreater than 5, where LogP \\nindicates solubility in octanol \\nrelative to water), more than 5 \\nhydrogen-bond donors and \\nmore than 10 hydrogen-bond \\nacceptors.The use of isogenic cell-line pairs to identify com-\\npounds that selectively kill cancer cells as a result \\nof synthetic interactions is a powerful approach for \\nthe following reason. It is not uncommon for ~1% of \\nthe compounds in a chemical library to inhibit the \\ngrowth of human cancer cells at the concentrations \\nused in typical high-throughput screens. This trans-\\nlates into thousands of potential anticancer drugs \\nfrom a screen conducted with 105 to 106 compounds \\n(such as might be found at a large pharmaceutical \\ncompany or public consortium). Without the use \\nof a filter, such as differential killing in a genotype-\\nspecific manner, there are too many ‘hits’ to pursue. \\nIn the past, this has led to ‘hits’ being prioritized on \\nthe basis of factors such as ease of synthesis, potency, \\nintellectual-property issues and the likelihood of hav-\\ning desirable absorption, distribution, metabolism \\nand excretion (ADME) properties based on accepted \\ncriteria such as ‘ LIPINSKI’S RULES ’75,76. Although they are \\nimportant, none of these latter considerations address \\nselectivity. Furthermore, these factors can sometimes \\nbe addressed by modifying the chemical structure \\nof the initial compound (medicinal chemistry). It \\nwould be ironic if chemicals that can selectively kill \\ncancer cells through synthetic lethal interactions \\nwere present but missed for this reason during the \\ncountless cytotoxic screens that have been conducted \\nsince the mid-twentieth century.\\nA generic problem for cell-based screening of \\nlibraries of chemical compounds relates to successful \\ntarget identification. In some cases, it is possible to use \\na chemical entity identified in such a screen to capture \\nits protein target by affinity chromatography77,78. For \\nchemicals that induce a phenotype in yeast, mutants \\nthat display increased or decreased resistance (fitness) \\ncan be sought79,80. Such mutants often provide clues as \\nto the pathways that are affected by a compound, and \\ntherefore its potential target (or targets). A conceptu-\\nally attractive approach to target identification would \\nbe to generate compendia of molecular signatures (for \\nexample, gene-expression profiles) for various loss-\\nof-function mutations in a suitable host (for example, \\nyeast or human cells)81. The signature generated by the \\ncompound of interest could then be compared in silico  \\nto the compendium, with the rationale that the com-\\npound signature and target-disruption signature should \\nbe near(est) neighbours in an ideal situation. The search \\nfor targets of chemicals identified in cell-based synthetic \\nlethal screens should also be expedited by a knowledge \\nof the genes that score as synthetic lethal in genetic \\nscreens carried out in model organisms and human \\ncells, as described below.\\nGenetic screens. In the past, genetic screens for syn-\\nthetic lethal interactors have been largely relegated to \\nmodel organisms such as yeast, the fruitfly Drosophila \\nmelanogaster  and the worm  Caenorhabditis elegans  that \\nare amenable to forward-genetic approaches. Typically, \\nthese approaches have combined random mutagenesis \\nwith phenotypic screens, reflecting the retention of the \\nquery gene linked to a suitable reporter. Synthetic lethal screens in yeast have been invaluable for elucidating \\ncertain principles surrounding synthetic lethal interac-\\ntions. Unfortunately, many tumour-suppressor genes \\nand oncogenes do not have clear yeast orthologues. \\nAlthough forward-genetic screens are more cumber-\\nsome in fruitflies and worms than in yeast, they offer \\nthe advantage that their genomes do contain ortho-\\nlogues of most human cancer genes. In worms the RB1 \\northologue, lin-35 , has been well studied in the context \\nof vulvar development82. Fay and co-workers reported \\nthat a gene encoding a ubiquitin-conjugating enzyme \\nrelated to human UBCH7 is synthetic lethal to lin-35  \\n\\uf6aeREF. 83\\uf6af , as is the worm homologue of CDH1  \\uf6aeREF. 84\\uf6af . \\nUsing a fruitfly-based screen in which the fruitfly \\nRB1-like gene Rbf1  was conditionally inactivated in \\nthe eye, Belvin and co-workers discovered that RBF1 \\nis synthetic lethal to a novel prolyl isomerase85. It is not \\nyet known whether these synthetic lethal interactions \\nwill hold true in all cell types, nor whether they will \\nhold true across species.\\nHowever, forward-genetic approaches such as these \\nare now giving way to genome-wide reverse-genetic \\napproaches. Successful studies have been carried out in \\nyeast \\uf6aeBOX 1\\uf6af  but, for the reasons cited above, metazoan \\nmodels are usually more appropriate than yeast for \\nsynthetic lethal screens for human cancer genes.\\nRNAi is a powerful method for silencing genes in \\nworms and fruitflies,  and collections of interfering \\nRNAs  have been created to facilitate high-throughput \\ngenome-wide screens in these organisms86–89 (for an \\nexcellent review, see REF. 90 ). RNAi can be conveniently \\nachieved in wild-type or mutant worms by growing \\nthem on lawns of Escherichia coli  carrying a plasmid \\nthat produces the interfering RNAs of interest, which \\nare then ingested. Alternatively, interfering RNAs can \\nbe delivered to worms by soaking them in a solution \\nthat contains the appropriate molecules. An interfering \\nRNA  that exacerbated the mutant phenotype without \\naffecting wild-type animals would indicate a synthetic \\nlethal, or synthetic sick, interaction. High-throughput \\nscreens have also been conducted to identify interfer-\\ning RNAs that inhibit the proliferation of fruitfly  cells \\ngrown in multiwell plates88. Such screens could easily \\nbe adapted to carry out synthetic lethal screens. In \\nthis scenario, the identification of interfering RNAs \\nthat do not affect wild-type fruitfly  cells but kill fruit-\\nfly cells in which the gene of interest was mutated or \\nsilenced would be desired. If required, silencing could \\nbe accomplished by simultaneously administering two \\ninterfering RNAs (one corresponding to the query gene \\nand one corresponding to the gene of interest).\\nMany cancer-relevant genes are linked to spe-\\ncific types of cancer despite being ubiquitously \\nexpressed and performing functions that are \\nthought to be generic rather than tissue specific. In \\naddition, there are now many examples where dif-\\nferent phenotypes have been observed following \\nheterozygous inactivation of a particular tumour-\\nsuppressor gene in both mice and humans. These \\nobservations indicate that context, with respect to \\ncell-type and species, is important. As a corollary, \\nNATURE REVIEWS | CANCER   VOLUME 5 | SEPTEMBER 2005 | 695REVIEWS', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), Document(id_='dd2eee98-72b8-45cc-adac-243d58cdc626', embedding=None, metadata={'page_label': '696', 'file_name': 'THE CONCEPT OF SYNTHETIC LETHALITY IN THE CONTEXT OF ANTICANCER THERAPY.pdf', 'file_path': 'files/Papers_FullText/THE CONCEPT OF SYNTHETIC LETHALITY IN THE CONTEXT OF ANTICANCER THERAPY.pdf', 'file_type': 'application/pdf', 'file_size': 600362, 'creation_date': '2024-01-10', 'last_modified_date': '2024-01-10', 'last_accessed_date': '2024-01-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, hash='d7b9c98423e69ea245b28a68e942c95db6dd6ba143449366cb44729b57321e3f', text='© 2005 Nature Publi shing Group Library of bar-coded\\ndeletion mutants or\\nshRNA vectors∆ Gene 1\\n∆ Gene 2\\n∆ Gene 3\\n∆ Gene n\\nTimeBar-code abundance\\nCondition A Condition B\\nRecover DNA bar codes\\nby PCR and label redRecover DNA bar codes\\nby PCR and label green\\nHybridize to microarray \\ncontaining probes for \\nDNA bar codesSelected against under condition B\\nEnriched for under condition BRelative fitnessthey indicate that synthetic lethal relationships ulti-\\nmately need to be discovered or validated in relevant \\nhuman cells, and that caution needs to be exercised \\nwhen extrapolating cell-culture results to intact organ-\\nisms. In the past, the use of RNAi in mammalian cells \\nwas problematic because double-stranded RNA elicits \\nan antiviral response on entry into mammalian cells. \\nIn 2001, however, Tuschl and co-workers showed that \\nsiRNAs can be used to silence genes in mammalian cells \\nwithout triggering a nonspecific host response91. Soon \\nthereafter several groups showed that the actions of \\nsiRNAs in cells can be mimicked with short hairpin \\nRNAs (shRNAs) encoded by plasmid or viral vectors92–96. \\nsiRNA libraries and shRNA vector libraries are being \\ncreated, and proof-of-concept experiments indicate \\nthat these libraries can be used to carry out genome-\\nwide phenotypic screens in mammalian cells (including \\nhuman cells)97–100. In theory, these libraries could be used \\nto carry out synthetic lethal screens using isogenic cell-\\nline pairs, scoring for siRNA (or shRNA) species that \\nspecifically kill cells with a cancer-relevant mutation in \\na one well/one siRNA (or shRNA) species format (FIG. 5) . \\nAlternatively, several groups are incorporating DNA ‘bar \\ncodes’ \\uf6aeBOX 1\\uf6af into shRNA vectors, modelled after the use \\nof DNA bar codes in yeast and E. coli  (or have used the \\nshRNA sequence itself as a bar code)97,98. If successful, it \\nshould be possible to infect isogenic cell-line pairs with \\npools of vectors encoding different shRNAs, and then \\nidentify those shRNAs that cause a fitness defect spe-\\ncifically in those cells that harbour the cancer-relevant \\nmutation under investigation.\\nCombination therapy\\nRandom mutations that lead to gene inactivation should \\ntheoretically decrease the genetic buffering capacity of \\nan individual cancer cell. As outlined above, therapies \\npredicated on synthetic lethal relationships are one way \\nto exploit this. At the same time, random mutations and \\ngenome plasticity, viewed at the level of a tumour, mark-\\nedly increase the likelihood that rare therapy-resistant \\nsubclones will emerge. Decades of clinical experience, \\nincluding recent examples of imatinib mesylate resist-\\nance101,102, as well as tumour models incorporating the \\nuse of conditionally expressed oncogenes103, support \\nthis view. A 1-cm3 tumour already contains >109 cells. \\nSo, the likelihood of clinical success will increase with \\nearly diagnosis (to minimize the number of cells in the \\npool from which resistant cells might arise) and the use \\nof effective drug combinations. The use of drug combi-\\nnations to minimize chemotherapeutic resistance is a \\nwell-established pharmaceutical principle. It is based on \\nthe knowledge that the probability of a given cell being \\nsimultaneously resistant to a combination of non-cross-\\nresistant drugs varies as the product of the probabilities of \\nbecoming resistant to each of the individual components. \\nThe choice of which drugs to combine might be based \\non a knowledge of cancer molecular biology (for exam-\\nple, by simultaneously targeting two or more cancer-\\nrelevant mutations), empirical testing (for example, by \\nsystematically testing combinations of active agents for \\nadditive or synergistic effects) or both.Box 1 | DNA bar code screens for genes that alter fitness\\nScreens in yeast\\nIn yeast,  synthetic lethal interactions have been identified by systematically \\ncreating double mutants (by mating or transformation) and studying their fitness, \\neither by robotically spotting them in arrays14,15 or following their behaviour in \\npools with the use of DNA ‘bar codes’106. Deletion mutants have been generated \\nfor almost every gene, with each strain carrying two unique DNA sequences or \\nbar codes — one immediately upstream and one immediately downstream of the \\ngene deletion — that serve as strain identifiers106–109. The upstream and \\ndownstream bar codes are each flanked by invariant sequences that can be used to \\nPCR amplify either all possible upstream bar codes or all possible downstream \\nbar codes. Pools of yeast representing the various deletion mutants can be grown \\nunder different sets of conditions. The abundance, and therefore the ‘fitness’ , of \\nindividual deletion strains can be monitored by recovering the bar codes using \\nPCR and hybridizing them to oligonucleotide arrays where each feature is \\ncomplementary to, and therefore interrogates, a different bar code106–108. Similar \\nassays were reported earlier using Escherichia coli  that had been mutagenized with \\ntransposons bearing unique DNA sequence tags110. This approach could be used \\nto look for yeast deletion mutants that become ‘unfit’ (synthetic sick or synthetic \\nlethal) in the context of a mutation of interest (for example, a DNA-repair enzyme \\nthat is deleted in human cancers).\\nScreens in mammalian cells\\nCells are created where inactivation of a particular gene is linked to the introduction \\nof a unique DNA sequence that serves as a molecular identifier or bar code. \\nInactivation can be accomplished by homologous recombination (as has been done \\nin yeast) or with short hairpin RNA (shRNA) vectors. In the example shown, two bar \\ncodes, one ‘upstream’ and one ‘downstream’ have been used per gene. Pools of bar-\\ncoded cells are grown under two sets of experimental conditions, such as presence or \\nabsence of a chemotherapeutic agent or of a query gene. The bar codes are then \\nrecovered by PCR (using primers that recognize invariant sequences flanking either \\nthe upstream or downstream bar codes), fluorescently labelled (red for one condition \\nand green for the other), and hybridized to microarrays containing probes \\ncomplementary to the individual bar codes. Selection for or against a particular \\ndeletion strain under the test conditions relative to the control conditions gives rise \\nto changes in the red/green ratio (bottom). The abundance of particular clones over \\ntime in response to a particular condition, as determined by normalized fluorescence \\nintensity, can also be used to monitor fitness (right). \\n696 | SEPTEMBER 2005 | VOLUME 5  www.nature.com/reviews/cancerREVIEWS', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), Document(id_='5804c3b6-dc3d-4254-9590-0782df753d89', embedding=None, metadata={'page_label': '697', 'file_name': 'THE CONCEPT OF SYNTHETIC LETHALITY IN THE CONTEXT OF ANTICANCER THERAPY.pdf', 'file_path': 'files/Papers_FullText/THE CONCEPT OF SYNTHETIC LETHALITY IN THE CONTEXT OF ANTICANCER THERAPY.pdf', 'file_type': 'application/pdf', 'file_size': 600362, 'creation_date': '2024-01-10', 'last_modified_date': '2024-01-10', 'last_accessed_date': '2024-01-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, hash='173d37052c0653aabb3468213968feaef369b7dbb99f10448ccc4bf356f4c6a3', text='© 2005 Nature Publi shing Group Implications and future directions\\nOver the decades, the medical therapy of metastatic \\ncancer has, with a few notable exceptions, been a frus-\\ntrating and often futile exercise. This has contributed \\nto the view that each mutation within a cancer cell \\nis another plate of armour that serves as a barrier to \\nsuccessful therapy. However, our empirical knowledge \\nof the susceptibility of cancer cells to drugs in humans \\nstems from an armamentarium that was largely dis-\\ncovered and developed using the same paradigm. \\nMoreover, there is every reason to believe that certain \\ngenetic changes within cancer cells will create liabilities \\nunder the appropriate conditions. There are now tools to systematically search for mutated oncogenes that \\nencode molecules, such as kinases, that can be targeted \\nby drugs, as well as the tools to reveal vulnerabilities \\ncreated by synthetic lethal interactions. Understanding \\nhow the phenotypes created by cancer genotypes (for \\nexample, tumour type and resistance to therapy), as \\nwell as synthetic lethal relationships, are influenced \\nby contextual differences (for example, cell type and \\nspecies) remains a formidable task. Nonetheless, we \\nare clearly poised to move away from empirically dis-\\ncovered cytotoxics and towards new agents that are \\nbased on a knowledge of cancer genetics and a more \\nsophisticated view of gene–gene interactions.\\n1.  Kaelin, W. G. Jr. Choosing anticancer drug targets in the \\npostgenomic era. J. Clin. Invest.  104, 1503–1506 \\n(1999).\\n2.  Reddy, A. & Kaelin, W. G. Jr. Using cancer genetics to \\nguide the selection of anticancer drug targets. Curr. Opin. \\nPharmacol.  2, 366–373 (2002).\\n3.  Kaelin, W. G. Jr. Gleevec: prototype or outlier? Sci. STKE  \\n2004, PE12 (2004).\\nReferences 1–3 provide counter-arguments to \\nnaysayers who suggest that genetically complex \\ncancers will never be successfully treated with \\ndrugs.\\n4.  Hartman, J. T., Garvik, B. & Hartwell, L. Principles for the \\nbuffering of genetic variation. Science  291, 1001–1004 \\n(2001).\\n5.  Guarente, L. Synthetic enhancement in gene interaction: a \\ngenetic tool come of age. Trends Genet . 9, 362–366 \\n(1993).\\n6.  Kamb, A. Mutation load, functional overlap, and synthetic \\nlethality in the evolution and treatment of cancer. J. Theor. \\nBiol. 223, 205–213 (2003).\\nThis paper and reference 58 are thoughtful essays \\non maladaptive genetic changes in cancer cells that \\nmight render them vunerable to pharmacological \\nattack.\\n7.  Friend, S. & Oliff, A. Emerging uses for genomic information \\nin drug discovery. N. Engl. J. Med . 338, 125–126 (1998).\\n8.  Dobzhansky, T. Genetics of natural populations. XIII. \\nRecombination and variability in populations of Drosophila  \\npseudoobscura . Genetics  31, 269–290 (1946).\\n9.  Lucchesi, J. C. Synthetic lethality and semi-lethality among \\nfunctionally related mutants of Drosophila melanogaster . \\nGenetics  59, 37–44 (1968).\\n10.  Sharom, J. R., Bellows, D. S. & Tyers, M. From large \\nnetworks to small molecules. Curr. Opin. Chem. Biol . 8, \\n81–90 (2004).\\nExcellent introduction to systems biology as applied \\nto cancer and cancer pharmacology.\\n11.  Kroll, E. S., Hyland, K. M., Hieter, P . & Li, J. J. Establishing \\ngenetic interactions by a synthetic dosage lethality \\nphenotype. Genetics  143, 95–102 (1996).\\n12.  Measday, V. & Hieter, P . Synthetic dosage lethality. \\nMethods Enzymol . 350, 316–326 (2002).\\n13.  Li, J. J. & Herskowitz, I. Isolation of ORC6, a component of \\nthe yeast origin recognition complex by a one-hybrid \\nsystem. Science  262, 1870–1874 (1993).\\n14.  Tong, A. H. et al. Systematic genetic analysis with ordered \\narrays of yeast deletion mutants. Science  294, 2364–2368 \\n(2001).\\n15.  Tong, A. H. et al. Global mapping of the yeast genetic \\ninteraction network. Science  303, 808–813 (2004).\\nReferences 14 and 15 provide a glimpse into the \\ncomplexity of synthetic lethal networks in yeast.\\n16.  Hartwell, L., Szankasi, P ., Roberts, C., Murray, A. & \\nFriend, S. Integrating genetic approaches into the \\ndiscovery of anticancer drugs. Science  278, 1064–1068 \\n(1997).\\nThis seminal paper argues that synthetic lethal \\ninteractions be exploited to arrive at safer, more \\nefficacious cancer drugs.\\n17.  Sellers, W. R. & Kaelin, W. G. Jr. Role of the retinoblastoma \\nprotein in the pathogenesis of human cancer. J. Clin. \\nOncol. 15, 3301–3312 (1997).\\n18.  Nip, J. et al. E2F-1 cooperates with topoisomerase II \\ninhibition and DNA damage to selectively augment \\np53-independent apoptosis. Mol. Cell. Biol.  17, \\n1049–1056 (1997).19.  Almasan, A. et al. Deficiency of retinoblastoma protein \\nleads to inappropriate S-phase entry, activation of \\nE2F-responsive genes, and apoptosis. Proc. Natl Acad. \\nSci. USA  92, 5436–5440 (1995).\\n20.  Banerjee, D. et al. Role of E2F-1 in chemosensitivity. \\nCancer Res.  58, 4292–4296 (1998).\\n21.  Dolma, S., Lessnick, S. L., Hahn, W. C. & Stockwell, B. R. \\nIdentification of genotype-selective antitumor agents using \\nsynthetic lethal chemical screening in engineered human \\ntumor cells. Cancer Cell  3, 285–296 (2003).\\n22.  Evan, G. I. & Vousden, K. H. Proliferation, cell cycle and \\napoptosis in cancer. Nature  411, 342–348 (2001).\\n23.  Zaika, A., Irwin, M., Sansome, C. & Moll, U. M. Oncogenes \\ninduce and activate endogenous p73 protein. J. Biol. \\nChem.  276, 11310–11316 (2001).\\n24.  Meng, R., Phillips, P . & El-Deiry, W. p53-independent \\nincrease in E2F-1 expression enhances the cytoxic effects of \\netoposide and of adriamycin. Intl J. Oncol. 14, 5–14 (1999).\\n25.  Irwin, M. S. et al. Chemosensitivity linked to p73 function. \\nCancer Cell  3, 403–410 (2003).\\n26.  Rutherford, S. L. & Lindquist, S. HSP90 as a capacitor for \\nmorphological evolution. Nature  396, 336–342 (1998).\\n27.  Isaacs, J. S., Xu, W. & Neckers, L. Heat shock protein 90 \\nas a molecular target for cancer therapeutics. Cancer Cell  \\n3, 213–217 (2003).\\n28.  Workman, P . Altered states: selectively drugging the \\nHSP90 cancer chaperone. Trends Mol. Med.  10, 47–51 \\n(2004).\\n29.  Neckers, L. & Neckers, K. Heat-shock protein 90 inhibitors \\nas novel cancer chemotherapeutics – an update. Expert \\nOpin. Emerg. Drugs  10, 137–149 (2005).\\n30.  Goldberg, A. L. Protein degradation and protection against \\nmisfolded or damaged proteins. Nature  426, 895–899 \\n(2003).\\n31.  Rajkumar, S. V., Richardson, P . G., Hideshima, T. & \\nAnderson, K. C. Proteasome inhibition as a novel \\ntherapeutic target in human cancer. J. Clin. Oncol.  23, \\n630–639 (2005).\\n32.  Krek, W., Xu, G., & Livingston, D. M. Cyclin A-kinase \\nregulation of E2F1 DNA binding function underlies \\nsuppression of an S phase checkpoint. Cell 83, \\n1149–1158 (1995).\\n33.  Dynlacht, B. D., Flores, O., Lees, J. A. & Harlow, E. \\nDifferential regulation of E2F transactivation by cyclin/CDK \\ncomplexes. Genes Dev . 8, 1772–1786 (1994).\\n34.  Krek, W. et al. Negative regulation of the growth-promoting \\ntranscription factor E2F-1 by a stably bound \\ncyclin A-d ependent protein kinase. Cell 78, 1–20 (1994).\\n35.  Xu, M., Sheppard, K. A., Peng, C-Y., Yee, A. S. & \\nPiwnica-Worms, H. Cyclin A/CDK2 binds directly to E2F1 \\nand inhibits the DNA-binding activity of E2F1/DP1 by \\nphosphorylation. Mol. Cell. Biol. 14, 8420–8431 (1994).\\n36.  Parr, M. J. et al. Tumor-selective transgene expression \\nin vivo  mediated by an E2F-responsive adenoviral vector. \\nNature Med.  3, 1145–1149 (1997).\\n37.  Chen, Y. et al. Selective killing of transformed cells by \\ncyclin/cyclin-dependent kinase 2 antagonists. Proc. Natl \\nAcad. Sci. USA  96, 4325–4329 (1999).\\n38.  Chen, W., Lee, J., Cho, S. Y. & Fine, H. A. Proteasome-\\nmediated destruction of the cyclin A/cyclin-dependent \\nkinase 2 complex suppresses tumor cell growth in vitro  \\nand in vivo . Cancer Res.  64, 3949–3957 (2004).\\n39.  Mendoza, N. et al. Selective cyclin-dependent kinase 2/\\ncyclin A antagonists that differ from ATP site inhibitors \\nblock tumor growth. Cancer Res.  63, 1020–1024 (2003).\\n40.  Tetsu, O. & McCormick, F. Proliferation of cancer cells \\ndespite CDK2 inhibition. Cancer Cell  3, 233–245 (2003).41.  Berthet, C., Aleem, E., Coppola, V., Tessarollo, L. & Kaldis, P . \\nCDK2 knockout mice are viable. Curr. Biol.  13, 1775–1785 \\n(2003).\\n42.  Ortega, S. et al. Cyclin-dependent kinase 2 is essential for \\nmeiosis but not for mitotic cell division in mice. Nature \\nGenet.  35, 25–31 (2003).\\n43.  Schlegel, R. & Pardee, A. B. Caffeine-induced uncoupling \\nof mitosis from the completion of DNA replication in \\nmammalian cells. Science  232, 1264–1266 (1986).\\n44.  Nishimoto, T., Ishida, R., Ajiro, K., Yamamoto, S. & \\nTakahashi, T. The synthesis of protein(s) for chromosome \\ncondensation may be regulated by a post-transcriptional \\nmechanism. J. Cell. Physiol.  109, 299–308 (1981).\\n45.  Hall-Jackson, C. A., Cross, D. A., Morrice, N. & Smythe, C. \\nATR is a caffeine-sensitive, DNA-activated protein kinase \\nwith a substrate specificity distinct from DNA-PK. \\nOncogene  18, 6707–6713 (1999).\\n46.  Sarkaria, J. N. et al. Inhibition of ATM and ATR kinase \\nactivities by the radiosensitizing agent, caffeine. Cancer \\nRes. 59, 4375–4382 (1999).\\n47.  Nghiem, P ., Park, P ., Kim, Y., Vaziri, C. & Schreiber, S. ATR \\ninhibition selectively sensitizes G1 checkpoint-deficient \\ncells to lethal premature chromatin condensation. Proc. \\nNatl Acad. Sci. USA  98, 9092–9097 (2001).\\n48.  Sordella, R., Bell, D. W., Haber, D. A. & Settleman, J. \\nGefitinib-sensitizing EGFR mutations in lung cancer \\nactivate anti-apoptotic pathways. Science  305, 1163–\\n1167 (2004).\\n49.  Lynch, T. J. et al. Activating mutations in the epidermal \\ngrowth factor receptor underlying responsiveness of non-\\nsmall-cell lung cancer to gefitinib. N. Engl. J. Med.  350, \\n2129–2139 (2004).\\n50.  Paez, J. G. et al. EGFR mutations in lung cancer: \\ncorrelation with clinical response to gefitinib therapy. \\nScience  304, 1497–1500 (2004).\\n51.  Pao, W. et al. EGF receptor gene mutations are common \\nin lung cancers from ‘never smokers’ and are associated \\nwith sensitivity of tumors to gefitinib and erlotinib. Proc. \\nNatl Acad. Sci. USA  101, 13306–13311 (2004).\\n52.  Weinstein, I. B. et al. Disorders in cell circuitry associated \\nwith multistage carcinogenesis: exploitable targets for \\ncancer prevention and therapy. Clin. Cancer Res.  3, 2696–\\n2702 (1997).\\n53.  Weinstein, I. B. Disorders in cell circuitry during multistage \\ncarcinogenesis: the role of homeostasis. Carcinogenesis  \\n21, 857–864 (2000).\\n54.  Weinstein, I. B. Cancer. Addiction to oncogenes — \\nthe Achilles heal of cancer. Science  297, 63–64 (2002).\\nReferences 52–54, introduced the term ‘oncogene \\naddiction’.\\n55.  Adams, P . & Kaelin, W. J. Jr. The cellular effects of E2F \\noverexpression. Curr. Top. Microbiol. Immunol.  208, 79–93 \\n(1996).\\n56.  Sherr, C. The Pezcoller lecture: cancer cell cycles revisited. \\nCancer Res.  60, 3689–3695 (2000).\\n57.  Mills, G., Lu, Y. & Kohn, E. Linking molecular therapeutics \\nto molecular diagnostics: inhibition of the FRAP/RAFT/TOR \\ncomponent of the PI3K pathway preferentially blocks \\nPTEN mutant cells in vitro  and in vivo . Proc. Natl Acad. \\nSci. USA  98, 10031–10033 (2001).\\n58.  Kamb, A. Consequences of nonadaptive alterations in \\ncancer. Mol. Biol. Cell  14, 2201–2205 (2003).\\n59.  Neshat, M. et al. Enhanced sensitivity of PTEN-deficient \\ntumors to inhibition of FRAP/mTOR. Proc. Natl Acad. Sci. \\nUSA 98, 10314–10319 (2001).\\n60.  Frei, E. D. Gene deletion: a new target for cancer \\nchemotherapy. Lancet  342, 662–664 (1993).\\nNATURE REVIEWS | CANCER   VOLUME 5 | SEPTEMBER 2005 | 697REVIEWS', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), Document(id_='ce4b986e-6629-4677-a928-a3fda5db35d9', embedding=None, metadata={'page_label': '698', 'file_name': 'THE CONCEPT OF SYNTHETIC LETHALITY IN THE CONTEXT OF ANTICANCER THERAPY.pdf', 'file_path': 'files/Papers_FullText/THE CONCEPT OF SYNTHETIC LETHALITY IN THE CONTEXT OF ANTICANCER THERAPY.pdf', 'file_type': 'application/pdf', 'file_size': 600362, 'creation_date': '2024-01-10', 'last_modified_date': '2024-01-10', 'last_accessed_date': '2024-01-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, hash='88c8c37e215886ca3ae6701e87a182d129572949b1f6ca916792b6bbccc83936', text='© 2005 Nature Publi shing Group 61.  Cairns, P . et al. Frequency of homozygous deletion at \\np16/CDKN2 in primary human tumours. Nature Genet.  11, \\n210–212 (1995).\\n62.  Li, W. et al. Status of methylthioadenosine phosphorylase \\nand its impact on cellular response to L-alanosine and \\nmethylmercaptopurine riboside in human soft tissue \\nsarcoma cells. Oncol. Res.  14, 373–379 (2004).\\n63.  Wong, S. L. et al. Combining biological networks to predict \\ngenetic interactions. Proc. Natl Acad. Sci. USA  101, \\n15682–15687 (2004).\\n64.  Simon, J. A. et al. Differential toxicities of anticancer agents \\namong DNA repair and checkpoint mutants of \\nSaccharomyces cerevisiae . Cancer Res.  60, 328–333 \\n(2000).\\n65.  Stockwell, B., Haggarty, S. & Schreiber, S. High-\\nthroughput screening of small molecules in miniaturized \\nmammalian cell-based assays involving post-translational \\nmodifications. Chem. Biol.  6, 71–83 (1999).\\nReferences 64 and 65 are two early examples of \\nusing isogenic cell lines to isolate compounds that \\nkill cells in a genotype-specific manner.\\n66.  Torrance, C., Agrawal, V., Vogelstein, B. & Kinzler, K. Use \\nof isogenic human cancer cells for high-throughput \\nscreening and drug discovery. Nature Biotechnol.  19, \\n940–945 (2001).\\n67.  Bender, A. & Pringle, J. R. Use of a screen for synthetic \\nlethal and multicopy suppressee mutants to identify two \\nnew genes involved in morphogenesis in Saccharomyces \\ncerevisiae . Mol. Cell. Biol.  11, 1295–1305 (1991).\\n68.  Simons, A., Dafni, N., Dotan, I., Oron, Y. & Canaani, D. \\nEstablishment of a chemical synthetic lethality screen in \\ncultured human cells. Genome Res.  11, 266–273 \\n(2001).\\n69.  Simons, A., Dafni, N., Dotan, I., Oron, Y. & Canaani, D. \\nGenetic synthetic lethality screen at the single gene level \\nin cultured human cells. Nucleic Acids Res.  29, E100 \\n(2001).\\n70.  Fantin, V. R. & Leder, P . F16, a mitochondriotoxic \\ncompound, triggers apoptosis or necrosis depending on \\nthe genetic background of the target carcinoma cell. \\nCancer Res.  64, 329–336 (2004).\\n71.  Fantin, V. R., Berardi, M. J., Scorrano, L., Korsmeyer, S. J. \\n& Leder, P . A novel mitochondriotoxic small molecule that \\nselectively inhibits tumor cell growth. Cancer Cell  2, 29–42 \\n(2002).\\n72.  Wang, Y. et al. Synthetic lethal targeting of MYC by \\nactivation of the DR5 death receptor pathway. Cancer Cell  \\n5, 501–512 (2004).\\n73.  Haggarty, S. J., Clemons, P . A. & Schreiber, S. L. \\nChemical genomic profiling of biological networks using \\ngraph theory and combinations of small molecule \\nperturbations. J. Am. Chem. Soc.  125, 10543–10545 \\n(2003).\\n74.  Borisy, A. A. et al. Systematic discovery of multicomponent \\ntherapeutics. Proc. Natl Acad. Sci. USA  100, 7977–7982 \\n(2003).\\n75.  Lipinski, C. A., Lombardo, F., Dominy, B. W. & Feeney, P . J. \\nExperimental and computational approaches to estimate \\nsolubility and permeability in drug discovery and \\ndevelopment settings. Adv. Drug Deliv. Rev.  46, 3–26 \\n(2001).\\n76.  Lipinski, C. A. Drug-like properties and the causes of poor \\nsolubility and poor permeability. J. Pharmacol. Toxicol. \\nMethods  44, 235–249 (2000).\\n77.  Knockaert, M. et al. Intracellular targets of cyclin-\\ndependent kinase inhibitors: identification by affinity \\nchromatography using immobilised inhibitors. Chem. Biol.  \\n7, 411–422 (2000).\\n78.  Hultsch, T., Albers, M. W., Schreiber, S. L. & Hohman, R. J. \\nImmunophilin ligands demonstrate common features of signal transduction leading to exocytosis or transcription. \\nProc. Natl Acad. Sci. USA  88, 6229–6233 (1991).\\n79.  Baetz, K. et al. Yeast genome-wide drug-induced \\nhaploinsufficiency screen to determine drug mode of \\naction. Proc. Natl Acad. Sci. USA  101, 4525–4230 (2004).\\n80.  Giaever, G. et al. Genomic profiling of drug sensitivities via \\ninduced haploinsufficiency. Nature Genet.  21, 278–283 \\n(1999).\\n81.  Marton, M. et al. Drug target validation and identification of \\nsecondary drug target effects using DNA microarrays. \\nNature Med.  4, 1293–1301 (1998).\\n82.  Lu, X. & Horvitz, H. R. lin-35  and lin-53 , two genes that \\nantagonize a C. elegans  Ras pathway, encode proteins \\nsimilar to RB and its binding protein RBAp48. Cell 95, 981–\\n991 (1998).\\n83.  Fay, D. S., Large, E., Han, M. & Darland, M. lin-35 /Rb and \\nubc-18 , an E2 ubiquitin-conjugating enzyme, function \\nredundantly to control pharyngeal morphogenesis in \\nC. elegans . Development  130, 3319–3330 (2003).\\n84.  Fay, D. S., Keenan, S. & Han, M. fzr-1 and lin-35/Rb \\nfunction redundantly to control cell proliferation in \\nC. elegans  as revealed by a nonbiased synthetic screen. \\nGenes Dev.  16, 503–517 (2002).\\n85.  Edgar, K. A. et al. Synthetic lethality of retinoblastoma mutant \\ncells in the Drosophila  eye by mutation of a novel peptidyl \\nprolyl isomerase gene. Genetics  170, 161–171 (2005).\\n86.  Kamath, R. S. et al. Systematic functional analysis of the \\nCaenorhabditis elegans  genome using RNAi. Nature  421, \\n231–237 (2003).\\n87.  Ashrafi, K. et al. Genome-wide RNAi analysis of \\nCaenorhabditis elegans  fat regulatory genes. Nature  421, \\n268–272 (2003).\\n88.  Cherry, S. et al. Genome-wide RNAi screen reveals a \\nspecific sensitivity of IRES-containing RNA viruses to host \\ntranslation inhibition. Genes Dev.  19, 445–452 (2005).\\n89.  Rual, J. F. et al. Toward improving Caenorhabditis elegans  \\nphenome mapping with an ORFeome-based RNAi library. \\nGenome Res.  14, 2162–2168 (2004).\\n90.  Willingham, A. T., Deveraux, Q. L., Hampton, G. M. & \\nAza-Blanc, P . RNAi and HTS: exploring cancer by systematic \\nloss-of-function. Oncogene  23, 8392–8400 (2004).\\n91.  Elbashir, S. et al. Duplexes of 21-nucleotide RNAs mediate \\nRNA interference in cultured mammalian cells. Nature  411, \\n494–498 (2001).\\n92.  Brummelkamp, T. R., Bernards, R. & Agami, R. Stable \\nsuppression of tumorigenicity by virus-mediated RNA \\ninterference. Cancer Cell  2, 243–247 (2002).\\n93.  Brummelkamp, T. R., Bernards, R. & Agami, R. A system \\nfor stable expression of short interfering RNAs in \\nmammalian cells. Science  296, 550–553 (2002).\\n94.  Lee, N. S. et al. Expression of small interfering RNAs \\ntargeted against HIV-1 rev transcripts in human cells. \\nNature Biotechnol.  20, 500–505 (2002).\\n95.  Paddison, P . J., Caudy, A. A., Bernstein, E., Hannon, G. J. \\n& Conklin, D. S. Short hairpin RNAs (shRNAs) induce \\nsequence-specific silencing in mammalian cells. Genes \\nDev. 16, 948–958 (2002).\\n96.  Sui, G. et al. A DNA vector-based RNAi technology to \\nsuppress gene expression in mammalian cells. Proc. Natl \\nAcad. Sci. USA  99, 5515–5520 (2002).\\n97.  Berns, K. et al. A large-scale RNAi screen in human cells \\nidentifies new components of the p53 pathway. Nature  \\n428, 431–437 (2004).\\n98.  Paddison, P . J. et al. A resource for large-scale RNA-\\ninterference-based screens in mammals. Nature  428, \\n427–431 (2004).\\nReferences 97 and 98 suggest that it should \\neventually be possible to carry out synthetic lethal \\nscreens in isogenic human cell-line pairs using bar-\\ncoded shRNA libraries.99.  Shirane, D. et al. Enzymatic production of RNAi libraries \\nfrom cDNAs. Nature Genet.  36, 190–196 (2004).\\n100.  Aza-Blanc, P . et al. Identification of modulators of TRAIL-\\ninduced apoptosis via RNAi-based phenotypic screening. \\nMol. Cell  12, 627–637 (2003).\\n101.  Gorre, M.  et al. Clinical resistance to STI-571 cancer \\ntherapy caused by BCR–ABL gene mutation or \\namplification. Science  293, 876–880 (2001).\\n102.  Shah, N. P . et al. L. Multiple BCR–ABL kinase domain \\nmutations confer polyclonal resistance to the tyrosine \\nkinase inhibitor imatinib (STI571) in chronic phase and \\nblast crisis chronic myeloid leukemia. Cancer Cell  2, \\n117–125 (2002).\\n103.  Jonkers, J. & Berns, A. Oncogene addiction: sometimes a \\ntemporary slavery. Cancer Cell  6, 535–538 (2004).\\n104.  Bailey, S. N., Sabatini, D. M. & Stockwell, B. R. \\nMicroarrays of small molecules embedded in \\nbiodegradable polymers for use in mammalian cell-based \\nscreens. Proc. Natl Acad. Sci. USA  101, 16144–16149 \\n(2004).\\n105.  Wheeler, D. B.  et al. RNAi living-cell microarrays for loss-\\nof-function screens in Drosophila melanogaster  cells. \\nNature Methods  1, 127–132 (2004).\\n106.  Ooi, S. L., Shoemaker, D. D. & Boeke, J. D. DNA helicase \\ngene interaction network defined using synthetic lethality \\nanalyzed by microarray. Nature Genet.  35, 277–286 \\n(2003).\\nDescribes the use of DNA bar codes coupled with \\noligonucleotide microarrays to conduct synthetic \\nlethal assays in yeast.\\n107.  Shoemaker, D. D., Lashkari, D. A., Morris, D., Mittmann, M. \\n& Davis, R. W. Quantitative phenotypic analysis of yeast \\ndeletion mutants using a highly parallel molecular bar-\\ncoding strategy. Nature Genet . 14, 450–456 (1996).\\n108.  Winzeler, E. A. et al. Functional characterization of the \\nS. cerevisiae  genome by gene deletion and parallel \\nanalysis. Science  285, 901–906 (1999).\\n109.  Eason, R. G. et al. Characterization of synthetic DNA bar \\ncodes in Saccharomyces cerevisiae  gene-deletion strains. \\nProc. Natl Acad. Sci. USA  101, 11046–11051 (2004).\\n110.  Hensel, M. et al. Simultaneous identification of bacterial \\nvirulence genes by negative selection. Science  269, \\n400–403 (1995).\\nAcknowledgements\\nI would like to thank S. Elledge, A. Reddy, M. Tyers, P . Silver and \\nM. Vidal for their critical reading of this manuscript and/or helpful \\ncomments. I apologize to colleagues whose work was not cited \\ndue to space limitations or my ignorance. Dedicated to the mem-\\nory of Nancy P . Kaelin.\\nCompeting interests statement\\nThe authors declare no competing financial interests.\\n Online links\\nDATABASES\\nThe following terms in this article are linked online to:\\nEntrez Gene:  http://www.ncbi.nlm.nih.gov/entrez/query.\\nfcgi?db=gene\\nABL | BCR | CDK2 | E2F1 | EGFR | ERBB2 | HSP90 | KRAS  | \\nMYC | p53 | pRB | PTEN | \\nNational Cancer Institute:  http://www.cancer.gov\\nchronic myelogenous leukaemia \\nFURTHER INFORMATION\\nAuthor’s homepage:  http://www.hhmi.org/research/\\ninvestigators/kaelin.html\\nAccess to this interactive links box is free online.\\n698 | SEPTEMBER 2005 | VOLUME 5  www.nature.com/reviews/cancerREVIEWS', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n')]\n"
     ]
    }
   ],
   "source": [
    "# RAG with our files\n",
    "from llama_index import VectorStoreIndex, SimpleDirectoryReader\n",
    "\n",
    "documents = SimpleDirectoryReader(\"files/Papers_FullText/\").load_data()\n",
    "database = SimpleDirectoryReader(\"files/db/\").load_data()\n",
    "index = VectorStoreIndex.from_documents(documents)\n",
    "query_engine = index.as_query_engine()\n",
    "response = query_engine.query(\"How many oncogenes and what impacts them?\")\n",
    "print(response)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
