{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Table: ccle_cn_gistic in files/db/CCLEGisticCNDB.db\n",
      "Schema:\n",
      "(0, 'Hugo_Symbol', 'TEXT', 0, None, 0)\n",
      "(1, 'CCLE_Name', 'TEXT', 0, None, 0)\n",
      "(2, 'gistic_cn', 'REAL', 0, None, 0)\n",
      "(3, 'DepMap_ID', 'TEXT', 0, None, 0)\n",
      "('A1BG', 'DMS53_LUNG', 0.0, 'ACH-000698')\n",
      "('A1BG', 'SW1116_LARGE_INTESTINE', 0.0, 'ACH-000489')\n",
      "('A1BG', 'NCIH1694_LUNG', 0.0, 'ACH-000431')\n",
      "('A1BG', 'P3HR1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE', 0.0, 'ACH-000707')\n",
      "('A1BG', 'HUT78_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE', 0.0, 'ACH-000509')\n",
      "MetaData()\n",
      "\n",
      "Table: ccle_mutation in files/db/CCLEMutDB.db\n",
      "Schema:\n",
      "(0, 'Hugo_Symbol', 'TEXT', 0, None, 0)\n",
      "(1, 'Entrez_Gene_Id', 'INTEGER', 0, None, 0)\n",
      "(2, 'NCBI_Build', 'INTEGER', 0, None, 0)\n",
      "(3, 'Chromosome', 'TEXT', 0, None, 0)\n",
      "(4, 'Start_position', 'INTEGER', 0, None, 0)\n",
      "(5, 'End_position', 'INTEGER', 0, None, 0)\n",
      "(6, 'Strand', 'TEXT', 0, None, 0)\n",
      "(7, 'Variant_Classification', 'TEXT', 0, None, 0)\n",
      "(8, 'Variant_Type', 'TEXT', 0, None, 0)\n",
      "(9, 'Reference_Allele', 'TEXT', 0, None, 0)\n",
      "(10, 'Alternate_Allele', 'TEXT', 0, None, 0)\n",
      "(11, 'dbSNP_RS', 'TEXT', 0, None, 0)\n",
      "(12, 'dbSNP_Val_Status', 'TEXT', 0, None, 0)\n",
      "(13, 'Genome_Change', 'TEXT', 0, None, 0)\n",
      "(14, 'Annotation_Transcript', 'TEXT', 0, None, 0)\n",
      "(15, 'DepMap_ID', 'TEXT', 0, None, 0)\n",
      "(16, 'cDNA_Change', 'TEXT', 0, None, 0)\n",
      "(17, 'Codon_Change', 'TEXT', 0, None, 0)\n",
      "(18, 'Protein_Change', 'TEXT', 0, None, 0)\n",
      "(19, 'isDeleterious', 'TEXT', 0, None, 0)\n",
      "(20, 'isTCGAhotspot', 'TEXT', 0, None, 0)\n",
      "(21, 'TCGAhsCnt', 'REAL', 0, None, 0)\n",
      "(22, 'isCOSMIChotspot', 'TEXT', 0, None, 0)\n",
      "(23, 'COSMIChsCnt', 'REAL', 0, None, 0)\n",
      "(24, 'ExAC_AF', 'REAL', 0, None, 0)\n",
      "(25, 'Variant_annotation', 'TEXT', 0, None, 0)\n",
      "(26, 'CGA_WES_AC', 'TEXT', 0, None, 0)\n",
      "(27, 'HC_AC', 'TEXT', 0, None, 0)\n",
      "(28, 'RD_AC', 'TEXT', 0, None, 0)\n",
      "(29, 'RNAseq_AC', 'TEXT', 0, None, 0)\n",
      "(30, 'SangerWES_AC', 'TEXT', 0, None, 0)\n",
      "(31, 'WGS_AC', 'TEXT', 0, None, 0)\n",
      "('VPS13D', 55187, 37, '1', 12359347, 12359347, '+', 'Nonsense_Mutation', 'SNP', 'C', 'A', '', '', 'g.chr1:12359347C>A', 'ENST00000358136.3', 'ACH-000001', 'c.6122C>A', 'c.(6121-6123)tCa>tAa', 'p.S2041*', 'True', 'False', None, 'False', 0.0, None, 'damaging', '34:213', '', '', '', '34:221', '')\n",
      "('AADACL4', 343066, 37, '1', 12726308, 12726322, '+', 'In_Frame_Del', 'DEL', 'CTGGCGTGACGCCAT', '-', 'rs58218425|rs139261871|rs369427733|rs560787141', 'byFrequency', 'g.chr1:12726308_12726322delCTGGCGTGACGCCAT', 'ENST00000376221.1', 'ACH-000001', 'c.786_800delCTGGCGTGACGCCAT', 'c.(784-801)tcctggcgtgacgccatc>tcc', 'p.WRDAI263del', 'False', 'False', None, 'False', 3.0, None, 'other non-conserving', '57:141', '', '', '', '9:0', '28:32')\n",
      "('IFNLR1', 163702, 37, '1', 24484172, 24484172, '+', 'Silent', 'SNP', 'G', 'A', '', '', 'g.chr1:24484172G>A', 'ENST00000327535.1', 'ACH-000001', 'c.1011C>T', 'c.(1009-1011)ggC>ggT', 'p.G337G', 'False', 'False', None, 'False', 0.0, None, 'silent', '118:0', '', '', '10:0', '118:0', '18:0')\n",
      "('TMEM57', 55219, 37, '1', 25785018, 25785019, '+', 'Frame_Shift_Ins', 'INS', '-', 'A', '', '', 'g.chr1:25785018_25785019insA', 'ENST00000374343.4', 'ACH-000001', 'c.789_790insA', 'c.(790-792)aaafs', 'p.K264fs', 'True', 'False', 0.0, 'False', 0.0, None, 'damaging', '', '', '', '6:28', '', '')\n",
      "('ZSCAN20', 7579, 37, '1', 33954141, 33954141, '+', 'Missense_Mutation', 'SNP', 'T', 'G', '', '', 'g.chr1:33954141T>G', 'ENST00000361328.3', 'ACH-000001', 'c.494T>G', 'c.(493-495)gTg>gGg', 'p.V165G', 'False', 'False', None, 'False', 0.0, None, 'other non-conserving', '28:62', '', '', '', '27:61', '')\n",
      "MetaData()\n",
      "\n",
      "Table: ccle_variants in files/db/CCLEVarDB.db\n",
      "Schema:\n",
      "(0, 'DepMap_ID', 'TEXT', 0, None, 0)\n",
      "(1, 'Chrom', 'TEXT', 0, None, 0)\n",
      "(2, 'Pos', 'INTEGER', 0, None, 0)\n",
      "(3, 'Ref', 'TEXT', 0, None, 0)\n",
      "(4, 'Alt', 'TEXT', 0, None, 0)\n",
      "(5, 'AF', 'REAL', 0, None, 0)\n",
      "(6, 'VariantType', 'TEXT', 0, None, 0)\n",
      "(7, 'VariantInfo', 'TEXT', 0, None, 0)\n",
      "(8, 'DNAChange', 'TEXT', 0, None, 0)\n",
      "(9, 'ProteinChange', 'TEXT', 0, None, 0)\n",
      "(10, 'gene_name', 'TEXT', 0, None, 0)\n",
      "(11, 'Transcript', 'TEXT', 0, None, 0)\n",
      "(12, 'CCLEDeleterious', 'TEXT', 0, None, 0)\n",
      "(13, 'CosmicHotspot', 'TEXT', 0, None, 0)\n",
      "(14, 'AssociatedWith', 'TEXT', 0, None, 0)\n",
      "(15, 'LoF', 'TEXT', 0, None, 0)\n",
      "(16, 'Driver', 'TEXT', 0, None, 0)\n",
      "(17, 'LikelyDriver', 'TEXT', 0, None, 0)\n",
      "(18, 'LikelyGoF', 'TEXT', 0, None, 0)\n",
      "(19, 'LikelyLoF', 'TEXT', 0, None, 0)\n",
      "(20, 'VariantAnnotation', 'TEXT', 0, None, 0)\n",
      "(21, 'copy_number', 'TEXT', 0, None, 0)\n",
      "('ACH-000839', 'chr1', 1242864, 'GC', 'CT', 0.31, 'DNP', 'MISSENSE', 'c.780_781GC>AG', 'p.L261V', 'C1QTNF12', 'ENST00000330388.2', 'False', 'False', '', 'False', 'False', 'False', 'False', 'False', 'Other', 'Diploid')\n",
      "('ACH-000839', 'chr1', 10647969, 'A', 'G', 0.4, 'SNP', 'MISSENSE', 'c.3329T>C', 'p.V1110A', 'CASZ1', 'ENST00000377022.8', 'False', 'False', '', 'False', 'False', 'False', 'False', 'False', 'Other', 'Diploid')\n",
      "('ACH-000839', 'chr1', 10648097, 'T', 'G', 0.349, 'SNP', 'SILENT', 'c.3201A>C', 'p.T1067T', 'CASZ1', 'ENST00000377022.8', 'False', 'False', '', 'False', 'False', 'False', 'False', 'False', 'Silent', 'Diploid')\n",
      "('ACH-000839', 'chr1', 13198424, 'G', 'A', 0.833, 'SNP', 'MISSENSE', 'c.770C>T', 'p.T257I', 'PRAMEF13', 'ENST00000625019.3', 'False', 'False', '', 'False', 'False', 'False', 'False', 'False', 'Other', 'Diploid')\n",
      "('ACH-000839', 'chr1', 13225068, 'A', 'G', 0.396, 'SNP', 'MISSENSE', 'c.653T>C', 'p.L218P', 'PRAMEF18', 'ENST00000624297.2', 'False', 'False', '', 'False', 'False', 'False', 'False', 'False', 'Other', 'Diploid')\n",
      "MetaData()\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "import openai\n",
    "from IPython.display import Markdown, display\n",
    "from llama_index import SQLDatabase, ServiceContext\n",
    "from llama_index.llms import OpenAI\n",
    "import sqlite3\n",
    "from sqlalchemy import (\n",
    "    create_engine,\n",
    "    MetaData,\n",
    "    Table,\n",
    "    Column,\n",
    "    String,\n",
    "    Integer,\n",
    "    select,\n",
    ")\n",
    "def inspect_database(db_path):\n",
    "    # Connect to the database\n",
    "    conn = sqlite3.connect(db_path)\n",
    "    cursor = conn.cursor()\n",
    "\n",
    "    # Get all tables\n",
    "    cursor.execute(\"SELECT name FROM sqlite_master WHERE type='table';\")\n",
    "    tables = cursor.fetchall()\n",
    "\n",
    "    for table in tables:\n",
    "        table_name = table[0]\n",
    "        print(f\"\\nTable: {table_name} in {db_path}\")\n",
    "\n",
    "        # Get table schema\n",
    "        print(\"Schema:\")\n",
    "        cursor.execute(f\"PRAGMA table_info({table_name});\")\n",
    "        schema = cursor.fetchall()\n",
    "        for column in schema:\n",
    "            print(column)\n",
    "\n",
    "        # Get the first five rows\n",
    "        # print(\"\\nFirst 5 rows:\")\n",
    "        cursor.execute(f\"SELECT * FROM {table_name} LIMIT 5;\")\n",
    "        rows = cursor.fetchall()\n",
    "        for row in rows:\n",
    "            print(row)\n",
    "\n",
    "    # Close the cursor and connection\n",
    "    cursor.close()\n",
    "    conn.close()\n",
    "\n",
    "# Paths to your database files\n",
    "db_files = ['files/db/CCLEGisticCNDB.db', 'files/db/CCLEMutDB.db', 'files/db/CCLEVarDB.db']\n",
    "\n",
    "# Iterate over the files and inspect tables\n",
    "for db_file in db_files:\n",
    "    inspect_database(db_file)\n",
    "    engine = create_engine('sqlite:///' + db_file)\n",
    "    metadata_obj = MetaData()\n",
    "    print(metadata_obj)\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Table: ccle_cn_gistic in files/db/CCLEGisticCNDB.db\n",
      "Schema:\n",
      "(0, 'Hugo_Symbol', 'TEXT', 0, None, 0)\n",
      "(1, 'CCLE_Name', 'TEXT', 0, None, 0)\n",
      "(2, 'gistic_cn', 'REAL', 0, None, 0)\n",
      "(3, 'DepMap_ID', 'TEXT', 0, None, 0)\n",
      "('A1BG', 'DMS53_LUNG', 0.0, 'ACH-000698')\n",
      "('A1BG', 'SW1116_LARGE_INTESTINE', 0.0, 'ACH-000489')\n",
      "('A1BG', 'NCIH1694_LUNG', 0.0, 'ACH-000431')\n",
      "('A1BG', 'P3HR1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE', 0.0, 'ACH-000707')\n",
      "('A1BG', 'HUT78_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE', 0.0, 'ACH-000509')\n",
      "\n",
      "Table: ccle_mutation in files/db/CCLEMutDB.db\n",
      "Schema:\n",
      "(0, 'Hugo_Symbol', 'TEXT', 0, None, 0)\n",
      "(1, 'Entrez_Gene_Id', 'INTEGER', 0, None, 0)\n",
      "(2, 'NCBI_Build', 'INTEGER', 0, None, 0)\n",
      "(3, 'Chromosome', 'TEXT', 0, None, 0)\n",
      "(4, 'Start_position', 'INTEGER', 0, None, 0)\n",
      "(5, 'End_position', 'INTEGER', 0, None, 0)\n",
      "(6, 'Strand', 'TEXT', 0, None, 0)\n",
      "(7, 'Variant_Classification', 'TEXT', 0, None, 0)\n",
      "(8, 'Variant_Type', 'TEXT', 0, None, 0)\n",
      "(9, 'Reference_Allele', 'TEXT', 0, None, 0)\n",
      "(10, 'Alternate_Allele', 'TEXT', 0, None, 0)\n",
      "(11, 'dbSNP_RS', 'TEXT', 0, None, 0)\n",
      "(12, 'dbSNP_Val_Status', 'TEXT', 0, None, 0)\n",
      "(13, 'Genome_Change', 'TEXT', 0, None, 0)\n",
      "(14, 'Annotation_Transcript', 'TEXT', 0, None, 0)\n",
      "(15, 'DepMap_ID', 'TEXT', 0, None, 0)\n",
      "(16, 'cDNA_Change', 'TEXT', 0, None, 0)\n",
      "(17, 'Codon_Change', 'TEXT', 0, None, 0)\n",
      "(18, 'Protein_Change', 'TEXT', 0, None, 0)\n",
      "(19, 'isDeleterious', 'TEXT', 0, None, 0)\n",
      "(20, 'isTCGAhotspot', 'TEXT', 0, None, 0)\n",
      "(21, 'TCGAhsCnt', 'REAL', 0, None, 0)\n",
      "(22, 'isCOSMIChotspot', 'TEXT', 0, None, 0)\n",
      "(23, 'COSMIChsCnt', 'REAL', 0, None, 0)\n",
      "(24, 'ExAC_AF', 'REAL', 0, None, 0)\n",
      "(25, 'Variant_annotation', 'TEXT', 0, None, 0)\n",
      "(26, 'CGA_WES_AC', 'TEXT', 0, None, 0)\n",
      "(27, 'HC_AC', 'TEXT', 0, None, 0)\n",
      "(28, 'RD_AC', 'TEXT', 0, None, 0)\n",
      "(29, 'RNAseq_AC', 'TEXT', 0, None, 0)\n",
      "(30, 'SangerWES_AC', 'TEXT', 0, None, 0)\n",
      "(31, 'WGS_AC', 'TEXT', 0, None, 0)\n",
      "('VPS13D', 55187, 37, '1', 12359347, 12359347, '+', 'Nonsense_Mutation', 'SNP', 'C', 'A', '', '', 'g.chr1:12359347C>A', 'ENST00000358136.3', 'ACH-000001', 'c.6122C>A', 'c.(6121-6123)tCa>tAa', 'p.S2041*', 'True', 'False', None, 'False', 0.0, None, 'damaging', '34:213', '', '', '', '34:221', '')\n",
      "('AADACL4', 343066, 37, '1', 12726308, 12726322, '+', 'In_Frame_Del', 'DEL', 'CTGGCGTGACGCCAT', '-', 'rs58218425|rs139261871|rs369427733|rs560787141', 'byFrequency', 'g.chr1:12726308_12726322delCTGGCGTGACGCCAT', 'ENST00000376221.1', 'ACH-000001', 'c.786_800delCTGGCGTGACGCCAT', 'c.(784-801)tcctggcgtgacgccatc>tcc', 'p.WRDAI263del', 'False', 'False', None, 'False', 3.0, None, 'other non-conserving', '57:141', '', '', '', '9:0', '28:32')\n",
      "('IFNLR1', 163702, 37, '1', 24484172, 24484172, '+', 'Silent', 'SNP', 'G', 'A', '', '', 'g.chr1:24484172G>A', 'ENST00000327535.1', 'ACH-000001', 'c.1011C>T', 'c.(1009-1011)ggC>ggT', 'p.G337G', 'False', 'False', None, 'False', 0.0, None, 'silent', '118:0', '', '', '10:0', '118:0', '18:0')\n",
      "('TMEM57', 55219, 37, '1', 25785018, 25785019, '+', 'Frame_Shift_Ins', 'INS', '-', 'A', '', '', 'g.chr1:25785018_25785019insA', 'ENST00000374343.4', 'ACH-000001', 'c.789_790insA', 'c.(790-792)aaafs', 'p.K264fs', 'True', 'False', 0.0, 'False', 0.0, None, 'damaging', '', '', '', '6:28', '', '')\n",
      "('ZSCAN20', 7579, 37, '1', 33954141, 33954141, '+', 'Missense_Mutation', 'SNP', 'T', 'G', '', '', 'g.chr1:33954141T>G', 'ENST00000361328.3', 'ACH-000001', 'c.494T>G', 'c.(493-495)gTg>gGg', 'p.V165G', 'False', 'False', None, 'False', 0.0, None, 'other non-conserving', '28:62', '', '', '', '27:61', '')\n",
      "\n",
      "Table: ccle_variants in files/db/CCLEVarDB.db\n",
      "Schema:\n",
      "(0, 'DepMap_ID', 'TEXT', 0, None, 0)\n",
      "(1, 'Chrom', 'TEXT', 0, None, 0)\n",
      "(2, 'Pos', 'INTEGER', 0, None, 0)\n",
      "(3, 'Ref', 'TEXT', 0, None, 0)\n",
      "(4, 'Alt', 'TEXT', 0, None, 0)\n",
      "(5, 'AF', 'REAL', 0, None, 0)\n",
      "(6, 'VariantType', 'TEXT', 0, None, 0)\n",
      "(7, 'VariantInfo', 'TEXT', 0, None, 0)\n",
      "(8, 'DNAChange', 'TEXT', 0, None, 0)\n",
      "(9, 'ProteinChange', 'TEXT', 0, None, 0)\n",
      "(10, 'gene_name', 'TEXT', 0, None, 0)\n",
      "(11, 'Transcript', 'TEXT', 0, None, 0)\n",
      "(12, 'CCLEDeleterious', 'TEXT', 0, None, 0)\n",
      "(13, 'CosmicHotspot', 'TEXT', 0, None, 0)\n",
      "(14, 'AssociatedWith', 'TEXT', 0, None, 0)\n",
      "(15, 'LoF', 'TEXT', 0, None, 0)\n",
      "(16, 'Driver', 'TEXT', 0, None, 0)\n",
      "(17, 'LikelyDriver', 'TEXT', 0, None, 0)\n",
      "(18, 'LikelyGoF', 'TEXT', 0, None, 0)\n",
      "(19, 'LikelyLoF', 'TEXT', 0, None, 0)\n",
      "(20, 'VariantAnnotation', 'TEXT', 0, None, 0)\n",
      "(21, 'copy_number', 'TEXT', 0, None, 0)\n",
      "('ACH-000839', 'chr1', 1242864, 'GC', 'CT', 0.31, 'DNP', 'MISSENSE', 'c.780_781GC>AG', 'p.L261V', 'C1QTNF12', 'ENST00000330388.2', 'False', 'False', '', 'False', 'False', 'False', 'False', 'False', 'Other', 'Diploid')\n",
      "('ACH-000839', 'chr1', 10647969, 'A', 'G', 0.4, 'SNP', 'MISSENSE', 'c.3329T>C', 'p.V1110A', 'CASZ1', 'ENST00000377022.8', 'False', 'False', '', 'False', 'False', 'False', 'False', 'False', 'Other', 'Diploid')\n",
      "('ACH-000839', 'chr1', 10648097, 'T', 'G', 0.349, 'SNP', 'SILENT', 'c.3201A>C', 'p.T1067T', 'CASZ1', 'ENST00000377022.8', 'False', 'False', '', 'False', 'False', 'False', 'False', 'False', 'Silent', 'Diploid')\n",
      "('ACH-000839', 'chr1', 13198424, 'G', 'A', 0.833, 'SNP', 'MISSENSE', 'c.770C>T', 'p.T257I', 'PRAMEF13', 'ENST00000625019.3', 'False', 'False', '', 'False', 'False', 'False', 'False', 'False', 'Other', 'Diploid')\n",
      "('ACH-000839', 'chr1', 13225068, 'A', 'G', 0.396, 'SNP', 'MISSENSE', 'c.653T>C', 'p.L218P', 'PRAMEF18', 'ENST00000624297.2', 'False', 'False', '', 'False', 'False', 'False', 'False', 'False', 'Other', 'Diploid')\n",
      "MetaData()\n"
     ]
    },
    {
     "data": {
      "text/markdown": [
       "<b>There are 691,749 genes that are diploid.</b>"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Paths to your database files\n",
    "db_files = ['files/db/CCLEGisticCNDB.db', 'files/db/CCLEMutDB.db', 'files/db/CCLEVarDB.db']\n",
    "\n",
    "# Iterate over the files and inspect tables\n",
    "for db_file in db_files:\n",
    "    inspect_database(db_file)\n",
    "\n",
    "engine = create_engine('sqlite:///' + 'files/db/CCLEGisticCNDB.db')\n",
    "metadata_obj = MetaData()\n",
    "print(metadata_obj)\n",
    "\n",
    "from llama_index.indices.struct_store.sql_query import (\n",
    "    SQLTableRetrieverQueryEngine,\n",
    ")\n",
    "from llama_index.objects import (\n",
    "    SQLTableNodeMapping,\n",
    "    ObjectIndex,\n",
    "    SQLTableSchema,\n",
    ")\n",
    "\n",
    "from llama_index import VectorStoreIndex\n",
    "llm = OpenAI(temperature=0.1, model=\"gpt-3.5-turbo\")\n",
    "service_context = ServiceContext.from_defaults(llm=llm)\n",
    "sql_database = SQLDatabase(engine)\n",
    "# set Logging to DEBUG for more detailed outputs\n",
    "table_node_mapping = SQLTableNodeMapping(sql_database)\n",
    "table_schema_objs = [\n",
    "    (SQLTableSchema(table_name=\"ccle_cn_gistic\"))\n",
    "]  # add a SQLTableSchema for each table\n",
    "\n",
    "obj_index = ObjectIndex.from_objects(\n",
    "    table_schema_objs,\n",
    "    table_node_mapping,\n",
    "    VectorStoreIndex,\n",
    ")\n",
    "query_engine = SQLTableRetrieverQueryEngine(\n",
    "    sql_database, obj_index.as_retriever(similarity_top_k=1)\n",
    ")\n",
    "response = query_engine.query(\"How many genes are Diploid?\")\n",
    "display(Markdown(f\"<b>{response}</b>\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# A check\n",
    "import re \n",
    "def extract_sql(self, llm_response: str) -> str:\n",
    "    # If the llm_response contains a markdown code block, with or without the sql tag, extract the sql from it\n",
    "    sql = re.search(r\"```sql\\n(.*)```\", llm_response, re.DOTALL)\n",
    "    if sql:\n",
    "        self.log(f\"Output from LLM: {llm_response} \\nExtracted SQL: {sql.group(1)}\")\n",
    "        return sql.group(1)\n",
    "\n",
    "    sql = re.search(r\"```(.*)```\", llm_response, re.DOTALL)\n",
    "    if sql:\n",
    "        self.log(f\"Output from LLM: {llm_response} \\nExtracted SQL: {sql.group(1)}\")\n",
    "        return sql.group(1)\n",
    "\n",
    "    return llm_response\n",
    "\n",
    "def is_sql_valid(self, sql: str) -> bool:\n",
    "    # This is a check to see the SQL is valid and should be run\n",
    "    # This simple function just checks if the SQL contains a SELECT statement\n",
    "\n",
    "    if \"SELECT\" in sql.upper():\n",
    "        return True\n",
    "    else:\n",
    "        return False\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
